{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Importing some shizzle.\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from matplotlib import pyplot as plt\n",
    "from scipy import stats\n",
    "import time\n",
    "import requests\n",
    "import datetime\n",
    "import sys\n",
    "import os\n",
    "import pathlib\n",
    "import seaborn as sns\n",
    "from sklearn.metrics import roc_auc_score\n",
    "import math\n",
    "import json\n",
    "import glob\n",
    "import gzip\n",
    "from utilities import perform_stats, calc_z_scores, get_header\n",
    "\n",
    "# Defining some import and export locations\n",
    "location = 'rjsietsma'\n",
    "read_loc = '/run/media/rjsietsma/evo2tb/linux/Datafiles/'\n",
    "data_expor_loc = '/home/'+location+'/Documents/School/Master_DSLS/Final_Thesis/Past_initial_data/'\n",
    "img_output_dir = '/home/'+location+'/PycharmProjects/dsls_master_thesis/side_scripts/output_img/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " # CTEPG (Catching The Elusive Predictable Genes) 15-06-2020 - 22-06-2020\n",
    "<br/><br/>\n",
    "\n",
    "What have I done:\n",
    "\n",
    "\n",
    "Planned for this week:\n",
    "- Make model on bad performing genes and investigate further.\n",
    "    - At first, make models of [the original training data](https://zenodo.org/record/3564757#.XudiZ2ixUUE), models of various gene panels (but keep the variants of those panels within the bigger model). See if they improve and how much.\n",
    "        - _Note: the training data needs to be converted to [variant calling file (VCF)](http://samtools.github.io/hts-specs/VCFv4.1.pdf) before using, which means it needs a fileformat. Or the annotated file can be downloaded directly from Shuang her GCC tmp storage._\n",
    "    - Then make a single model of the original training data for CADD 1.6 annotations.\n",
    "    - Then make multiple models of the original training data for CADD 1.6 annotations.\n",
    "    - Investigate improvements.\n",
    "    - Then choose between:\n",
    "        - newer / larger dataset or \n",
    "        - Custom features that might be relevant to the cause (prefered)\n",
    "- Make single plot of panel vs all others combined (except for 5GPM).\n",
    "- Usefull, [convert 37 to 38](https://genome.ucsc.edu/cgi-bin/hgLiftOver).\n",
    "- Start investigating what is required for the data sources to be used in the training phase.\n",
    "- Create outline for article.\n",
    "     - For static plots: make labels on each point with the term name and the number of genes in (n=x) behind that.\n",
    "     - For first concept, it's due date is the last week of the resit exams on the 6th of July.\n",
    "- _Optionally_ :\n",
    "    - Find possible way to annotate genes for co-factors, sub-family etc.\n",
    "    - Interactive plot of all variants labeled as benign or pathogenic with their capice score (y-axis) and location (x-axis), to show the variant itself.\n",
    "<br/><br/>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Allergy/Immunology/Infectious</th>\n",
       "      <th>Alt</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Audiologic/Otolaryngologic</th>\n",
       "      <th>Biochemical</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>CDSpos</th>\n",
       "      <th>Cardiovascular</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>revel</th>\n",
       "      <th>sift</th>\n",
       "      <th>source</th>\n",
       "      <th>tOverlapMotifs</th>\n",
       "      <th>targetScan</th>\n",
       "      <th>to_be_deleted</th>\n",
       "      <th>verPhCons</th>\n",
       "      <th>verPhyloP</th>\n",
       "      <th>inTest</th>\n",
       "      <th>sample_weight</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14</td>\n",
       "      <td>False</td>\n",
       "      <td>G</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS9787.1</td>\n",
       "      <td>806.0</td>\n",
       "      <td>False</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.843</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS13112.1</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift,stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>4.670</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>C</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS13112.1</td>\n",
       "      <td>2118.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.043</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS13112.1</td>\n",
       "      <td>1586.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>6.221</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>6.368</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334596</th>\n",
       "      <td>17</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS32642.1</td>\n",
       "      <td>1563.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>6.031</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334597</th>\n",
       "      <td>17</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS32642.1</td>\n",
       "      <td>2029.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>4.100</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334598</th>\n",
       "      <td>10</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS7431.1</td>\n",
       "      <td>1216.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.852</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334599</th>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS2382.1</td>\n",
       "      <td>2998.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.031</td>\n",
       "      <td>2.213</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334600</th>\n",
       "      <td>5</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS3952.1</td>\n",
       "      <td>1221.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.528</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>334601 rows × 152 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Chrom  Allergy/Immunology/Infectious Alt          AnnoType  \\\n",
       "0          14                          False   G  CodingTranscript   \n",
       "1          20                          False   T  CodingTranscript   \n",
       "2          20                          False   C  CodingTranscript   \n",
       "3          20                          False   A  CodingTranscript   \n",
       "4          20                          False   A        Intergenic   \n",
       "...       ...                            ...  ..               ...   \n",
       "334596     17                          False   A  CodingTranscript   \n",
       "334597     17                          False   T  CodingTranscript   \n",
       "334598     10                          False   T  CodingTranscript   \n",
       "334599      2                          False   T  CodingTranscript   \n",
       "334600      5                          False   T  CodingTranscript   \n",
       "\n",
       "        Audiologic/Otolaryngologic  Biochemical         CCDS  CDSpos  \\\n",
       "0                            False        False   CCDS9787.1   806.0   \n",
       "1                             True        False  CCDS13112.1  1899.0   \n",
       "2                             True        False  CCDS13112.1  2118.0   \n",
       "3                             True        False  CCDS13112.1  1586.0   \n",
       "4                             True        False          NaN     NaN   \n",
       "...                            ...          ...          ...     ...   \n",
       "334596                       False        False  CCDS32642.1  1563.0   \n",
       "334597                       False        False  CCDS32642.1  2029.0   \n",
       "334598                       False        False   CCDS7431.1  1216.0   \n",
       "334599                       False        False   CCDS2382.1  2998.0   \n",
       "334600                       False        False   CCDS3952.1  1221.0   \n",
       "\n",
       "        Cardiovascular              ConsDetail  ...  revel sift   source  \\\n",
       "0                False              frameshift  ...    NaN  NaN     vkgl   \n",
       "1                 True  frameshift,stop_gained  ...    NaN  NaN     vkgl   \n",
       "2                 True              frameshift  ...    NaN  NaN     vkgl   \n",
       "3                 True              frameshift  ...    NaN  NaN     vkgl   \n",
       "4                 True              downstream  ...    NaN  NaN     vkgl   \n",
       "...                ...                     ...  ...    ...  ...      ...   \n",
       "334596           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334597           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334598           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334599           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334600           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "\n",
       "        tOverlapMotifs  targetScan  to_be_deleted  verPhCons verPhyloP inTest  \\\n",
       "0                  NaN         NaN          False      1.000     5.843  False   \n",
       "1                  NaN         NaN          False      1.000     4.670  False   \n",
       "2                  NaN         NaN          False      1.000     5.043  False   \n",
       "3                  NaN         NaN          False      1.000     6.221  False   \n",
       "4                  NaN         NaN          False      1.000     6.368  False   \n",
       "...                ...         ...            ...        ...       ...    ...   \n",
       "334596             NaN         NaN          False      1.000     6.031  False   \n",
       "334597             NaN         NaN          False      1.000     4.100  False   \n",
       "334598             NaN         NaN          False      1.000     5.852  False   \n",
       "334599             NaN         NaN          False      0.031     2.213  False   \n",
       "334600             NaN         NaN          False      1.000     0.528  False   \n",
       "\n",
       "        sample_weight  \n",
       "0                 1.0  \n",
       "1                 1.0  \n",
       "2                 1.0  \n",
       "3                 1.0  \n",
       "4                 1.0  \n",
       "...               ...  \n",
       "334596            0.8  \n",
       "334597            0.8  \n",
       "334598            0.8  \n",
       "334599            0.8  \n",
       "334600            0.8  \n",
       "\n",
       "[334601 rows x 152 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_loc = os.path.join(read_loc, 'train.txt.gz')\n",
    "header = get_header(file_loc, '#Chrom')\n",
    "train = pd.read_csv(file_loc, compression='gzip', names=header, comment='#', sep='\\t', low_memory=False)\n",
    "train"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
