{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Index\n",
    "- [Imports](#Imports)\n",
    "- [CTEPG](#CTEPG-(Catching-The-Elusive-Predictable-Genes)-15-06-2020---22-06-2020)\n",
    "- [Dyslipid database creation](#Creating-dyslipid-dataset)\n",
    "- [Match / unmatch new results to old results (benign/pathogenic)](#AUC-validation-vanilla-new-model)\n",
    "- [Cross validated models](#CV-Models)\n",
    "- [XGB models out of CV models](#New-Models)\n",
    "- [Correct new threshold](#Correct-probability-prediction)\n",
    "- [AUC preparation & analysis](#Preparing-for-AUC-analysis)\n",
    "- [Model on best performing panel, preparation](#Model-on-best-performing-panel,-prepartation)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Importing some shizzle.\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from matplotlib import pyplot as plt\n",
    "from scipy import stats\n",
    "import time\n",
    "import requests\n",
    "import datetime\n",
    "import sys\n",
    "import os\n",
    "import pathlib\n",
    "import seaborn as sns\n",
    "from sklearn.metrics import roc_auc_score\n",
    "import math\n",
    "import json\n",
    "import glob\n",
    "import gzip\n",
    "import pickle\n",
    "import xgboost as xgb\n",
    "from utilities import perform_stats, calc_z_scores, get_header\n",
    "from sklearn.metrics import recall_score, roc_auc_score, precision_score, f1_score\n",
    "\n",
    "# Defining some import and export locations\n",
    "location = 'rjsietsma'\n",
    "read_loc = '/run/media/rjsietsma/evo2tb/linux/Datafiles/'\n",
    "data_expor_loc = '/home/'+location+'/Documents/School/Master_DSLS/Final_Thesis/Past_initial_data/'\n",
    "img_output_dir = '/home/'+location+'/PycharmProjects/dsls_master_thesis/side_scripts/output_img/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " # CTEPG (Catching The Elusive Predictable Genes) 15-06-2020 - 22-06-2020\n",
    "<br/><br/>\n",
    "\n",
    "What have I done:\n",
    "\n",
    "- Made plot of Panel vs all combined:\n",
    "![Panel vs all combined](img/allvs1boxplot_umcggenepanels.png)\n",
    "\n",
    "- Issued [pull request](https://github.com/molgenis/capice/pull/1) for Dennis and Bart to be able to work with CAPICE locally.\n",
    "- Refactored CAPICE code _(not finished)_.\n",
    "- Retrained new models.\n",
    "- Found new Pathogenic / Benign threshold within CAPICE.\n",
    "- Made fancy new notebook with [indexing](#Index).\n",
    "\n",
    "\n",
    "\n",
    "Planned for this week:\n",
    "- Make barplot of type of variants for each panel.\n",
    "- Feature importances for new base model vs feature importances specific model.\n",
    "    - Scatter plot of base on the y axis and specific models on the x axis.\n",
    "    - Do make multiple plots.\n",
    "    - Make a feature importance rank delta (so how much do the feature importance ranks change?)\n",
    "- Rebuild benchmarking dataset to the full dataset in Initial Data Exploration.\n",
    "- Model refinement:\n",
    "    - AUC analysis of new model (model xgb_booster37.pickle.dat)\n",
    "    - Check for improvements.\n",
    "    - Then split the training / testing data in the different models. (xgb_booster37.pickle.dat and xgb_booster_dyslipid.pickle.dat)\n",
    "        - Also do the full dataset on dyslipid. Do not try to split the data into the 2 models. \n",
    "        - Try to build new models on average performing and well performing panels aswell, try to see if everything improves nicely.\n",
    "            - Could conclude from this: performance improves a lot for bad performing panels when trained on those bad performing panels, not so much for good performing panels (but do improve). If trained on good performing models, does that do the same?\n",
    "    - AUC analysis again.\n",
    "    - Upgrade to CADD 1.6, train again.\n",
    "    - AUC analysis.\n",
    "    - [Convert to genome build 38](https://genome.ucsc.edu/cgi-bin/hgLiftOver)\n",
    "    - Make [pseudo-VCF](http://samtools.github.io/hts-specs/VCFv4.1.pdf) of build 38 train file and test file.\n",
    "    - Repeat.\n",
    "    - Then choose between:\n",
    "        - Newer / larger datasets or\n",
    "        - Adding features specific to the benefits of the different models (prefered)\n",
    "        - _Note: Shuang her data has almost half the variants than the newer VGKL and Clinvar datasets, with 3913 benign and 1116 malignant samples._\n",
    "- Start investigating what is required for the data sources to be used in the training phase.\n",
    "- Create outline for article.\n",
    "     - For static plots: make labels on each point with the term name and the number of genes in (n=x) behind that.\n",
    "     - For first concept, it's due date is the last week of the resit exams on the 6th of July.\n",
    "- _Optionally_ :\n",
    "    - Find possible way to annotate genes for co-factors, sub-family etc.\n",
    "    - Interactive plot of all variants labeled as benign or pathogenic with their capice score (y-axis) and location (x-axis), to show the variant itself.\n",
    "<br/><br/>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Creating dyslipid dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Allergy/Immunology/Infectious</th>\n",
       "      <th>Alt</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Audiologic/Otolaryngologic</th>\n",
       "      <th>Biochemical</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>CDSpos</th>\n",
       "      <th>Cardiovascular</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>revel</th>\n",
       "      <th>sift</th>\n",
       "      <th>source</th>\n",
       "      <th>tOverlapMotifs</th>\n",
       "      <th>targetScan</th>\n",
       "      <th>to_be_deleted</th>\n",
       "      <th>verPhCons</th>\n",
       "      <th>verPhyloP</th>\n",
       "      <th>inTest</th>\n",
       "      <th>sample_weight</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14</td>\n",
       "      <td>False</td>\n",
       "      <td>G</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS9787.1</td>\n",
       "      <td>806.0</td>\n",
       "      <td>False</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.843</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS13112.1</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift,stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>4.670</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>C</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS13112.1</td>\n",
       "      <td>2118.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.043</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS13112.1</td>\n",
       "      <td>1586.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>6.221</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>6.368</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334596</th>\n",
       "      <td>17</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS32642.1</td>\n",
       "      <td>1563.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>6.031</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334597</th>\n",
       "      <td>17</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS32642.1</td>\n",
       "      <td>2029.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>4.100</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334598</th>\n",
       "      <td>10</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS7431.1</td>\n",
       "      <td>1216.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.852</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334599</th>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS2382.1</td>\n",
       "      <td>2998.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.031</td>\n",
       "      <td>2.213</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334600</th>\n",
       "      <td>5</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS3952.1</td>\n",
       "      <td>1221.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.528</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>334601 rows × 152 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Chrom  Allergy/Immunology/Infectious Alt          AnnoType  \\\n",
       "0          14                          False   G  CodingTranscript   \n",
       "1          20                          False   T  CodingTranscript   \n",
       "2          20                          False   C  CodingTranscript   \n",
       "3          20                          False   A  CodingTranscript   \n",
       "4          20                          False   A        Intergenic   \n",
       "...       ...                            ...  ..               ...   \n",
       "334596     17                          False   A  CodingTranscript   \n",
       "334597     17                          False   T  CodingTranscript   \n",
       "334598     10                          False   T  CodingTranscript   \n",
       "334599      2                          False   T  CodingTranscript   \n",
       "334600      5                          False   T  CodingTranscript   \n",
       "\n",
       "        Audiologic/Otolaryngologic  Biochemical         CCDS  CDSpos  \\\n",
       "0                            False        False   CCDS9787.1   806.0   \n",
       "1                             True        False  CCDS13112.1  1899.0   \n",
       "2                             True        False  CCDS13112.1  2118.0   \n",
       "3                             True        False  CCDS13112.1  1586.0   \n",
       "4                             True        False          NaN     NaN   \n",
       "...                            ...          ...          ...     ...   \n",
       "334596                       False        False  CCDS32642.1  1563.0   \n",
       "334597                       False        False  CCDS32642.1  2029.0   \n",
       "334598                       False        False   CCDS7431.1  1216.0   \n",
       "334599                       False        False   CCDS2382.1  2998.0   \n",
       "334600                       False        False   CCDS3952.1  1221.0   \n",
       "\n",
       "        Cardiovascular              ConsDetail  ...  revel sift   source  \\\n",
       "0                False              frameshift  ...    NaN  NaN     vkgl   \n",
       "1                 True  frameshift,stop_gained  ...    NaN  NaN     vkgl   \n",
       "2                 True              frameshift  ...    NaN  NaN     vkgl   \n",
       "3                 True              frameshift  ...    NaN  NaN     vkgl   \n",
       "4                 True              downstream  ...    NaN  NaN     vkgl   \n",
       "...                ...                     ...  ...    ...  ...      ...   \n",
       "334596           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334597           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334598           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334599           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334600           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "\n",
       "        tOverlapMotifs  targetScan  to_be_deleted  verPhCons verPhyloP inTest  \\\n",
       "0                  NaN         NaN          False      1.000     5.843  False   \n",
       "1                  NaN         NaN          False      1.000     4.670  False   \n",
       "2                  NaN         NaN          False      1.000     5.043  False   \n",
       "3                  NaN         NaN          False      1.000     6.221  False   \n",
       "4                  NaN         NaN          False      1.000     6.368  False   \n",
       "...                ...         ...            ...        ...       ...    ...   \n",
       "334596             NaN         NaN          False      1.000     6.031  False   \n",
       "334597             NaN         NaN          False      1.000     4.100  False   \n",
       "334598             NaN         NaN          False      1.000     5.852  False   \n",
       "334599             NaN         NaN          False      0.031     2.213  False   \n",
       "334600             NaN         NaN          False      1.000     0.528  False   \n",
       "\n",
       "        sample_weight  \n",
       "0                 1.0  \n",
       "1                 1.0  \n",
       "2                 1.0  \n",
       "3                 1.0  \n",
       "4                 1.0  \n",
       "...               ...  \n",
       "334596            0.8  \n",
       "334597            0.8  \n",
       "334598            0.8  \n",
       "334599            0.8  \n",
       "334600            0.8  \n",
       "\n",
       "[334601 rows x 152 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_loc = os.path.join(read_loc, 'train.txt.gz')\n",
    "header = get_header(file_loc, '#Chrom')\n",
    "train = pd.read_csv(file_loc, compression='gzip', names=header, comment='#', sep='\\t', low_memory=False)\n",
    "train"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('./umcg_genepanels.json', 'r') as json_file:\n",
    "    genes = json.load(json_file)\n",
    "dislipid_genes = genes['Hart- en vaatziekten']\n",
    "genelist = []\n",
    "for key, value in dislipid_genes.items():\n",
    "    if key.lower().startswith('dyslipid'):\n",
    "        for g in value:\n",
    "            if g not in genelist:\n",
    "                genelist.append(g)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Allergy/Immunology/Infectious</th>\n",
       "      <th>Alt</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Audiologic/Otolaryngologic</th>\n",
       "      <th>Biochemical</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>CDSpos</th>\n",
       "      <th>Cardiovascular</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>revel</th>\n",
       "      <th>sift</th>\n",
       "      <th>source</th>\n",
       "      <th>tOverlapMotifs</th>\n",
       "      <th>targetScan</th>\n",
       "      <th>to_be_deleted</th>\n",
       "      <th>verPhCons</th>\n",
       "      <th>verPhyloP</th>\n",
       "      <th>inTest</th>\n",
       "      <th>sample_weight</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>16</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS10772.1</td>\n",
       "      <td>848.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.928</td>\n",
       "      <td>2.614</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>19</td>\n",
       "      <td>False</td>\n",
       "      <td>CCGGCGAGGTGCAGGCCATGCT</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS12647.1</td>\n",
       "      <td>409.0</td>\n",
       "      <td>True</td>\n",
       "      <td>protein_altering</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>0.863</td>\n",
       "      <td>0.839</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>C</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS1703.1</td>\n",
       "      <td>13028.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.058</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>G</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS1703.1</td>\n",
       "      <td>28.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.021</td>\n",
       "      <td>-0.103</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>C</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS1703.1</td>\n",
       "      <td>2534.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.653</td>\n",
       "      <td>0.251</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334284</th>\n",
       "      <td>12</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS8685.1</td>\n",
       "      <td>1093.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.999</td>\n",
       "      <td>3.072</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334285</th>\n",
       "      <td>12</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS8685.1</td>\n",
       "      <td>1475.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.580</td>\n",
       "      <td>0.563</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334286</th>\n",
       "      <td>12</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS8685.1</td>\n",
       "      <td>1537.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.643</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334287</th>\n",
       "      <td>12</td>\n",
       "      <td>False</td>\n",
       "      <td>G</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS8685.1</td>\n",
       "      <td>1553.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.510</td>\n",
       "      <td>2.765</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334288</th>\n",
       "      <td>12</td>\n",
       "      <td>False</td>\n",
       "      <td>G</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS8685.1</td>\n",
       "      <td>1634.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.326</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5029 rows × 152 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Chrom  Allergy/Immunology/Infectious                     Alt  \\\n",
       "142        16                          False                       T   \n",
       "148        19                          False  CCGGCGAGGTGCAGGCCATGCT   \n",
       "149         2                          False                       C   \n",
       "150         2                          False                       G   \n",
       "151         2                          False                       C   \n",
       "...       ...                            ...                     ...   \n",
       "334284     12                          False                       T   \n",
       "334285     12                          False                       A   \n",
       "334286     12                          False                       T   \n",
       "334287     12                          False                       G   \n",
       "334288     12                          False                       G   \n",
       "\n",
       "                AnnoType  Audiologic/Otolaryngologic  Biochemical  \\\n",
       "142     CodingTranscript                       False        False   \n",
       "148     CodingTranscript                       False        False   \n",
       "149     CodingTranscript                       False        False   \n",
       "150     CodingTranscript                       False        False   \n",
       "151     CodingTranscript                       False        False   \n",
       "...                  ...                         ...          ...   \n",
       "334284  CodingTranscript                       False        False   \n",
       "334285  CodingTranscript                       False        False   \n",
       "334286  CodingTranscript                       False        False   \n",
       "334287  CodingTranscript                       False        False   \n",
       "334288  CodingTranscript                       False        False   \n",
       "\n",
       "               CCDS   CDSpos  Cardiovascular        ConsDetail  ...  revel  \\\n",
       "142     CCDS10772.1    848.0            True        frameshift  ...    NaN   \n",
       "148     CCDS12647.1    409.0            True  protein_altering  ...    NaN   \n",
       "149      CCDS1703.1  13028.0            True        frameshift  ...    NaN   \n",
       "150      CCDS1703.1     28.0            True        frameshift  ...    NaN   \n",
       "151      CCDS1703.1   2534.0            True        frameshift  ...    NaN   \n",
       "...             ...      ...             ...               ...  ...    ...   \n",
       "334284   CCDS8685.1   1093.0           False       stop_gained  ...    NaN   \n",
       "334285   CCDS8685.1   1475.0           False       stop_gained  ...    NaN   \n",
       "334286   CCDS8685.1   1537.0           False       stop_gained  ...    NaN   \n",
       "334287   CCDS8685.1   1553.0           False       stop_gained  ...    NaN   \n",
       "334288   CCDS8685.1   1634.0           False       stop_gained  ...    NaN   \n",
       "\n",
       "       sift   source  tOverlapMotifs  targetScan  to_be_deleted  verPhCons  \\\n",
       "142     NaN     vkgl             NaN         NaN          False      0.928   \n",
       "148     NaN     vkgl             NaN         NaN           True      0.863   \n",
       "149     NaN     vkgl             NaN         NaN          False      0.000   \n",
       "150     NaN     vkgl             NaN         NaN          False      0.021   \n",
       "151     NaN     vkgl             NaN         NaN          False      0.653   \n",
       "...     ...      ...             ...         ...            ...        ...   \n",
       "334284  NaN  unknown             NaN         NaN          False      0.999   \n",
       "334285  NaN  unknown             NaN         NaN          False      0.580   \n",
       "334286  NaN  unknown             NaN         NaN          False      0.003   \n",
       "334287  NaN  unknown             NaN         NaN          False      0.510   \n",
       "334288  NaN  unknown             NaN         NaN          False      0.000   \n",
       "\n",
       "       verPhyloP inTest  sample_weight  \n",
       "142        2.614  False            1.0  \n",
       "148        0.839  False            1.0  \n",
       "149        0.058  False            1.0  \n",
       "150       -0.103  False            1.0  \n",
       "151        0.251  False            1.0  \n",
       "...          ...    ...            ...  \n",
       "334284     3.072  False            0.8  \n",
       "334285     0.563  False            0.8  \n",
       "334286     0.643  False            0.8  \n",
       "334287     2.765  False            0.8  \n",
       "334288    -0.326  False            0.8  \n",
       "\n",
       "[5029 rows x 152 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dislipid_subset = train.loc[train['GeneName'].isin(genelist)]\n",
    "dislipid_subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Benign        3913\n",
       "Pathogenic    1116\n",
       "Name: label, dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dislipid_subset['label'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CV Models"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'objective': 'binary:logistic',\n",
       " 'base_score': 0.5,\n",
       " 'booster': 'gbtree',\n",
       " 'colsample_bylevel': 1,\n",
       " 'colsample_bynode': 1,\n",
       " 'colsample_bytree': 1,\n",
       " 'gamma': 0,\n",
       " 'gpu_id': -1,\n",
       " 'importance_type': 'gain',\n",
       " 'interaction_constraints': '',\n",
       " 'learning_rate': 0.12794736796290956,\n",
       " 'max_delta_step': 0,\n",
       " 'max_depth': 15,\n",
       " 'min_child_weight': 1,\n",
       " 'monotone_constraints': '()',\n",
       " 'n_estimators': 420,\n",
       " 'n_jobs': 8,\n",
       " 'num_parallel_tree': 1,\n",
       " 'random_state': 0,\n",
       " 'reg_alpha': 0,\n",
       " 'reg_lambda': 1,\n",
       " 'scale_pos_weight': 1,\n",
       " 'subsample': 1,\n",
       " 'tree_method': 'exact',\n",
       " 'validate_parameters': 1,\n",
       " 'verbosity': 1}"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model = pickle.load(open('./models/xgb_weightedSample_randomsearch_v2.pickle.dat', 'rb'))\n",
    "xgbmodel = model.best_estimator_\n",
    "optimal_params = xgbmodel.get_params()\n",
    "optimal_params.pop('missing')\n",
    "with open('./optimal_capice_v2_params.json', 'w+') as json_file:\n",
    "    json.dump(optimal_params, json_file, indent=4)\n",
    "optimal_params"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "model = xgb.XGBClassifier(optimal_params)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'objective': 'binary:logistic',\n",
       " 'base_score': 0.5,\n",
       " 'booster': 'gbtree',\n",
       " 'colsample_bylevel': 1,\n",
       " 'colsample_bynode': 1,\n",
       " 'colsample_bytree': 1,\n",
       " 'gamma': 0,\n",
       " 'gpu_id': -1,\n",
       " 'importance_type': 'gain',\n",
       " 'interaction_constraints': '',\n",
       " 'learning_rate': 0.1968770013672272,\n",
       " 'max_delta_step': 0,\n",
       " 'max_depth': 16,\n",
       " 'min_child_weight': 1,\n",
       " 'monotone_constraints': '()',\n",
       " 'n_estimators': 427,\n",
       " 'n_jobs': 8,\n",
       " 'num_parallel_tree': 1,\n",
       " 'random_state': 0,\n",
       " 'reg_alpha': 0,\n",
       " 'reg_lambda': 1,\n",
       " 'scale_pos_weight': 1,\n",
       " 'subsample': 1,\n",
       " 'tree_method': 'exact',\n",
       " 'validate_parameters': 1,\n",
       " 'verbosity': 1}"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model_dislipid = pickle.load(open('./models/xgb_weightedSample_randomsearch_dislipid.pickle.dat', 'rb'))\n",
    "xgbmodel_dislipid = model_dislipid.best_estimator_\n",
    "optimal_params_dislipid = xgbmodel_dislipid.get_params()\n",
    "optimal_params_dislipid.pop('missing')\n",
    "with open('./optimal_capice_dislipid_params.json', 'w+') as json_file:\n",
    "    json.dump(optimal_params_dislipid, json_file, indent=4)\n",
    "optimal_params_dislipid"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# New Models"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([0.00282909, 0.00207738, 0.00183909, 0.00320904, 0.00281486,\n",
       "       0.00190347, 0.00192447, 0.00191927, 0.00175739, 0.00351866,\n",
       "       0.00253019, 0.00301028, 0.00308079, 0.00301589, 0.00242207,\n",
       "       0.00284953, 0.00254809, 0.00301885, 0.00276217, 0.00329895,\n",
       "       0.0029472 , 0.00288387, 0.00196225, 0.00324255, 0.00249287,\n",
       "       0.00308638, 0.00275804, 0.00326703, 0.003321  , 0.02822421,\n",
       "       0.01098935, 0.00849544, 0.0026135 , 0.0021958 , 0.00212344,\n",
       "       0.02957111, 0.00307086, 0.00227875, 0.00224516, 0.00332749,\n",
       "       0.00291972, 0.00251198, 0.00405553, 0.00422445, 0.00319547,\n",
       "       0.00250752, 0.00358728, 0.0023705 , 0.00314594, 0.00369558,\n",
       "       0.00300821, 0.00350186, 0.00383268, 0.00286349, 0.00239821,\n",
       "       0.00318689, 0.00370395, 0.00359068, 0.00305595, 0.01804393,\n",
       "       0.00381216, 0.00197791, 0.0018304 , 0.0023901 , 0.00244684,\n",
       "       0.00294106, 0.00317492, 0.00303257, 0.00199907, 0.00196537,\n",
       "       0.00232091, 0.00312337, 0.00172605, 0.00178056, 0.        ,\n",
       "       0.00235593, 0.00206226, 0.00217503, 0.00434332, 0.0020939 ,\n",
       "       0.00449965, 0.01930718, 0.0018447 , 0.00274791, 0.00609136,\n",
       "       0.00580421, 0.00188665, 0.00931865, 0.00234432, 0.00315913,\n",
       "       0.00402672, 0.00646648, 0.00910742, 0.00322477, 0.00176483,\n",
       "       0.00343788, 0.00176548, 0.07158165, 0.0033759 , 0.        ,\n",
       "       0.00535233, 0.0063484 , 0.00330518, 0.00307293, 0.00348948,\n",
       "       0.00726572, 0.01224917, 0.01062934, 0.02123239, 0.00196283,\n",
       "       0.00444081, 0.00216935, 0.00285928, 0.0026971 , 0.00244203,\n",
       "       0.00233031, 0.14946337, 0.02471637, 0.02496736, 0.01712855,\n",
       "       0.00207521, 0.00310974, 0.00341634, 0.16352823, 0.00382764,\n",
       "       0.01300227, 0.00745142, 0.01435524, 0.01285369, 0.00542477,\n",
       "       0.00070084], dtype=float32)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model = pickle.load(open('./xgbmodels/xgb_booster_v2.pickle.dat', 'rb'))\n",
    "model.feature_importances_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([0.00278866, 0.00208772, 0.00174322, 0.00296931, 0.00269667,\n",
       "       0.00179555, 0.00186813, 0.00186437, 0.00175499, 0.00395201,\n",
       "       0.00240749, 0.00280701, 0.00290935, 0.00285559, 0.0023123 ,\n",
       "       0.0027687 , 0.0024073 , 0.00306501, 0.00286279, 0.00350749,\n",
       "       0.00261141, 0.00282689, 0.00188535, 0.00320358, 0.00238541,\n",
       "       0.00293148, 0.00261692, 0.00331071, 0.00368271, 0.02656523,\n",
       "       0.01037843, 0.00846208, 0.00255091, 0.00212624, 0.00206606,\n",
       "       0.03021989, 0.0027948 , 0.00225078, 0.00219586, 0.00343691,\n",
       "       0.00298393, 0.00254942, 0.00387159, 0.00394147, 0.00327632,\n",
       "       0.00257063, 0.00359947, 0.00250112, 0.00339626, 0.00355939,\n",
       "       0.0030033 , 0.00344496, 0.00338445, 0.00294264, 0.00213853,\n",
       "       0.00313247, 0.00377457, 0.00349594, 0.00298174, 0.01875002,\n",
       "       0.00379864, 0.00185824, 0.00179364, 0.00239529, 0.00231335,\n",
       "       0.00317147, 0.00292523, 0.00316273, 0.00262851, 0.00228997,\n",
       "       0.00601442, 0.00521736, 0.00169761, 0.00183883, 0.        ,\n",
       "       0.00234997, 0.00158524, 0.00213381, 0.00315469, 0.00248724,\n",
       "       0.00430137, 0.01801617, 0.0018445 , 0.00280421, 0.00624151,\n",
       "       0.00602516, 0.00171275, 0.00894723, 0.00171685, 0.00311938,\n",
       "       0.00349789, 0.00585964, 0.00859777, 0.00276252, 0.00209822,\n",
       "       0.00323153, 0.00161626, 0.06846787, 0.0036844 , 0.        ,\n",
       "       0.00444979, 0.00538614, 0.00277188, 0.00351531, 0.00360888,\n",
       "       0.00591644, 0.01264253, 0.00823699, 0.0243496 , 0.00018329,\n",
       "       0.00428347, 0.00217733, 0.00302215, 0.0026641 , 0.00253335,\n",
       "       0.00206614, 0.16421454, 0.02516109, 0.02144627, 0.02164953,\n",
       "       0.00223931, 0.00240782, 0.00254637, 0.16032705, 0.00204109,\n",
       "       0.01342087, 0.00806119, 0.01339926, 0.01314391, 0.005304  ,\n",
       "       0.00024527], dtype=float32)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model_dislipid = pickle.load(open('./xgbmodels/xgb_booster_dyslipid.pickle.dat', 'rb'))\n",
    "model_dislipid.feature_importances_"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# AUC validation vanilla new model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>probabilities</th>\n",
       "      <th>prediction</th>\n",
       "      <th>combined_prediction</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17</td>\n",
       "      <td>41246652</td>\n",
       "      <td>ACATTC</td>\n",
       "      <td>GA</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>26.600</td>\n",
       "      <td>9.999933e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>19</td>\n",
       "      <td>11216246</td>\n",
       "      <td>TGCAAGGACAAATCTGAC</td>\n",
       "      <td>CCGACTG</td>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999907e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>19</td>\n",
       "      <td>11216251</td>\n",
       "      <td>GGACAAATCTGACGA</td>\n",
       "      <td>AACTGCGGTAAACTGCGGTAAACT</td>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999896e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>19</td>\n",
       "      <td>11216262</td>\n",
       "      <td>ACG</td>\n",
       "      <td>CA</td>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999894e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>47702328</td>\n",
       "      <td>GTTGA</td>\n",
       "      <td>TTTC</td>\n",
       "      <td>MSH2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999891e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334596</th>\n",
       "      <td>14</td>\n",
       "      <td>64653189</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>MIR548AZ</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>14.370</td>\n",
       "      <td>2.765540e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334597</th>\n",
       "      <td>17</td>\n",
       "      <td>10419945</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>0.611</td>\n",
       "      <td>2.524258e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334598</th>\n",
       "      <td>17</td>\n",
       "      <td>10419849</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>9.827</td>\n",
       "      <td>2.164549e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334599</th>\n",
       "      <td>16</td>\n",
       "      <td>88804658</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>LOC100289580</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>14.100</td>\n",
       "      <td>1.984229e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334600</th>\n",
       "      <td>17</td>\n",
       "      <td>10419849</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>10.040</td>\n",
       "      <td>1.227600e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>334601 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr       pos                 ref                       alt  \\\n",
       "0       17  41246652              ACATTC                        GA   \n",
       "1       19  11216246  TGCAAGGACAAATCTGAC                   CCGACTG   \n",
       "2       19  11216251     GGACAAATCTGACGA  AACTGCGGTAAACTGCGGTAAACT   \n",
       "3       19  11216262                 ACG                        CA   \n",
       "4        2  47702328               GTTGA                      TTTC   \n",
       "...     ..       ...                 ...                       ...   \n",
       "334596  14  64653189                   T                         C   \n",
       "334597  17  10419945                   A                         G   \n",
       "334598  17  10419849                   T                         G   \n",
       "334599  16  88804658                   G                         A   \n",
       "334600  17  10419849                   T                         C   \n",
       "\n",
       "            GeneName  Consequence   PHRED  probabilities  prediction  \\\n",
       "0              BRCA1  FRAME_SHIFT  26.600   9.999933e-01  Pathogenic   \n",
       "1               LDLR  FRAME_SHIFT  35.000   9.999907e-01  Pathogenic   \n",
       "2               LDLR  FRAME_SHIFT  34.000   9.999896e-01  Pathogenic   \n",
       "3               LDLR  FRAME_SHIFT  34.000   9.999894e-01  Pathogenic   \n",
       "4               MSH2  FRAME_SHIFT  35.000   9.999891e-01  Pathogenic   \n",
       "...              ...          ...     ...            ...         ...   \n",
       "334596      MIR548AZ   SYNONYMOUS  14.370   2.765540e-07     Neutral   \n",
       "334597         MYHAS   SYNONYMOUS   0.611   2.524258e-07     Neutral   \n",
       "334598         MYHAS   SYNONYMOUS   9.827   2.164549e-07     Neutral   \n",
       "334599  LOC100289580   SYNONYMOUS  14.100   1.984229e-07     Neutral   \n",
       "334600         MYHAS   SYNONYMOUS  10.040   1.227600e-07     Neutral   \n",
       "\n",
       "       combined_prediction  \n",
       "0               Pathogenic  \n",
       "1               Pathogenic  \n",
       "2               Pathogenic  \n",
       "3               Pathogenic  \n",
       "4               Pathogenic  \n",
       "...                    ...  \n",
       "334596             Neutral  \n",
       "334597             Neutral  \n",
       "334598             Neutral  \n",
       "334599             Neutral  \n",
       "334600             Neutral  \n",
       "\n",
       "[334601 rows x 10 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_original = pd.read_csv('/home/rjsietsma/Documents/School/Master_DSLS/Final_Thesis/Initial_Data_exploration/train_results.txt', sep='\\t', low_memory=False)\n",
    "train_original"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>probabilities</th>\n",
       "      <th>prediction</th>\n",
       "      <th>combined_prediction</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999982e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216252</td>\n",
       "      <td>GACAAA</td>\n",
       "      <td>TTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999978e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32936775</td>\n",
       "      <td>GAATTT</td>\n",
       "      <td>AG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999976e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216246</td>\n",
       "      <td>TGCAAGGACAAATCTGAC</td>\n",
       "      <td>CCGACTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999967e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216262</td>\n",
       "      <td>ACG</td>\n",
       "      <td>CA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>20.800</td>\n",
       "      <td>9.999956e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32906847</td>\n",
       "      <td>TACCCCTATTG</td>\n",
       "      <td>ACAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334596</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>10.130</td>\n",
       "      <td>7.769997e-08</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10429043</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334597</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>15.280</td>\n",
       "      <td>7.578360e-08</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10415744</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334598</th>\n",
       "      <td>MIR1273H</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>13.040</td>\n",
       "      <td>6.680243e-08</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>4</td>\n",
       "      <td>39448569</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334599</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>12.770</td>\n",
       "      <td>5.909591e-08</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10318882</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334600</th>\n",
       "      <td>MYH8</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>2.474</td>\n",
       "      <td>4.037909e-08</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10318897</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>334601 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        GeneName  Consequence   PHRED  probabilities  prediction  \\\n",
       "0           LDLR  FRAME_SHIFT  35.000   9.999982e-01  Pathogenic   \n",
       "1          BRCA2  FRAME_SHIFT  35.000   9.999978e-01  Pathogenic   \n",
       "2           LDLR  FRAME_SHIFT  35.000   9.999976e-01  Pathogenic   \n",
       "3           LDLR  FRAME_SHIFT  34.000   9.999967e-01  Pathogenic   \n",
       "4          BRCA2  FRAME_SHIFT  20.800   9.999956e-01  Pathogenic   \n",
       "...          ...          ...     ...            ...         ...   \n",
       "334596     MYHAS   SYNONYMOUS  10.130   7.769997e-08     Neutral   \n",
       "334597     MYHAS   SYNONYMOUS  15.280   7.578360e-08     Neutral   \n",
       "334598  MIR1273H   SYNONYMOUS  13.040   6.680243e-08     Neutral   \n",
       "334599     MYHAS   SYNONYMOUS  12.770   5.909591e-08     Neutral   \n",
       "334600      MYH8  SPLICE_SITE   2.474   4.037909e-08     Neutral   \n",
       "\n",
       "       combined_prediction chr       pos                 ref      alt  \n",
       "0               Pathogenic  19  11216252              GACAAA      TTT  \n",
       "1               Pathogenic  13  32936775              GAATTT       AG  \n",
       "2               Pathogenic  19  11216246  TGCAAGGACAAATCTGAC  CCGACTG  \n",
       "3               Pathogenic  19  11216262                 ACG       CA  \n",
       "4               Pathogenic  13  32906847         TACCCCTATTG     ACAT  \n",
       "...                    ...  ..       ...                 ...      ...  \n",
       "334596             Neutral  17  10429043                   G        A  \n",
       "334597             Neutral  17  10415744                   A        G  \n",
       "334598             Neutral   4  39448569                   G        A  \n",
       "334599             Neutral  17  10318882                   G        A  \n",
       "334600             Neutral  17  10318897                   G        A  \n",
       "\n",
       "[334601 rows x 10 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_new = pd.read_csv('./datafiles/train_results_v4.txt.gz', compression='gzip',\n",
    "                       sep='\\t', low_memory=False)\n",
    "important_info = train_new['chr_pos_ref_alt'].str.split(\"_\", expand=True)\n",
    "train_new['chr'] = important_info[0]\n",
    "train_new['pos'] = important_info[1].astype(np.int64)\n",
    "train_new['ref'] = important_info[2]\n",
    "train_new['alt'] = important_info[3]\n",
    "train_new.drop(columns=['chr_pos_ref_alt'], inplace=True)\n",
    "train_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There is a 4.3523480204781215% mismatch.\n"
     ]
    }
   ],
   "source": [
    "merge = train_original[['chr', 'pos','ref','alt','prediction']].merge(train_new[['chr', 'pos','ref','alt','prediction']],\n",
    "                                                                     on=['chr', 'pos','ref','alt'])\n",
    "merge[merge['prediction_x'] != merge['prediction_y']]\n",
    "print(f\"There is a \"\n",
    "      f\"{merge[merge['prediction_x'] != merge['prediction_y']].shape[0] / train_original.shape[0] * 100}% mismatch.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>max_AF</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>label</th>\n",
       "      <th>revel</th>\n",
       "      <th>clinpred</th>\n",
       "      <th>sift</th>\n",
       "      <th>provean</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>fathmm_score</th>\n",
       "      <th>capice</th>\n",
       "      <th>ponp2</th>\n",
       "      <th>prediction</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>21</td>\n",
       "      <td>33974174</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000058</td>\n",
       "      <td>STOP_LOST</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.050</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003275</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>X</td>\n",
       "      <td>99661625</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.806</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000075</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17</td>\n",
       "      <td>29509638</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.647</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>10.760</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001511</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>21</td>\n",
       "      <td>35742999</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000133</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>18.640</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000621</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>2160973</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>16.340</td>\n",
       "      <td>0.008252</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10837</th>\n",
       "      <td>17</td>\n",
       "      <td>29556342</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>11.450</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000139</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10838</th>\n",
       "      <td>11</td>\n",
       "      <td>5248177</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000344</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>17.970</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.069934</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10839</th>\n",
       "      <td>15</td>\n",
       "      <td>48787324</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.664</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>12.170</td>\n",
       "      <td>0.019530</td>\n",
       "      <td>0.909190</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10840</th>\n",
       "      <td>19</td>\n",
       "      <td>17947957</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>10.990</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001238</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10841</th>\n",
       "      <td>16</td>\n",
       "      <td>223597</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001252</td>\n",
       "      <td>STOP_LOST</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18.020</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.971616</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10842 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr       pos ref alt    max_AF Consequence label  revel  clinpred  \\\n",
       "0      21  33974174   C   G  0.000058   STOP_LOST  LB/B    NaN       NaN   \n",
       "1       X  99661625   G   C  0.000037  SYNONYMOUS  LB/B    NaN       NaN   \n",
       "2      17  29509638   C   T  0.000000  SYNONYMOUS  LB/B    NaN       NaN   \n",
       "3      21  35742999   C   T  0.000133  SYNONYMOUS  LB/B    NaN       NaN   \n",
       "4       1   2160973   G   A  0.000000  SYNONYMOUS  LB/B    NaN       NaN   \n",
       "...    ..       ...  ..  ..       ...         ...   ...    ...       ...   \n",
       "10837  17  29556342   G   A  0.000000  SYNONYMOUS  LP/P    NaN       NaN   \n",
       "10838  11   5248177   A   T  0.000344  SYNONYMOUS  LP/P    NaN       NaN   \n",
       "10839  15  48787324   T   C  0.000000  SYNONYMOUS  LP/P    NaN       NaN   \n",
       "10840  19  17947957   G   A  0.000015  SYNONYMOUS  LP/P    NaN       NaN   \n",
       "10841  16    223597   T   C  0.001252   STOP_LOST  LP/P    NaN       NaN   \n",
       "\n",
       "        sift  provean   PHRED  fathmm_score    capice  ponp2  prediction  \n",
       "0        NaN      NaN  15.050           NaN  0.003275    NaN     Neutral  \n",
       "1      0.000     -0.0   0.806           NaN  0.000075    NaN     Neutral  \n",
       "2      0.647     -0.0  10.760           NaN  0.001511    NaN     Neutral  \n",
       "3      0.000     -0.0  18.640           NaN  0.000621    NaN     Neutral  \n",
       "4      0.000     -0.0  16.340      0.008252  0.000012    NaN     Neutral  \n",
       "...      ...      ...     ...           ...       ...    ...         ...  \n",
       "10837  0.000     -0.0  11.450           NaN  0.000139    NaN     Neutral  \n",
       "10838  0.000     -0.0  17.970           NaN  0.069934    NaN  Pathogenic  \n",
       "10839  0.664     -0.0  12.170      0.019530  0.909190    NaN  Pathogenic  \n",
       "10840  0.000     -0.0  10.990           NaN  0.001238    NaN     Neutral  \n",
       "10841    NaN      NaN  18.020           NaN  0.971616    NaN  Pathogenic  \n",
       "\n",
       "[10842 rows x 16 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_original = pd.read_csv('/home/rjsietsma/Documents/School/Master_DSLS/Final_Thesis/Initial_Data_exploration/test_results.txt', sep='\\t', low_memory=False)\n",
    "tellPathogenic_pred = lambda x: \"Pathogenic\" if x > 0.02 else 'Neutral'\n",
    "test_original['prediction'] = [tellPathogenic_pred(probability) for probability in test_original['capice']]\n",
    "test_original.rename(columns={'#Chrom': 'chr', 'Pos':'pos', 'Ref': 'ref', 'Alt':'alt'}, inplace=True)\n",
    "test_original"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>probabilities</th>\n",
       "      <th>prediction</th>\n",
       "      <th>combined_prediction</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999962e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>17</td>\n",
       "      <td>41243705</td>\n",
       "      <td>CTGAG</td>\n",
       "      <td>GCCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>26.900</td>\n",
       "      <td>9.999943e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32912883</td>\n",
       "      <td>CTG</td>\n",
       "      <td>TT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MSH2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999934e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>2</td>\n",
       "      <td>47639582</td>\n",
       "      <td>AGAAA</td>\n",
       "      <td>TAAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>24.100</td>\n",
       "      <td>9.999933e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32912590</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>25.600</td>\n",
       "      <td>9.999924e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32913442</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10837</th>\n",
       "      <td>MIR548AZ</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>5.553</td>\n",
       "      <td>1.214339e-06</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>14</td>\n",
       "      <td>64537567</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10838</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.068</td>\n",
       "      <td>8.699849e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10432854</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10839</th>\n",
       "      <td>NBPF20</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>3.320</td>\n",
       "      <td>6.732982e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>1</td>\n",
       "      <td>145439717</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10840</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>0.035</td>\n",
       "      <td>6.130708e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10442488</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10841</th>\n",
       "      <td>MIR548AZ</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>1.347</td>\n",
       "      <td>4.053531e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>14</td>\n",
       "      <td>64519932</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10842 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       GeneName     Consequence   PHRED  probabilities  prediction  \\\n",
       "0         BRCA1     STOP_GAINED  34.000   9.999962e-01  Pathogenic   \n",
       "1         BRCA2     FRAME_SHIFT  26.900   9.999943e-01  Pathogenic   \n",
       "2          MSH2     FRAME_SHIFT  34.000   9.999934e-01  Pathogenic   \n",
       "3         BRCA2     FRAME_SHIFT  24.100   9.999933e-01  Pathogenic   \n",
       "4         BRCA2     FRAME_SHIFT  25.600   9.999924e-01  Pathogenic   \n",
       "...         ...             ...     ...            ...         ...   \n",
       "10837  MIR548AZ  NON_SYNONYMOUS   5.553   1.214339e-06     Neutral   \n",
       "10838     MYHAS        INTRONIC   0.068   8.699849e-07     Neutral   \n",
       "10839    NBPF20      REGULATORY   3.320   6.732982e-07     Neutral   \n",
       "10840     MYHAS      DOWNSTREAM   0.035   6.130708e-07     Neutral   \n",
       "10841  MIR548AZ  NON_SYNONYMOUS   1.347   4.053531e-07     Neutral   \n",
       "\n",
       "      combined_prediction chr        pos    ref   alt  \n",
       "0              Pathogenic  17   41243705  CTGAG  GCCT  \n",
       "1              Pathogenic  13   32912883    CTG    TT  \n",
       "2              Pathogenic   2   47639582  AGAAA  TAAT  \n",
       "3              Pathogenic  13   32912590     TA     T  \n",
       "4              Pathogenic  13   32913442     TC     T  \n",
       "...                   ...  ..        ...    ...   ...  \n",
       "10837             Neutral  14   64537567      G     C  \n",
       "10838             Neutral  17   10432854      G     A  \n",
       "10839             Neutral   1  145439717      C     G  \n",
       "10840             Neutral  17   10442488      G     A  \n",
       "10841             Neutral  14   64519932      A     G  \n",
       "\n",
       "[10842 rows x 10 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_new = pd.read_csv('./datafiles/test_results_v4.txt.gz', compression='gzip',\n",
    "                       sep='\\t', low_memory=False)\n",
    "important_info = test_new['chr_pos_ref_alt'].str.split(\"_\", expand=True)\n",
    "test_new['chr'] = important_info[0]\n",
    "test_new['pos'] = important_info[1].astype(np.int64)\n",
    "test_new['ref'] = important_info[2]\n",
    "test_new['alt'] = important_info[3]\n",
    "test_new.drop(columns=['chr_pos_ref_alt'], inplace=True)\n",
    "test_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There is a 14.25936174137613% mismatch.\n"
     ]
    }
   ],
   "source": [
    "merge = test_original[['chr', 'pos','ref','alt','prediction']].merge(test_new[['chr', 'pos','ref','alt','prediction']],\n",
    "                                                                     on=['chr', 'pos','ref','alt'])\n",
    "merge[merge['prediction_x'] != merge['prediction_y']]\n",
    "print(f\"There is a \"\n",
    "      f\"{merge[merge['prediction_x'] != merge['prediction_y']].shape[0] / test_original.shape[0] * 100}% mismatch.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Correct probability prediction"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Currently marked as markdown to prevent executing again when re-launching the notebook"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "thresholds = np.arange(0,1,0.001)\n",
    "data = pd.read_csv('/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/test_output/train_results_v3.txt',\n",
    "                  sep='\\t', low_memory=False)\n",
    "important_info = data['chr_pos_ref_alt'].str.split(\"_\", expand=True)\n",
    "data['chr'] = important_info[0]\n",
    "data['pos'] = important_info[1].astype(np.int64)\n",
    "data['ref'] = important_info[2]\n",
    "data['alt'] = important_info[3]\n",
    "data.drop(columns=['chr_pos_ref_alt'], inplace=True)\n",
    "data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "train_in = pd.read_csv('/run/media/rjsietsma/evo2tb/linux/Datafiles/train.txt.gz',\n",
    "                      compression='gzip', sep='\\t', low_memory=False)\n",
    "train_in"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "data = data.merge(train_in[['#Chrom', 'Pos', 'Ref', 'Alt', 'label']],\n",
    "                           left_on=['chr','pos','ref','alt'],\n",
    "                           right_on=['#Chrom','Pos','Ref','Alt'])\n",
    "data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "drop_labels = ['#Chrom', 'Pos', 'Ref', 'Alt']\n",
    "for x in data.columns:\n",
    "    if x.endswith('_x') or x.endswith('_y'):\n",
    "        drop_labels.append(x)\n",
    "data.drop(columns=drop_labels, inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "def apply_func_thresholding(probability, threshold):\n",
    "    return_value = 0\n",
    "    if probability > threshold:\n",
    "        return_value = 1\n",
    "    return return_value"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "data['label'].replace({'Pathogenic': 1, 'Benign': 0}, inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "true_recall = 0\n",
    "true_threshold = 0\n",
    "for threshold in thresholds:\n",
    "    data['pred_label'] = data['probabilities'].apply(lambda x: apply_func_thresholding(x, threshold))\n",
    "    y_pred = np.array(data['pred_label'])\n",
    "    y_true = np.array(data['label'])\n",
    "    recall = recall_score(y_pred, y_true, zero_division=0)\n",
    "    if 0.94 <= recall <= 0.96:\n",
    "        true_recall = recall\n",
    "        true_threshold = threshold\n",
    "        break\n",
    "print(f'Recall {true_recall} is found at threshold {true_threshold}.')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Preparing for AUC analysis\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>probabilities</th>\n",
       "      <th>prediction</th>\n",
       "      <th>combined_prediction</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999982e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216252</td>\n",
       "      <td>GACAAA</td>\n",
       "      <td>TTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999978e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32936775</td>\n",
       "      <td>GAATTT</td>\n",
       "      <td>AG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999976e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216246</td>\n",
       "      <td>TGCAAGGACAAATCTGAC</td>\n",
       "      <td>CCGACTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999967e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216262</td>\n",
       "      <td>ACG</td>\n",
       "      <td>CA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>20.800</td>\n",
       "      <td>9.999956e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32906847</td>\n",
       "      <td>TACCCCTATTG</td>\n",
       "      <td>ACAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334596</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>10.130</td>\n",
       "      <td>7.769997e-08</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10429043</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334597</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>15.280</td>\n",
       "      <td>7.578360e-08</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10415744</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334598</th>\n",
       "      <td>MIR1273H</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>13.040</td>\n",
       "      <td>6.680243e-08</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>4</td>\n",
       "      <td>39448569</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334599</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>12.770</td>\n",
       "      <td>5.909591e-08</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10318882</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334600</th>\n",
       "      <td>MYH8</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>2.474</td>\n",
       "      <td>4.037909e-08</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10318897</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>334601 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        GeneName  Consequence   PHRED  probabilities  prediction  \\\n",
       "0           LDLR  FRAME_SHIFT  35.000   9.999982e-01  Pathogenic   \n",
       "1          BRCA2  FRAME_SHIFT  35.000   9.999978e-01  Pathogenic   \n",
       "2           LDLR  FRAME_SHIFT  35.000   9.999976e-01  Pathogenic   \n",
       "3           LDLR  FRAME_SHIFT  34.000   9.999967e-01  Pathogenic   \n",
       "4          BRCA2  FRAME_SHIFT  20.800   9.999956e-01  Pathogenic   \n",
       "...          ...          ...     ...            ...         ...   \n",
       "334596     MYHAS   SYNONYMOUS  10.130   7.769997e-08     Neutral   \n",
       "334597     MYHAS   SYNONYMOUS  15.280   7.578360e-08     Neutral   \n",
       "334598  MIR1273H   SYNONYMOUS  13.040   6.680243e-08     Neutral   \n",
       "334599     MYHAS   SYNONYMOUS  12.770   5.909591e-08     Neutral   \n",
       "334600      MYH8  SPLICE_SITE   2.474   4.037909e-08     Neutral   \n",
       "\n",
       "       combined_prediction chr       pos                 ref      alt  \n",
       "0               Pathogenic  19  11216252              GACAAA      TTT  \n",
       "1               Pathogenic  13  32936775              GAATTT       AG  \n",
       "2               Pathogenic  19  11216246  TGCAAGGACAAATCTGAC  CCGACTG  \n",
       "3               Pathogenic  19  11216262                 ACG       CA  \n",
       "4               Pathogenic  13  32906847         TACCCCTATTG     ACAT  \n",
       "...                    ...  ..       ...                 ...      ...  \n",
       "334596             Neutral  17  10429043                   G        A  \n",
       "334597             Neutral  17  10415744                   A        G  \n",
       "334598             Neutral   4  39448569                   G        A  \n",
       "334599             Neutral  17  10318882                   G        A  \n",
       "334600             Neutral  17  10318897                   G        A  \n",
       "\n",
       "[334601 rows x 10 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_out = train_new.copy()\n",
    "train_out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Allergy/Immunology/Infectious</th>\n",
       "      <th>Alt</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Audiologic/Otolaryngologic</th>\n",
       "      <th>Biochemical</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>CDSpos</th>\n",
       "      <th>Cardiovascular</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>revel</th>\n",
       "      <th>sift</th>\n",
       "      <th>source</th>\n",
       "      <th>tOverlapMotifs</th>\n",
       "      <th>targetScan</th>\n",
       "      <th>to_be_deleted</th>\n",
       "      <th>verPhCons</th>\n",
       "      <th>verPhyloP</th>\n",
       "      <th>inTest</th>\n",
       "      <th>sample_weight</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14</td>\n",
       "      <td>False</td>\n",
       "      <td>G</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS9787.1</td>\n",
       "      <td>806.0</td>\n",
       "      <td>False</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.843</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS13112.1</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift,stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>4.670</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>C</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS13112.1</td>\n",
       "      <td>2118.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.043</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS13112.1</td>\n",
       "      <td>1586.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>6.221</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>6.368</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334596</th>\n",
       "      <td>17</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS32642.1</td>\n",
       "      <td>1563.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>6.031</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334597</th>\n",
       "      <td>17</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS32642.1</td>\n",
       "      <td>2029.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>4.100</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334598</th>\n",
       "      <td>10</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS7431.1</td>\n",
       "      <td>1216.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.852</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334599</th>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS2382.1</td>\n",
       "      <td>2998.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.031</td>\n",
       "      <td>2.213</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334600</th>\n",
       "      <td>5</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS3952.1</td>\n",
       "      <td>1221.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.528</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>334601 rows × 152 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Chrom  Allergy/Immunology/Infectious Alt          AnnoType  \\\n",
       "0          14                          False   G  CodingTranscript   \n",
       "1          20                          False   T  CodingTranscript   \n",
       "2          20                          False   C  CodingTranscript   \n",
       "3          20                          False   A  CodingTranscript   \n",
       "4          20                          False   A        Intergenic   \n",
       "...       ...                            ...  ..               ...   \n",
       "334596     17                          False   A  CodingTranscript   \n",
       "334597     17                          False   T  CodingTranscript   \n",
       "334598     10                          False   T  CodingTranscript   \n",
       "334599      2                          False   T  CodingTranscript   \n",
       "334600      5                          False   T  CodingTranscript   \n",
       "\n",
       "        Audiologic/Otolaryngologic  Biochemical         CCDS  CDSpos  \\\n",
       "0                            False        False   CCDS9787.1   806.0   \n",
       "1                             True        False  CCDS13112.1  1899.0   \n",
       "2                             True        False  CCDS13112.1  2118.0   \n",
       "3                             True        False  CCDS13112.1  1586.0   \n",
       "4                             True        False          NaN     NaN   \n",
       "...                            ...          ...          ...     ...   \n",
       "334596                       False        False  CCDS32642.1  1563.0   \n",
       "334597                       False        False  CCDS32642.1  2029.0   \n",
       "334598                       False        False   CCDS7431.1  1216.0   \n",
       "334599                       False        False   CCDS2382.1  2998.0   \n",
       "334600                       False        False   CCDS3952.1  1221.0   \n",
       "\n",
       "        Cardiovascular              ConsDetail  ...  revel sift   source  \\\n",
       "0                False              frameshift  ...    NaN  NaN     vkgl   \n",
       "1                 True  frameshift,stop_gained  ...    NaN  NaN     vkgl   \n",
       "2                 True              frameshift  ...    NaN  NaN     vkgl   \n",
       "3                 True              frameshift  ...    NaN  NaN     vkgl   \n",
       "4                 True              downstream  ...    NaN  NaN     vkgl   \n",
       "...                ...                     ...  ...    ...  ...      ...   \n",
       "334596           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334597           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334598           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334599           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334600           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "\n",
       "        tOverlapMotifs  targetScan  to_be_deleted  verPhCons verPhyloP inTest  \\\n",
       "0                  NaN         NaN          False      1.000     5.843  False   \n",
       "1                  NaN         NaN          False      1.000     4.670  False   \n",
       "2                  NaN         NaN          False      1.000     5.043  False   \n",
       "3                  NaN         NaN          False      1.000     6.221  False   \n",
       "4                  NaN         NaN          False      1.000     6.368  False   \n",
       "...                ...         ...            ...        ...       ...    ...   \n",
       "334596             NaN         NaN          False      1.000     6.031  False   \n",
       "334597             NaN         NaN          False      1.000     4.100  False   \n",
       "334598             NaN         NaN          False      1.000     5.852  False   \n",
       "334599             NaN         NaN          False      0.031     2.213  False   \n",
       "334600             NaN         NaN          False      1.000     0.528  False   \n",
       "\n",
       "        sample_weight  \n",
       "0                 1.0  \n",
       "1                 1.0  \n",
       "2                 1.0  \n",
       "3                 1.0  \n",
       "4                 1.0  \n",
       "...               ...  \n",
       "334596            0.8  \n",
       "334597            0.8  \n",
       "334598            0.8  \n",
       "334599            0.8  \n",
       "334600            0.8  \n",
       "\n",
       "[334601 rows x 152 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_in = pd.read_csv('./datafiles/train.txt.gz',\n",
    "                      compression='gzip', sep='\\t', low_memory=False)\n",
    "train_in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>probabilities</th>\n",
       "      <th>prediction</th>\n",
       "      <th>combined_prediction</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999982e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216252</td>\n",
       "      <td>GACAAA</td>\n",
       "      <td>TTT</td>\n",
       "      <td>19</td>\n",
       "      <td>11216252</td>\n",
       "      <td>GACAAA</td>\n",
       "      <td>TTT</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999978e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32936775</td>\n",
       "      <td>GAATTT</td>\n",
       "      <td>AG</td>\n",
       "      <td>13</td>\n",
       "      <td>32936775</td>\n",
       "      <td>GAATTT</td>\n",
       "      <td>AG</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999976e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216246</td>\n",
       "      <td>TGCAAGGACAAATCTGAC</td>\n",
       "      <td>CCGACTG</td>\n",
       "      <td>19</td>\n",
       "      <td>11216246</td>\n",
       "      <td>TGCAAGGACAAATCTGAC</td>\n",
       "      <td>CCGACTG</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999967e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216262</td>\n",
       "      <td>ACG</td>\n",
       "      <td>CA</td>\n",
       "      <td>19</td>\n",
       "      <td>11216262</td>\n",
       "      <td>ACG</td>\n",
       "      <td>CA</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>20.800</td>\n",
       "      <td>9.999956e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32906847</td>\n",
       "      <td>TACCCCTATTG</td>\n",
       "      <td>ACAT</td>\n",
       "      <td>13</td>\n",
       "      <td>32906847</td>\n",
       "      <td>TACCCCTATTG</td>\n",
       "      <td>ACAT</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334596</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>10.130</td>\n",
       "      <td>7.769997e-08</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10429043</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>17</td>\n",
       "      <td>10429043</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334597</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>15.280</td>\n",
       "      <td>7.578360e-08</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10415744</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>17</td>\n",
       "      <td>10415744</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334598</th>\n",
       "      <td>MIR1273H</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>13.040</td>\n",
       "      <td>6.680243e-08</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>4</td>\n",
       "      <td>39448569</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>4</td>\n",
       "      <td>39448569</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334599</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>12.770</td>\n",
       "      <td>5.909591e-08</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10318882</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>17</td>\n",
       "      <td>10318882</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334600</th>\n",
       "      <td>MYH8</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>2.474</td>\n",
       "      <td>4.037909e-08</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10318897</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>17</td>\n",
       "      <td>10318897</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>334601 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        GeneName  Consequence   PHRED  probabilities  prediction  \\\n",
       "0           LDLR  FRAME_SHIFT  35.000   9.999982e-01  Pathogenic   \n",
       "1          BRCA2  FRAME_SHIFT  35.000   9.999978e-01  Pathogenic   \n",
       "2           LDLR  FRAME_SHIFT  35.000   9.999976e-01  Pathogenic   \n",
       "3           LDLR  FRAME_SHIFT  34.000   9.999967e-01  Pathogenic   \n",
       "4          BRCA2  FRAME_SHIFT  20.800   9.999956e-01  Pathogenic   \n",
       "...          ...          ...     ...            ...         ...   \n",
       "334596     MYHAS   SYNONYMOUS  10.130   7.769997e-08     Neutral   \n",
       "334597     MYHAS   SYNONYMOUS  15.280   7.578360e-08     Neutral   \n",
       "334598  MIR1273H   SYNONYMOUS  13.040   6.680243e-08     Neutral   \n",
       "334599     MYHAS   SYNONYMOUS  12.770   5.909591e-08     Neutral   \n",
       "334600      MYH8  SPLICE_SITE   2.474   4.037909e-08     Neutral   \n",
       "\n",
       "       combined_prediction chr       pos                 ref      alt #Chrom  \\\n",
       "0               Pathogenic  19  11216252              GACAAA      TTT     19   \n",
       "1               Pathogenic  13  32936775              GAATTT       AG     13   \n",
       "2               Pathogenic  19  11216246  TGCAAGGACAAATCTGAC  CCGACTG     19   \n",
       "3               Pathogenic  19  11216262                 ACG       CA     19   \n",
       "4               Pathogenic  13  32906847         TACCCCTATTG     ACAT     13   \n",
       "...                    ...  ..       ...                 ...      ...    ...   \n",
       "334596             Neutral  17  10429043                   G        A     17   \n",
       "334597             Neutral  17  10415744                   A        G     17   \n",
       "334598             Neutral   4  39448569                   G        A      4   \n",
       "334599             Neutral  17  10318882                   G        A     17   \n",
       "334600             Neutral  17  10318897                   G        A     17   \n",
       "\n",
       "             Pos                 Ref      Alt       label  \n",
       "0       11216252              GACAAA      TTT  Pathogenic  \n",
       "1       32936775              GAATTT       AG  Pathogenic  \n",
       "2       11216246  TGCAAGGACAAATCTGAC  CCGACTG  Pathogenic  \n",
       "3       11216262                 ACG       CA  Pathogenic  \n",
       "4       32906847         TACCCCTATTG     ACAT  Pathogenic  \n",
       "...          ...                 ...      ...         ...  \n",
       "334596  10429043                   G        A      Benign  \n",
       "334597  10415744                   A        G      Benign  \n",
       "334598  39448569                   G        A      Benign  \n",
       "334599  10318882                   G        A      Benign  \n",
       "334600  10318897                   G        A      Benign  \n",
       "\n",
       "[334601 rows x 15 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_out = train_out.merge(train_in[['#Chrom', 'Pos', 'Ref', 'Alt', 'label']],\n",
    "                           left_on=['chr','pos','ref','alt'],\n",
    "                           right_on=['#Chrom','Pos','Ref','Alt'])\n",
    "train_out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['#Chrom', 'Pos', 'Ref', 'Alt']"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drop_labels = ['#Chrom', 'Pos', 'Ref', 'Alt']\n",
    "for x in train_out.columns:\n",
    "    if x.endswith('_x') or x.endswith('_y'):\n",
    "        drop_labels.append(x)\n",
    "drop_labels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "train_out.drop(columns=drop_labels, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>probabilities</th>\n",
       "      <th>prediction</th>\n",
       "      <th>combined_prediction</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999982e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216252</td>\n",
       "      <td>GACAAA</td>\n",
       "      <td>TTT</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999978e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32936775</td>\n",
       "      <td>GAATTT</td>\n",
       "      <td>AG</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999976e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216246</td>\n",
       "      <td>TGCAAGGACAAATCTGAC</td>\n",
       "      <td>CCGACTG</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999967e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216262</td>\n",
       "      <td>ACG</td>\n",
       "      <td>CA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>20.800</td>\n",
       "      <td>9.999956e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32906847</td>\n",
       "      <td>TACCCCTATTG</td>\n",
       "      <td>ACAT</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334596</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>10.130</td>\n",
       "      <td>7.769997e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10429043</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334597</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>15.280</td>\n",
       "      <td>7.578360e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10415744</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334598</th>\n",
       "      <td>MIR1273H</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>13.040</td>\n",
       "      <td>6.680243e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>4</td>\n",
       "      <td>39448569</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334599</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>12.770</td>\n",
       "      <td>5.909591e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10318882</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334600</th>\n",
       "      <td>MYH8</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>2.474</td>\n",
       "      <td>4.037909e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10318897</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>334601 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        GeneName  Consequence   PHRED  probabilities  prediction  \\\n",
       "0           LDLR  FRAME_SHIFT  35.000   9.999982e-01           1   \n",
       "1          BRCA2  FRAME_SHIFT  35.000   9.999978e-01           1   \n",
       "2           LDLR  FRAME_SHIFT  35.000   9.999976e-01           1   \n",
       "3           LDLR  FRAME_SHIFT  34.000   9.999967e-01           1   \n",
       "4          BRCA2  FRAME_SHIFT  20.800   9.999956e-01           1   \n",
       "...          ...          ...     ...            ...         ...   \n",
       "334596     MYHAS   SYNONYMOUS  10.130   7.769997e-08           0   \n",
       "334597     MYHAS   SYNONYMOUS  15.280   7.578360e-08           0   \n",
       "334598  MIR1273H   SYNONYMOUS  13.040   6.680243e-08           0   \n",
       "334599     MYHAS   SYNONYMOUS  12.770   5.909591e-08           0   \n",
       "334600      MYH8  SPLICE_SITE   2.474   4.037909e-08           0   \n",
       "\n",
       "       combined_prediction chr       pos                 ref      alt  label  \n",
       "0               Pathogenic  19  11216252              GACAAA      TTT      1  \n",
       "1               Pathogenic  13  32936775              GAATTT       AG      1  \n",
       "2               Pathogenic  19  11216246  TGCAAGGACAAATCTGAC  CCGACTG      1  \n",
       "3               Pathogenic  19  11216262                 ACG       CA      1  \n",
       "4               Pathogenic  13  32906847         TACCCCTATTG     ACAT      1  \n",
       "...                    ...  ..       ...                 ...      ...    ...  \n",
       "334596             Neutral  17  10429043                   G        A      0  \n",
       "334597             Neutral  17  10415744                   A        G      0  \n",
       "334598             Neutral   4  39448569                   G        A      0  \n",
       "334599             Neutral  17  10318882                   G        A      0  \n",
       "334600             Neutral  17  10318897                   G        A      0  \n",
       "\n",
       "[334601 rows x 11 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_out['prediction'].replace({'Pathogenic': 1, 'Neutral': 0}, inplace=True)\n",
    "train_out['label'].replace({'Pathogenic': 1, 'Benign': 0}, inplace=True)\n",
    "train_out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.9822742260567594"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "y_true = np.array(train_out['label'])\n",
    "y_pred = np.array(train_out['prediction'])\n",
    "train_auc = roc_auc_score(y_true = y_true, y_score=y_pred)\n",
    "train_auc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "test = test_new.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>probabilities</th>\n",
       "      <th>prediction</th>\n",
       "      <th>combined_prediction</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999962e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>17</td>\n",
       "      <td>41243705</td>\n",
       "      <td>CTGAG</td>\n",
       "      <td>GCCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>26.900</td>\n",
       "      <td>9.999943e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32912883</td>\n",
       "      <td>CTG</td>\n",
       "      <td>TT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MSH2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999934e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>2</td>\n",
       "      <td>47639582</td>\n",
       "      <td>AGAAA</td>\n",
       "      <td>TAAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>24.100</td>\n",
       "      <td>9.999933e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32912590</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>25.600</td>\n",
       "      <td>9.999924e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32913442</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10837</th>\n",
       "      <td>MIR548AZ</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>5.553</td>\n",
       "      <td>1.214339e-06</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>14</td>\n",
       "      <td>64537567</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10838</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.068</td>\n",
       "      <td>8.699849e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10432854</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10839</th>\n",
       "      <td>NBPF20</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>3.320</td>\n",
       "      <td>6.732982e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>1</td>\n",
       "      <td>145439717</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10840</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>0.035</td>\n",
       "      <td>6.130708e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10442488</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10841</th>\n",
       "      <td>MIR548AZ</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>1.347</td>\n",
       "      <td>4.053531e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>14</td>\n",
       "      <td>64519932</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10842 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       GeneName     Consequence   PHRED  probabilities  prediction  \\\n",
       "0         BRCA1     STOP_GAINED  34.000   9.999962e-01  Pathogenic   \n",
       "1         BRCA2     FRAME_SHIFT  26.900   9.999943e-01  Pathogenic   \n",
       "2          MSH2     FRAME_SHIFT  34.000   9.999934e-01  Pathogenic   \n",
       "3         BRCA2     FRAME_SHIFT  24.100   9.999933e-01  Pathogenic   \n",
       "4         BRCA2     FRAME_SHIFT  25.600   9.999924e-01  Pathogenic   \n",
       "...         ...             ...     ...            ...         ...   \n",
       "10837  MIR548AZ  NON_SYNONYMOUS   5.553   1.214339e-06     Neutral   \n",
       "10838     MYHAS        INTRONIC   0.068   8.699849e-07     Neutral   \n",
       "10839    NBPF20      REGULATORY   3.320   6.732982e-07     Neutral   \n",
       "10840     MYHAS      DOWNSTREAM   0.035   6.130708e-07     Neutral   \n",
       "10841  MIR548AZ  NON_SYNONYMOUS   1.347   4.053531e-07     Neutral   \n",
       "\n",
       "      combined_prediction chr        pos    ref   alt  \n",
       "0              Pathogenic  17   41243705  CTGAG  GCCT  \n",
       "1              Pathogenic  13   32912883    CTG    TT  \n",
       "2              Pathogenic   2   47639582  AGAAA  TAAT  \n",
       "3              Pathogenic  13   32912590     TA     T  \n",
       "4              Pathogenic  13   32913442     TC     T  \n",
       "...                   ...  ..        ...    ...   ...  \n",
       "10837             Neutral  14   64537567      G     C  \n",
       "10838             Neutral  17   10432854      G     A  \n",
       "10839             Neutral   1  145439717      C     G  \n",
       "10840             Neutral  17   10442488      G     A  \n",
       "10841             Neutral  14   64519932      A     G  \n",
       "\n",
       "[10842 rows x 10 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>21</td>\n",
       "      <td>33974174</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>X</td>\n",
       "      <td>99661625</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17</td>\n",
       "      <td>29509638</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>21</td>\n",
       "      <td>35742999</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>2160973</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10837</th>\n",
       "      <td>17</td>\n",
       "      <td>29556342</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10838</th>\n",
       "      <td>11</td>\n",
       "      <td>5248177</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10839</th>\n",
       "      <td>15</td>\n",
       "      <td>48787324</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10840</th>\n",
       "      <td>19</td>\n",
       "      <td>17947957</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10841</th>\n",
       "      <td>16</td>\n",
       "      <td>223597</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10842 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr       pos ref alt       label\n",
       "0      21  33974174   C   G      Benign\n",
       "1       X  99661625   G   C      Benign\n",
       "2      17  29509638   C   T      Benign\n",
       "3      21  35742999   C   T      Benign\n",
       "4       1   2160973   G   A      Benign\n",
       "...    ..       ...  ..  ..         ...\n",
       "10837  17  29556342   G   A  Pathogenic\n",
       "10838  11   5248177   A   T  Pathogenic\n",
       "10839  15  48787324   T   C  Pathogenic\n",
       "10840  19  17947957   G   A  Pathogenic\n",
       "10841  16    223597   T   C  Pathogenic\n",
       "\n",
       "[10842 rows x 5 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_in = pd.read_csv('./datafiles/test.txt.gz',\n",
    "                            compression='gzip',\n",
    "                            low_memory=False,\n",
    "                           sep='\\t')\n",
    "test_in = test_in[['#Chrom', 'Pos', 'Ref', 'Alt', 'label']]\n",
    "test_in.columns = ['chr', 'pos', 'ref', 'alt', 'label']\n",
    "test_in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>probabilities</th>\n",
       "      <th>prediction</th>\n",
       "      <th>combined_prediction</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999962e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>17</td>\n",
       "      <td>41243705</td>\n",
       "      <td>CTGAG</td>\n",
       "      <td>GCCT</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>26.900</td>\n",
       "      <td>9.999943e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32912883</td>\n",
       "      <td>CTG</td>\n",
       "      <td>TT</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MSH2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999934e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>2</td>\n",
       "      <td>47639582</td>\n",
       "      <td>AGAAA</td>\n",
       "      <td>TAAT</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>24.100</td>\n",
       "      <td>9.999933e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32912590</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>25.600</td>\n",
       "      <td>9.999924e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32913442</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10837</th>\n",
       "      <td>MIR548AZ</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>5.553</td>\n",
       "      <td>1.214339e-06</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>14</td>\n",
       "      <td>64537567</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10838</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.068</td>\n",
       "      <td>8.699849e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10432854</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10839</th>\n",
       "      <td>NBPF20</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>3.320</td>\n",
       "      <td>6.732982e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>1</td>\n",
       "      <td>145439717</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10840</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>0.035</td>\n",
       "      <td>6.130708e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10442488</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10841</th>\n",
       "      <td>MIR548AZ</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>1.347</td>\n",
       "      <td>4.053531e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>14</td>\n",
       "      <td>64519932</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10842 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       GeneName     Consequence   PHRED  probabilities  prediction  \\\n",
       "0         BRCA1     STOP_GAINED  34.000   9.999962e-01  Pathogenic   \n",
       "1         BRCA2     FRAME_SHIFT  26.900   9.999943e-01  Pathogenic   \n",
       "2          MSH2     FRAME_SHIFT  34.000   9.999934e-01  Pathogenic   \n",
       "3         BRCA2     FRAME_SHIFT  24.100   9.999933e-01  Pathogenic   \n",
       "4         BRCA2     FRAME_SHIFT  25.600   9.999924e-01  Pathogenic   \n",
       "...         ...             ...     ...            ...         ...   \n",
       "10837  MIR548AZ  NON_SYNONYMOUS   5.553   1.214339e-06     Neutral   \n",
       "10838     MYHAS        INTRONIC   0.068   8.699849e-07     Neutral   \n",
       "10839    NBPF20      REGULATORY   3.320   6.732982e-07     Neutral   \n",
       "10840     MYHAS      DOWNSTREAM   0.035   6.130708e-07     Neutral   \n",
       "10841  MIR548AZ  NON_SYNONYMOUS   1.347   4.053531e-07     Neutral   \n",
       "\n",
       "      combined_prediction chr        pos    ref   alt       label  \n",
       "0              Pathogenic  17   41243705  CTGAG  GCCT  Pathogenic  \n",
       "1              Pathogenic  13   32912883    CTG    TT  Pathogenic  \n",
       "2              Pathogenic   2   47639582  AGAAA  TAAT  Pathogenic  \n",
       "3              Pathogenic  13   32912590     TA     T  Pathogenic  \n",
       "4              Pathogenic  13   32913442     TC     T  Pathogenic  \n",
       "...                   ...  ..        ...    ...   ...         ...  \n",
       "10837             Neutral  14   64537567      G     C      Benign  \n",
       "10838             Neutral  17   10432854      G     A      Benign  \n",
       "10839             Neutral   1  145439717      C     G      Benign  \n",
       "10840             Neutral  17   10442488      G     A      Benign  \n",
       "10841             Neutral  14   64519932      A     G      Benign  \n",
       "\n",
       "[10842 rows x 11 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = test.merge(test_in)\n",
    "test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "test['prediction'].replace({'Pathogenic': 1, 'Neutral': 0}, inplace=True)\n",
    "test['label'].replace({'Pathogenic': 1, 'Benign': 0}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>probabilities</th>\n",
       "      <th>prediction</th>\n",
       "      <th>combined_prediction</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999962e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>17</td>\n",
       "      <td>41243705</td>\n",
       "      <td>CTGAG</td>\n",
       "      <td>GCCT</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>26.900</td>\n",
       "      <td>9.999943e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32912883</td>\n",
       "      <td>CTG</td>\n",
       "      <td>TT</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MSH2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999934e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>2</td>\n",
       "      <td>47639582</td>\n",
       "      <td>AGAAA</td>\n",
       "      <td>TAAT</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>24.100</td>\n",
       "      <td>9.999933e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32912590</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>25.600</td>\n",
       "      <td>9.999924e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32913442</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10837</th>\n",
       "      <td>MIR548AZ</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>5.553</td>\n",
       "      <td>1.214339e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>14</td>\n",
       "      <td>64537567</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10838</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.068</td>\n",
       "      <td>8.699849e-07</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10432854</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10839</th>\n",
       "      <td>NBPF20</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>3.320</td>\n",
       "      <td>6.732982e-07</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>1</td>\n",
       "      <td>145439717</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10840</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>0.035</td>\n",
       "      <td>6.130708e-07</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10442488</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10841</th>\n",
       "      <td>MIR548AZ</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>1.347</td>\n",
       "      <td>4.053531e-07</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>14</td>\n",
       "      <td>64519932</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10842 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       GeneName     Consequence   PHRED  probabilities  prediction  \\\n",
       "0         BRCA1     STOP_GAINED  34.000   9.999962e-01           1   \n",
       "1         BRCA2     FRAME_SHIFT  26.900   9.999943e-01           1   \n",
       "2          MSH2     FRAME_SHIFT  34.000   9.999934e-01           1   \n",
       "3         BRCA2     FRAME_SHIFT  24.100   9.999933e-01           1   \n",
       "4         BRCA2     FRAME_SHIFT  25.600   9.999924e-01           1   \n",
       "...         ...             ...     ...            ...         ...   \n",
       "10837  MIR548AZ  NON_SYNONYMOUS   5.553   1.214339e-06           0   \n",
       "10838     MYHAS        INTRONIC   0.068   8.699849e-07           0   \n",
       "10839    NBPF20      REGULATORY   3.320   6.732982e-07           0   \n",
       "10840     MYHAS      DOWNSTREAM   0.035   6.130708e-07           0   \n",
       "10841  MIR548AZ  NON_SYNONYMOUS   1.347   4.053531e-07           0   \n",
       "\n",
       "      combined_prediction chr        pos    ref   alt  label  \n",
       "0              Pathogenic  17   41243705  CTGAG  GCCT      1  \n",
       "1              Pathogenic  13   32912883    CTG    TT      1  \n",
       "2              Pathogenic   2   47639582  AGAAA  TAAT      1  \n",
       "3              Pathogenic  13   32912590     TA     T      1  \n",
       "4              Pathogenic  13   32913442     TC     T      1  \n",
       "...                   ...  ..        ...    ...   ...    ...  \n",
       "10837             Neutral  14   64537567      G     C      0  \n",
       "10838             Neutral  17   10432854      G     A      0  \n",
       "10839             Neutral   1  145439717      C     G      0  \n",
       "10840             Neutral  17   10442488      G     A      0  \n",
       "10841             Neutral  14   64519932      A     G      0  \n",
       "\n",
       "[10842 rows x 11 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>probabilities</th>\n",
       "      <th>prediction</th>\n",
       "      <th>combined_prediction</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999982e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216252</td>\n",
       "      <td>GACAAA</td>\n",
       "      <td>TTT</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999978e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32936775</td>\n",
       "      <td>GAATTT</td>\n",
       "      <td>AG</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999976e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216246</td>\n",
       "      <td>TGCAAGGACAAATCTGAC</td>\n",
       "      <td>CCGACTG</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999967e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216262</td>\n",
       "      <td>ACG</td>\n",
       "      <td>CA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>20.800</td>\n",
       "      <td>9.999956e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32906847</td>\n",
       "      <td>TACCCCTATTG</td>\n",
       "      <td>ACAT</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334596</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>10.130</td>\n",
       "      <td>7.769997e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10429043</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334597</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>15.280</td>\n",
       "      <td>7.578360e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10415744</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334598</th>\n",
       "      <td>MIR1273H</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>13.040</td>\n",
       "      <td>6.680243e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>4</td>\n",
       "      <td>39448569</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334599</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>12.770</td>\n",
       "      <td>5.909591e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10318882</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334600</th>\n",
       "      <td>MYH8</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>2.474</td>\n",
       "      <td>4.037909e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10318897</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>334601 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        GeneName  Consequence   PHRED  probabilities  prediction  \\\n",
       "0           LDLR  FRAME_SHIFT  35.000   9.999982e-01           1   \n",
       "1          BRCA2  FRAME_SHIFT  35.000   9.999978e-01           1   \n",
       "2           LDLR  FRAME_SHIFT  35.000   9.999976e-01           1   \n",
       "3           LDLR  FRAME_SHIFT  34.000   9.999967e-01           1   \n",
       "4          BRCA2  FRAME_SHIFT  20.800   9.999956e-01           1   \n",
       "...          ...          ...     ...            ...         ...   \n",
       "334596     MYHAS   SYNONYMOUS  10.130   7.769997e-08           0   \n",
       "334597     MYHAS   SYNONYMOUS  15.280   7.578360e-08           0   \n",
       "334598  MIR1273H   SYNONYMOUS  13.040   6.680243e-08           0   \n",
       "334599     MYHAS   SYNONYMOUS  12.770   5.909591e-08           0   \n",
       "334600      MYH8  SPLICE_SITE   2.474   4.037909e-08           0   \n",
       "\n",
       "       combined_prediction chr       pos                 ref      alt  label  \n",
       "0               Pathogenic  19  11216252              GACAAA      TTT      1  \n",
       "1               Pathogenic  13  32936775              GAATTT       AG      1  \n",
       "2               Pathogenic  19  11216246  TGCAAGGACAAATCTGAC  CCGACTG      1  \n",
       "3               Pathogenic  19  11216262                 ACG       CA      1  \n",
       "4               Pathogenic  13  32906847         TACCCCTATTG     ACAT      1  \n",
       "...                    ...  ..       ...                 ...      ...    ...  \n",
       "334596             Neutral  17  10429043                   G        A      0  \n",
       "334597             Neutral  17  10415744                   A        G      0  \n",
       "334598             Neutral   4  39448569                   G        A      0  \n",
       "334599             Neutral  17  10318882                   G        A      0  \n",
       "334600             Neutral  17  10318897                   G        A      0  \n",
       "\n",
       "[334601 rows x 11 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>probabilities</th>\n",
       "      <th>prediction</th>\n",
       "      <th>combined_prediction</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>label</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999982e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216252</td>\n",
       "      <td>GACAAA</td>\n",
       "      <td>TTT</td>\n",
       "      <td>1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999978e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32936775</td>\n",
       "      <td>GAATTT</td>\n",
       "      <td>AG</td>\n",
       "      <td>1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999976e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216246</td>\n",
       "      <td>TGCAAGGACAAATCTGAC</td>\n",
       "      <td>CCGACTG</td>\n",
       "      <td>1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999967e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>19</td>\n",
       "      <td>11216262</td>\n",
       "      <td>ACG</td>\n",
       "      <td>CA</td>\n",
       "      <td>1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>20.800</td>\n",
       "      <td>9.999956e-01</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>13</td>\n",
       "      <td>32906847</td>\n",
       "      <td>TACCCCTATTG</td>\n",
       "      <td>ACAT</td>\n",
       "      <td>1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345438</th>\n",
       "      <td>MIR548AZ</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>5.553</td>\n",
       "      <td>1.214339e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>14</td>\n",
       "      <td>64537567</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345439</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.068</td>\n",
       "      <td>8.699849e-07</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10432854</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345440</th>\n",
       "      <td>NBPF20</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>3.320</td>\n",
       "      <td>6.732982e-07</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>1</td>\n",
       "      <td>145439717</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345441</th>\n",
       "      <td>MYHAS</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>0.035</td>\n",
       "      <td>6.130708e-07</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>10442488</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345442</th>\n",
       "      <td>MIR548AZ</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>1.347</td>\n",
       "      <td>4.053531e-07</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>14</td>\n",
       "      <td>64519932</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>345443 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        GeneName     Consequence   PHRED  probabilities  prediction  \\\n",
       "0           LDLR     FRAME_SHIFT  35.000   9.999982e-01           1   \n",
       "1          BRCA2     FRAME_SHIFT  35.000   9.999978e-01           1   \n",
       "2           LDLR     FRAME_SHIFT  35.000   9.999976e-01           1   \n",
       "3           LDLR     FRAME_SHIFT  34.000   9.999967e-01           1   \n",
       "4          BRCA2     FRAME_SHIFT  20.800   9.999956e-01           1   \n",
       "...          ...             ...     ...            ...         ...   \n",
       "345438  MIR548AZ  NON_SYNONYMOUS   5.553   1.214339e-06           0   \n",
       "345439     MYHAS        INTRONIC   0.068   8.699849e-07           0   \n",
       "345440    NBPF20      REGULATORY   3.320   6.732982e-07           0   \n",
       "345441     MYHAS      DOWNSTREAM   0.035   6.130708e-07           0   \n",
       "345442  MIR548AZ  NON_SYNONYMOUS   1.347   4.053531e-07           0   \n",
       "\n",
       "       combined_prediction chr        pos                 ref      alt  label  \\\n",
       "0               Pathogenic  19   11216252              GACAAA      TTT      1   \n",
       "1               Pathogenic  13   32936775              GAATTT       AG      1   \n",
       "2               Pathogenic  19   11216246  TGCAAGGACAAATCTGAC  CCGACTG      1   \n",
       "3               Pathogenic  19   11216262                 ACG       CA      1   \n",
       "4               Pathogenic  13   32906847         TACCCCTATTG     ACAT      1   \n",
       "...                    ...  ..        ...                 ...      ...    ...   \n",
       "345438             Neutral  14   64537567                   G        C      0   \n",
       "345439             Neutral  17   10432854                   G        A      0   \n",
       "345440             Neutral   1  145439717                   C        G      0   \n",
       "345441             Neutral  17   10442488                   G        A      0   \n",
       "345442             Neutral  14   64519932                   A        G      0   \n",
       "\n",
       "       source  \n",
       "0       train  \n",
       "1       train  \n",
       "2       train  \n",
       "3       train  \n",
       "4       train  \n",
       "...       ...  \n",
       "345438   test  \n",
       "345439   test  \n",
       "345440   test  \n",
       "345441   test  \n",
       "345442   test  \n",
       "\n",
       "[345443 rows x 12 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_out['source'] = 'train'\n",
    "test['source'] = 'test'\n",
    "full = train_out.append(test, ignore_index=True)\n",
    "full"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.7743036340158642"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "y_true = np.array(test['label'])\n",
    "y_pred = np.array(test['prediction'])\n",
    "test_auc = roc_auc_score(y_true = y_true, y_score=y_pred)\n",
    "test_auc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "I am stilling running, done 4.913115432354158%\n",
      "I am stilling running, done 9.805544062887877%\n",
      "I am stilling running, done 14.687629292511378%\n",
      "I am stilling running, done 19.631774927596194%\n",
      "I am stilling running, done 24.627637567232107%\n",
      "I am stilling running, done 30.388911874224245%\n",
      "I am stilling running, done 36.150186181216384%\n",
      "I am stilling running, done 41.84940008274721%\n",
      "I am stilling running, done 47.62101779064957%\n",
      "I am stilling running, done 53.44435250310302%\n",
      "I am stilling running, done 59.28837401737691%\n",
      "I am stilling running, done 65.13239553165081%\n",
      "I am stilling running, done 71.01779064956558%\n",
      "I am stilling running, done 76.91352916839057%\n",
      "I am stilling running, done 82.77823748448489%\n",
      "I am stilling running, done 88.67397600330989%\n",
      "I am stilling running, done 94.5800579230451%\n"
     ]
    }
   ],
   "source": [
    "if not os.path.exists('./not_saving_directory/'):\n",
    "    os.mkdir('./not_saving_directory/')\n",
    "auc_analysis_output_filename = './not_saving_directory/auc_analysis_37.csv'\n",
    "if not os.path.isfile(auc_analysis_output_filename):\n",
    "    auc_analysis = pd.DataFrame(columns = ['gene', 'auc', 'f1', 'recall', 'fpr', 'precision',\n",
    "                                          'n_benign','n_malign','n_tot','n_train','n_test'])\n",
    "    n_tot_iters = full['GeneName'].unique().size\n",
    "    done_iters = 0\n",
    "    t_fls = time.time()\n",
    "    for gene in full['GeneName'].unique():\n",
    "        done_iters += 1\n",
    "        t_ifl = time.time()\n",
    "        if t_ifl - t_fls > 10:\n",
    "            print(f'I am stilling running, done {done_iters / n_tot_iters * 100}%')\n",
    "            t_fls = time.time()\n",
    "        subset = full[full['GeneName'] == gene]\n",
    "        y_true = np.array(subset['label'])\n",
    "        y_pred = np.array(subset['prediction'])\n",
    "        n_train = subset[subset['source'] == 'train'].shape[0]\n",
    "        n_test = subset[subset['source'] == 'test'].shape[0]\n",
    "        try:\n",
    "            auc = roc_auc_score(y_true, y_pred)\n",
    "            f1 = f1_score(y_true, y_pred)\n",
    "            recall = recall_score(y_true, y_pred, zero_division=0)\n",
    "            fpr = 1 - recall\n",
    "            precision = precision_score(y_true, y_pred, zero_division=0)\n",
    "            n_benign = y_true[y_true == 0].size\n",
    "            n_malign = y_true[y_true == 1].size\n",
    "            n_tot = y_true.size\n",
    "        except Exception:\n",
    "            continue\n",
    "        auc_analysis = auc_analysis.append(\n",
    "            pd.DataFrame({\n",
    "                'gene': [gene],\n",
    "                'auc': [auc],\n",
    "                'f1': [f1],\n",
    "                'recall': [recall],\n",
    "                'fpr': [fpr],\n",
    "                'precision': [precision],\n",
    "                'n_benign': [n_benign],\n",
    "                'n_malign': [n_malign],\n",
    "                'n_tot': [n_tot],\n",
    "                'n_train': [n_train],\n",
    "                'n_test': [n_test]\n",
    "            }, index=[0]), ignore_index=True)\n",
    "    auc_analysis.to_csv(auc_analysis_output_filename)\n",
    "else:\n",
    "    auc_analysis = pd.read_csv(auc_analysis_output_filename)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>auc</th>\n",
       "      <th>f1</th>\n",
       "      <th>recall</th>\n",
       "      <th>fpr</th>\n",
       "      <th>precision</th>\n",
       "      <th>n_benign</th>\n",
       "      <th>n_malign</th>\n",
       "      <th>n_tot</th>\n",
       "      <th>n_train</th>\n",
       "      <th>n_test</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2218</th>\n",
       "      <td>MFAP5</td>\n",
       "      <td>0.437500</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>1</td>\n",
       "      <td>41</td>\n",
       "      <td>38</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2221</th>\n",
       "      <td>GFM2</td>\n",
       "      <td>0.447368</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>114</td>\n",
       "      <td>2</td>\n",
       "      <td>116</td>\n",
       "      <td>107</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2257</th>\n",
       "      <td>SLC10A2</td>\n",
       "      <td>0.471429</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35</td>\n",
       "      <td>1</td>\n",
       "      <td>36</td>\n",
       "      <td>30</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2270</th>\n",
       "      <td>C2</td>\n",
       "      <td>0.472222</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>19</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2280</th>\n",
       "      <td>AFF4</td>\n",
       "      <td>0.475000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2261</th>\n",
       "      <td>TSHZ1</td>\n",
       "      <td>0.477612</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>67</td>\n",
       "      <td>1</td>\n",
       "      <td>68</td>\n",
       "      <td>66</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2278</th>\n",
       "      <td>KCNK9</td>\n",
       "      <td>0.487179</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>39</td>\n",
       "      <td>1</td>\n",
       "      <td>40</td>\n",
       "      <td>39</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2248</th>\n",
       "      <td>SPTLC2</td>\n",
       "      <td>0.490385</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>208</td>\n",
       "      <td>2</td>\n",
       "      <td>210</td>\n",
       "      <td>208</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2223</th>\n",
       "      <td>ATP6AP1</td>\n",
       "      <td>0.492188</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>64</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>63</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2282</th>\n",
       "      <td>GJB3</td>\n",
       "      <td>0.494318</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>88</td>\n",
       "      <td>2</td>\n",
       "      <td>90</td>\n",
       "      <td>90</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene       auc   f1  recall  fpr  precision n_benign n_malign n_tot  \\\n",
       "2218    MFAP5  0.437500  0.0     0.0  1.0        0.0       40        1    41   \n",
       "2221     GFM2  0.447368  0.0     0.0  1.0        0.0      114        2   116   \n",
       "2257  SLC10A2  0.471429  0.0     0.0  1.0        0.0       35        1    36   \n",
       "2270       C2  0.472222  0.0     0.0  1.0        0.0       18        1    19   \n",
       "2280     AFF4  0.475000  0.0     0.0  1.0        0.0       20        1    21   \n",
       "2261    TSHZ1  0.477612  0.0     0.0  1.0        0.0       67        1    68   \n",
       "2278    KCNK9  0.487179  0.0     0.0  1.0        0.0       39        1    40   \n",
       "2248   SPTLC2  0.490385  0.0     0.0  1.0        0.0      208        2   210   \n",
       "2223  ATP6AP1  0.492188  0.0     0.0  1.0        0.0       64        1    65   \n",
       "2282     GJB3  0.494318  0.0     0.0  1.0        0.0       88        2    90   \n",
       "\n",
       "     n_train n_test  \n",
       "2218      38      3  \n",
       "2221     107      9  \n",
       "2257      30      6  \n",
       "2270      18      1  \n",
       "2280      21      0  \n",
       "2261      66      2  \n",
       "2278      39      1  \n",
       "2248     208      2  \n",
       "2223      63      2  \n",
       "2282      90      0  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# The 10 worst performing genes in the new model. SPTLC2 is again amongst them.\n",
    "\n",
    "auc_analysis.sort_values(by='auc').head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mean of new model: 0.9512713923971363 \n",
      "Median of new model: 0.9893045069778083\n"
     ]
    }
   ],
   "source": [
    "# Average auc is pretty good though.\n",
    "\n",
    "print(f\"Mean of new model: {auc_analysis['auc'].mean()} \\nMedian of new model: {auc_analysis['auc'].median()}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "previous_model_auc = pd.read_csv('/home/rjsietsma/Documents/School/Master_DSLS/Final_Thesis/Initial_Data_exploration/auc_results.csv', index_col=0)\n",
    "previous_model_auc = previous_model_auc[['Gene', 'AUC']]\n",
    "previous_model_auc.columns = ['gene', 'auc_previous']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mean of previous model: 0.9448573343008154 \n",
      "Median of previous model: 0.96875\n"
     ]
    }
   ],
   "source": [
    "# On average, the new model does perform better\n",
    "\n",
    "print(f\"Mean of previous model: {previous_model_auc['auc_previous'].mean()} \\nMedian of previous model: {previous_model_auc['auc_previous'].median()}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>auc</th>\n",
       "      <th>f1</th>\n",
       "      <th>recall</th>\n",
       "      <th>fpr</th>\n",
       "      <th>precision</th>\n",
       "      <th>n_benign</th>\n",
       "      <th>n_malign</th>\n",
       "      <th>n_tot</th>\n",
       "      <th>n_train</th>\n",
       "      <th>n_test</th>\n",
       "      <th>auc_previous</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>0.983725</td>\n",
       "      <td>0.992126</td>\n",
       "      <td>0.991259</td>\n",
       "      <td>0.008741</td>\n",
       "      <td>0.992995</td>\n",
       "      <td>336</td>\n",
       "      <td>1144</td>\n",
       "      <td>1480</td>\n",
       "      <td>1325</td>\n",
       "      <td>155</td>\n",
       "      <td>0.905925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>0.992003</td>\n",
       "      <td>0.995261</td>\n",
       "      <td>0.997110</td>\n",
       "      <td>0.002890</td>\n",
       "      <td>0.993418</td>\n",
       "      <td>1221</td>\n",
       "      <td>2422</td>\n",
       "      <td>3643</td>\n",
       "      <td>2973</td>\n",
       "      <td>670</td>\n",
       "      <td>0.946305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>0.986892</td>\n",
       "      <td>0.992528</td>\n",
       "      <td>0.993808</td>\n",
       "      <td>0.006192</td>\n",
       "      <td>0.991251</td>\n",
       "      <td>849</td>\n",
       "      <td>1938</td>\n",
       "      <td>2787</td>\n",
       "      <td>2226</td>\n",
       "      <td>561</td>\n",
       "      <td>0.949156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MSH2</td>\n",
       "      <td>0.989096</td>\n",
       "      <td>0.990902</td>\n",
       "      <td>0.988449</td>\n",
       "      <td>0.011551</td>\n",
       "      <td>0.993367</td>\n",
       "      <td>390</td>\n",
       "      <td>606</td>\n",
       "      <td>996</td>\n",
       "      <td>850</td>\n",
       "      <td>146</td>\n",
       "      <td>0.955128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>USH2A</td>\n",
       "      <td>0.977106</td>\n",
       "      <td>0.970238</td>\n",
       "      <td>0.970238</td>\n",
       "      <td>0.029762</td>\n",
       "      <td>0.970238</td>\n",
       "      <td>312</td>\n",
       "      <td>168</td>\n",
       "      <td>480</td>\n",
       "      <td>459</td>\n",
       "      <td>21</td>\n",
       "      <td>0.944790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2349</th>\n",
       "      <td>NLRP1</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2350</th>\n",
       "      <td>APOA1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2351</th>\n",
       "      <td>IL2RG</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2352</th>\n",
       "      <td>SGSM3</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353</th>\n",
       "      <td>ANKRD17</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2354 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene       auc        f1    recall       fpr  precision n_benign  \\\n",
       "0        LDLR  0.983725  0.992126  0.991259  0.008741   0.992995      336   \n",
       "1       BRCA2  0.992003  0.995261  0.997110  0.002890   0.993418     1221   \n",
       "2       BRCA1  0.986892  0.992528  0.993808  0.006192   0.991251      849   \n",
       "3        MSH2  0.989096  0.990902  0.988449  0.011551   0.993367      390   \n",
       "4       USH2A  0.977106  0.970238  0.970238  0.029762   0.970238      312   \n",
       "...       ...       ...       ...       ...       ...        ...      ...   \n",
       "2349    NLRP1  0.500000  0.000000  0.000000  1.000000   0.000000       14   \n",
       "2350    APOA1  1.000000  1.000000  1.000000  0.000000   1.000000        9   \n",
       "2351    IL2RG  0.500000  0.000000  0.000000  1.000000   0.000000        6   \n",
       "2352    SGSM3  1.000000  1.000000  1.000000  0.000000   1.000000        3   \n",
       "2353  ANKRD17  1.000000  1.000000  1.000000  0.000000   1.000000        1   \n",
       "\n",
       "     n_malign n_tot n_train n_test  auc_previous  \n",
       "0        1144  1480    1325    155      0.905925  \n",
       "1        2422  3643    2973    670      0.946305  \n",
       "2        1938  2787    2226    561      0.949156  \n",
       "3         606   996     850    146      0.955128  \n",
       "4         168   480     459     21      0.944790  \n",
       "...       ...   ...     ...    ...           ...  \n",
       "2349        1    15      15      0           NaN  \n",
       "2350        1    10       9      1           NaN  \n",
       "2351        1     7       6      1           NaN  \n",
       "2352        1     4       3      1           NaN  \n",
       "2353        1     2       1      1           NaN  \n",
       "\n",
       "[2354 rows x 12 columns]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "old_new = auc_analysis.merge(previous_model_auc, how='left')\n",
    "old_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unmerged = previous_model_auc.merge(auc_analysis, how='left').dropna(subset=['auc_previous'])\n",
    "unmerged['auc'].isnull().any()\n",
    "\n",
    "# No genes are missing from the previous dataset, so essentially only more genes got an AUC now."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are now 62 new genes that have AUC data.\n"
     ]
    }
   ],
   "source": [
    "n_genes_now_auc = old_new.where(\n",
    "    (old_new['auc_previous'].isnull()) & \n",
    "    (old_new['auc'].notnull())\n",
    ").dropna(subset=['gene']).shape[0]\n",
    "print(f'There are now {n_genes_now_auc} new genes that have AUC data.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "old_new_2 = old_new.where(\n",
    "    (old_new['auc'] != 0.5) & \n",
    "    (old_new['auc'] != 1.0) & \n",
    "    (old_new['auc_previous'] != 0.5) & \n",
    "    (old_new['auc_previous'] != 1.0)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>auc</th>\n",
       "      <th>f1</th>\n",
       "      <th>recall</th>\n",
       "      <th>fpr</th>\n",
       "      <th>precision</th>\n",
       "      <th>n_benign</th>\n",
       "      <th>n_malign</th>\n",
       "      <th>n_tot</th>\n",
       "      <th>n_train</th>\n",
       "      <th>n_test</th>\n",
       "      <th>auc_previous</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>0.983725</td>\n",
       "      <td>0.992126</td>\n",
       "      <td>0.991259</td>\n",
       "      <td>0.008741</td>\n",
       "      <td>0.992995</td>\n",
       "      <td>336</td>\n",
       "      <td>1144</td>\n",
       "      <td>1480</td>\n",
       "      <td>1325</td>\n",
       "      <td>155</td>\n",
       "      <td>0.905925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>0.992003</td>\n",
       "      <td>0.995261</td>\n",
       "      <td>0.997110</td>\n",
       "      <td>0.002890</td>\n",
       "      <td>0.993418</td>\n",
       "      <td>1221</td>\n",
       "      <td>2422</td>\n",
       "      <td>3643</td>\n",
       "      <td>2973</td>\n",
       "      <td>670</td>\n",
       "      <td>0.946305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>0.986892</td>\n",
       "      <td>0.992528</td>\n",
       "      <td>0.993808</td>\n",
       "      <td>0.006192</td>\n",
       "      <td>0.991251</td>\n",
       "      <td>849</td>\n",
       "      <td>1938</td>\n",
       "      <td>2787</td>\n",
       "      <td>2226</td>\n",
       "      <td>561</td>\n",
       "      <td>0.949156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MSH2</td>\n",
       "      <td>0.989096</td>\n",
       "      <td>0.990902</td>\n",
       "      <td>0.988449</td>\n",
       "      <td>0.011551</td>\n",
       "      <td>0.993367</td>\n",
       "      <td>390</td>\n",
       "      <td>606</td>\n",
       "      <td>996</td>\n",
       "      <td>850</td>\n",
       "      <td>146</td>\n",
       "      <td>0.955128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>USH2A</td>\n",
       "      <td>0.977106</td>\n",
       "      <td>0.970238</td>\n",
       "      <td>0.970238</td>\n",
       "      <td>0.029762</td>\n",
       "      <td>0.970238</td>\n",
       "      <td>312</td>\n",
       "      <td>168</td>\n",
       "      <td>480</td>\n",
       "      <td>459</td>\n",
       "      <td>21</td>\n",
       "      <td>0.944790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2280</th>\n",
       "      <td>AFF4</td>\n",
       "      <td>0.475000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>0.400000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2282</th>\n",
       "      <td>GJB3</td>\n",
       "      <td>0.494318</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>88</td>\n",
       "      <td>2</td>\n",
       "      <td>90</td>\n",
       "      <td>90</td>\n",
       "      <td>0</td>\n",
       "      <td>0.494318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2283</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>0.992248</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>129</td>\n",
       "      <td>1</td>\n",
       "      <td>130</td>\n",
       "      <td>124</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2284</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>0.497093</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>172</td>\n",
       "      <td>1</td>\n",
       "      <td>173</td>\n",
       "      <td>172</td>\n",
       "      <td>1</td>\n",
       "      <td>0.485380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2338</th>\n",
       "      <td>TMPRSS6</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1176 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene       auc        f1    recall       fpr  precision n_benign  \\\n",
       "0        LDLR  0.983725  0.992126  0.991259  0.008741   0.992995      336   \n",
       "1       BRCA2  0.992003  0.995261  0.997110  0.002890   0.993418     1221   \n",
       "2       BRCA1  0.986892  0.992528  0.993808  0.006192   0.991251      849   \n",
       "3        MSH2  0.989096  0.990902  0.988449  0.011551   0.993367      390   \n",
       "4       USH2A  0.977106  0.970238  0.970238  0.029762   0.970238      312   \n",
       "...       ...       ...       ...       ...       ...        ...      ...   \n",
       "2280     AFF4  0.475000  0.000000  0.000000  1.000000   0.000000       20   \n",
       "2282     GJB3  0.494318  0.000000  0.000000  1.000000   0.000000       88   \n",
       "2283    A2ML1  0.992248  0.500000  1.000000  0.000000   0.333333      129   \n",
       "2284     JAK2  0.497093  0.000000  0.000000  1.000000   0.000000      172   \n",
       "2338  TMPRSS6  0.750000  0.666667  0.500000  0.500000   1.000000       14   \n",
       "\n",
       "     n_malign n_tot n_train n_test  auc_previous  \n",
       "0        1144  1480    1325    155      0.905925  \n",
       "1        2422  3643    2973    670      0.946305  \n",
       "2        1938  2787    2226    561      0.949156  \n",
       "3         606   996     850    146      0.955128  \n",
       "4         168   480     459     21      0.944790  \n",
       "...       ...   ...     ...    ...           ...  \n",
       "2280        1    21      21      0      0.400000  \n",
       "2282        2    90      90      0      0.494318  \n",
       "2283        1   130     124      6           NaN  \n",
       "2284        1   173     172      1      0.485380  \n",
       "2338        2    16      15      1           NaN  \n",
       "\n",
       "[1176 rows x 12 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "old_new_2.dropna(subset=['gene'], inplace=True)\n",
    "old_new_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>auc</th>\n",
       "      <th>f1</th>\n",
       "      <th>recall</th>\n",
       "      <th>fpr</th>\n",
       "      <th>precision</th>\n",
       "      <th>n_benign</th>\n",
       "      <th>n_malign</th>\n",
       "      <th>n_tot</th>\n",
       "      <th>n_train</th>\n",
       "      <th>n_test</th>\n",
       "      <th>auc_previous</th>\n",
       "      <th>delta</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2261</th>\n",
       "      <td>TSHZ1</td>\n",
       "      <td>0.477612</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>67</td>\n",
       "      <td>1</td>\n",
       "      <td>68</td>\n",
       "      <td>66</td>\n",
       "      <td>2</td>\n",
       "      <td>0.976923</td>\n",
       "      <td>0.499311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2223</th>\n",
       "      <td>ATP6AP1</td>\n",
       "      <td>0.492188</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>64</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>63</td>\n",
       "      <td>2</td>\n",
       "      <td>0.943548</td>\n",
       "      <td>0.451361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2120</th>\n",
       "      <td>ATP6V1B2</td>\n",
       "      <td>0.553571</td>\n",
       "      <td>0.285714</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.75</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>0.857143</td>\n",
       "      <td>0.303571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2182</th>\n",
       "      <td>COL4A2</td>\n",
       "      <td>0.681788</td>\n",
       "      <td>0.222222</td>\n",
       "      <td>0.40</td>\n",
       "      <td>0.60</td>\n",
       "      <td>0.153846</td>\n",
       "      <td>302</td>\n",
       "      <td>5</td>\n",
       "      <td>307</td>\n",
       "      <td>295</td>\n",
       "      <td>12</td>\n",
       "      <td>0.962069</td>\n",
       "      <td>0.280281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1563</th>\n",
       "      <td>CACNA1S</td>\n",
       "      <td>0.690798</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>0.40</td>\n",
       "      <td>0.60</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>163</td>\n",
       "      <td>5</td>\n",
       "      <td>168</td>\n",
       "      <td>162</td>\n",
       "      <td>6</td>\n",
       "      <td>0.953125</td>\n",
       "      <td>0.262327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2270</th>\n",
       "      <td>C2</td>\n",
       "      <td>0.472222</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>19</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2277</th>\n",
       "      <td>SMPX</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2278</th>\n",
       "      <td>KCNK9</td>\n",
       "      <td>0.487179</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>39</td>\n",
       "      <td>1</td>\n",
       "      <td>40</td>\n",
       "      <td>39</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2283</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>0.992248</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>129</td>\n",
       "      <td>1</td>\n",
       "      <td>130</td>\n",
       "      <td>124</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2338</th>\n",
       "      <td>TMPRSS6</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>0.50</td>\n",
       "      <td>0.50</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1176 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene       auc        f1  recall   fpr  precision n_benign n_malign  \\\n",
       "2261     TSHZ1  0.477612  0.000000    0.00  1.00   0.000000       67        1   \n",
       "2223   ATP6AP1  0.492188  0.000000    0.00  1.00   0.000000       64        1   \n",
       "2120  ATP6V1B2  0.553571  0.285714    0.25  0.75   0.333333       14        4   \n",
       "2182    COL4A2  0.681788  0.222222    0.40  0.60   0.153846      302        5   \n",
       "1563   CACNA1S  0.690798  0.400000    0.40  0.60   0.400000      163        5   \n",
       "...        ...       ...       ...     ...   ...        ...      ...      ...   \n",
       "2270        C2  0.472222  0.000000    0.00  1.00   0.000000       18        1   \n",
       "2277      SMPX  0.900000  0.666667    1.00  0.00   0.500000        5        1   \n",
       "2278     KCNK9  0.487179  0.000000    0.00  1.00   0.000000       39        1   \n",
       "2283     A2ML1  0.992248  0.500000    1.00  0.00   0.333333      129        1   \n",
       "2338   TMPRSS6  0.750000  0.666667    0.50  0.50   1.000000       14        2   \n",
       "\n",
       "     n_tot n_train n_test  auc_previous     delta  \n",
       "2261    68      66      2      0.976923  0.499311  \n",
       "2223    65      63      2      0.943548  0.451361  \n",
       "2120    18      18      0      0.857143  0.303571  \n",
       "2182   307     295     12      0.962069  0.280281  \n",
       "1563   168     162      6      0.953125  0.262327  \n",
       "...    ...     ...    ...           ...       ...  \n",
       "2270    19      18      1           NaN       NaN  \n",
       "2277     6       5      1           NaN       NaN  \n",
       "2278    40      39      1           NaN       NaN  \n",
       "2283   130     124      6           NaN       NaN  \n",
       "2338    16      15      1           NaN       NaN  \n",
       "\n",
       "[1176 rows x 13 columns]"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "old_new_2['delta'] = old_new_2['auc_previous'] - old_new_2['auc']\n",
    "old_new_2.sort_values(by=['delta'], ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>auc</th>\n",
       "      <th>f1</th>\n",
       "      <th>recall</th>\n",
       "      <th>fpr</th>\n",
       "      <th>precision</th>\n",
       "      <th>n_benign</th>\n",
       "      <th>n_malign</th>\n",
       "      <th>n_tot</th>\n",
       "      <th>n_train</th>\n",
       "      <th>n_test</th>\n",
       "      <th>auc_previous</th>\n",
       "      <th>delta</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>COL6A2</td>\n",
       "      <td>0.961697</td>\n",
       "      <td>0.953846</td>\n",
       "      <td>0.939394</td>\n",
       "      <td>0.060606</td>\n",
       "      <td>0.968750</td>\n",
       "      <td>125</td>\n",
       "      <td>66</td>\n",
       "      <td>191</td>\n",
       "      <td>185</td>\n",
       "      <td>6</td>\n",
       "      <td>0.975675</td>\n",
       "      <td>0.013978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>NEB</td>\n",
       "      <td>0.958708</td>\n",
       "      <td>0.943820</td>\n",
       "      <td>0.923077</td>\n",
       "      <td>0.076923</td>\n",
       "      <td>0.965517</td>\n",
       "      <td>530</td>\n",
       "      <td>91</td>\n",
       "      <td>621</td>\n",
       "      <td>604</td>\n",
       "      <td>17</td>\n",
       "      <td>0.983902</td>\n",
       "      <td>0.025193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>PKHD1</td>\n",
       "      <td>0.975460</td>\n",
       "      <td>0.975779</td>\n",
       "      <td>0.959184</td>\n",
       "      <td>0.040816</td>\n",
       "      <td>0.992958</td>\n",
       "      <td>121</td>\n",
       "      <td>147</td>\n",
       "      <td>268</td>\n",
       "      <td>248</td>\n",
       "      <td>20</td>\n",
       "      <td>0.979339</td>\n",
       "      <td>0.003879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>CACNA1A</td>\n",
       "      <td>0.950121</td>\n",
       "      <td>0.933333</td>\n",
       "      <td>0.909091</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>0.958904</td>\n",
       "      <td>339</td>\n",
       "      <td>77</td>\n",
       "      <td>416</td>\n",
       "      <td>407</td>\n",
       "      <td>9</td>\n",
       "      <td>0.956997</td>\n",
       "      <td>0.006876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>BLM</td>\n",
       "      <td>0.982456</td>\n",
       "      <td>0.982143</td>\n",
       "      <td>0.964912</td>\n",
       "      <td>0.035088</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>124</td>\n",
       "      <td>57</td>\n",
       "      <td>181</td>\n",
       "      <td>172</td>\n",
       "      <td>9</td>\n",
       "      <td>0.983871</td>\n",
       "      <td>0.001415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2223</th>\n",
       "      <td>ATP6AP1</td>\n",
       "      <td>0.492188</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>64</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>63</td>\n",
       "      <td>2</td>\n",
       "      <td>0.943548</td>\n",
       "      <td>0.451361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2225</th>\n",
       "      <td>B4GALT7</td>\n",
       "      <td>0.833333</td>\n",
       "      <td>0.800000</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "      <td>1</td>\n",
       "      <td>0.978261</td>\n",
       "      <td>0.144928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2227</th>\n",
       "      <td>CLASP1</td>\n",
       "      <td>0.665365</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>384</td>\n",
       "      <td>3</td>\n",
       "      <td>387</td>\n",
       "      <td>384</td>\n",
       "      <td>3</td>\n",
       "      <td>0.748691</td>\n",
       "      <td>0.083327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2239</th>\n",
       "      <td>RPL11</td>\n",
       "      <td>0.770833</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>8</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>0.812500</td>\n",
       "      <td>0.041667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2261</th>\n",
       "      <td>TSHZ1</td>\n",
       "      <td>0.477612</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>67</td>\n",
       "      <td>1</td>\n",
       "      <td>68</td>\n",
       "      <td>66</td>\n",
       "      <td>2</td>\n",
       "      <td>0.976923</td>\n",
       "      <td>0.499311</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>285 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene       auc        f1    recall       fpr  precision n_benign  \\\n",
       "60     COL6A2  0.961697  0.953846  0.939394  0.060606   0.968750      125   \n",
       "61        NEB  0.958708  0.943820  0.923077  0.076923   0.965517      530   \n",
       "64      PKHD1  0.975460  0.975779  0.959184  0.040816   0.992958      121   \n",
       "65    CACNA1A  0.950121  0.933333  0.909091  0.090909   0.958904      339   \n",
       "120       BLM  0.982456  0.982143  0.964912  0.035088   1.000000      124   \n",
       "...       ...       ...       ...       ...       ...        ...      ...   \n",
       "2223  ATP6AP1  0.492188  0.000000  0.000000  1.000000   0.000000       64   \n",
       "2225  B4GALT7  0.833333  0.800000  0.666667  0.333333   1.000000       23   \n",
       "2227   CLASP1  0.665365  0.400000  0.333333  0.666667   0.500000      384   \n",
       "2239    RPL11  0.770833  0.666667  0.666667  0.333333   0.666667        8   \n",
       "2261    TSHZ1  0.477612  0.000000  0.000000  1.000000   0.000000       67   \n",
       "\n",
       "     n_malign n_tot n_train n_test  auc_previous     delta  \n",
       "60         66   191     185      6      0.975675  0.013978  \n",
       "61         91   621     604     17      0.983902  0.025193  \n",
       "64        147   268     248     20      0.979339  0.003879  \n",
       "65         77   416     407      9      0.956997  0.006876  \n",
       "120        57   181     172      9      0.983871  0.001415  \n",
       "...       ...   ...     ...    ...           ...       ...  \n",
       "2223        1    65      63      2      0.943548  0.451361  \n",
       "2225        3    26      25      1      0.978261  0.144928  \n",
       "2227        3   387     384      3      0.748691  0.083327  \n",
       "2239        3    11      11      0      0.812500  0.041667  \n",
       "2261        1    68      66      2      0.976923  0.499311  \n",
       "\n",
       "[285 rows x 13 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "old_new_2[old_new_2['delta'] > 0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Model on best performing panel, prepartation\n",
    "[Index](#Index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Borstkanker': ['ATM', 'BRCA1', 'BRCA2', 'CHEK2', 'PALB2'],\n",
       " 'Borstkanker & Li Fraumeni syndroom': ['ATM',\n",
       "  'BRCA1',\n",
       "  'BRCA2',\n",
       "  'CHEK2',\n",
       "  'PALB2',\n",
       "  'TP53'],\n",
       " 'Darmkanker': ['EPCAM',\n",
       "  'MLH1',\n",
       "  'MSH2',\n",
       "  'MSH6',\n",
       "  'MUTYH',\n",
       "  'PMS2',\n",
       "  'POLD1',\n",
       "  'POLE'],\n",
       " 'Darmpoliepen': ['APC',\n",
       "  'AXIN2',\n",
       "  'BMPR1A',\n",
       "  'ENG',\n",
       "  'MSH3',\n",
       "  'MUTYH',\n",
       "  'NTHL1',\n",
       "  'POLD1',\n",
       "  'POLE',\n",
       "  'PTEN',\n",
       "  'RNF43',\n",
       "  'SMAD4',\n",
       "  'SMAD4',\n",
       "  'STK11'],\n",
       " 'Eierstokkanker': ['BRCA1',\n",
       "  'BRCA2',\n",
       "  'BRIP1',\n",
       "  'EPCAM',\n",
       "  'MLH1',\n",
       "  'MSH2',\n",
       "  'MSH6',\n",
       "  'PMS2',\n",
       "  'RAD51C',\n",
       "  'RAD51D'],\n",
       " 'Endocrinologie-feochromocytomen': ['FH',\n",
       "  'MAX',\n",
       "  'MDH2',\n",
       "  'NF1',\n",
       "  'RET',\n",
       "  'SDHA',\n",
       "  'SDHAF2',\n",
       "  'SDHB',\n",
       "  'SDHC',\n",
       "  'SDHD',\n",
       "  'TMEM127',\n",
       "  'VHL'],\n",
       " 'Endocrinologie-NET': ['AIP',\n",
       "  'CDKN1A',\n",
       "  'CDKN1B',\n",
       "  'CDKN2B',\n",
       "  'CDKN2C',\n",
       "  'MEN1',\n",
       "  'NF1',\n",
       "  'PRKAR1A'],\n",
       " 'Huidkanker': ['ACD',\n",
       "  'ACTRT1',\n",
       "  'BAP1',\n",
       "  'CDK4',\n",
       "  'CDKN2A',\n",
       "  'ERCC2',\n",
       "  'MITF',\n",
       "  'POLH',\n",
       "  'POT1',\n",
       "  'PTCH1',\n",
       "  'PTCH2',\n",
       "  'SUFU',\n",
       "  'TERF2IP',\n",
       "  'TERT'],\n",
       " 'Kinderonco-medulloblastoom': ['DICER1', 'PTCH1', 'PTCH2', 'SUFU'],\n",
       " 'Kinderonco-meningeoma/schwannoma': ['LZTR1',\n",
       "  'NF1',\n",
       "  'NF2',\n",
       "  'SMARCA4',\n",
       "  'SMARCB1',\n",
       "  'SMARCE1',\n",
       "  'SUFU',\n",
       "  'TSC1',\n",
       "  'TSC2'],\n",
       " 'Kinderonco-neuroblastoom': ['ALK', 'DICER1', 'PHOX2B'],\n",
       " 'Kinderonco-nierkanker aanvullend': ['DICER1', 'TSC1', 'TSC2', 'WT1'],\n",
       " 'Leukemie / lymfoom': ['CEBPA',\n",
       "  'DDX41',\n",
       "  'GATA2',\n",
       "  'NF1',\n",
       "  'PAX5',\n",
       "  'RUNX1',\n",
       "  'TERT'],\n",
       " 'Li Fraumeni syndroom': ['TP53'],\n",
       " 'Maagkanker': ['CDH1', 'CTNNA1', 'EPCAM', 'MLH1', 'MSH2', 'MSH6', 'PMS2'],\n",
       " 'Nierkanker': ['BAP1',\n",
       "  'FH',\n",
       "  'FLCN',\n",
       "  'MET',\n",
       "  'PTEN',\n",
       "  'SDHA',\n",
       "  'SDHAF2',\n",
       "  'SDHB',\n",
       "  'SDHC',\n",
       "  'SDHD',\n",
       "  'VHL'],\n",
       " 'Niet-endocriene alvleesklierkanker': ['ATM',\n",
       "  'BRCA1',\n",
       "  'BRCA2',\n",
       "  'CDKN2A',\n",
       "  'EPCAM',\n",
       "  'MLH1',\n",
       "  'MSH2',\n",
       "  'MSH6',\n",
       "  'PALB2',\n",
       "  'PALLD',\n",
       "  'PMS2',\n",
       "  'STK11'],\n",
       " 'Prostaatkanker': ['BRCA1', 'BRCA2', 'HOXB13'],\n",
       " 'Retinoblastoom': ['RB1'],\n",
       " 'Longkanker': ['EGFR'],\n",
       " 'Endocrinologie-(bij)schildklierkanker': ['APC',\n",
       "  'CDC73',\n",
       "  'DICER1',\n",
       "  'MEN1',\n",
       "  'PTEN',\n",
       "  'RET'],\n",
       " 'GIST (gastrointestinale stromaceltumor)': ['KIT',\n",
       "  'NF1',\n",
       "  'PDGFRA',\n",
       "  'SDHA',\n",
       "  'SDHB',\n",
       "  'SDHC',\n",
       "  'SDHD']}"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with open('./umcg_genepanels.json', 'r') as panels:\n",
    "    genepanels = json.load(panels)\n",
    "ek = genepanels['Erfelijke Kanker']\n",
    "ek"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "87"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ek_genes = []\n",
    "for key, values in ek.items():\n",
    "    for value in values:\n",
    "        if value not in ek_genes:\n",
    "            ek_genes.append(value)\n",
    "len(ek_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Allergy/Immunology/Infectious</th>\n",
       "      <th>Alt</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Audiologic/Otolaryngologic</th>\n",
       "      <th>Biochemical</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>CDSpos</th>\n",
       "      <th>Cardiovascular</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>revel</th>\n",
       "      <th>sift</th>\n",
       "      <th>source</th>\n",
       "      <th>tOverlapMotifs</th>\n",
       "      <th>targetScan</th>\n",
       "      <th>to_be_deleted</th>\n",
       "      <th>verPhCons</th>\n",
       "      <th>verPhyloP</th>\n",
       "      <th>inTest</th>\n",
       "      <th>sample_weight</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS1836.1</td>\n",
       "      <td>741.0</td>\n",
       "      <td>False</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.839</td>\n",
       "      <td>-0.283</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>TA</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS1836.1</td>\n",
       "      <td>1944.0</td>\n",
       "      <td>False</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.466</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>CC</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS1836.1</td>\n",
       "      <td>3259.0</td>\n",
       "      <td>False</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>1.000</td>\n",
       "      <td>3.249</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>C</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS1836.1</td>\n",
       "      <td>102.0</td>\n",
       "      <td>False</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.743</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>AT</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS1836.1</td>\n",
       "      <td>395.0</td>\n",
       "      <td>False</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.138</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334504</th>\n",
       "      <td>5</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS4107.1</td>\n",
       "      <td>3925.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>3.528</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334505</th>\n",
       "      <td>5</td>\n",
       "      <td>False</td>\n",
       "      <td>GACAAAGA</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS4107.1</td>\n",
       "      <td>4160.0</td>\n",
       "      <td>False</td>\n",
       "      <td>frameshift,stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.909</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334533</th>\n",
       "      <td>17</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS45646.1</td>\n",
       "      <td>271.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.639</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334534</th>\n",
       "      <td>17</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS45646.1</td>\n",
       "      <td>184.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.266</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334535</th>\n",
       "      <td>17</td>\n",
       "      <td>False</td>\n",
       "      <td>C</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS45646.1</td>\n",
       "      <td>161.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>0.017</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.004</td>\n",
       "      <td>0.318</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26776 rows × 152 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Chrom  Allergy/Immunology/Infectious       Alt          AnnoType  \\\n",
       "55          2                          False         A  CodingTranscript   \n",
       "56          2                          False        TA  CodingTranscript   \n",
       "57          2                          False        CC  CodingTranscript   \n",
       "58          2                          False         C  CodingTranscript   \n",
       "59          2                          False        AT  CodingTranscript   \n",
       "...       ...                            ...       ...               ...   \n",
       "334504      5                          False         T  CodingTranscript   \n",
       "334505      5                          False  GACAAAGA  CodingTranscript   \n",
       "334533     17                          False         A  CodingTranscript   \n",
       "334534     17                          False         A  CodingTranscript   \n",
       "334535     17                          False         C  CodingTranscript   \n",
       "\n",
       "        Audiologic/Otolaryngologic  Biochemical         CCDS  CDSpos  \\\n",
       "55                           False        False   CCDS1836.1   741.0   \n",
       "56                           False        False   CCDS1836.1  1944.0   \n",
       "57                           False        False   CCDS1836.1  3259.0   \n",
       "58                           False        False   CCDS1836.1   102.0   \n",
       "59                           False        False   CCDS1836.1   395.0   \n",
       "...                            ...          ...          ...     ...   \n",
       "334504                       False        False   CCDS4107.1  3925.0   \n",
       "334505                       False        False   CCDS4107.1  4160.0   \n",
       "334533                       False        False  CCDS45646.1   271.0   \n",
       "334534                       False        False  CCDS45646.1   184.0   \n",
       "334535                       False        False  CCDS45646.1   161.0   \n",
       "\n",
       "        Cardiovascular              ConsDetail  ...  revel sift   source  \\\n",
       "55               False              frameshift  ...    NaN  NaN     vkgl   \n",
       "56               False              frameshift  ...    NaN  NaN     vkgl   \n",
       "57               False              frameshift  ...    NaN  NaN     vkgl   \n",
       "58               False              frameshift  ...    NaN  NaN     vkgl   \n",
       "59               False              frameshift  ...    NaN  NaN     vkgl   \n",
       "...                ...                     ...  ...    ...  ...      ...   \n",
       "334504           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334505           False  frameshift,stop_gained  ...    NaN  NaN  unknown   \n",
       "334533           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334534           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334535           False             stop_gained  ...  0.017  NaN  unknown   \n",
       "\n",
       "        tOverlapMotifs  targetScan  to_be_deleted  verPhCons verPhyloP inTest  \\\n",
       "55                 NaN         NaN          False      0.839    -0.283  False   \n",
       "56                 NaN         NaN           True      0.001     0.466  False   \n",
       "57                 NaN         NaN           True      1.000     3.249  False   \n",
       "58                 NaN         NaN          False      0.000    -0.743  False   \n",
       "59                 NaN         NaN           True      1.000     5.138  False   \n",
       "...                ...         ...            ...        ...       ...    ...   \n",
       "334504             NaN         NaN          False      1.000     3.528  False   \n",
       "334505             NaN         NaN          False      1.000     5.909  False   \n",
       "334533             NaN         NaN          False      0.003     0.639  False   \n",
       "334534             NaN         NaN          False      0.003     0.266  False   \n",
       "334535             NaN         NaN          False      0.004     0.318  False   \n",
       "\n",
       "        sample_weight  \n",
       "55                1.0  \n",
       "56                1.0  \n",
       "57                1.0  \n",
       "58                1.0  \n",
       "59                1.0  \n",
       "...               ...  \n",
       "334504            0.8  \n",
       "334505            0.8  \n",
       "334533            0.8  \n",
       "334534            0.8  \n",
       "334535            0.8  \n",
       "\n",
       "[26776 rows x 152 columns]"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ek_subset = train.loc[train['GeneName'].isin(ek_genes)]\n",
    "ek_subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Benign        16869\n",
       "Pathogenic     9907\n",
       "Name: label, dtype: int64"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ek_subset['label'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "ek_subset.to_csv('./datafiles/erfelijke_kanker_subset.tsv.gz', sep='\\t', index=False, compression='gzip')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
