{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Index\n",
    "- [Imports](#Imports)\n",
    "- [CTEPG](#CTEPG-(Catching-The-Elusive-Predictable-Genes)-15-06-2020---22-06-2020)\n",
    "- [Dyslipid database creation](#Creating-dyslipid-dataset)\n",
    "- [New vanilla model validation](#AUC-validation-vanilla-new-model)\n",
    "- [Cross validated models](#CV-Models)\n",
    "- [XGB models out of CV models](#New-Models)\n",
    "- [Changes in predition new model](#AUC-validation-vanilla-new-model)\n",
    "- [Correct new threshold](#Correct-probability-prediction)\n",
    "- [AUC preparation](#Preparing-for-AUC-analysis)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Importing some shizzle.\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from matplotlib import pyplot as plt\n",
    "from scipy import stats\n",
    "import time\n",
    "import requests\n",
    "import datetime\n",
    "import sys\n",
    "import os\n",
    "import pathlib\n",
    "import seaborn as sns\n",
    "from sklearn.metrics import roc_auc_score\n",
    "import math\n",
    "import json\n",
    "import glob\n",
    "import gzip\n",
    "import pickle\n",
    "import xgboost as xgb\n",
    "from utilities import perform_stats, calc_z_scores, get_header\n",
    "from sklearn.metrics import recall_score \n",
    "\n",
    "# Defining some import and export locations\n",
    "location = 'rjsietsma'\n",
    "read_loc = '/run/media/rjsietsma/evo2tb/linux/Datafiles/'\n",
    "data_expor_loc = '/home/'+location+'/Documents/School/Master_DSLS/Final_Thesis/Past_initial_data/'\n",
    "img_output_dir = '/home/'+location+'/PycharmProjects/dsls_master_thesis/side_scripts/output_img/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " # CTEPG (Catching The Elusive Predictable Genes) 15-06-2020 - 22-06-2020\n",
    "<br/><br/>\n",
    "\n",
    "What have I done:\n",
    "\n",
    "\n",
    "Planned for this week:\n",
    "- Make model on bad performing genes and investigate further.\n",
    "    - At first, make models of [the original training data](https://zenodo.org/record/3564757#.XudiZ2ixUUE), models of various gene panels (but keep the variants of those panels within the bigger model). See if they improve and how much.\n",
    "        - _Note: the training data needs to be converted to [variant calling file (VCF)](http://samtools.github.io/hts-specs/VCFv4.1.pdf) before using, which means it needs a fileformat. Or the annotated file can be downloaded directly from Shuang her GCC tmp storage._\n",
    "        - ___Note 2: Shuang her data has almost half the variants than the newer VGKL and Clinvar datasets, with 3913 benign and 1116 malignant samples.___\n",
    "    - Then make a single model of the original training data for CADD 1.6 annotations.\n",
    "    - Then make multiple models of the original training data for CADD 1.6 annotations.\n",
    "    - Investigate improvements.\n",
    "    - Then choose between:\n",
    "        - newer / larger dataset or \n",
    "        - Custom features that might be relevant to the cause (prefered)\n",
    "- Make single plot of panel vs all others combined (except for 5GPM).\n",
    "- Usefull, [convert 37 to 38](https://genome.ucsc.edu/cgi-bin/hgLiftOver).\n",
    "- Start investigating what is required for the data sources to be used in the training phase.\n",
    "- Create outline for article.\n",
    "     - For static plots: make labels on each point with the term name and the number of genes in (n=x) behind that.\n",
    "     - For first concept, it's due date is the last week of the resit exams on the 6th of July.\n",
    "- _Optionally_ :\n",
    "    - Find possible way to annotate genes for co-factors, sub-family etc.\n",
    "    - Interactive plot of all variants labeled as benign or pathogenic with their capice score (y-axis) and location (x-axis), to show the variant itself.\n",
    "<br/><br/>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Creating dyslipid dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Allergy/Immunology/Infectious</th>\n",
       "      <th>Alt</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Audiologic/Otolaryngologic</th>\n",
       "      <th>Biochemical</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>CDSpos</th>\n",
       "      <th>Cardiovascular</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>revel</th>\n",
       "      <th>sift</th>\n",
       "      <th>source</th>\n",
       "      <th>tOverlapMotifs</th>\n",
       "      <th>targetScan</th>\n",
       "      <th>to_be_deleted</th>\n",
       "      <th>verPhCons</th>\n",
       "      <th>verPhyloP</th>\n",
       "      <th>inTest</th>\n",
       "      <th>sample_weight</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14</td>\n",
       "      <td>False</td>\n",
       "      <td>G</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS9787.1</td>\n",
       "      <td>806.0</td>\n",
       "      <td>False</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.843</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS13112.1</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift,stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>4.670</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>C</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS13112.1</td>\n",
       "      <td>2118.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.043</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS13112.1</td>\n",
       "      <td>1586.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>6.221</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>6.368</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334596</th>\n",
       "      <td>17</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS32642.1</td>\n",
       "      <td>1563.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>6.031</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334597</th>\n",
       "      <td>17</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS32642.1</td>\n",
       "      <td>2029.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>4.100</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334598</th>\n",
       "      <td>10</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS7431.1</td>\n",
       "      <td>1216.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.852</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334599</th>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS2382.1</td>\n",
       "      <td>2998.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.031</td>\n",
       "      <td>2.213</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334600</th>\n",
       "      <td>5</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS3952.1</td>\n",
       "      <td>1221.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.528</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>334601 rows × 152 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Chrom  Allergy/Immunology/Infectious Alt          AnnoType  \\\n",
       "0          14                          False   G  CodingTranscript   \n",
       "1          20                          False   T  CodingTranscript   \n",
       "2          20                          False   C  CodingTranscript   \n",
       "3          20                          False   A  CodingTranscript   \n",
       "4          20                          False   A        Intergenic   \n",
       "...       ...                            ...  ..               ...   \n",
       "334596     17                          False   A  CodingTranscript   \n",
       "334597     17                          False   T  CodingTranscript   \n",
       "334598     10                          False   T  CodingTranscript   \n",
       "334599      2                          False   T  CodingTranscript   \n",
       "334600      5                          False   T  CodingTranscript   \n",
       "\n",
       "        Audiologic/Otolaryngologic  Biochemical         CCDS  CDSpos  \\\n",
       "0                            False        False   CCDS9787.1   806.0   \n",
       "1                             True        False  CCDS13112.1  1899.0   \n",
       "2                             True        False  CCDS13112.1  2118.0   \n",
       "3                             True        False  CCDS13112.1  1586.0   \n",
       "4                             True        False          NaN     NaN   \n",
       "...                            ...          ...          ...     ...   \n",
       "334596                       False        False  CCDS32642.1  1563.0   \n",
       "334597                       False        False  CCDS32642.1  2029.0   \n",
       "334598                       False        False   CCDS7431.1  1216.0   \n",
       "334599                       False        False   CCDS2382.1  2998.0   \n",
       "334600                       False        False   CCDS3952.1  1221.0   \n",
       "\n",
       "        Cardiovascular              ConsDetail  ...  revel sift   source  \\\n",
       "0                False              frameshift  ...    NaN  NaN     vkgl   \n",
       "1                 True  frameshift,stop_gained  ...    NaN  NaN     vkgl   \n",
       "2                 True              frameshift  ...    NaN  NaN     vkgl   \n",
       "3                 True              frameshift  ...    NaN  NaN     vkgl   \n",
       "4                 True              downstream  ...    NaN  NaN     vkgl   \n",
       "...                ...                     ...  ...    ...  ...      ...   \n",
       "334596           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334597           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334598           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334599           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334600           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "\n",
       "        tOverlapMotifs  targetScan  to_be_deleted  verPhCons verPhyloP inTest  \\\n",
       "0                  NaN         NaN          False      1.000     5.843  False   \n",
       "1                  NaN         NaN          False      1.000     4.670  False   \n",
       "2                  NaN         NaN          False      1.000     5.043  False   \n",
       "3                  NaN         NaN          False      1.000     6.221  False   \n",
       "4                  NaN         NaN          False      1.000     6.368  False   \n",
       "...                ...         ...            ...        ...       ...    ...   \n",
       "334596             NaN         NaN          False      1.000     6.031  False   \n",
       "334597             NaN         NaN          False      1.000     4.100  False   \n",
       "334598             NaN         NaN          False      1.000     5.852  False   \n",
       "334599             NaN         NaN          False      0.031     2.213  False   \n",
       "334600             NaN         NaN          False      1.000     0.528  False   \n",
       "\n",
       "        sample_weight  \n",
       "0                 1.0  \n",
       "1                 1.0  \n",
       "2                 1.0  \n",
       "3                 1.0  \n",
       "4                 1.0  \n",
       "...               ...  \n",
       "334596            0.8  \n",
       "334597            0.8  \n",
       "334598            0.8  \n",
       "334599            0.8  \n",
       "334600            0.8  \n",
       "\n",
       "[334601 rows x 152 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_loc = os.path.join(read_loc, 'train.txt.gz')\n",
    "header = get_header(file_loc, '#Chrom')\n",
    "train = pd.read_csv(file_loc, compression='gzip', names=header, comment='#', sep='\\t', low_memory=False)\n",
    "train"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('./umcg_genepanels.json', 'r') as json_file:\n",
    "    genes = json.load(json_file)\n",
    "dislipid_genes = genes['Hart- en vaatziekten']\n",
    "genelist = []\n",
    "for key, value in dislipid_genes.items():\n",
    "    if key.lower().startswith('dyslipid'):\n",
    "        for g in value:\n",
    "            if g not in genelist:\n",
    "                genelist.append(g)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Allergy/Immunology/Infectious</th>\n",
       "      <th>Alt</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Audiologic/Otolaryngologic</th>\n",
       "      <th>Biochemical</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>CDSpos</th>\n",
       "      <th>Cardiovascular</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>revel</th>\n",
       "      <th>sift</th>\n",
       "      <th>source</th>\n",
       "      <th>tOverlapMotifs</th>\n",
       "      <th>targetScan</th>\n",
       "      <th>to_be_deleted</th>\n",
       "      <th>verPhCons</th>\n",
       "      <th>verPhyloP</th>\n",
       "      <th>inTest</th>\n",
       "      <th>sample_weight</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>16</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS10772.1</td>\n",
       "      <td>848.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.928</td>\n",
       "      <td>2.614</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>19</td>\n",
       "      <td>False</td>\n",
       "      <td>CCGGCGAGGTGCAGGCCATGCT</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS12647.1</td>\n",
       "      <td>409.0</td>\n",
       "      <td>True</td>\n",
       "      <td>protein_altering</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>0.863</td>\n",
       "      <td>0.839</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>C</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS1703.1</td>\n",
       "      <td>13028.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.058</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>G</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS1703.1</td>\n",
       "      <td>28.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.021</td>\n",
       "      <td>-0.103</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>C</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS1703.1</td>\n",
       "      <td>2534.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.653</td>\n",
       "      <td>0.251</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334284</th>\n",
       "      <td>12</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS8685.1</td>\n",
       "      <td>1093.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.999</td>\n",
       "      <td>3.072</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334285</th>\n",
       "      <td>12</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS8685.1</td>\n",
       "      <td>1475.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.580</td>\n",
       "      <td>0.563</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334286</th>\n",
       "      <td>12</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS8685.1</td>\n",
       "      <td>1537.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.643</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334287</th>\n",
       "      <td>12</td>\n",
       "      <td>False</td>\n",
       "      <td>G</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS8685.1</td>\n",
       "      <td>1553.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.510</td>\n",
       "      <td>2.765</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334288</th>\n",
       "      <td>12</td>\n",
       "      <td>False</td>\n",
       "      <td>G</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS8685.1</td>\n",
       "      <td>1634.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.326</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5029 rows × 152 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Chrom  Allergy/Immunology/Infectious                     Alt  \\\n",
       "142        16                          False                       T   \n",
       "148        19                          False  CCGGCGAGGTGCAGGCCATGCT   \n",
       "149         2                          False                       C   \n",
       "150         2                          False                       G   \n",
       "151         2                          False                       C   \n",
       "...       ...                            ...                     ...   \n",
       "334284     12                          False                       T   \n",
       "334285     12                          False                       A   \n",
       "334286     12                          False                       T   \n",
       "334287     12                          False                       G   \n",
       "334288     12                          False                       G   \n",
       "\n",
       "                AnnoType  Audiologic/Otolaryngologic  Biochemical  \\\n",
       "142     CodingTranscript                       False        False   \n",
       "148     CodingTranscript                       False        False   \n",
       "149     CodingTranscript                       False        False   \n",
       "150     CodingTranscript                       False        False   \n",
       "151     CodingTranscript                       False        False   \n",
       "...                  ...                         ...          ...   \n",
       "334284  CodingTranscript                       False        False   \n",
       "334285  CodingTranscript                       False        False   \n",
       "334286  CodingTranscript                       False        False   \n",
       "334287  CodingTranscript                       False        False   \n",
       "334288  CodingTranscript                       False        False   \n",
       "\n",
       "               CCDS   CDSpos  Cardiovascular        ConsDetail  ...  revel  \\\n",
       "142     CCDS10772.1    848.0            True        frameshift  ...    NaN   \n",
       "148     CCDS12647.1    409.0            True  protein_altering  ...    NaN   \n",
       "149      CCDS1703.1  13028.0            True        frameshift  ...    NaN   \n",
       "150      CCDS1703.1     28.0            True        frameshift  ...    NaN   \n",
       "151      CCDS1703.1   2534.0            True        frameshift  ...    NaN   \n",
       "...             ...      ...             ...               ...  ...    ...   \n",
       "334284   CCDS8685.1   1093.0           False       stop_gained  ...    NaN   \n",
       "334285   CCDS8685.1   1475.0           False       stop_gained  ...    NaN   \n",
       "334286   CCDS8685.1   1537.0           False       stop_gained  ...    NaN   \n",
       "334287   CCDS8685.1   1553.0           False       stop_gained  ...    NaN   \n",
       "334288   CCDS8685.1   1634.0           False       stop_gained  ...    NaN   \n",
       "\n",
       "       sift   source  tOverlapMotifs  targetScan  to_be_deleted  verPhCons  \\\n",
       "142     NaN     vkgl             NaN         NaN          False      0.928   \n",
       "148     NaN     vkgl             NaN         NaN           True      0.863   \n",
       "149     NaN     vkgl             NaN         NaN          False      0.000   \n",
       "150     NaN     vkgl             NaN         NaN          False      0.021   \n",
       "151     NaN     vkgl             NaN         NaN          False      0.653   \n",
       "...     ...      ...             ...         ...            ...        ...   \n",
       "334284  NaN  unknown             NaN         NaN          False      0.999   \n",
       "334285  NaN  unknown             NaN         NaN          False      0.580   \n",
       "334286  NaN  unknown             NaN         NaN          False      0.003   \n",
       "334287  NaN  unknown             NaN         NaN          False      0.510   \n",
       "334288  NaN  unknown             NaN         NaN          False      0.000   \n",
       "\n",
       "       verPhyloP inTest  sample_weight  \n",
       "142        2.614  False            1.0  \n",
       "148        0.839  False            1.0  \n",
       "149        0.058  False            1.0  \n",
       "150       -0.103  False            1.0  \n",
       "151        0.251  False            1.0  \n",
       "...          ...    ...            ...  \n",
       "334284     3.072  False            0.8  \n",
       "334285     0.563  False            0.8  \n",
       "334286     0.643  False            0.8  \n",
       "334287     2.765  False            0.8  \n",
       "334288    -0.326  False            0.8  \n",
       "\n",
       "[5029 rows x 152 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dislipid_subset = train.loc[train['GeneName'].isin(genelist)]\n",
    "dislipid_subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Benign        3913\n",
       "Pathogenic    1116\n",
       "Name: label, dtype: int64"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dislipid_subset['label'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CV Models"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'objective': 'binary:logistic',\n",
       " 'base_score': 0.5,\n",
       " 'booster': 'gbtree',\n",
       " 'colsample_bylevel': 1,\n",
       " 'colsample_bynode': 1,\n",
       " 'colsample_bytree': 1,\n",
       " 'gamma': 0,\n",
       " 'gpu_id': -1,\n",
       " 'importance_type': 'gain',\n",
       " 'interaction_constraints': '',\n",
       " 'learning_rate': 0.12794736796290956,\n",
       " 'max_delta_step': 0,\n",
       " 'max_depth': 15,\n",
       " 'min_child_weight': 1,\n",
       " 'monotone_constraints': '()',\n",
       " 'n_estimators': 420,\n",
       " 'n_jobs': 8,\n",
       " 'num_parallel_tree': 1,\n",
       " 'random_state': 0,\n",
       " 'reg_alpha': 0,\n",
       " 'reg_lambda': 1,\n",
       " 'scale_pos_weight': 1,\n",
       " 'subsample': 1,\n",
       " 'tree_method': 'exact',\n",
       " 'validate_parameters': 1,\n",
       " 'verbosity': 1}"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model = pickle.load(open('./models/xgb_weightedSample_randomsearch_v2.pickle.dat', 'rb'))\n",
    "xgbmodel = model.best_estimator_\n",
    "optimal_params = xgbmodel.get_params()\n",
    "optimal_params.pop('missing')\n",
    "with open('./optimal_capice_v2_params.json', 'w+') as json_file:\n",
    "    json.dump(optimal_params, json_file, indent=4)\n",
    "optimal_params"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "model = xgb.XGBClassifier(optimal_params)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'objective': 'binary:logistic',\n",
       " 'base_score': 0.5,\n",
       " 'booster': 'gbtree',\n",
       " 'colsample_bylevel': 1,\n",
       " 'colsample_bynode': 1,\n",
       " 'colsample_bytree': 1,\n",
       " 'gamma': 0,\n",
       " 'gpu_id': -1,\n",
       " 'importance_type': 'gain',\n",
       " 'interaction_constraints': '',\n",
       " 'learning_rate': 0.1968770013672272,\n",
       " 'max_delta_step': 0,\n",
       " 'max_depth': 16,\n",
       " 'min_child_weight': 1,\n",
       " 'monotone_constraints': '()',\n",
       " 'n_estimators': 427,\n",
       " 'n_jobs': 8,\n",
       " 'num_parallel_tree': 1,\n",
       " 'random_state': 0,\n",
       " 'reg_alpha': 0,\n",
       " 'reg_lambda': 1,\n",
       " 'scale_pos_weight': 1,\n",
       " 'subsample': 1,\n",
       " 'tree_method': 'exact',\n",
       " 'validate_parameters': 1,\n",
       " 'verbosity': 1}"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model_dislipid = pickle.load(open('./models/xgb_weightedSample_randomsearch_dislipid.pickle.dat', 'rb'))\n",
    "xgbmodel_dislipid = model_dislipid.best_estimator_\n",
    "optimal_params_dislipid = xgbmodel_dislipid.get_params()\n",
    "optimal_params_dislipid.pop('missing')\n",
    "with open('./optimal_capice_dislipid_params.json', 'w+') as json_file:\n",
    "    json.dump(optimal_params_dislipid, json_file, indent=4)\n",
    "optimal_params_dislipid"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# New Models"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,\n",
       "              colsample_bynode=1, colsample_bytree=1, gamma=0, gpu_id=-1,\n",
       "              importance_type='gain', interaction_constraints='',\n",
       "              learning_rate=0.12794736796290956, max_delta_step=0, max_depth=15,\n",
       "              min_child_weight=1, missing=nan, monotone_constraints='()',\n",
       "              n_estimators=420, n_jobs=8, num_parallel_tree=1, random_state=0,\n",
       "              reg_alpha=0, reg_lambda=1, scale_pos_weight=1, subsample=1,\n",
       "              tree_method='exact', validate_parameters=1, verbosity=1)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model = pickle.load(open('./xgbmodels/xgb_booster_v2.pickle.dat', 'rb'))\n",
    "model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,\n",
       "              colsample_bynode=1, colsample_bytree=1, gamma=0, gpu_id=-1,\n",
       "              importance_type='gain', interaction_constraints='',\n",
       "              learning_rate=0.1968770013672272, max_delta_step=0, max_depth=16,\n",
       "              min_child_weight=1, missing=nan, monotone_constraints='()',\n",
       "              n_estimators=427, n_jobs=8, num_parallel_tree=1, random_state=0,\n",
       "              reg_alpha=0, reg_lambda=1, scale_pos_weight=1, subsample=1,\n",
       "              tree_method='exact', validate_parameters=1, verbosity=1)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model_dislipid = pickle.load(open('./xgbmodels/xgb_booster_dyslipid.pickle.dat', 'rb'))\n",
    "model_dislipid"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# AUC validation vanilla new model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>probabilities</th>\n",
       "      <th>prediction</th>\n",
       "      <th>combined_prediction</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17</td>\n",
       "      <td>41246652</td>\n",
       "      <td>ACATTC</td>\n",
       "      <td>GA</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>26.600</td>\n",
       "      <td>9.999933e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>19</td>\n",
       "      <td>11216246</td>\n",
       "      <td>TGCAAGGACAAATCTGAC</td>\n",
       "      <td>CCGACTG</td>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999907e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>19</td>\n",
       "      <td>11216251</td>\n",
       "      <td>GGACAAATCTGACGA</td>\n",
       "      <td>AACTGCGGTAAACTGCGGTAAACT</td>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999896e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>19</td>\n",
       "      <td>11216262</td>\n",
       "      <td>ACG</td>\n",
       "      <td>CA</td>\n",
       "      <td>LDLR</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>34.000</td>\n",
       "      <td>9.999894e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>47702328</td>\n",
       "      <td>GTTGA</td>\n",
       "      <td>TTTC</td>\n",
       "      <td>MSH2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>9.999891e-01</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334596</th>\n",
       "      <td>14</td>\n",
       "      <td>64653189</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>MIR548AZ</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>14.370</td>\n",
       "      <td>2.765540e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334597</th>\n",
       "      <td>17</td>\n",
       "      <td>10419945</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>0.611</td>\n",
       "      <td>2.524258e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334598</th>\n",
       "      <td>17</td>\n",
       "      <td>10419849</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>9.827</td>\n",
       "      <td>2.164549e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334599</th>\n",
       "      <td>16</td>\n",
       "      <td>88804658</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>LOC100289580</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>14.100</td>\n",
       "      <td>1.984229e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334600</th>\n",
       "      <td>17</td>\n",
       "      <td>10419849</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>MYHAS</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>10.040</td>\n",
       "      <td>1.227600e-07</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>334601 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr       pos                 ref                       alt  \\\n",
       "0       17  41246652              ACATTC                        GA   \n",
       "1       19  11216246  TGCAAGGACAAATCTGAC                   CCGACTG   \n",
       "2       19  11216251     GGACAAATCTGACGA  AACTGCGGTAAACTGCGGTAAACT   \n",
       "3       19  11216262                 ACG                        CA   \n",
       "4        2  47702328               GTTGA                      TTTC   \n",
       "...     ..       ...                 ...                       ...   \n",
       "334596  14  64653189                   T                         C   \n",
       "334597  17  10419945                   A                         G   \n",
       "334598  17  10419849                   T                         G   \n",
       "334599  16  88804658                   G                         A   \n",
       "334600  17  10419849                   T                         C   \n",
       "\n",
       "            GeneName  Consequence   PHRED  probabilities  prediction  \\\n",
       "0              BRCA1  FRAME_SHIFT  26.600   9.999933e-01  Pathogenic   \n",
       "1               LDLR  FRAME_SHIFT  35.000   9.999907e-01  Pathogenic   \n",
       "2               LDLR  FRAME_SHIFT  34.000   9.999896e-01  Pathogenic   \n",
       "3               LDLR  FRAME_SHIFT  34.000   9.999894e-01  Pathogenic   \n",
       "4               MSH2  FRAME_SHIFT  35.000   9.999891e-01  Pathogenic   \n",
       "...              ...          ...     ...            ...         ...   \n",
       "334596      MIR548AZ   SYNONYMOUS  14.370   2.765540e-07     Neutral   \n",
       "334597         MYHAS   SYNONYMOUS   0.611   2.524258e-07     Neutral   \n",
       "334598         MYHAS   SYNONYMOUS   9.827   2.164549e-07     Neutral   \n",
       "334599  LOC100289580   SYNONYMOUS  14.100   1.984229e-07     Neutral   \n",
       "334600         MYHAS   SYNONYMOUS  10.040   1.227600e-07     Neutral   \n",
       "\n",
       "       combined_prediction  \n",
       "0               Pathogenic  \n",
       "1               Pathogenic  \n",
       "2               Pathogenic  \n",
       "3               Pathogenic  \n",
       "4               Pathogenic  \n",
       "...                    ...  \n",
       "334596             Neutral  \n",
       "334597             Neutral  \n",
       "334598             Neutral  \n",
       "334599             Neutral  \n",
       "334600             Neutral  \n",
       "\n",
       "[334601 rows x 10 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_original = pd.read_csv('/home/rjsietsma/Documents/School/Master_DSLS/Final_Thesis/Initial_Data_exploration/train_results.txt', sep='\\t', low_memory=False)\n",
    "train_original"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>probabilities</th>\n",
       "      <th>prediction</th>\n",
       "      <th>combined_prediction</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RDH12</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>14</td>\n",
       "      <td>68196054</td>\n",
       "      <td>GCCCTG</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ANK1</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>32.000</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>8</td>\n",
       "      <td>41554106</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>35.000</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>12</td>\n",
       "      <td>121431481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>39.000</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>12</td>\n",
       "      <td>121435345</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>INFRAME</td>\n",
       "      <td>19.860</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>12</td>\n",
       "      <td>121431471</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334596</th>\n",
       "      <td>SPG11</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>15.230</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>15</td>\n",
       "      <td>44952677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334597</th>\n",
       "      <td>SPG11</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>19.320</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>15</td>\n",
       "      <td>44952632</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334598</th>\n",
       "      <td>SPG11</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>9.169</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>15</td>\n",
       "      <td>44943927</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334599</th>\n",
       "      <td>SPG11</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>13.460</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>15</td>\n",
       "      <td>44943897</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334600</th>\n",
       "      <td>HCN1</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>36.000</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>5</td>\n",
       "      <td>45396603</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>334601 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       GeneName       Consequence   PHRED  probabilities  prediction  \\\n",
       "0         RDH12       FRAME_SHIFT  35.000              1  Pathogenic   \n",
       "1          ANK1  CANONICAL_SPLICE  32.000              1  Pathogenic   \n",
       "2         HNF1A       STOP_GAINED  35.000              1  Pathogenic   \n",
       "3         HNF1A       STOP_GAINED  39.000              1  Pathogenic   \n",
       "4         HNF1A           INFRAME  19.860              1  Pathogenic   \n",
       "...         ...               ...     ...            ...         ...   \n",
       "334596    SPG11    NON_SYNONYMOUS  15.230              0     Neutral   \n",
       "334597    SPG11    NON_SYNONYMOUS  19.320              0     Neutral   \n",
       "334598    SPG11        SYNONYMOUS   9.169              0     Neutral   \n",
       "334599    SPG11    NON_SYNONYMOUS  13.460              0     Neutral   \n",
       "334600     HCN1       STOP_GAINED  36.000              0     Neutral   \n",
       "\n",
       "       combined_prediction chr        pos     ref alt  \n",
       "0               Pathogenic  14   68196054  GCCCTG   G  \n",
       "1               Pathogenic   8   41554106       C   T  \n",
       "2               Pathogenic  12  121431481       C   T  \n",
       "3               Pathogenic  12  121435345       C   T  \n",
       "4               Pathogenic  12  121431471    CAAG   C  \n",
       "...                    ...  ..        ...     ...  ..  \n",
       "334596             Neutral  15   44952677       C   T  \n",
       "334597             Neutral  15   44952632       A   G  \n",
       "334598             Neutral  15   44943927       G   C  \n",
       "334599             Neutral  15   44943897       C   A  \n",
       "334600          Pathogenic   5   45396603       A   T  \n",
       "\n",
       "[334601 rows x 10 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_new = pd.read_csv('/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/test_output/train_results_v2.txt',\n",
    "                       sep='\\t', low_memory=False)\n",
    "important_info = train_new['chr_pos_ref_alt'].str.split(\"_\", expand=True)\n",
    "train_new['chr'] = important_info[0]\n",
    "train_new['pos'] = important_info[1].astype(np.int64)\n",
    "train_new['ref'] = important_info[2]\n",
    "train_new['alt'] = important_info[3]\n",
    "train_new.drop(columns=['chr_pos_ref_alt'], inplace=True)\n",
    "train_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "chr                     object\n",
       "pos                      int64\n",
       "ref                     object\n",
       "alt                     object\n",
       "GeneName                object\n",
       "Consequence             object\n",
       "PHRED                  float64\n",
       "probabilities          float64\n",
       "prediction              object\n",
       "combined_prediction     object\n",
       "dtype: object"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_original.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GeneName                object\n",
       "Consequence             object\n",
       "PHRED                  float64\n",
       "probabilities            int64\n",
       "prediction              object\n",
       "combined_prediction     object\n",
       "chr                     object\n",
       "pos                      int64\n",
       "ref                     object\n",
       "alt                     object\n",
       "dtype: object"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_new.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There is a 5.093230444619114% mismatch.\n"
     ]
    }
   ],
   "source": [
    "merge = train_original[['chr', 'pos','ref','alt','prediction']].merge(train_new[['chr', 'pos','ref','alt','prediction']],\n",
    "                                                                     on=['chr', 'pos','ref','alt'])\n",
    "merge[merge['prediction_x'] != merge['prediction_y']]\n",
    "print(f\"There is a \"\n",
    "      f\"{merge[merge['prediction_x'] != merge['prediction_y']].shape[0] / train_original.shape[0] * 100}% mismatch.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>max_AF</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>label</th>\n",
       "      <th>revel</th>\n",
       "      <th>clinpred</th>\n",
       "      <th>sift</th>\n",
       "      <th>provean</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>fathmm_score</th>\n",
       "      <th>capice</th>\n",
       "      <th>ponp2</th>\n",
       "      <th>prediction</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>21</td>\n",
       "      <td>33974174</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000058</td>\n",
       "      <td>STOP_LOST</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.050</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003275</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>X</td>\n",
       "      <td>99661625</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.806</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000075</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17</td>\n",
       "      <td>29509638</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.647</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>10.760</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001511</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>21</td>\n",
       "      <td>35742999</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000133</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>18.640</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000621</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>2160973</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>16.340</td>\n",
       "      <td>0.008252</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10837</th>\n",
       "      <td>17</td>\n",
       "      <td>29556342</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>11.450</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000139</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10838</th>\n",
       "      <td>11</td>\n",
       "      <td>5248177</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000344</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>17.970</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.069934</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10839</th>\n",
       "      <td>15</td>\n",
       "      <td>48787324</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.664</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>12.170</td>\n",
       "      <td>0.019530</td>\n",
       "      <td>0.909190</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10840</th>\n",
       "      <td>19</td>\n",
       "      <td>17947957</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>10.990</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001238</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Neutral</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10841</th>\n",
       "      <td>16</td>\n",
       "      <td>223597</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001252</td>\n",
       "      <td>STOP_LOST</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18.020</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.971616</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10842 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr       pos ref alt    max_AF Consequence label  revel  clinpred  \\\n",
       "0      21  33974174   C   G  0.000058   STOP_LOST  LB/B    NaN       NaN   \n",
       "1       X  99661625   G   C  0.000037  SYNONYMOUS  LB/B    NaN       NaN   \n",
       "2      17  29509638   C   T  0.000000  SYNONYMOUS  LB/B    NaN       NaN   \n",
       "3      21  35742999   C   T  0.000133  SYNONYMOUS  LB/B    NaN       NaN   \n",
       "4       1   2160973   G   A  0.000000  SYNONYMOUS  LB/B    NaN       NaN   \n",
       "...    ..       ...  ..  ..       ...         ...   ...    ...       ...   \n",
       "10837  17  29556342   G   A  0.000000  SYNONYMOUS  LP/P    NaN       NaN   \n",
       "10838  11   5248177   A   T  0.000344  SYNONYMOUS  LP/P    NaN       NaN   \n",
       "10839  15  48787324   T   C  0.000000  SYNONYMOUS  LP/P    NaN       NaN   \n",
       "10840  19  17947957   G   A  0.000015  SYNONYMOUS  LP/P    NaN       NaN   \n",
       "10841  16    223597   T   C  0.001252   STOP_LOST  LP/P    NaN       NaN   \n",
       "\n",
       "        sift  provean   PHRED  fathmm_score    capice  ponp2  prediction  \n",
       "0        NaN      NaN  15.050           NaN  0.003275    NaN     Neutral  \n",
       "1      0.000     -0.0   0.806           NaN  0.000075    NaN     Neutral  \n",
       "2      0.647     -0.0  10.760           NaN  0.001511    NaN     Neutral  \n",
       "3      0.000     -0.0  18.640           NaN  0.000621    NaN     Neutral  \n",
       "4      0.000     -0.0  16.340      0.008252  0.000012    NaN     Neutral  \n",
       "...      ...      ...     ...           ...       ...    ...         ...  \n",
       "10837  0.000     -0.0  11.450           NaN  0.000139    NaN     Neutral  \n",
       "10838  0.000     -0.0  17.970           NaN  0.069934    NaN  Pathogenic  \n",
       "10839  0.664     -0.0  12.170      0.019530  0.909190    NaN  Pathogenic  \n",
       "10840  0.000     -0.0  10.990           NaN  0.001238    NaN     Neutral  \n",
       "10841    NaN      NaN  18.020           NaN  0.971616    NaN  Pathogenic  \n",
       "\n",
       "[10842 rows x 16 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_original = pd.read_csv('/home/rjsietsma/Documents/School/Master_DSLS/Final_Thesis/Initial_Data_exploration/test_results.txt', sep='\\t', low_memory=False)\n",
    "tellPathogenic_pred = lambda x: \"Pathogenic\" if x > 0.02 else 'Neutral'\n",
    "test_original['prediction'] = [tellPathogenic_pred(probability) for probability in test_original['capice']]\n",
    "test_original.rename(columns={'#Chrom': 'chr', 'Pos':'pos', 'Ref': 'ref', 'Alt':'alt'}, inplace=True)\n",
    "test_original"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>probabilities</th>\n",
       "      <th>prediction</th>\n",
       "      <th>combined_prediction</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MSH2</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>24.30</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>2</td>\n",
       "      <td>47672768</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PHIP</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>34.00</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>6</td>\n",
       "      <td>79692624</td>\n",
       "      <td>CTTCT</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>24.20</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>17</td>\n",
       "      <td>41244488</td>\n",
       "      <td>TGGAA</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CHEK2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>26.40</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>22</td>\n",
       "      <td>29121049</td>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>32.00</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>17</td>\n",
       "      <td>41243468</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10837</th>\n",
       "      <td>CDCP2</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>23.00</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>1</td>\n",
       "      <td>54605318</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10838</th>\n",
       "      <td>ASXL1</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>15.46</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>20</td>\n",
       "      <td>31022934</td>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10839</th>\n",
       "      <td>KIAA0196</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.00</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>8</td>\n",
       "      <td>126051130</td>\n",
       "      <td>AGT</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10840</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>28.10</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>17</td>\n",
       "      <td>41203088</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10841</th>\n",
       "      <td>C21orf59</td>\n",
       "      <td>STOP_LOST</td>\n",
       "      <td>15.05</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>21</td>\n",
       "      <td>33974174</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10842 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       GeneName     Consequence  PHRED  probabilities  prediction  \\\n",
       "0          MSH2  NON_SYNONYMOUS  24.30              1  Pathogenic   \n",
       "1          PHIP     FRAME_SHIFT  34.00              1  Pathogenic   \n",
       "2         BRCA1     FRAME_SHIFT  24.20              1  Pathogenic   \n",
       "3         CHEK2     FRAME_SHIFT  26.40              1  Pathogenic   \n",
       "4         BRCA1     FRAME_SHIFT  32.00              1  Pathogenic   \n",
       "...         ...             ...    ...            ...         ...   \n",
       "10837     CDCP2     FRAME_SHIFT  23.00              0     Neutral   \n",
       "10838     ASXL1     FRAME_SHIFT  15.46              0     Neutral   \n",
       "10839  KIAA0196     FRAME_SHIFT  35.00              0     Neutral   \n",
       "10840     BRCA1  NON_SYNONYMOUS  28.10              0     Neutral   \n",
       "10841  C21orf59       STOP_LOST  15.05              0     Neutral   \n",
       "\n",
       "      combined_prediction chr        pos    ref alt  \n",
       "0              Pathogenic   2   47672768      T   A  \n",
       "1              Pathogenic   6   79692624  CTTCT   C  \n",
       "2              Pathogenic  17   41244488  TGGAA   T  \n",
       "3              Pathogenic  22   29121049     CA   C  \n",
       "4              Pathogenic  17   41243468     GC   G  \n",
       "...                   ...  ..        ...    ...  ..  \n",
       "10837             Neutral   1   54605318      T  TC  \n",
       "10838             Neutral  20   31022934     GT   G  \n",
       "10839          Pathogenic   8  126051130    AGT   A  \n",
       "10840             Neutral  17   41203088      A   C  \n",
       "10841             Neutral  21   33974174      C   G  \n",
       "\n",
       "[10842 rows x 10 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_new = pd.read_csv('/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/test_output/test_results_v2.txt', sep='\\t', low_memory=False)\n",
    "important_info = test_new['chr_pos_ref_alt'].str.split(\"_\", expand=True)\n",
    "test_new['chr'] = important_info[0]\n",
    "test_new['pos'] = important_info[1].astype(np.int64)\n",
    "test_new['ref'] = important_info[2]\n",
    "test_new['alt'] = important_info[3]\n",
    "test_new.drop(columns=['chr_pos_ref_alt'], inplace=True)\n",
    "test_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There is a 27.23667220070098% mismatch.\n"
     ]
    }
   ],
   "source": [
    "merge = test_original[['chr', 'pos','ref','alt','prediction']].merge(test_new[['chr', 'pos','ref','alt','prediction']],\n",
    "                                                                     on=['chr', 'pos','ref','alt'])\n",
    "merge[merge['prediction_x'] != merge['prediction_y']]\n",
    "print(f\"There is a \"\n",
    "      f\"{merge[merge['prediction_x'] != merge['prediction_y']].shape[0] / test_original.shape[0] * 100}% mismatch.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Correct probability prediction"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Currently marked as markdown to prevent executing again when re-launching the notebook"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "thresholds = np.arange(0,1,0.001)\n",
    "data = pd.read_csv('/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/test_output/train_results_v3.txt',\n",
    "                  sep='\\t', low_memory=False)\n",
    "important_info = data['chr_pos_ref_alt'].str.split(\"_\", expand=True)\n",
    "data['chr'] = important_info[0]\n",
    "data['pos'] = important_info[1].astype(np.int64)\n",
    "data['ref'] = important_info[2]\n",
    "data['alt'] = important_info[3]\n",
    "data.drop(columns=['chr_pos_ref_alt'], inplace=True)\n",
    "data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "train_in = pd.read_csv('/run/media/rjsietsma/evo2tb/linux/Datafiles/train.txt.gz',\n",
    "                      compression='gzip', sep='\\t', low_memory=False)\n",
    "train_in"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "data = data.merge(train_in[['#Chrom', 'Pos', 'Ref', 'Alt', 'label']],\n",
    "                           left_on=['chr','pos','ref','alt'],\n",
    "                           right_on=['#Chrom','Pos','Ref','Alt'])\n",
    "data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "drop_labels = ['#Chrom', 'Pos', 'Ref', 'Alt']\n",
    "for x in data.columns:\n",
    "    if x.endswith('_x') or x.endswith('_y'):\n",
    "        drop_labels.append(x)\n",
    "data.drop(columns=drop_labels, inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "def apply_func_thresholding(probability, threshold):\n",
    "    return_value = 0\n",
    "    if probability > threshold:\n",
    "        return_value = 1\n",
    "    return return_value"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "data['label'].replace({'Pathogenic': 1, 'Benign': 0}, inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "true_recall = 0\n",
    "true_threshold = 0\n",
    "for threshold in thresholds:\n",
    "    data['pred_label'] = data['probabilities'].apply(lambda x: apply_func_thresholding(x, threshold))\n",
    "    y_pred = np.array(data['pred_label'])\n",
    "    y_true = np.array(data['label'])\n",
    "    recall = recall_score(y_pred, y_true, zero_division=0)\n",
    "    if 0.94 <= recall <= 0.96:\n",
    "        true_recall = recall\n",
    "        true_threshold = threshold\n",
    "        break\n",
    "print(f'Recall {true_recall} is found at threshold {true_threshold}.')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Preparing for AUC analysis\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>probabilities</th>\n",
       "      <th>prediction</th>\n",
       "      <th>combined_prediction</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RDH12</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>14</td>\n",
       "      <td>68196054</td>\n",
       "      <td>GCCCTG</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ANK1</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>32.000</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>8</td>\n",
       "      <td>41554106</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>35.000</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>12</td>\n",
       "      <td>121431481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>39.000</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>12</td>\n",
       "      <td>121435345</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>INFRAME</td>\n",
       "      <td>19.860</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>12</td>\n",
       "      <td>121431471</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334596</th>\n",
       "      <td>SPG11</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>15.230</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>15</td>\n",
       "      <td>44952677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334597</th>\n",
       "      <td>SPG11</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>19.320</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>15</td>\n",
       "      <td>44952632</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334598</th>\n",
       "      <td>SPG11</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>9.169</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>15</td>\n",
       "      <td>44943927</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334599</th>\n",
       "      <td>SPG11</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>13.460</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>15</td>\n",
       "      <td>44943897</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334600</th>\n",
       "      <td>HCN1</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>36.000</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>5</td>\n",
       "      <td>45396603</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>334601 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       GeneName       Consequence   PHRED  probabilities  prediction  \\\n",
       "0         RDH12       FRAME_SHIFT  35.000              1  Pathogenic   \n",
       "1          ANK1  CANONICAL_SPLICE  32.000              1  Pathogenic   \n",
       "2         HNF1A       STOP_GAINED  35.000              1  Pathogenic   \n",
       "3         HNF1A       STOP_GAINED  39.000              1  Pathogenic   \n",
       "4         HNF1A           INFRAME  19.860              1  Pathogenic   \n",
       "...         ...               ...     ...            ...         ...   \n",
       "334596    SPG11    NON_SYNONYMOUS  15.230              0     Neutral   \n",
       "334597    SPG11    NON_SYNONYMOUS  19.320              0     Neutral   \n",
       "334598    SPG11        SYNONYMOUS   9.169              0     Neutral   \n",
       "334599    SPG11    NON_SYNONYMOUS  13.460              0     Neutral   \n",
       "334600     HCN1       STOP_GAINED  36.000              0     Neutral   \n",
       "\n",
       "       combined_prediction chr        pos     ref alt  \n",
       "0               Pathogenic  14   68196054  GCCCTG   G  \n",
       "1               Pathogenic   8   41554106       C   T  \n",
       "2               Pathogenic  12  121431481       C   T  \n",
       "3               Pathogenic  12  121435345       C   T  \n",
       "4               Pathogenic  12  121431471    CAAG   C  \n",
       "...                    ...  ..        ...     ...  ..  \n",
       "334596             Neutral  15   44952677       C   T  \n",
       "334597             Neutral  15   44952632       A   G  \n",
       "334598             Neutral  15   44943927       G   C  \n",
       "334599             Neutral  15   44943897       C   A  \n",
       "334600          Pathogenic   5   45396603       A   T  \n",
       "\n",
       "[334601 rows x 10 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_out = train_new.copy()\n",
    "train_out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Allergy/Immunology/Infectious</th>\n",
       "      <th>Alt</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Audiologic/Otolaryngologic</th>\n",
       "      <th>Biochemical</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>CDSpos</th>\n",
       "      <th>Cardiovascular</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>revel</th>\n",
       "      <th>sift</th>\n",
       "      <th>source</th>\n",
       "      <th>tOverlapMotifs</th>\n",
       "      <th>targetScan</th>\n",
       "      <th>to_be_deleted</th>\n",
       "      <th>verPhCons</th>\n",
       "      <th>verPhyloP</th>\n",
       "      <th>inTest</th>\n",
       "      <th>sample_weight</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14</td>\n",
       "      <td>False</td>\n",
       "      <td>G</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS9787.1</td>\n",
       "      <td>806.0</td>\n",
       "      <td>False</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.843</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS13112.1</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift,stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>4.670</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>C</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS13112.1</td>\n",
       "      <td>2118.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.043</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS13112.1</td>\n",
       "      <td>1586.0</td>\n",
       "      <td>True</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>6.221</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vkgl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>6.368</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334596</th>\n",
       "      <td>17</td>\n",
       "      <td>False</td>\n",
       "      <td>A</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS32642.1</td>\n",
       "      <td>1563.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>6.031</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334597</th>\n",
       "      <td>17</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS32642.1</td>\n",
       "      <td>2029.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>4.100</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334598</th>\n",
       "      <td>10</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS7431.1</td>\n",
       "      <td>1216.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.852</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334599</th>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS2382.1</td>\n",
       "      <td>2998.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.031</td>\n",
       "      <td>2.213</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334600</th>\n",
       "      <td>5</td>\n",
       "      <td>False</td>\n",
       "      <td>T</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>CCDS3952.1</td>\n",
       "      <td>1221.0</td>\n",
       "      <td>False</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.528</td>\n",
       "      <td>False</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>334601 rows × 152 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Chrom  Allergy/Immunology/Infectious Alt          AnnoType  \\\n",
       "0          14                          False   G  CodingTranscript   \n",
       "1          20                          False   T  CodingTranscript   \n",
       "2          20                          False   C  CodingTranscript   \n",
       "3          20                          False   A  CodingTranscript   \n",
       "4          20                          False   A        Intergenic   \n",
       "...       ...                            ...  ..               ...   \n",
       "334596     17                          False   A  CodingTranscript   \n",
       "334597     17                          False   T  CodingTranscript   \n",
       "334598     10                          False   T  CodingTranscript   \n",
       "334599      2                          False   T  CodingTranscript   \n",
       "334600      5                          False   T  CodingTranscript   \n",
       "\n",
       "        Audiologic/Otolaryngologic  Biochemical         CCDS  CDSpos  \\\n",
       "0                            False        False   CCDS9787.1   806.0   \n",
       "1                             True        False  CCDS13112.1  1899.0   \n",
       "2                             True        False  CCDS13112.1  2118.0   \n",
       "3                             True        False  CCDS13112.1  1586.0   \n",
       "4                             True        False          NaN     NaN   \n",
       "...                            ...          ...          ...     ...   \n",
       "334596                       False        False  CCDS32642.1  1563.0   \n",
       "334597                       False        False  CCDS32642.1  2029.0   \n",
       "334598                       False        False   CCDS7431.1  1216.0   \n",
       "334599                       False        False   CCDS2382.1  2998.0   \n",
       "334600                       False        False   CCDS3952.1  1221.0   \n",
       "\n",
       "        Cardiovascular              ConsDetail  ...  revel sift   source  \\\n",
       "0                False              frameshift  ...    NaN  NaN     vkgl   \n",
       "1                 True  frameshift,stop_gained  ...    NaN  NaN     vkgl   \n",
       "2                 True              frameshift  ...    NaN  NaN     vkgl   \n",
       "3                 True              frameshift  ...    NaN  NaN     vkgl   \n",
       "4                 True              downstream  ...    NaN  NaN     vkgl   \n",
       "...                ...                     ...  ...    ...  ...      ...   \n",
       "334596           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334597           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334598           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334599           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "334600           False             stop_gained  ...    NaN  NaN  unknown   \n",
       "\n",
       "        tOverlapMotifs  targetScan  to_be_deleted  verPhCons verPhyloP inTest  \\\n",
       "0                  NaN         NaN          False      1.000     5.843  False   \n",
       "1                  NaN         NaN          False      1.000     4.670  False   \n",
       "2                  NaN         NaN          False      1.000     5.043  False   \n",
       "3                  NaN         NaN          False      1.000     6.221  False   \n",
       "4                  NaN         NaN          False      1.000     6.368  False   \n",
       "...                ...         ...            ...        ...       ...    ...   \n",
       "334596             NaN         NaN          False      1.000     6.031  False   \n",
       "334597             NaN         NaN          False      1.000     4.100  False   \n",
       "334598             NaN         NaN          False      1.000     5.852  False   \n",
       "334599             NaN         NaN          False      0.031     2.213  False   \n",
       "334600             NaN         NaN          False      1.000     0.528  False   \n",
       "\n",
       "        sample_weight  \n",
       "0                 1.0  \n",
       "1                 1.0  \n",
       "2                 1.0  \n",
       "3                 1.0  \n",
       "4                 1.0  \n",
       "...               ...  \n",
       "334596            0.8  \n",
       "334597            0.8  \n",
       "334598            0.8  \n",
       "334599            0.8  \n",
       "334600            0.8  \n",
       "\n",
       "[334601 rows x 152 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_in = pd.read_csv('/run/media/rjsietsma/evo2tb/linux/Datafiles/train.txt.gz',\n",
    "                      compression='gzip', sep='\\t', low_memory=False)\n",
    "train_in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>probabilities</th>\n",
       "      <th>prediction</th>\n",
       "      <th>combined_prediction</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RDH12</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>14</td>\n",
       "      <td>68196054</td>\n",
       "      <td>GCCCTG</td>\n",
       "      <td>G</td>\n",
       "      <td>14</td>\n",
       "      <td>68196054</td>\n",
       "      <td>GCCCTG</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ANK1</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>32.000</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>8</td>\n",
       "      <td>41554106</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>8</td>\n",
       "      <td>41554106</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>35.000</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>12</td>\n",
       "      <td>121431481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>12</td>\n",
       "      <td>121431481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>39.000</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>12</td>\n",
       "      <td>121435345</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>12</td>\n",
       "      <td>121435345</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>INFRAME</td>\n",
       "      <td>19.860</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>12</td>\n",
       "      <td>121431471</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>C</td>\n",
       "      <td>12</td>\n",
       "      <td>121431471</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334596</th>\n",
       "      <td>SPG11</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>15.230</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>15</td>\n",
       "      <td>44952677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>15</td>\n",
       "      <td>44952677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334597</th>\n",
       "      <td>SPG11</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>19.320</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>15</td>\n",
       "      <td>44952632</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>15</td>\n",
       "      <td>44952632</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334598</th>\n",
       "      <td>SPG11</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>9.169</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>15</td>\n",
       "      <td>44943927</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>15</td>\n",
       "      <td>44943927</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334599</th>\n",
       "      <td>SPG11</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>13.460</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>15</td>\n",
       "      <td>44943897</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>15</td>\n",
       "      <td>44943897</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334600</th>\n",
       "      <td>HCN1</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>36.000</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>5</td>\n",
       "      <td>45396603</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>5</td>\n",
       "      <td>45396603</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>334601 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       GeneName       Consequence   PHRED  probabilities  prediction  \\\n",
       "0         RDH12       FRAME_SHIFT  35.000              1  Pathogenic   \n",
       "1          ANK1  CANONICAL_SPLICE  32.000              1  Pathogenic   \n",
       "2         HNF1A       STOP_GAINED  35.000              1  Pathogenic   \n",
       "3         HNF1A       STOP_GAINED  39.000              1  Pathogenic   \n",
       "4         HNF1A           INFRAME  19.860              1  Pathogenic   \n",
       "...         ...               ...     ...            ...         ...   \n",
       "334596    SPG11    NON_SYNONYMOUS  15.230              0     Neutral   \n",
       "334597    SPG11    NON_SYNONYMOUS  19.320              0     Neutral   \n",
       "334598    SPG11        SYNONYMOUS   9.169              0     Neutral   \n",
       "334599    SPG11    NON_SYNONYMOUS  13.460              0     Neutral   \n",
       "334600     HCN1       STOP_GAINED  36.000              0     Neutral   \n",
       "\n",
       "       combined_prediction chr        pos     ref alt #Chrom        Pos  \\\n",
       "0               Pathogenic  14   68196054  GCCCTG   G     14   68196054   \n",
       "1               Pathogenic   8   41554106       C   T      8   41554106   \n",
       "2               Pathogenic  12  121431481       C   T     12  121431481   \n",
       "3               Pathogenic  12  121435345       C   T     12  121435345   \n",
       "4               Pathogenic  12  121431471    CAAG   C     12  121431471   \n",
       "...                    ...  ..        ...     ...  ..    ...        ...   \n",
       "334596             Neutral  15   44952677       C   T     15   44952677   \n",
       "334597             Neutral  15   44952632       A   G     15   44952632   \n",
       "334598             Neutral  15   44943927       G   C     15   44943927   \n",
       "334599             Neutral  15   44943897       C   A     15   44943897   \n",
       "334600          Pathogenic   5   45396603       A   T      5   45396603   \n",
       "\n",
       "           Ref Alt       label  \n",
       "0       GCCCTG   G  Pathogenic  \n",
       "1            C   T  Pathogenic  \n",
       "2            C   T  Pathogenic  \n",
       "3            C   T  Pathogenic  \n",
       "4         CAAG   C  Pathogenic  \n",
       "...        ...  ..         ...  \n",
       "334596       C   T      Benign  \n",
       "334597       A   G      Benign  \n",
       "334598       G   C      Benign  \n",
       "334599       C   A      Benign  \n",
       "334600       A   T      Benign  \n",
       "\n",
       "[334601 rows x 15 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_out = train_out.merge(train_in[['#Chrom', 'Pos', 'Ref', 'Alt', 'label']],\n",
    "                           left_on=['chr','pos','ref','alt'],\n",
    "                           right_on=['#Chrom','Pos','Ref','Alt'])\n",
    "train_out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['#Chrom', 'Pos', 'Ref', 'Alt']"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drop_labels = ['#Chrom', 'Pos', 'Ref', 'Alt']\n",
    "for x in train_out.columns:\n",
    "    if x.endswith('_x') or x.endswith('_y'):\n",
    "        drop_labels.append(x)\n",
    "drop_labels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "train_out.drop(columns=drop_labels, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>probabilities</th>\n",
       "      <th>prediction</th>\n",
       "      <th>combined_prediction</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RDH12</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>35.000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>14</td>\n",
       "      <td>68196054</td>\n",
       "      <td>GCCCTG</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ANK1</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>32.000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>8</td>\n",
       "      <td>41554106</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>35.000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>12</td>\n",
       "      <td>121431481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>39.000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>12</td>\n",
       "      <td>121435345</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>INFRAME</td>\n",
       "      <td>19.860</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>12</td>\n",
       "      <td>121431471</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334596</th>\n",
       "      <td>SPG11</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>15.230</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>15</td>\n",
       "      <td>44952677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334597</th>\n",
       "      <td>SPG11</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>19.320</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>15</td>\n",
       "      <td>44952632</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334598</th>\n",
       "      <td>SPG11</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>9.169</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>15</td>\n",
       "      <td>44943927</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334599</th>\n",
       "      <td>SPG11</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>13.460</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Neutral</td>\n",
       "      <td>15</td>\n",
       "      <td>44943897</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334600</th>\n",
       "      <td>HCN1</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>36.000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>5</td>\n",
       "      <td>45396603</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>334601 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       GeneName       Consequence   PHRED  probabilities  prediction  \\\n",
       "0         RDH12       FRAME_SHIFT  35.000              1           1   \n",
       "1          ANK1  CANONICAL_SPLICE  32.000              1           1   \n",
       "2         HNF1A       STOP_GAINED  35.000              1           1   \n",
       "3         HNF1A       STOP_GAINED  39.000              1           1   \n",
       "4         HNF1A           INFRAME  19.860              1           1   \n",
       "...         ...               ...     ...            ...         ...   \n",
       "334596    SPG11    NON_SYNONYMOUS  15.230              0           0   \n",
       "334597    SPG11    NON_SYNONYMOUS  19.320              0           0   \n",
       "334598    SPG11        SYNONYMOUS   9.169              0           0   \n",
       "334599    SPG11    NON_SYNONYMOUS  13.460              0           0   \n",
       "334600     HCN1       STOP_GAINED  36.000              0           0   \n",
       "\n",
       "       combined_prediction chr        pos     ref alt  label  \n",
       "0               Pathogenic  14   68196054  GCCCTG   G      1  \n",
       "1               Pathogenic   8   41554106       C   T      1  \n",
       "2               Pathogenic  12  121431481       C   T      1  \n",
       "3               Pathogenic  12  121435345       C   T      1  \n",
       "4               Pathogenic  12  121431471    CAAG   C      1  \n",
       "...                    ...  ..        ...     ...  ..    ...  \n",
       "334596             Neutral  15   44952677       C   T      0  \n",
       "334597             Neutral  15   44952632       A   G      0  \n",
       "334598             Neutral  15   44943927       G   C      0  \n",
       "334599             Neutral  15   44943897       C   A      0  \n",
       "334600          Pathogenic   5   45396603       A   T      0  \n",
       "\n",
       "[334601 rows x 11 columns]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_out['prediction'].replace({'Pathogenic': 1, 'Neutral': 0}, inplace=True)\n",
    "train_out['label'].replace({'Pathogenic': 1, 'Benign': 0}, inplace=True)\n",
    "train_out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    294979\n",
       "1     39622\n",
       "Name: probabilities, dtype: int64"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_out['probabilities'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
