{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Importing some shizzle.\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from matplotlib import pyplot as plt\n",
    "from scipy import stats\n",
    "import time\n",
    "import requests\n",
    "import datetime\n",
    "import sys\n",
    "import os\n",
    "import pathlib\n",
    "import seaborn as sns\n",
    "from sklearn.metrics import roc_auc_score\n",
    "import math\n",
    "import json\n",
    "import glob\n",
    "\n",
    "# Defining some import and export locations\n",
    "location = 'rjsietsma'\n",
    "read_loc = '/home/'+location+'/Documents/School/Master_DSLS/Final_Thesis/Initial_Data_exploration/'\n",
    "data_expor_loc = '/home/'+location+'/Documents/School/Master_DSLS/Final_Thesis/Past_initial_data/'\n",
    "img_output_dir = '/home/'+location+'/PycharmProjects/dsls_master_thesis/side_scripts/output_img/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Some bokeh stuff\n",
    "\n",
    "from bokeh.plotting import figure, ColumnDataSource, output_file, show, save\n",
    "from bokeh.models import HoverTool, WheelZoomTool, PanTool, BoxZoomTool, ResetTool, TapTool, SaveTool, FactorRange\n",
    "from bokeh.palettes import inferno, Spectral5\n",
    "from bokeh.transform import factor_cmap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define function to perform Shapiro-Wilk, Kolmogorov-Smirnov, Wilcoxon-/Mann-Whitney U -test, and a pearson correlation test.\n",
    "\n",
    "def perform_stats(x, y):\n",
    "    shapiro_x = stats.shapiro(x)[1]\n",
    "    shapiro_y = stats.shapiro(y)[1]\n",
    "    if shapiro_x >= 0.05:\n",
    "        print(\"X is likely normally distributed.\")\n",
    "    else:\n",
    "        print(\"X is likely not normally distributed.\")\n",
    "    if shapiro_y >= 0.05:\n",
    "        print(\"Y is likely normally distributed.\")\n",
    "    else:\n",
    "        print(\"Y is likely not normally distributed.\")\n",
    "    try:\n",
    "        p_ks_xy = stats.ks_2samp(x, y)[1]\n",
    "        print(f\"The Kolmogorov-Smirnov test p-value: {p_ks_xy}\")\n",
    "    except Exception:\n",
    "        print(\"Kolmogorov-Smirnov could not be performed!\")\n",
    "    try:\n",
    "        p_wc_xy = stats.wilcoxon(x, y)[1]\n",
    "        print(f\"The Wilcoxon test p-value: {p_wc_xy}\")\n",
    "    except ValueError:\n",
    "        p_mw_xy = stats.mannwhitneyu(x, y)[1]\n",
    "        print(f\"Wilcoxon could not be performed, \\n\"\n",
    "             f\"Using Mann-Whitney rank test p-value: {p_mw_xy}\")\n",
    "    except Exception:\n",
    "        print(\"Neither Wilcoxon nor Mann-Whitney tests could be performed!\")\n",
    "    try:\n",
    "        p_pears_xy = stats.pearsonr(x, y)\n",
    "        print(f\"The Pearson correlation: {p_pears_xy[0]},\\n\"\n",
    "             f\"p-value: {p_pears_xy[1]}\")\n",
    "    except Exception:\n",
    "        print(\"Pearson correlation could not be performed!\")\n",
    "\n",
    "# Define function to calculate the Z-scores of given data.\n",
    "\n",
    "def calc_z_scores(data):\n",
    "    centered = data - data.mean(axis=0)\n",
    "    return centered / centered.std(axis=0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Sprint 28-04-2020 - 19-05-2020 (?)\n",
    "=====================\n",
    "\n",
    "- What I want to do:\n",
    "  1. [Focus on fixing missing CAPICE precomputed scores.](#Fixing-CAPICE-precomputed-scores) __Done__\n",
    "      - _Note: Find out how the script of Shuang works on a smaller scale, get an example where a CAPICE score is missing._\n",
    "  2. [Normalize BM_ratio](#Normalize-BM_ratio) for gene length, so $\\frac{n_{malignant}}{n_{benign}}$ per 1000 bases in a gene, instead of [$\\frac{n_{malign}}{n_{benign}} \\times n_{total}$](output_img/bm_ratio.png \"BM_ratio\").\n",
    "  3. _(questionable)_ Fix PCA by identifying categorical features _(Questionable because should I still put time and focus in on the input data?)_\n",
    "      - _Ask Shuang, Joeri and Krista in the PRU 28-04-2020 @ 13:00-14:00_\n",
    "  4. [See what is required](#Required-for-data-sources) for the [data sources](https://docs.google.com/document/d/1D5SiNbeDEfY2hTWquS88MGUrLv6IgZzfCzflppRGb5k/edit) to be implemented into the input data.\n",
    "      - _Note: as this is research, I can use sources that only have precomputed scores available, but just to keep in mind that I should mention that in the discussion._\n",
    "\n",
    "- Notes:\n",
    "  - If I find myself unfocussed:\n",
    "      - Win a game of [solitaire](https://play.google.com/store/apps/details?id=com.lemongame.klondike.solitaire&hl=nl) (game is rigged, not every match is win-able).\n",
    "      - Clean out my closet (it's messy, maybe to clean up the mess inside my head too).\n",
    "      - Go for a walk around the neighbourhood.\n",
    "\n",
    "<br/><br/>\n",
    "<br/><br/>\n",
    "### Fixing CAPICE precomputed scores\n",
    "\n",
    "Problem: CAPICE precomputed file of all single nucleotide level variants contains caps.\n",
    "\n",
    "Solution: Fix the underlying problem that causes these gaps.\n",
    "\n",
    "Required: \n",
    "- Script to make and batch the CAPICE computing __(positive check)__\n",
    "- Access to GCC cluster __(positive check)__\n",
    "- Knowledge of job scheduler on GCC cluster __(positive check)__\n",
    "- Smaller batches of SNV files to run locally  __(positive check)__\n",
    "\n",
    "Possible cause:\n",
    "- Incorrect batching on tabix.\n",
    "\n",
    "31-04-2020 Problem:\n",
    "- Got it working on Gearshift, but with a problem:\n",
    "    - AttributeError: Can't get attribute 'DeprecationDict' on <module 'sklearn.utils.deprecation' from '/apps/software/PythonPlus/3.7.4-foss-2018b-v20.02.1/lib/python3.7/site-packages/sklearn/utils/deprecation.py'>\n",
    "- Which let's me to beleive that the model was trained in a previous version of sklearn.\n",
    "\n",
    "<br/><br/>\n",
    "\n",
    "- Update 10:51 : Model compatibility issue. Might have to retrain CAPICE on Gearshift using the latest xgboost, sklearn, scikit-learn, numpy and pandas just to be safe.\n",
    "\n",
    "- Update 12:01 : Managed to get it working using python3.6 on calculon, essentially downgrading my own script (I only used python3.7 for F-strings)\n",
    "\n",
    "- Update 15:55 : Found a very weird bug in the chunking. Seems like the script doesn't like to start at 0 with skiprows and 1000 nrows, solution: nrows - 2 when first iteration, then + 1, then start += nrows + 1...\n",
    "\n",
    "- Update 04-05-2020 : Found out that the first iteration, an overhead of 2 variants was present. Adjusted the start with 2 to compensate. Seemed fine in the first 400000 CADD entries. (no duplicates nor missed entries). __Started full processing__\n",
    "\n",
    "- Update 04-05-2020 : \n",
    "    - Add functionality to continue where left off if it crashes.\n",
    "    - Make dummy script to check for duplicates or if the files are fully complete.\n",
    "    \n",
    "- Update 06-05-2020 : \n",
    "    - Functionaility to continue where left off is required. After 40 hours, it's still processing chromosome 1 after 7 million processed variants. \n",
    "        - Ideas: write json file with start and stop each iteration once that part is processed and added to the larger files. Should overwrite.\n",
    "        - As for now, since this idea is not implemented yet, count the number of rows in the already processed files and set that as start.\n",
    "\n",
    "- Update 11-05-2020:\n",
    "    - The 4 days have ended, chromosome 1 has not been finished yet.\n",
    "    - Output file of chromosome 1 has been renamed to a .tsv file and gzipped (saving about 5 gb of storage).\n",
    "    - Adjustments to the original script have to be made.\n",
    "   \n",
    "\n",
    "#### Version 2.0\n",
    "\n",
    "- Added gzip output by default\n",
    "- Currently no function yet to continue where left off.\n",
    "    - Ideas:\n",
    "        - for line in gzip.open(file.gz): nlines+=1 (to trace back where was left off)\n",
    "            - Do not know if this lead to any complications of duplicated entries. Saving the previous results in the RAM memory might solve this to compare to.\n",
    "    - Then in the future:\n",
    "        - Save a json of start and batch size each iteration to mark where the file was left of.\n",
    "\n",
    "- Update 15-05-2020:\n",
    "    - Program did not load in the previous results correctly, resulting in 12 duplicates (111400001-111400012) (including start and stop). Discuss with Joeri and Shuang what to do. It takes roughly 200 MB of ram to load in 2.1 million processed variants, calculating this for 111.4 million variants (that's how much there is in the file now), would come down to ~11GB of ram usage.\n",
    "\n",
    "- Update 18-05-2020:\n",
    "    - Script to remove duplicates from the CAPICE processed file has been run and duplicates have been removed.\n",
    "    - First test runs have been ran, only for me to realised I messed up. I kept copying the main class, instead of the calculate_capice class. Final release should be on this date.\n",
    "    \n",
    "\n",
    "<br/><br/>\n",
    "<br/><br/>\n",
    "### Normalize BM_ratio\n",
    "\n",
    "Problem: Larger genes have more benign and malignant samples, because they are larger and thus the chanses of variation on the reference read is higher.\n",
    "\n",
    "Solution: Make a $n_{malignant}$ and $n_{benign}$ per $x$ bases in a gene, where {$x \\in \\mathbb{N}\\ | x \\geq 1 | x \\leq 10000$}\n",
    "\n",
    "<br/><br/>\n",
    "<br/><br/>\n",
    "### Required for data sources\n",
    "\n",
    "Problem: Now that a lot of data sources have been found, annotation and merging has to be performed in order to use it as training data.\n",
    "- _Additional problem: once a source has been proven beneficial, it has te be implemented to be annotated before capice fits and predicts a new sample._\n",
    "\n",
    "Solution: Find out, feature by feature if genes that have a low AUC improve *__and that genes that have a high AUC do not perform worse__*\n",
    "\n",
    "Required:\n",
    "- Storage \n",
    "    - On GCC cluster for precomputed score files.\n",
    "    - Locally IF file isn't that big or an API can be used.\n",
    "- Data sources __([positive check](https://docs.google.com/document/d/1D5SiNbeDEfY2hTWquS88MGUrLv6IgZzfCzflppRGb5k/edit))__\n",
    "- Script to annotate / merge\n",
    "\n",
    "\n",
    "      "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CTEPG (Catching The Elusive Predictable Genes) 28-04-2020 - 04-05-2020\n",
    "<br/><br/>\n",
    "\n",
    "What have I done:\n",
    "\n",
    "- [Enrichr Genes < 0.6](https://amp.pharm.mssm.edu/Enrichr/enrich?dataset=b7a959357214069d1905719220ed3e3b)\n",
    "- [Enrichr Genes < 0.7](https://amp.pharm.mssm.edu/Enrichr/enrich?dataset=7d252a3e1df275859228fd562e964344)\n",
    "    - For reference: [Enrichr Genes < 0.85](https://amp.pharm.mssm.edu/Enrichr/enrich?dataset=60c74741c967c25331419bd2b5e7183e)\n",
    "- [Finished investigating initial data sources](https://docs.google.com/document/d/1D5SiNbeDEfY2hTWquS88MGUrLv6IgZzfCzflppRGb5k/edit)\n",
    "\n",
    "<br/><br/>\n",
    "<br/><br/>\n",
    "\n",
    "What I'm planning to do this week:\n",
    "\n",
    "- [Fixing missing values in precomputed CAPICE scoring file](#Fixing-CAPICE-precomputed-scores). __Done and not done, script has to have some additonal features__\n",
    "    - If stuck, do not keep walking into the wall, continue below and contact Shuang and Joeri\n",
    "- [Run enrichr on increasing cutoffs in AUC (.5, .6, .7 etc.)](#Enrichr-automated-API). __Done__\n",
    "    - Read into the [Enrichr API documentation](https://amp.pharm.mssm.edu/Enrichr/help#api)\n",
    "- Mail Tsjerk that a meeting is apreciated. __Done, no response yet__\n",
    "- Make [suggested plot](https://docs.google.com/presentation/d/1_woKoD9nqWnxueyFYfwyX6B4owg8vYF7rw_nKpMPux4/edit#slide=id.p). __Done__\n",
    "    - Select from enrichr all the genes involved in (lipid) metabolism and see if this explains all / most of the bad performing genes.\n",
    "- Make model on bad performing genes and investigate further.\n",
    "- Start investigating what is required for the data sources to be used in the training phase.\n",
    "- Look into the genes that do perform well without a CGD category, also look into the 1/2 genes that do not perform well __with__ a CGD category\n",
    "- _Optionally_ :\n",
    "    - Create outline for article.\n",
    "    - Adjust BM_ratio plot to normalize for gene length.\n",
    "    - Get the UMCG gene panel lists into python\n",
    "        - Then: investigate which gene panel is performing the worst.\n",
    "        - It would be nice to have this on the side for diagnostics, do __not__ focus on this\n",
    "    - Fix PCA.\n",
    "    - Find possible way to annotate genes for co-factors, sub-family etc.\n",
    "<br/><br/>\n",
    "\n",
    "# CTEPG (Catching The Elusive Predictable Genes) 04-05-2020 - 11-05-2020\n",
    "<br/><br/>\n",
    "\n",
    "What have I done:\n",
    "- [Fixing missing values in precomputed CAPICE scoring file](#Fixing-CAPICE-precomputed-scores).\n",
    "- [Run enrichr on increasing cutoffs in AUC (.5, .6, .7 etc.)](#Enrichr-automated-API).\n",
    "- Make [suggested plot](https://docs.google.com/presentation/d/1_woKoD9nqWnxueyFYfwyX6B4owg8vYF7rw_nKpMPux4/edit#slide=id.p).\n",
    "<br/><br/>\n",
    "\n",
    "Planned for this week:\n",
    "- Release precomputed CAPICE script v2.0\n",
    "- Investigate Bokeh plots on commonalities between Adjusted P-value of 0-0.2 and AUC of 0.5-~0.75 and try to find features / links that would be beneficial to train a separate model on. (or prehaps just train a new model without features and see if performance improves)\n",
    "    - Ion transport / transmembrane related genes seems to perform bad. \n",
    "    - Find commonalities between y-axis 0-0.2 and x-axis 0-0.7\n",
    "    - Make model on bad performing genes and investigate further.\n",
    "        - Make plots combined of all sources, but color them by source.\n",
    "- Start investigating what is required for the data sources to be used in the training phase.\n",
    "- Look into the genes that do perform well without a CGD category, also look into the 1/2 genes that do not perform well __with__ a CGD category\n",
    "- _Optionally_ :\n",
    "    - Create outline for article.\n",
    "        - For static plots: make labels on each point with the term name and the number of genes in (n=x) behind that.\n",
    "    - Adjust BM_ratio plot to normalize for gene length.\n",
    "    - Get the UMCG gene panel lists into python (calculate average AUC for each panel, take the worst performing onces and retain).\n",
    "        - Then: investigate which gene panel is performing the worst.\n",
    "        - It would be nice to have this on the side for diagnostics, do __not__ focus on this\n",
    "    - Find possible way to annotate genes for co-factors, sub-family etc.\n",
    "    - Interactive plot of all variants labeled as benign or pathogenic with their capice score (y-axis) and location (x-axis), to show the variant itself.\n",
    "<br/><br/>\n",
    "\n",
    "# CTEPG (Catching The Elusive Predictable Genes) 11-05-2020 - 18-05-2020\n",
    "<br/><br/>\n",
    "\n",
    "What have I done:\n",
    "- [Fixing missing values in precomputed CAPICE scoring file](#Fixing-CAPICE-precomputed-scores).\n",
    "- Make more bokeh plots on enrichr results\n",
    "    - Investigate correlations between AUC's and Adjusted P-value.  (spearman / pearson, preverably spearman). Keep in mind that the distribution is not linear, but more like a normal distribution like.\n",
    "        - Make legend interactive to confirm normal distribution like.\n",
    "            - To make legend interactive, I have to scatterplot each component separetly, like [this](https://docs.bokeh.org/en/latest/docs/user_guide/interaction/legends.html#hiding-glyphs)\n",
    "    - Chromosome X seems enriched for higher AUC thresholds. Bias in the source data? (Put in discussion). Lot's of developmental delay reseach is linked / correlated to chromosome X.\n",
    "\n",
    "Planned for this week:\n",
    "- __Release precomputed CAPICE script v2.0__\n",
    "- Run correlation test for all possible sources on Enrichr to maximize chanses of finding stuff.\n",
    "    - Look into a good way to get all possible results from the enrichr API\n",
    "- __Escalate mail contact with Tsjerk to Martijn and Fenna.__\n",
    "- Investigate Bokeh plots on commonalities between Adjusted P-value of 0-0.2 and AUC of 0.5-~0.75 and try to find features / links that would be beneficial to train a separate model on. (or prehaps just train a new model without features and see if performance improves)\n",
    "    - Ion transport / transmembrane related genes seems to perform bad. \n",
    "    - Find commonalities between y-axis 0-0.2 and x-axis 0-0.7\n",
    "    - __Make model on bad performing genes and investigate further.__\n",
    "        - Make plots combined of all sources, but color them by source. __Done__\n",
    "- Start investigating what is required for the data sources to be used in the training phase.\n",
    "- Look into the genes that do perform well without a CGD category, also look into the 1/2 genes that do not perform well __with__ a CGD category\n",
    "- _Optionally_ :\n",
    "    - __Create outline for article.__\n",
    "        - For static plots: make labels on each point with the term name and the number of genes in (n=x) behind that.\n",
    "    - Adjust BM_ratio plot to normalize for gene length.\n",
    "    - Get the UMCG gene panel lists into python (calculate average AUC for each panel, take the worst performing onces and retain).\n",
    "        - Then: investigate which gene panel is performing the worst.\n",
    "        - It would be nice to have this on the side for diagnostics, do __not__ focus on this\n",
    "    - Find possible way to annotate genes for co-factors, sub-family etc.\n",
    "    - Interactive plot of all variants labeled as benign or pathogenic with their capice score (y-axis) and location (x-axis), to show the variant itself.\n",
    "<br/><br/>\n",
    "    \n",
    "## For after I graduate:\n",
    "<br/><br/>\n",
    "Make a script that for every gene, for every gene makes a plot like [these](https://molgenis26.gcc.rug.nl/downloads/capice/plots/), but then interactive that if you hover over a point you see it's chromosomal location, ref and alt and maybe an RS number?\n",
    " \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "  \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Current Tickets:\n",
    "<br/><br/>\n",
    "\n",
    "## As student / Robert I want to map requirements for SNV and gene features.\n",
    "\n",
    "- Content:\n",
    "\n",
    "Overview of features has now been collected and processed (https://docs.google.com/document/d/1D5SiNbeDEfY2hTWquS88MGUrLv6IgZzfCzflppRGb5k/edit). In order to use these features, knowledge has to be obtained for the requirements on said features including, but not limited to merging to the trianing dataset and storage.\n",
    "\n",
    "- Acceptance criterea:\n",
    "    - Does not take up terrabytes of storage.\n",
    "        - _If it does, data sources might have to be merged 1 by 1, removing the source after the merge._\n",
    "    - Can be merged without 3rd party software (tabix, python etc. allowed).\n",
    "    - Data can easily be downloaded.\n",
    "\n",
    "<br/><br/>\n",
    "\n",
    "## As student / Robert I want to find commonalities between bad performing genes in CAPICE\n",
    "\n",
    "- Content:\n",
    "\n",
    "The enrichr tool has proven to be quite beneficial to understanding a set of bad performing genes. Further processing of these enrichr results is required in order to fully understand the underlying mechanics and commonalities that these bad performing genes share.\n",
    "\n",
    "- Acceptance criterea:\n",
    "    - Enrichr is performed on multiple gene sets of different AUC thresholds (0.5, 0.6, 0.7 etc.).\n",
    "    - Results of various data sources (kegg, bioplanet, wikipathways etc.) have been obtained in an automated way using the Enrichr API.\n",
    "    - Results from the Enrichr API have been processed in an understandable plot, an mockup can be found [here](https://docs.google.com/presentation/d/1_woKoD9nqWnxueyFYfwyX6B4owg8vYF7rw_nKpMPux4/edit#slide=id.p).\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Enrichr automated API"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "enrichr_url = 'http://amp.pharm.mssm.edu/Enrichr/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>default_auc</th>\n",
       "      <th>default_f1</th>\n",
       "      <th>default_recall</th>\n",
       "      <th>default_fpr</th>\n",
       "      <th>default_spec</th>\n",
       "      <th>optimal_c</th>\n",
       "      <th>optimal_auc</th>\n",
       "      <th>optimal_f1</th>\n",
       "      <th>optimal_recall</th>\n",
       "      <th>optimal_fpr</th>\n",
       "      <th>optimal_spec</th>\n",
       "      <th>n_tot</th>\n",
       "      <th>n_benign</th>\n",
       "      <th>n_malign</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2775</th>\n",
       "      <td>PRKN</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1659</th>\n",
       "      <td>BUB1B</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>73</td>\n",
       "      <td>66</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2102</th>\n",
       "      <td>GALNT3</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1655</th>\n",
       "      <td>PNPLA1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.8</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.8</td>\n",
       "      <td>31</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2782</th>\n",
       "      <td>TMEM80</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2625</th>\n",
       "      <td>C10orf2</td>\n",
       "      <td>0.469697</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.469697</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>133</td>\n",
       "      <td>132</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2330</th>\n",
       "      <td>SPTLC2</td>\n",
       "      <td>0.455157</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.011</td>\n",
       "      <td>0.750374</td>\n",
       "      <td>0.170213</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>229</td>\n",
       "      <td>223</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2615</th>\n",
       "      <td>PLCD1</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2611</th>\n",
       "      <td>UBE2A</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2806</th>\n",
       "      <td>SOX9-AS1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2807 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene  default_auc  default_f1  default_recall  default_fpr  \\\n",
       "2775      PRKN     1.000000         1.0             1.0          0.0   \n",
       "1659     BUB1B     1.000000         1.0             1.0          0.0   \n",
       "2102    GALNT3     1.000000         1.0             1.0          0.0   \n",
       "1655    PNPLA1     1.000000         1.0             1.0          0.0   \n",
       "2782    TMEM80     1.000000         1.0             1.0          0.0   \n",
       "...        ...          ...         ...             ...          ...   \n",
       "2625   C10orf2     0.469697         0.0             0.0          1.0   \n",
       "2330    SPTLC2     0.455157         0.0             0.0          1.0   \n",
       "2615     PLCD1     0.500000         0.0             0.0          1.0   \n",
       "2611     UBE2A     0.500000         0.0             0.0          1.0   \n",
       "2806  SOX9-AS1     0.000000         0.0             0.0          1.0   \n",
       "\n",
       "      default_spec  optimal_c  optimal_auc  optimal_f1  optimal_recall  \\\n",
       "2775           1.0      0.020     1.000000    1.000000             1.0   \n",
       "1659           1.0      0.020     1.000000    1.000000             1.0   \n",
       "2102           1.0      0.020     1.000000    1.000000             1.0   \n",
       "1655           1.0      0.020     1.000000    1.000000             0.8   \n",
       "2782           1.0      0.020     1.000000    1.000000             1.0   \n",
       "...            ...        ...          ...         ...             ...   \n",
       "2625           0.0      0.020     0.469697    0.000000             0.0   \n",
       "2330           0.0      0.011     0.750374    0.170213             0.0   \n",
       "2615           0.0      0.020     0.500000    0.000000             0.0   \n",
       "2611           0.0      0.020     0.500000    0.000000             0.0   \n",
       "2806           0.0      0.020     0.000000    0.000000             0.0   \n",
       "\n",
       "      optimal_fpr  optimal_spec  n_tot  n_benign  n_malign  \n",
       "2775          0.0           1.0     11         7         4  \n",
       "1659          0.0           1.0     73        66         7  \n",
       "2102          0.0           1.0     12        10         2  \n",
       "1655          0.2           0.8     31        26         5  \n",
       "2782          0.0           1.0      5         4         1  \n",
       "...           ...           ...    ...       ...       ...  \n",
       "2625          1.0           0.0    133       132         1  \n",
       "2330          1.0           0.0    229       223         6  \n",
       "2615          1.0           0.0      8         7         1  \n",
       "2611          1.0           0.0      3         2         1  \n",
       "2806          1.0           0.0      3         2         1  \n",
       "\n",
       "[2807 rows x 15 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load in initial data\n",
    "data = pd.read_csv(read_loc+'optimal_f1_full_ds_v2.csv', header=0)\n",
    "data.sort_values(by='default_f1', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2807"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Make a dictionary of lists of genes of interest with as key the 0.x threshold\n",
    "\n",
    "auc_dict = {}\n",
    "thresholds = np.arange(0.5, 1.01, 0.05).round(decimals=2)\n",
    "for t in thresholds:\n",
    "    list_of_genes = data[data['default_auc'] < t]['gene'].tolist()\n",
    "    auc_dict[t.astype('str')] = '\\n'.join(list_of_genes)\n",
    "    del list_of_genes\n",
    "del thresholds\n",
    "\n",
    "# Save it to a file so I don't bombard the Enrichr API with requests\n",
    "if not os.path.isfile('./user_id_list.json'):\n",
    "    file = open('./user_id_list.json', 'w')\n",
    "    file.close()\n",
    "\n",
    "with open('./user_id_list.json') as json_file:\n",
    "    try:\n",
    "        user_list_ids = json.load(json_file)\n",
    "        json_was_empty = False\n",
    "    except json.JSONDecodeError:\n",
    "        user_list_ids = {}\n",
    "        json_was_empty = True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get list ID's for the different AUC's\n",
    "\n",
    "for threshold in auc_dict.keys():\n",
    "    if threshold not in user_list_ids.keys():\n",
    "        description = f'CAPICE performance of genes with an AUC lower than {threshold}'\n",
    "        payload = {\n",
    "            'list': (None, auc_dict[threshold]),\n",
    "            'description': (None, description)\n",
    "        }\n",
    "        response = requests.post(enrichr_url + '/addList', files=payload)\n",
    "        if not response.ok:\n",
    "            print(f'Something went wrong with {threshold}!')\n",
    "        else:\n",
    "            enrichr_addlist_response = json.loads(response.text)\n",
    "            user_list_ids[threshold] = enrichr_addlist_response['userListId']\n",
    "\n",
    "# Write it to a file in case I didn't have any.\n",
    "if json_was_empty:\n",
    "    with open('./user_id_list.json', 'w') as json_file:\n",
    "        json.dump(user_list_ids, json_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Now to get the enrichment results.\n",
    "\n",
    "enrichr_sources_oi = ['BioPlanet_2019', 'WikiPathways_2019_Human', 'KEGG_2019_Human', 'Elsevier_Pathway_Collection',\n",
    "                     'GO_Biological_Process_2018','Human_Phenotype_Ontology','Jensen_DISEASES',\n",
    "                     'PheWeb_2019', 'ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET','OMIM_Disease','OMIM_Expanded',\n",
    "                     'Human_Gene_Atlas', 'ARCHS4_Tissues',\n",
    "                     'HMDB_Metabolites','Chromosome_Location_hg19']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Now get the enrichr results.\n",
    "def get_enrichr_results(enrichr_sources, number_of_hits):\n",
    "    if len(enrichr_sources) == 0:\n",
    "        print('You should give some sources for me to get data from.')\n",
    "    else:\n",
    "\n",
    "        # Prepare output dataframes for each source\n",
    "        output_dir = '/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/'\n",
    "        \n",
    "        for iteration, source in enumerate(enrichr_sources):\n",
    "            print(f'Working on source: {source}')\n",
    "            if iteration == len(enrichr_sources):\n",
    "                sources_left = []\n",
    "            else:\n",
    "                sources_left = enrichr_sources[iteration+1::]\n",
    "            print(f'Sources left: {sources_left}')\n",
    "            # Prepare the output dict\n",
    "            current_source_df = pd.DataFrame(columns=['AUC',\n",
    "                                                      'Rank',\n",
    "                                                      'Term_Name',\n",
    "                                                      'Overlapping_Genes',\n",
    "                                                      'Adjusted_P-value',\n",
    "                                                      'Source'])\n",
    "            for (threshold, user_id) in zip(user_list_ids.keys(),user_list_ids.values()):\n",
    "                print(f'\\tCurrently working on threshold: {threshold}')\n",
    "                req = enrichr_url + f'enrich?userListId={user_id}&backgroundType={source}'\n",
    "                response = requests.get(req)\n",
    "                if not response.ok:\n",
    "                    print(f'Userid {user_id} (threshold: {threshold})'\n",
    "                          f'received error on: {source}!')\n",
    "                else:\n",
    "                    enrichr_response_data = json.loads(response.text)\n",
    "                    content = enrichr_response_data[source]\n",
    "                    auc = threshold\n",
    "                    content_df = pd.DataFrame(columns=['AUC', 'Rank', 'Term_Name',\n",
    "                                                       'Combined_score',\n",
    "                                                       'Overlapping_Genes',\n",
    "                                                       'Adjusted_P-value',\n",
    "                                                       'Source'])\n",
    "                    for c in content:\n",
    "                        rank = c[0]\n",
    "                        term_name = c[1]\n",
    "                        combined_score = c[4]\n",
    "                        overlapping_genes = ', '.join(c[5])\n",
    "                        a_p_value = c[6]\n",
    "                        local_df = pd.DataFrame({'AUC':auc, 'Rank':rank,\n",
    "                                                 'Term_Name': term_name,\n",
    "                                                 'Combined_score': combined_score,\n",
    "                                                 'Overlapping_Genes': overlapping_genes,\n",
    "                                                 'Adjusted_P-value': a_p_value,\n",
    "                                                'Source': source}, index=[0])\n",
    "                        content_df = content_df.append(local_df, ignore_index=True)\n",
    "                    content_df = content_df.sort_values(by=['Adjusted_P-value',\n",
    "                                                            'Combined_score'],\n",
    "                                                        ascending=[True, False])\n",
    "                    max_rows = 0\n",
    "                    if content_df.shape[0] < number_of_hits:\n",
    "                        max_rows = content_df.shape[0] - 1\n",
    "                    else:\n",
    "                        max_rows = number_of_hits\n",
    "                    out_df = content_df[:max_rows]\n",
    "                    out_df.drop('Combined_score', axis=1, inplace=True)\n",
    "                    current_source_df = current_source_df.append(out_df, ignore_index=True)\n",
    "            output_filename = output_dir + source + '.csv'\n",
    "            current_source_df.to_csv(output_filename, index=False, sep=',')\n",
    "            print(f'\\tSource {source} has been exported to csv (\\n\\t\\t{output_filename}\\n\\t)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "output_source_link = {}\n",
    "for source in enrichr_sources_oi:\n",
    "    output_filename = source + '.csv'\n",
    "    output_source_link[source] = output_filename"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "glob_files_present = glob.glob('./EnrichrAPIResults/*.csv')\n",
    "list_of_files_present = []\n",
    "if len(glob_files_present) > 0:\n",
    "    for file in glob_files_present:\n",
    "        list_of_files_present.append(file.split('/')[-1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "sources_required = []\n",
    "for source, output_filename in output_source_link.items():\n",
    "    if output_filename not in list_of_files_present:\n",
    "        sources_required.append(source)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['BioPlanet_2019',\n",
       " 'WikiPathways_2019_Human',\n",
       " 'KEGG_2019_Human',\n",
       " 'Elsevier_Pathway_Collection',\n",
       " 'GO_Biological_Process_2018',\n",
       " 'Human_Phenotype_Ontology',\n",
       " 'Jensen_DISEASES',\n",
       " 'PheWeb_2019',\n",
       " 'ClinVar_2019',\n",
       " 'GWAS_Catalog_2019',\n",
       " 'DisGeNET',\n",
       " 'OMIM_Disease',\n",
       " 'OMIM_Expanded',\n",
       " 'Human_Gene_Atlas',\n",
       " 'ARCHS4_Tissues',\n",
       " 'HMDB_Metabolites',\n",
       " 'Chromosome_Location_hg19']"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sources_required"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Working on source: BioPlanet_2019\n",
      "Sources left: ['WikiPathways_2019_Human', 'KEGG_2019_Human', 'Elsevier_Pathway_Collection', 'GO_Biological_Process_2018', 'Human_Phenotype_Ontology', 'Jensen_DISEASES', 'PheWeb_2019', 'ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/rjsietsma/PycharmProjects/dsls_master_thesis/venv/lib/python3.8/site-packages/pandas/core/frame.py:3990: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  return super().drop(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource BioPlanet_2019 has been exported to csv (\n",
      "\t\t/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/BioPlanet_2019.csv\n",
      "\t)\n",
      "Working on source: WikiPathways_2019_Human\n",
      "Sources left: ['KEGG_2019_Human', 'Elsevier_Pathway_Collection', 'GO_Biological_Process_2018', 'Human_Phenotype_Ontology', 'Jensen_DISEASES', 'PheWeb_2019', 'ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource WikiPathways_2019_Human has been exported to csv (\n",
      "\t\t/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/WikiPathways_2019_Human.csv\n",
      "\t)\n",
      "Working on source: KEGG_2019_Human\n",
      "Sources left: ['Elsevier_Pathway_Collection', 'GO_Biological_Process_2018', 'Human_Phenotype_Ontology', 'Jensen_DISEASES', 'PheWeb_2019', 'ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource KEGG_2019_Human has been exported to csv (\n",
      "\t\t/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/KEGG_2019_Human.csv\n",
      "\t)\n",
      "Working on source: Elsevier_Pathway_Collection\n",
      "Sources left: ['GO_Biological_Process_2018', 'Human_Phenotype_Ontology', 'Jensen_DISEASES', 'PheWeb_2019', 'ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource Elsevier_Pathway_Collection has been exported to csv (\n",
      "\t\t/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/Elsevier_Pathway_Collection.csv\n",
      "\t)\n",
      "Working on source: GO_Biological_Process_2018\n",
      "Sources left: ['Human_Phenotype_Ontology', 'Jensen_DISEASES', 'PheWeb_2019', 'ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource GO_Biological_Process_2018 has been exported to csv (\n",
      "\t\t/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/GO_Biological_Process_2018.csv\n",
      "\t)\n",
      "Working on source: Human_Phenotype_Ontology\n",
      "Sources left: ['Jensen_DISEASES', 'PheWeb_2019', 'ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource Human_Phenotype_Ontology has been exported to csv (\n",
      "\t\t/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/Human_Phenotype_Ontology.csv\n",
      "\t)\n",
      "Working on source: Jensen_DISEASES\n",
      "Sources left: ['PheWeb_2019', 'ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource Jensen_DISEASES has been exported to csv (\n",
      "\t\t/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/Jensen_DISEASES.csv\n",
      "\t)\n",
      "Working on source: PheWeb_2019\n",
      "Sources left: ['ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource PheWeb_2019 has been exported to csv (\n",
      "\t\t/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/PheWeb_2019.csv\n",
      "\t)\n",
      "Working on source: ClinVar_2019\n",
      "Sources left: ['GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource ClinVar_2019 has been exported to csv (\n",
      "\t\t/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/ClinVar_2019.csv\n",
      "\t)\n",
      "Working on source: GWAS_Catalog_2019\n",
      "Sources left: ['DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource GWAS_Catalog_2019 has been exported to csv (\n",
      "\t\t/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/GWAS_Catalog_2019.csv\n",
      "\t)\n",
      "Working on source: DisGeNET\n",
      "Sources left: ['OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource DisGeNET has been exported to csv (\n",
      "\t\t/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/DisGeNET.csv\n",
      "\t)\n",
      "Working on source: OMIM_Disease\n",
      "Sources left: ['OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource OMIM_Disease has been exported to csv (\n",
      "\t\t/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/OMIM_Disease.csv\n",
      "\t)\n",
      "Working on source: OMIM_Expanded\n",
      "Sources left: ['Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource OMIM_Expanded has been exported to csv (\n",
      "\t\t/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/OMIM_Expanded.csv\n",
      "\t)\n",
      "Working on source: Human_Gene_Atlas\n",
      "Sources left: ['ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource Human_Gene_Atlas has been exported to csv (\n",
      "\t\t/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/Human_Gene_Atlas.csv\n",
      "\t)\n",
      "Working on source: ARCHS4_Tissues\n",
      "Sources left: ['HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource ARCHS4_Tissues has been exported to csv (\n",
      "\t\t/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/ARCHS4_Tissues.csv\n",
      "\t)\n",
      "Working on source: HMDB_Metabolites\n",
      "Sources left: ['Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource HMDB_Metabolites has been exported to csv (\n",
      "\t\t/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/HMDB_Metabolites.csv\n",
      "\t)\n",
      "Working on source: Chromosome_Location_hg19\n",
      "Sources left: []\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource Chromosome_Location_hg19 has been exported to csv (\n",
      "\t\t/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/Chromosome_Location_hg19.csv\n",
      "\t)\n"
     ]
    }
   ],
   "source": [
    "get_enrichr_results(sources_required, 20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "enrichr_data_loc = './EnrichrAPIResults/'\n",
    "\n",
    "enrichr_dataset = {}\n",
    "\n",
    "for source in enrichr_sources_oi:\n",
    "    dataset = enrichr_data_loc + source + '.csv'\n",
    "    enrichr_dataset[source] = pd.read_csv(dataset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.5</td>\n",
       "      <td>1</td>\n",
       "      <td>Synaptic proteins at the synaptic junction</td>\n",
       "      <td>NRXN3, SPTAN1</td>\n",
       "      <td>0.90841</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.5</td>\n",
       "      <td>10</td>\n",
       "      <td>Activation of TRKA receptors</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.5</td>\n",
       "      <td>12</td>\n",
       "      <td>FGFR3b ligand binding and activation</td>\n",
       "      <td>FGF9</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.5</td>\n",
       "      <td>13</td>\n",
       "      <td>Folate and pterine metabolism</td>\n",
       "      <td>SLC46A1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.5</td>\n",
       "      <td>14</td>\n",
       "      <td>Tat-mediated elongation of the HIV-1 transcript</td>\n",
       "      <td>CTDP1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   AUC  Rank                                        Term_Name  \\\n",
       "0  0.5     1       Synaptic proteins at the synaptic junction   \n",
       "1  0.5    10                     Activation of TRKA receptors   \n",
       "2  0.5    12             FGFR3b ligand binding and activation   \n",
       "3  0.5    13                    Folate and pterine metabolism   \n",
       "4  0.5    14  Tat-mediated elongation of the HIV-1 transcript   \n",
       "\n",
       "  Overlapping_Genes  Adjusted_P-value          Source  \n",
       "0     NRXN3, SPTAN1           0.90841  BioPlanet_2019  \n",
       "1             NTRK2           1.00000  BioPlanet_2019  \n",
       "2              FGF9           1.00000  BioPlanet_2019  \n",
       "3           SLC46A1           1.00000  BioPlanet_2019  \n",
       "4             CTDP1           1.00000  BioPlanet_2019  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enrichr_dataset['BioPlanet_2019'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_results(source, n=10, print_messages=False, single_plot = False):\n",
    "    source_name = source['Source'][0]\n",
    "    if not os.path.exists(f'./EnrichrAPIResults/Plots_for_{n}_categories'):\n",
    "        os.mkdir(f'./EnrichrAPIResults/Plots_for_{n}_categories')\n",
    "    output_file(filename=f'./EnrichrAPIResults/Plots_for_{n}_categories/n_{n}_results_from_{source_name}.html',\n",
    "               title=f'Bokeh plot of {source_name} (n={n})')\n",
    "    term_values = source['Term_Name'].value_counts()\n",
    "    if print_messages:\n",
    "        print(f'The most occuring term names in {source_name} is: \\n{term_values[:n]}')\n",
    "    plot_source = pd.DataFrame(columns=source.columns)\n",
    "    for auc in source['AUC'].unique():\n",
    "        subsource = source[source['AUC'] == auc][:n]\n",
    "        plot_source = plot_source.append(subsource)\n",
    "    category = 'Term_Name'\n",
    "    category_items = plot_source[category].unique()\n",
    "    palette = inferno(len(category_items) + 1)\n",
    "    colormap = dict(zip(category_items, palette))\n",
    "    plot_source['color'] = plot_source[category].map(colormap)\n",
    "    \n",
    "    title = f'Term names from {source_name}, n={n}'\n",
    "    source_bokeh = ColumnDataSource(plot_source)\n",
    "    hover = HoverTool(tooltips=[('Term_name', '@Term_Name'), ('Genes', '@Overlapping_Genes')])\n",
    "    tools = [hover, WheelZoomTool(), PanTool(), BoxZoomTool(), ResetTool(), SaveTool()]\n",
    "    \n",
    "    p = figure(tools=tools, title=title, plot_width=2000, plot_height=1200,toolbar_location='right', toolbar_sticky=False,)\n",
    "    if n > 5:\n",
    "        p.scatter(x='AUC',y='Adjusted_P-value', source=source_bokeh, size=10, color='color')\n",
    "    else:\n",
    "        p.scatter(x='AUC',y='Adjusted_P-value', source=source_bokeh, size=10, color='color', legend=category)\n",
    "    p.xaxis.axis_label = 'AUC'\n",
    "    p.yaxis.axis_label = 'Adjusted P-value'\n",
    "    return p\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "for key in enrichr_dataset.keys():\n",
    "    plot = plot_results(enrichr_dataset[key], 10)\n",
    "    save(plot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.5</td>\n",
       "      <td>1</td>\n",
       "      <td>Synaptic proteins at the synaptic junction</td>\n",
       "      <td>NRXN3, SPTAN1</td>\n",
       "      <td>0.90841</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.5</td>\n",
       "      <td>10</td>\n",
       "      <td>Activation of TRKA receptors</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.5</td>\n",
       "      <td>12</td>\n",
       "      <td>FGFR3b ligand binding and activation</td>\n",
       "      <td>FGF9</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.5</td>\n",
       "      <td>13</td>\n",
       "      <td>Folate and pterine metabolism</td>\n",
       "      <td>SLC46A1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.5</td>\n",
       "      <td>14</td>\n",
       "      <td>Tat-mediated elongation of the HIV-1 transcript</td>\n",
       "      <td>CTDP1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1864</th>\n",
       "      <td>1.0</td>\n",
       "      <td>6</td>\n",
       "      <td>chr21</td>\n",
       "      <td>APP, COL18A1, KCNE1, KCNE2, AIRE, TMPRSS3, ITG...</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1865</th>\n",
       "      <td>1.0</td>\n",
       "      <td>7</td>\n",
       "      <td>chr14</td>\n",
       "      <td>POMT2, ABCD4, SERPINA1, CHD8, PYGL, GPHN, IRF2...</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1866</th>\n",
       "      <td>1.0</td>\n",
       "      <td>8</td>\n",
       "      <td>chr22</td>\n",
       "      <td>SMARCB1, NCF4, MLC1, TCF20, TRIOBP, PEX26, CYP...</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1867</th>\n",
       "      <td>1.0</td>\n",
       "      <td>9</td>\n",
       "      <td>chr6 ssto hap7</td>\n",
       "      <td>SYNGAP1, TNXB, CYP21A2, TUBB, CSNK2B, NEU1, CO...</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1868</th>\n",
       "      <td>1.0</td>\n",
       "      <td>10</td>\n",
       "      <td>chr9</td>\n",
       "      <td>INVS, LRSAM1, PIGO, LAMC3, GLDC, DOCK8, POMT1,...</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1869 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      AUC Rank                                        Term_Name  \\\n",
       "0     0.5    1       Synaptic proteins at the synaptic junction   \n",
       "1     0.5   10                     Activation of TRKA receptors   \n",
       "2     0.5   12             FGFR3b ligand binding and activation   \n",
       "3     0.5   13                    Folate and pterine metabolism   \n",
       "4     0.5   14  Tat-mediated elongation of the HIV-1 transcript   \n",
       "...   ...  ...                                              ...   \n",
       "1864  1.0    6                                            chr21   \n",
       "1865  1.0    7                                            chr14   \n",
       "1866  1.0    8                                            chr22   \n",
       "1867  1.0    9                                   chr6 ssto hap7   \n",
       "1868  1.0   10                                             chr9   \n",
       "\n",
       "                                      Overlapping_Genes  Adjusted_P-value  \\\n",
       "0                                         NRXN3, SPTAN1           0.90841   \n",
       "1                                                 NTRK2           1.00000   \n",
       "2                                                  FGF9           1.00000   \n",
       "3                                               SLC46A1           1.00000   \n",
       "4                                                 CTDP1           1.00000   \n",
       "...                                                 ...               ...   \n",
       "1864  APP, COL18A1, KCNE1, KCNE2, AIRE, TMPRSS3, ITG...           1.00000   \n",
       "1865  POMT2, ABCD4, SERPINA1, CHD8, PYGL, GPHN, IRF2...           1.00000   \n",
       "1866  SMARCB1, NCF4, MLC1, TCF20, TRIOBP, PEX26, CYP...           1.00000   \n",
       "1867  SYNGAP1, TNXB, CYP21A2, TUBB, CSNK2B, NEU1, CO...           1.00000   \n",
       "1868  INVS, LRSAM1, PIGO, LAMC3, GLDC, DOCK8, POMT1,...           1.00000   \n",
       "\n",
       "                        Source  \n",
       "0               BioPlanet_2019  \n",
       "1               BioPlanet_2019  \n",
       "2               BioPlanet_2019  \n",
       "3               BioPlanet_2019  \n",
       "4               BioPlanet_2019  \n",
       "...                        ...  \n",
       "1864  Chromosome_Location_hg19  \n",
       "1865  Chromosome_Location_hg19  \n",
       "1866  Chromosome_Location_hg19  \n",
       "1867  Chromosome_Location_hg19  \n",
       "1868  Chromosome_Location_hg19  \n",
       "\n",
       "[1869 rows x 6 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df = pd.DataFrame(columns=enrichr_dataset['BioPlanet_2019'].columns)\n",
    "for key, value in enrichr_dataset.items():\n",
    "    subdf = pd.DataFrame(columns=combined_df.columns)\n",
    "    for auc in value['AUC'].unique():\n",
    "        subdf = subdf.append(value[value['AUC'] == auc][:10], ignore_index=True)\n",
    "    combined_df = combined_df.append(subdf, ignore_index=True)\n",
    "combined_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.50</td>\n",
       "      <td>1</td>\n",
       "      <td>Synaptic proteins at the synaptic junction</td>\n",
       "      <td>NRXN3, SPTAN1</td>\n",
       "      <td>0.90841</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.50</td>\n",
       "      <td>10</td>\n",
       "      <td>Activation of TRKA receptors</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.50</td>\n",
       "      <td>12</td>\n",
       "      <td>FGFR3b ligand binding and activation</td>\n",
       "      <td>FGF9</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.50</td>\n",
       "      <td>13</td>\n",
       "      <td>Folate and pterine metabolism</td>\n",
       "      <td>SLC46A1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.50</td>\n",
       "      <td>14</td>\n",
       "      <td>Tat-mediated elongation of the HIV-1 transcript</td>\n",
       "      <td>CTDP1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.50</td>\n",
       "      <td>15</td>\n",
       "      <td>Elevation of cytosolic calcium levels</td>\n",
       "      <td>TRPC6</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.50</td>\n",
       "      <td>18</td>\n",
       "      <td>Second messenger role in netrin-1 signaling</td>\n",
       "      <td>TRPC6</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.50</td>\n",
       "      <td>20</td>\n",
       "      <td>Signaling by activated point mutants of FGFR1</td>\n",
       "      <td>FGF9</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0.50</td>\n",
       "      <td>21</td>\n",
       "      <td>Caspase-mediated cleavage of cytoskeletal prot...</td>\n",
       "      <td>SPTAN1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>0.50</td>\n",
       "      <td>22</td>\n",
       "      <td>Tandem pore domain potassium channels</td>\n",
       "      <td>KCNK9</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>0.55</td>\n",
       "      <td>14</td>\n",
       "      <td>Phospholipase C delta-1 interactions in phosph...</td>\n",
       "      <td>PLCD1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>0.55</td>\n",
       "      <td>2</td>\n",
       "      <td>Synaptic proteins at the synaptic junction</td>\n",
       "      <td>NRXN3, SPTAN1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>0.55</td>\n",
       "      <td>18</td>\n",
       "      <td>Activation of TRKA receptors</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>0.55</td>\n",
       "      <td>19</td>\n",
       "      <td>Nicotine metabolism</td>\n",
       "      <td>CYP2A6</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>0.55</td>\n",
       "      <td>1</td>\n",
       "      <td>Netrin-1 signaling</td>\n",
       "      <td>TRPC6, HFE2, ROBO1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>0.55</td>\n",
       "      <td>22</td>\n",
       "      <td>Anandamide metabolism</td>\n",
       "      <td>BAAT</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>0.55</td>\n",
       "      <td>23</td>\n",
       "      <td>Caffeine metabolism</td>\n",
       "      <td>CYP2A6</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>0.55</td>\n",
       "      <td>24</td>\n",
       "      <td>FGFR3b ligand binding and activation</td>\n",
       "      <td>FGF9</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>0.55</td>\n",
       "      <td>4</td>\n",
       "      <td>Hypoxia and p53 in the cardiovascular system</td>\n",
       "      <td>CDKN1A, UBE2A</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>0.55</td>\n",
       "      <td>5</td>\n",
       "      <td>Triacylglyceride biosynthesis</td>\n",
       "      <td>LIPE, GPD1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     AUC Rank                                          Term_Name  \\\n",
       "0   0.50    1         Synaptic proteins at the synaptic junction   \n",
       "1   0.50   10                       Activation of TRKA receptors   \n",
       "2   0.50   12               FGFR3b ligand binding and activation   \n",
       "3   0.50   13                      Folate and pterine metabolism   \n",
       "4   0.50   14    Tat-mediated elongation of the HIV-1 transcript   \n",
       "5   0.50   15              Elevation of cytosolic calcium levels   \n",
       "6   0.50   18        Second messenger role in netrin-1 signaling   \n",
       "7   0.50   20      Signaling by activated point mutants of FGFR1   \n",
       "8   0.50   21  Caspase-mediated cleavage of cytoskeletal prot...   \n",
       "9   0.50   22              Tandem pore domain potassium channels   \n",
       "10  0.55   14  Phospholipase C delta-1 interactions in phosph...   \n",
       "11  0.55    2         Synaptic proteins at the synaptic junction   \n",
       "12  0.55   18                       Activation of TRKA receptors   \n",
       "13  0.55   19                                Nicotine metabolism   \n",
       "14  0.55    1                                 Netrin-1 signaling   \n",
       "15  0.55   22                              Anandamide metabolism   \n",
       "16  0.55   23                                Caffeine metabolism   \n",
       "17  0.55   24               FGFR3b ligand binding and activation   \n",
       "18  0.55    4       Hypoxia and p53 in the cardiovascular system   \n",
       "19  0.55    5                      Triacylglyceride biosynthesis   \n",
       "\n",
       "     Overlapping_Genes  Adjusted_P-value          Source  \n",
       "0        NRXN3, SPTAN1           0.90841  BioPlanet_2019  \n",
       "1                NTRK2           1.00000  BioPlanet_2019  \n",
       "2                 FGF9           1.00000  BioPlanet_2019  \n",
       "3              SLC46A1           1.00000  BioPlanet_2019  \n",
       "4                CTDP1           1.00000  BioPlanet_2019  \n",
       "5                TRPC6           1.00000  BioPlanet_2019  \n",
       "6                TRPC6           1.00000  BioPlanet_2019  \n",
       "7                 FGF9           1.00000  BioPlanet_2019  \n",
       "8               SPTAN1           1.00000  BioPlanet_2019  \n",
       "9                KCNK9           1.00000  BioPlanet_2019  \n",
       "10               PLCD1           1.00000  BioPlanet_2019  \n",
       "11       NRXN3, SPTAN1           1.00000  BioPlanet_2019  \n",
       "12               NTRK2           1.00000  BioPlanet_2019  \n",
       "13              CYP2A6           1.00000  BioPlanet_2019  \n",
       "14  TRPC6, HFE2, ROBO1           1.00000  BioPlanet_2019  \n",
       "15                BAAT           1.00000  BioPlanet_2019  \n",
       "16              CYP2A6           1.00000  BioPlanet_2019  \n",
       "17                FGF9           1.00000  BioPlanet_2019  \n",
       "18       CDKN1A, UBE2A           1.00000  BioPlanet_2019  \n",
       "19          LIPE, GPD1           1.00000  BioPlanet_2019  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_results_in_singleplot(source):\n",
    "    output_file(filename='./EnrichrAPIResults/combined_enrichr_plot_results.html',\n",
    "               title='Combined enrichr data sources plot')\n",
    "    category = 'Source'\n",
    "    category_items = source[category].unique()\n",
    "    palette = inferno(len(category_items) + 1)\n",
    "    colormap = dict(zip(category_items, palette))\n",
    "    print(colormap)\n",
    "    source['color'] = source[category].map(colormap)\n",
    "    \n",
    "    title = f'Combined enrichr plot results.'\n",
    "    source_bokeh = ColumnDataSource(source)\n",
    "    hover = HoverTool(tooltips=[('Term_name', '@Term_Name'),\n",
    "                                ('Genes', '@Overlapping_Genes'),\n",
    "                               ('Source', '@Source')])\n",
    "    tools = [hover, WheelZoomTool(), PanTool(), BoxZoomTool(), ResetTool(), SaveTool()]\n",
    "    \n",
    "    p = figure(tools=tools, title=title, plot_width=2000, plot_height=1200,toolbar_location='right', toolbar_sticky=False,)\n",
    "    p.scatter(x='AUC',y='Adjusted_P-value', source=source_bokeh, size=10, color='color', legend=category)\n",
    "    p.xaxis.axis_label = 'AUC'\n",
    "    p.yaxis.axis_label = 'Adjusted P-value'\n",
    "    p.legend.click_policy=\"hide\"\n",
    "    return p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "plot = plot_results_in_singleplot(combined_df)\n",
    "save(plot)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "combined_df_ext = pd.DataFrame(columns=enrichr_dataset['BioPlanet_2019'].columns)\n",
    "for key, value in enrichr_dataset.items():\n",
    "    subdf = pd.DataFrame(columns=combined_df_ext.columns)\n",
    "    for auc in value['AUC'].unique():\n",
    "        subdf = subdf.append(value[value['AUC'] == auc], ignore_index=True)\n",
    "    combined_df_ext = combined_df_ext.append(subdf, ignore_index=True)\n",
    "combined_df_ext"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "combined_df_ext.where((combined_df_ext['AUC'] == 0.5) & (combined_df_ext['Source'] == 'BioPlanet_2019')).dropna()"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "# https://gist.github.com/deekayen/4148741#file-1-1000-txt\n",
    "\n",
    "common_used_words = []\n",
    "with open('/home/rjsietsma/Documents/1-1000.txt') as file:\n",
    "    lines = file.readlines()\n",
    "    for line in lines:\n",
    "        line = line.strip()\n",
    "        common_used_words.append(line)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "word_count_df = pd.DataFrame(columns=['word', 'count', 'auc'])\n",
    "for auc in combined_df_ext['AUC'].unique():\n",
    "    word_list = []\n",
    "    for source in combined_df_ext['Source'].unique():\n",
    "        subset = combined_df_ext.where((combined_df_ext['AUC'] == auc) & (combined_df_ext['Source'] == source)).dropna()\n",
    "        for i, row in subset.iterrows():\n",
    "            term_name = row['Term_Name']\n",
    "            word_list += term_name.lower().split()\n",
    "    word_count_dict = {}\n",
    "    for word in word_list:\n",
    "        if word in common_used_words:\n",
    "            continue\n",
    "        if word not in word_count_dict.keys():\n",
    "            word_count_dict[word] = 1\n",
    "        else:\n",
    "            word_count_dict[word] += 1\n",
    "    temp_df = pd.DataFrame(columns=word_count_df.columns)\n",
    "    for key in word_count_dict.keys():\n",
    "        temp_temp_df = pd.DataFrame({'word':key, 'count': word_count_dict[key],\n",
    "                                    'auc': auc}, index=[0])\n",
    "        temp_df = temp_df.append(temp_temp_df, ignore_index=True)\n",
    "    word_count_df = word_count_df.append(temp_df, ignore_index=True)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "word_count_df.sort_values(by=['auc','count'], ascending=[True, False], inplace=True)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "nwkdf = pd.DataFrame(columns=word_count_df.columns)\n",
    "for auc in word_count_df['auc'].unique():\n",
    "    nwkdf = nwkdf.append(word_count_df[word_count_df['auc'] == auc][:5])\n",
    "nwkdf"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "def plot_word_count_results(source):\n",
    "    output_file(filename='./EnrichrAPIResults/enrichr_word_count_bokeh.html',\n",
    "                title='Word count plot')\n",
    "    category = 'word'\n",
    "    category_items = source[category].unique()\n",
    "    palette = inferno(len(category_items) + 1)\n",
    "    colormap = dict(zip(category_items, palette))\n",
    "    source['color'] = source[category].map(colormap)\n",
    "    \n",
    "    source['auc'] = source['auc'].astype(str)\n",
    "    \n",
    "    group = source.groupby(by=['auc', 'word']).sum()\n",
    "    group.sort_values(by=['auc','count'], ascending=[True,False], inplace=True)\n",
    "    \n",
    "    title = f'Word count bar plot'\n",
    "    \n",
    "    factors = group.index.tolist()\n",
    "    x = group['count'].tolist()\n",
    "    x_str = group['count'].astype(str).tolist()\n",
    "    colors = group['color'].tolist()\n",
    "    \n",
    "    temp_source = ColumnDataSource(source)\n",
    "        \n",
    "    tooltips = [('Counts', '@count')]\n",
    "    \n",
    "    p = figure(x_range=FactorRange(*factors),\n",
    "               title=title,\n",
    "               plot_width=2000,\n",
    "               plot_height=1200,\n",
    "               toolbar_location='right',\n",
    "               toolbar_sticky=False)\n",
    "    p.vbar(x=factors,\n",
    "           top=x,\n",
    "           color=colors,\n",
    "           width=0.9)\n",
    "    p.add_tools(HoverTool(tooltips=[('Counts','@top')]))\n",
    "    p.xaxis.axis_label = 'Word'\n",
    "    p.yaxis.axis_label = 'Count'\n",
    "    p.xaxis.major_label_orientation = 1.5\n",
    "    return p"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "plot = plot_word_count_results(nwkdf)\n",
    "save(plot)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "gene_count_df = pd.DataFrame(columns=['gene', 'count', 'auc'])\n",
    "for auc in combined_df_ext['AUC'].unique():\n",
    "    subset_df = combined_df_ext[combined_df_ext['AUC'] == auc]\n",
    "    gene_list = []\n",
    "    unsplitted_gene_list = subset_df['Overlapping_Genes'].tolist()\n",
    "    for unsplitted_gene in unsplitted_gene_list:\n",
    "        gene_list += unsplitted_gene.split(', ')\n",
    "    gene_count_dict = {}\n",
    "    for gene in gene_list:\n",
    "        if gene not in gene_count_dict.keys():\n",
    "            gene_count_dict[gene] = 1\n",
    "        else:\n",
    "            gene_count_dict[gene] += 1\n",
    "    temp_df = pd.DataFrame(columns=gene_count_df.columns)\n",
    "    for key in gene_count_dict.keys():\n",
    "        temp_temp_df = pd.DataFrame({'gene':key, 'count': gene_count_dict[key],\n",
    "                                    'auc': auc}, index=[0])\n",
    "        temp_df = temp_df.append(temp_temp_df, ignore_index=True)\n",
    "    gene_count_df = gene_count_df.append(temp_df, ignore_index=True)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "gene_count_df.sort_values(by=['auc','count'], ascending=[True, False], inplace=True)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "ngkdf = pd.DataFrame(columns=gene_count_df.columns)\n",
    "for auc in gene_count_df['auc'].unique():\n",
    "    ngkdf = ngkdf.append(gene_count_df[gene_count_df['auc'] == auc][:5])"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "def plot_gene_count_results(source):\n",
    "    output_file(filename='./EnrichrAPIResults/enrichr_gene_count_bokeh.html',\n",
    "                title='Gene count plot')\n",
    "    \n",
    "    category = 'gene'\n",
    "    category_items = source[category].unique()\n",
    "    palette = inferno(len(category_items) + 1)\n",
    "    colormap = dict(zip(category_items, palette))\n",
    "    source['color'] = source[category].map(colormap)\n",
    "    \n",
    "    source['auc'] = source['auc'].astype(str)\n",
    "    \n",
    "    group = source.groupby(by=['auc', 'gene']).sum()\n",
    "    group.sort_values(by=['auc','count'], ascending=[True,False], inplace=True)\n",
    "    \n",
    "    title = f'Gene count bar plot'\n",
    "    \n",
    "    factors = group.index.tolist()\n",
    "    x = group['count'].tolist()\n",
    "    x_str = group['count'].astype(str).tolist()\n",
    "    colors = group['color'].tolist()\n",
    "    \n",
    "    temp_source = ColumnDataSource(source)\n",
    "        \n",
    "    tooltips = [('Counts', '@count')]\n",
    "    \n",
    "    p = figure(x_range=FactorRange(*factors),\n",
    "               title=title,\n",
    "               plot_width=2000,\n",
    "               plot_height=1200,\n",
    "               toolbar_location='right',\n",
    "               toolbar_sticky=False)\n",
    "    p.vbar(x=factors,\n",
    "           top=x,\n",
    "           color=colors,\n",
    "           width=0.9)\n",
    "    p.add_tools(HoverTool(tooltips=[('Counts','@top')]))\n",
    "    p.xaxis.axis_label = 'Gene'\n",
    "    p.yaxis.axis_label = 'Count'\n",
    "    p.xaxis.major_label_orientation = 1.5\n",
    "    return p"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "plot = plot_gene_count_results(ngkdf)\n",
    "save(plot)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "cdfe = combined_df_ext.copy()\n",
    "cdfed = cdfe.copy()\n",
    "cdfe.drop(['Overlapping_Genes', 'Rank'], axis=1, inplace=True)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "fcdfe = pd.DataFrame(columns=cdfe.columns)\n",
    "for auc in cdfe['AUC'].unique():\n",
    "    subset = cdfe[cdfe['AUC'] == auc]\n",
    "    subset.sort_values(by='Adjusted_P-value', ascending=True, inplace=True)\n",
    "    subset = subset[:5]\n",
    "    fcdfe = fcdfe.append(subset, ignore_index=True)\n",
    "fcdfe.to_csv('./EnrichrAPIResults/combined_df_nondetailed_paper.csv', index=False)\n",
    "fcdfe"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "fcdfed = pd.DataFrame(columns=cdfed.columns)\n",
    "for auc in cdfed['AUC'].unique():\n",
    "    subset = cdfed[cdfed['AUC'] == auc]\n",
    "    subset.sort_values(by='Adjusted_P-value', ascending=True, inplace=True)\n",
    "    subset = subset[:5]\n",
    "    fcdfed = fcdfed.append(subset, ignore_index=True)\n",
    "fcdfed.to_csv('./EnrichrAPIResults/combined_df_detailed_paper.csv', index=False)\n",
    "fcdfed"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "fcdfede = pd.DataFrame(columns=cdfed.columns)\n",
    "for auc in cdfed['AUC'].unique():\n",
    "    subset = cdfed[cdfed['AUC'] == auc]\n",
    "    subset.sort_values(by='Adjusted_P-value', ascending=True, inplace=True)\n",
    "    fcdfede = fcdfede.append(subset, ignore_index=True)\n",
    "fcdfede.to_csv('./EnrichrAPIResults/combined_df_detailed_paper_extended.csv', index=False)\n",
    "fcdfede"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "## Interpretation\n",
    "It seems that a normal distribution like pattern is formed with a steeper slope after the maximum, starting at rougly AUC of 0.3 and ending at roughly 0.8.\n",
    "<br/><br/>\n",
    "- DisGeNET seems to outline an interesting picture with the top 10 term names:\n",
    "    - For it seems that in the really bad performing genes the main cause could be infant development related. Later on, more neurological causes are identified.\n",
    "        - Neurological seems clear to me... There is still so much unknown about the human mind.\n",
    "        - Infant development related is a bit less clear, since there is a growing amount of research towards this infant development.\n",
    "- GO_biological_process seems to have primarily the same genes for AUC 0.55-0.7 that are most enriched\n",
    "    - It seems linked to ion channeling\n",
    "- Feet and toe abnormalities seem to occur once or twice too\n",
    "- Could this be an indication to genes that are related to very small molecules? (like steroids, ions, neuro-molecules?)\n",
    "- Most significant: autosomal dominant inheritence\n",
    "    - Very interesting plot, the HPO bokeh plot of n=10\n",
    "- \n",
    "\n",
    "<br/><br/>\n",
    "TODO: make word counts of genes and term names?\n"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'BioPlanet_2019': '#000003', 'WikiPathways_2019_Human': '#090621', 'KEGG_2019_Human': '#1D0C45', 'Elsevier_Pathway_Collection': '#370961', 'GO_Biological_Process_2018': '#500D6C', 'Human_Phenotype_Ontology': '#68166E', 'Jensen_DISEASES': '#801F6B', 'PheWeb_2019': '#982765', 'ClinVar_2019': '#AF315B', 'GWAS_Catalog_2019': '#C53D4D', 'DisGeNET': '#D94D3D', 'OMIM_Disease': '#E8612B', 'OMIM_Expanded': '#F37918', 'Human_Gene_Atlas': '#FA9306', 'ARCHS4_Tissues': '#FBB014', 'HMDB_Metabolites': '#F8CD37', 'Chromosome_Location_hg19': '#F1E968'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "BokehDeprecationWarning: 'legend' keyword is deprecated, use explicit 'legend_label', 'legend_field', or 'legend_group' keywords instead\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/combined_enrichr_plot_results.html'"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Interpretation\n",
    "It seems that a normal distribution like pattern is formed with a steeper slope after the maximum, starting at rougly AUC of 0.3 and ending at roughly 0.8.\n",
    "<br/><br/>\n",
    "- DisGeNET seems to outline an interesting picture with the top 10 term names:\n",
    "    - For it seems that in the really bad performing genes the main cause could be infant development related. Later on, more neurological causes are identified.\n",
    "        - Neurological seems clear to me... There is still so much unknown about the human mind.\n",
    "        - Infant development related is a bit less clear, since there is a growing amount of research towards this infant development.\n",
    "- GO_biological_process seems to have primarily the same genes for AUC 0.55-0.7 that are most enriched\n",
    "    - It seems linked to ion channeling\n",
    "- Feet and toe abnormalities seem to occur once or twice too\n",
    "- Could this be an indication to genes that are related to very small molecules? (like steroids, ions, neuro-molecules?)\n",
    "- Most significant: autosomal dominant inheritence\n",
    "    - Very interesting plot, the HPO bokeh plot of n=10\n",
    "- \n",
    "\n",
    "<br/><br/>\n",
    "TODO: make word counts of genes and term names?\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.5</td>\n",
       "      <td>1</td>\n",
       "      <td>Synaptic proteins at the synaptic junction</td>\n",
       "      <td>NRXN3, SPTAN1</td>\n",
       "      <td>0.90841</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.5</td>\n",
       "      <td>10</td>\n",
       "      <td>Activation of TRKA receptors</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.5</td>\n",
       "      <td>12</td>\n",
       "      <td>FGFR3b ligand binding and activation</td>\n",
       "      <td>FGF9</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.5</td>\n",
       "      <td>13</td>\n",
       "      <td>Folate and pterine metabolism</td>\n",
       "      <td>SLC46A1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.5</td>\n",
       "      <td>14</td>\n",
       "      <td>Tat-mediated elongation of the HIV-1 transcript</td>\n",
       "      <td>CTDP1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3698</th>\n",
       "      <td>1.0</td>\n",
       "      <td>15</td>\n",
       "      <td>chr6 dbb hap3</td>\n",
       "      <td>TNXB, CYP21A2, TUBB, CSNK2B, NEU1, COL11A2, SK...</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3699</th>\n",
       "      <td>1.0</td>\n",
       "      <td>19</td>\n",
       "      <td>chr20</td>\n",
       "      <td>PTPRT, SNAP25, PANK2, PIGT, MOCS3, ABHD12, JPH...</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3700</th>\n",
       "      <td>1.0</td>\n",
       "      <td>17</td>\n",
       "      <td>chr6 qbl hap6</td>\n",
       "      <td>TNXB, CYP21A2, TUBB, CSNK2B, NEU1, COL11A2, SK...</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3701</th>\n",
       "      <td>1.0</td>\n",
       "      <td>20</td>\n",
       "      <td>chr11</td>\n",
       "      <td>CARS, DPAGT1, CLPB, CTNND1, DNHD1, SBF2-AS1, B...</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3702</th>\n",
       "      <td>1.0</td>\n",
       "      <td>18</td>\n",
       "      <td>chr6 apd hap1</td>\n",
       "      <td>TUBB, NEU1, PSMB8</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3703 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      AUC Rank                                        Term_Name  \\\n",
       "0     0.5    1       Synaptic proteins at the synaptic junction   \n",
       "1     0.5   10                     Activation of TRKA receptors   \n",
       "2     0.5   12             FGFR3b ligand binding and activation   \n",
       "3     0.5   13                    Folate and pterine metabolism   \n",
       "4     0.5   14  Tat-mediated elongation of the HIV-1 transcript   \n",
       "...   ...  ...                                              ...   \n",
       "3698  1.0   15                                    chr6 dbb hap3   \n",
       "3699  1.0   19                                            chr20   \n",
       "3700  1.0   17                                    chr6 qbl hap6   \n",
       "3701  1.0   20                                            chr11   \n",
       "3702  1.0   18                                    chr6 apd hap1   \n",
       "\n",
       "                                      Overlapping_Genes  Adjusted_P-value  \\\n",
       "0                                         NRXN3, SPTAN1           0.90841   \n",
       "1                                                 NTRK2           1.00000   \n",
       "2                                                  FGF9           1.00000   \n",
       "3                                               SLC46A1           1.00000   \n",
       "4                                                 CTDP1           1.00000   \n",
       "...                                                 ...               ...   \n",
       "3698  TNXB, CYP21A2, TUBB, CSNK2B, NEU1, COL11A2, SK...           1.00000   \n",
       "3699  PTPRT, SNAP25, PANK2, PIGT, MOCS3, ABHD12, JPH...           1.00000   \n",
       "3700  TNXB, CYP21A2, TUBB, CSNK2B, NEU1, COL11A2, SK...           1.00000   \n",
       "3701  CARS, DPAGT1, CLPB, CTNND1, DNHD1, SBF2-AS1, B...           1.00000   \n",
       "3702                                  TUBB, NEU1, PSMB8           1.00000   \n",
       "\n",
       "                        Source  \n",
       "0               BioPlanet_2019  \n",
       "1               BioPlanet_2019  \n",
       "2               BioPlanet_2019  \n",
       "3               BioPlanet_2019  \n",
       "4               BioPlanet_2019  \n",
       "...                        ...  \n",
       "3698  Chromosome_Location_hg19  \n",
       "3699  Chromosome_Location_hg19  \n",
       "3700  Chromosome_Location_hg19  \n",
       "3701  Chromosome_Location_hg19  \n",
       "3702  Chromosome_Location_hg19  \n",
       "\n",
       "[3703 rows x 6 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df_ext = pd.DataFrame(columns=enrichr_dataset['BioPlanet_2019'].columns)\n",
    "for key, value in enrichr_dataset.items():\n",
    "    subdf = pd.DataFrame(columns=combined_df_ext.columns)\n",
    "    for auc in value['AUC'].unique():\n",
    "        subdf = subdf.append(value[value['AUC'] == auc], ignore_index=True)\n",
    "    combined_df_ext = combined_df_ext.append(subdf, ignore_index=True)\n",
    "combined_df_ext"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.5</td>\n",
       "      <td>1</td>\n",
       "      <td>Synaptic proteins at the synaptic junction</td>\n",
       "      <td>NRXN3, SPTAN1</td>\n",
       "      <td>0.90841</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.5</td>\n",
       "      <td>10</td>\n",
       "      <td>Activation of TRKA receptors</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.5</td>\n",
       "      <td>12</td>\n",
       "      <td>FGFR3b ligand binding and activation</td>\n",
       "      <td>FGF9</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.5</td>\n",
       "      <td>13</td>\n",
       "      <td>Folate and pterine metabolism</td>\n",
       "      <td>SLC46A1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.5</td>\n",
       "      <td>14</td>\n",
       "      <td>Tat-mediated elongation of the HIV-1 transcript</td>\n",
       "      <td>CTDP1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.5</td>\n",
       "      <td>15</td>\n",
       "      <td>Elevation of cytosolic calcium levels</td>\n",
       "      <td>TRPC6</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.5</td>\n",
       "      <td>18</td>\n",
       "      <td>Second messenger role in netrin-1 signaling</td>\n",
       "      <td>TRPC6</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.5</td>\n",
       "      <td>20</td>\n",
       "      <td>Signaling by activated point mutants of FGFR1</td>\n",
       "      <td>FGF9</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0.5</td>\n",
       "      <td>21</td>\n",
       "      <td>Caspase-mediated cleavage of cytoskeletal prot...</td>\n",
       "      <td>SPTAN1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>0.5</td>\n",
       "      <td>22</td>\n",
       "      <td>Tandem pore domain potassium channels</td>\n",
       "      <td>KCNK9</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>0.5</td>\n",
       "      <td>23</td>\n",
       "      <td>Opening of calcium channels triggered by depol...</td>\n",
       "      <td>CACNB2</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>0.5</td>\n",
       "      <td>24</td>\n",
       "      <td>FGFR3 ligand binding and activation</td>\n",
       "      <td>FGF9</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>0.5</td>\n",
       "      <td>25</td>\n",
       "      <td>Ghrelin-mediated regulation of food intake and...</td>\n",
       "      <td>IGF1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>0.5</td>\n",
       "      <td>26</td>\n",
       "      <td>FGFR4 ligand binding and activation</td>\n",
       "      <td>FGF9</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>0.5</td>\n",
       "      <td>2</td>\n",
       "      <td>Netrin-1 signaling</td>\n",
       "      <td>TRPC6, HFE2</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>0.5</td>\n",
       "      <td>27</td>\n",
       "      <td>HIV-1 early elongation complex formation</td>\n",
       "      <td>CTDP1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>0.5</td>\n",
       "      <td>28</td>\n",
       "      <td>HDL-mediated lipid transport</td>\n",
       "      <td>SCARB1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>0.5</td>\n",
       "      <td>29</td>\n",
       "      <td>Mu-calpain pathway</td>\n",
       "      <td>SPTAN1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>0.5</td>\n",
       "      <td>30</td>\n",
       "      <td>Validated nuclear estrogen receptor beta network</td>\n",
       "      <td>ESR2</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>0.5</td>\n",
       "      <td>31</td>\n",
       "      <td>Erythrocyte differentiation pathway</td>\n",
       "      <td>IGF1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    AUC Rank                                          Term_Name  \\\n",
       "0   0.5    1         Synaptic proteins at the synaptic junction   \n",
       "1   0.5   10                       Activation of TRKA receptors   \n",
       "2   0.5   12               FGFR3b ligand binding and activation   \n",
       "3   0.5   13                      Folate and pterine metabolism   \n",
       "4   0.5   14    Tat-mediated elongation of the HIV-1 transcript   \n",
       "5   0.5   15              Elevation of cytosolic calcium levels   \n",
       "6   0.5   18        Second messenger role in netrin-1 signaling   \n",
       "7   0.5   20      Signaling by activated point mutants of FGFR1   \n",
       "8   0.5   21  Caspase-mediated cleavage of cytoskeletal prot...   \n",
       "9   0.5   22              Tandem pore domain potassium channels   \n",
       "10  0.5   23  Opening of calcium channels triggered by depol...   \n",
       "11  0.5   24                FGFR3 ligand binding and activation   \n",
       "12  0.5   25  Ghrelin-mediated regulation of food intake and...   \n",
       "13  0.5   26                FGFR4 ligand binding and activation   \n",
       "14  0.5    2                                 Netrin-1 signaling   \n",
       "15  0.5   27           HIV-1 early elongation complex formation   \n",
       "16  0.5   28                       HDL-mediated lipid transport   \n",
       "17  0.5   29                                 Mu-calpain pathway   \n",
       "18  0.5   30   Validated nuclear estrogen receptor beta network   \n",
       "19  0.5   31                Erythrocyte differentiation pathway   \n",
       "\n",
       "   Overlapping_Genes  Adjusted_P-value          Source  \n",
       "0      NRXN3, SPTAN1           0.90841  BioPlanet_2019  \n",
       "1              NTRK2           1.00000  BioPlanet_2019  \n",
       "2               FGF9           1.00000  BioPlanet_2019  \n",
       "3            SLC46A1           1.00000  BioPlanet_2019  \n",
       "4              CTDP1           1.00000  BioPlanet_2019  \n",
       "5              TRPC6           1.00000  BioPlanet_2019  \n",
       "6              TRPC6           1.00000  BioPlanet_2019  \n",
       "7               FGF9           1.00000  BioPlanet_2019  \n",
       "8             SPTAN1           1.00000  BioPlanet_2019  \n",
       "9              KCNK9           1.00000  BioPlanet_2019  \n",
       "10            CACNB2           1.00000  BioPlanet_2019  \n",
       "11              FGF9           1.00000  BioPlanet_2019  \n",
       "12              IGF1           1.00000  BioPlanet_2019  \n",
       "13              FGF9           1.00000  BioPlanet_2019  \n",
       "14       TRPC6, HFE2           1.00000  BioPlanet_2019  \n",
       "15             CTDP1           1.00000  BioPlanet_2019  \n",
       "16            SCARB1           1.00000  BioPlanet_2019  \n",
       "17            SPTAN1           1.00000  BioPlanet_2019  \n",
       "18              ESR2           1.00000  BioPlanet_2019  \n",
       "19              IGF1           1.00000  BioPlanet_2019  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df_ext.where((combined_df_ext['AUC'] == 0.5) & (combined_df_ext['Source'] == 'BioPlanet_2019')).dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# https://gist.github.com/deekayen/4148741#file-1-1000-txt\n",
    "\n",
    "common_used_words = []\n",
    "with open('/home/rjsietsma/Documents/1-1000.txt') as file:\n",
    "    lines = file.readlines()\n",
    "    for line in lines:\n",
    "        line = line.strip()\n",
    "        common_used_words.append(line)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "word_count_df = pd.DataFrame(columns=['word', 'count', 'auc'])\n",
    "for auc in combined_df_ext['AUC'].unique():\n",
    "    word_list = []\n",
    "    for source in combined_df_ext['Source'].unique():\n",
    "        subset = combined_df_ext.where((combined_df_ext['AUC'] == auc) & (combined_df_ext['Source'] == source)).dropna()\n",
    "        for i, row in subset.iterrows():\n",
    "            term_name = row['Term_Name']\n",
    "            word_list += term_name.lower().split()\n",
    "    word_count_dict = {}\n",
    "    for word in word_list:\n",
    "        if word in common_used_words:\n",
    "            continue\n",
    "        if word not in word_count_dict.keys():\n",
    "            word_count_dict[word] = 1\n",
    "        else:\n",
    "            word_count_dict[word] += 1\n",
    "    temp_df = pd.DataFrame(columns=word_count_df.columns)\n",
    "    for key in word_count_dict.keys():\n",
    "        temp_temp_df = pd.DataFrame({'word':key, 'count': word_count_dict[key],\n",
    "                                    'auc': auc}, index=[0])\n",
    "        temp_df = temp_df.append(temp_temp_df, ignore_index=True)\n",
    "    word_count_df = word_count_df.append(temp_df, ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "word_count_df.sort_values(by=['auc','count'], ascending=[True, False], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>count</th>\n",
       "      <th>auc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>signaling</td>\n",
       "      <td>14</td>\n",
       "      <td>0.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>pathway</td>\n",
       "      <td>14</td>\n",
       "      <td>0.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>regulation</td>\n",
       "      <td>12</td>\n",
       "      <td>0.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>disease</td>\n",
       "      <td>12</td>\n",
       "      <td>0.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>317</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>9</td>\n",
       "      <td>0.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>592</th>\n",
       "      <td>signaling</td>\n",
       "      <td>16</td>\n",
       "      <td>0.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>628</th>\n",
       "      <td>pathway</td>\n",
       "      <td>13</td>\n",
       "      <td>0.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>691</th>\n",
       "      <td>disease</td>\n",
       "      <td>12</td>\n",
       "      <td>0.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>761</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>11</td>\n",
       "      <td>0.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>600</th>\n",
       "      <td>metabolism</td>\n",
       "      <td>10</td>\n",
       "      <td>0.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1222</th>\n",
       "      <td>signaling</td>\n",
       "      <td>13</td>\n",
       "      <td>0.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1263</th>\n",
       "      <td>pathway</td>\n",
       "      <td>12</td>\n",
       "      <td>0.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1381</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>11</td>\n",
       "      <td>0.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1236</th>\n",
       "      <td>metabolism</td>\n",
       "      <td>10</td>\n",
       "      <td>0.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1322</th>\n",
       "      <td>disease</td>\n",
       "      <td>10</td>\n",
       "      <td>0.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1988</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>14</td>\n",
       "      <td>0.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1873</th>\n",
       "      <td>regulation</td>\n",
       "      <td>12</td>\n",
       "      <td>0.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1878</th>\n",
       "      <td>metabolism</td>\n",
       "      <td>11</td>\n",
       "      <td>0.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2131</th>\n",
       "      <td>disease</td>\n",
       "      <td>11</td>\n",
       "      <td>0.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1995</th>\n",
       "      <td>ion</td>\n",
       "      <td>10</td>\n",
       "      <td>0.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2449</th>\n",
       "      <td>acid</td>\n",
       "      <td>14</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2618</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>14</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2450</th>\n",
       "      <td>metabolism</td>\n",
       "      <td>12</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2458</th>\n",
       "      <td>signaling</td>\n",
       "      <td>12</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2481</th>\n",
       "      <td>pathway</td>\n",
       "      <td>12</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3058</th>\n",
       "      <td>pathway</td>\n",
       "      <td>14</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3127</th>\n",
       "      <td>cholesterol</td>\n",
       "      <td>13</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3083</th>\n",
       "      <td>signaling</td>\n",
       "      <td>11</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3189</th>\n",
       "      <td>involved</td>\n",
       "      <td>9</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3223</th>\n",
       "      <td>disease</td>\n",
       "      <td>9</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3662</th>\n",
       "      <td>pathway</td>\n",
       "      <td>20</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3688</th>\n",
       "      <td>signaling</td>\n",
       "      <td>15</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3778</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>13</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3672</th>\n",
       "      <td>regulation</td>\n",
       "      <td>11</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3712</th>\n",
       "      <td>cancer</td>\n",
       "      <td>9</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4215</th>\n",
       "      <td>pathway</td>\n",
       "      <td>20</td>\n",
       "      <td>0.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4219</th>\n",
       "      <td>signaling</td>\n",
       "      <td>17</td>\n",
       "      <td>0.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4244</th>\n",
       "      <td>regulation</td>\n",
       "      <td>11</td>\n",
       "      <td>0.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4295</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>10</td>\n",
       "      <td>0.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4319</th>\n",
       "      <td>cholesterol</td>\n",
       "      <td>10</td>\n",
       "      <td>0.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4773</th>\n",
       "      <td>cardiomyopathy</td>\n",
       "      <td>20</td>\n",
       "      <td>0.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4784</th>\n",
       "      <td>pathway</td>\n",
       "      <td>14</td>\n",
       "      <td>0.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4794</th>\n",
       "      <td>regulation</td>\n",
       "      <td>13</td>\n",
       "      <td>0.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4816</th>\n",
       "      <td>cancer</td>\n",
       "      <td>12</td>\n",
       "      <td>0.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4772</th>\n",
       "      <td>dilated</td>\n",
       "      <td>11</td>\n",
       "      <td>0.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5320</th>\n",
       "      <td>cancer</td>\n",
       "      <td>29</td>\n",
       "      <td>0.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5339</th>\n",
       "      <td>cardiomyopathy</td>\n",
       "      <td>18</td>\n",
       "      <td>0.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5337</th>\n",
       "      <td>pathway</td>\n",
       "      <td>12</td>\n",
       "      <td>0.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5338</th>\n",
       "      <td>dilated</td>\n",
       "      <td>11</td>\n",
       "      <td>0.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5377</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>10</td>\n",
       "      <td>0.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5816</th>\n",
       "      <td>cancer</td>\n",
       "      <td>23</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5837</th>\n",
       "      <td>cardiomyopathy</td>\n",
       "      <td>18</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5825</th>\n",
       "      <td>signaling</td>\n",
       "      <td>12</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5836</th>\n",
       "      <td>dilated</td>\n",
       "      <td>11</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5852</th>\n",
       "      <td>pathway</td>\n",
       "      <td>11</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                word count   auc\n",
       "23         signaling    14  0.50\n",
       "52           pathway    14  0.50\n",
       "42        regulation    12  0.50\n",
       "85           disease    12  0.50\n",
       "317         syndrome     9  0.50\n",
       "592        signaling    16  0.55\n",
       "628          pathway    13  0.55\n",
       "691          disease    12  0.55\n",
       "761         syndrome    11  0.55\n",
       "600       metabolism    10  0.55\n",
       "1222       signaling    13  0.60\n",
       "1263         pathway    12  0.60\n",
       "1381        syndrome    11  0.60\n",
       "1236      metabolism    10  0.60\n",
       "1322         disease    10  0.60\n",
       "1988        syndrome    14  0.65\n",
       "1873      regulation    12  0.65\n",
       "1878      metabolism    11  0.65\n",
       "2131         disease    11  0.65\n",
       "1995             ion    10  0.65\n",
       "2449            acid    14  0.70\n",
       "2618        syndrome    14  0.70\n",
       "2450      metabolism    12  0.70\n",
       "2458       signaling    12  0.70\n",
       "2481         pathway    12  0.70\n",
       "3058         pathway    14  0.75\n",
       "3127     cholesterol    13  0.75\n",
       "3083       signaling    11  0.75\n",
       "3189        involved     9  0.75\n",
       "3223         disease     9  0.75\n",
       "3662         pathway    20  0.80\n",
       "3688       signaling    15  0.80\n",
       "3778        syndrome    13  0.80\n",
       "3672      regulation    11  0.80\n",
       "3712          cancer     9  0.80\n",
       "4215         pathway    20  0.85\n",
       "4219       signaling    17  0.85\n",
       "4244      regulation    11  0.85\n",
       "4295        syndrome    10  0.85\n",
       "4319     cholesterol    10  0.85\n",
       "4773  cardiomyopathy    20  0.90\n",
       "4784         pathway    14  0.90\n",
       "4794      regulation    13  0.90\n",
       "4816          cancer    12  0.90\n",
       "4772         dilated    11  0.90\n",
       "5320          cancer    29  0.95\n",
       "5339  cardiomyopathy    18  0.95\n",
       "5337         pathway    12  0.95\n",
       "5338         dilated    11  0.95\n",
       "5377        syndrome    10  0.95\n",
       "5816          cancer    23  1.00\n",
       "5837  cardiomyopathy    18  1.00\n",
       "5825       signaling    12  1.00\n",
       "5836         dilated    11  1.00\n",
       "5852         pathway    11  1.00"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nwkdf = pd.DataFrame(columns=word_count_df.columns)\n",
    "for auc in word_count_df['auc'].unique():\n",
    "    nwkdf = nwkdf.append(word_count_df[word_count_df['auc'] == auc][:5])\n",
    "nwkdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_word_count_results(source):\n",
    "    output_file(filename='./EnrichrAPIResults/enrichr_word_count_bokeh.html',\n",
    "                title='Word count plot')\n",
    "    category = 'word'\n",
    "    category_items = source[category].unique()\n",
    "    palette = inferno(len(category_items) + 1)\n",
    "    colormap = dict(zip(category_items, palette))\n",
    "    source['color'] = source[category].map(colormap)\n",
    "    \n",
    "    source['auc'] = source['auc'].astype(str)\n",
    "    \n",
    "    group = source.groupby(by=['auc', 'word']).sum()\n",
    "    group.sort_values(by=['auc','count'], ascending=[True,False], inplace=True)\n",
    "    \n",
    "    title = f'Word count bar plot'\n",
    "    \n",
    "    factors = group.index.tolist()\n",
    "    x = group['count'].tolist()\n",
    "    x_str = group['count'].astype(str).tolist()\n",
    "    colors = group['color'].tolist()\n",
    "    \n",
    "    temp_source = ColumnDataSource(source)\n",
    "        \n",
    "    tooltips = [('Counts', '@count')]\n",
    "    \n",
    "    p = figure(x_range=FactorRange(*factors),\n",
    "               title=title,\n",
    "               plot_width=2000,\n",
    "               plot_height=1200,\n",
    "               toolbar_location='right',\n",
    "               toolbar_sticky=False)\n",
    "    p.vbar(x=factors,\n",
    "           top=x,\n",
    "           color=colors,\n",
    "           width=0.9)\n",
    "    p.add_tools(HoverTool(tooltips=[('Counts','@top')]))\n",
    "    p.xaxis.axis_label = 'Word'\n",
    "    p.yaxis.axis_label = 'Count'\n",
    "    p.xaxis.major_label_orientation = 1.5\n",
    "    return p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/rjsietsma/PycharmProjects/dsls_master_thesis/venv/lib/python3.8/site-packages/bokeh/models/plots.py:764: UserWarning: \n",
      "You are attempting to set `plot.legend.click_policy` on a plot that has zero legends added, this will have no effect.\n",
      "\n",
      "Before legend properties can be set, you must add a Legend explicitly, or call a glyph method with a legend parameter set.\n",
      "\n",
      "  warnings.warn(_LEGEND_EMPTY_WARNING % attr)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/enrichr_word_count_bokeh.html'"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "plot = plot_word_count_results(nwkdf)\n",
    "save(plot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_count_df = pd.DataFrame(columns=['gene', 'count', 'auc'])\n",
    "for auc in combined_df_ext['AUC'].unique():\n",
    "    subset_df = combined_df_ext[combined_df_ext['AUC'] == auc]\n",
    "    gene_list = []\n",
    "    unsplitted_gene_list = subset_df['Overlapping_Genes'].tolist()\n",
    "    for unsplitted_gene in unsplitted_gene_list:\n",
    "        gene_list += unsplitted_gene.split(', ')\n",
    "    gene_count_dict = {}\n",
    "    for gene in gene_list:\n",
    "        if gene not in gene_count_dict.keys():\n",
    "            gene_count_dict[gene] = 1\n",
    "        else:\n",
    "            gene_count_dict[gene] += 1\n",
    "    temp_df = pd.DataFrame(columns=gene_count_df.columns)\n",
    "    for key in gene_count_dict.keys():\n",
    "        temp_temp_df = pd.DataFrame({'gene':key, 'count': gene_count_dict[key],\n",
    "                                    'auc': auc}, index=[0])\n",
    "        temp_df = temp_df.append(temp_temp_df, ignore_index=True)\n",
    "    gene_count_df = gene_count_df.append(temp_df, ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_count_df.sort_values(by=['auc','count'], ascending=[True, False], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "ngkdf = pd.DataFrame(columns=gene_count_df.columns)\n",
    "for auc in gene_count_df['auc'].unique():\n",
    "    ngkdf = ngkdf.append(gene_count_df[gene_count_df['auc'] == auc][:5])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_gene_count_results(source):\n",
    "    output_file(filename='./EnrichrAPIResults/enrichr_gene_count_bokeh.html',\n",
    "                title='Gene count plot')\n",
    "    \n",
    "    category = 'gene'\n",
    "    category_items = source[category].unique()\n",
    "    palette = inferno(len(category_items) + 1)\n",
    "    colormap = dict(zip(category_items, palette))\n",
    "    source['color'] = source[category].map(colormap)\n",
    "    \n",
    "    source['auc'] = source['auc'].astype(str)\n",
    "    \n",
    "    group = source.groupby(by=['auc', 'gene']).sum()\n",
    "    group.sort_values(by=['auc','count'], ascending=[True,False], inplace=True)\n",
    "    \n",
    "    title = f'Gene count bar plot'\n",
    "    \n",
    "    factors = group.index.tolist()\n",
    "    x = group['count'].tolist()\n",
    "    x_str = group['count'].astype(str).tolist()\n",
    "    colors = group['color'].tolist()\n",
    "    \n",
    "    temp_source = ColumnDataSource(source)\n",
    "        \n",
    "    tooltips = [('Counts', '@count')]\n",
    "    \n",
    "    p = figure(x_range=FactorRange(*factors),\n",
    "               title=title,\n",
    "               plot_width=2000,\n",
    "               plot_height=1200,\n",
    "               toolbar_location='right',\n",
    "               toolbar_sticky=False)\n",
    "    p.vbar(x=factors,\n",
    "           top=x,\n",
    "           color=colors,\n",
    "           width=0.9)\n",
    "    p.add_tools(HoverTool(tooltips=[('Counts','@top')]))\n",
    "    p.xaxis.axis_label = 'Gene'\n",
    "    p.yaxis.axis_label = 'Count'\n",
    "    p.xaxis.major_label_orientation = 1.5\n",
    "    return p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/enrichr_gene_count_bokeh.html'"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "plot = plot_gene_count_results(ngkdf)\n",
    "save(plot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "cdfe = combined_df_ext.copy()\n",
    "cdfed = cdfe.copy()\n",
    "cdfe.drop(['Overlapping_Genes', 'Rank'], axis=1, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-54-cf2ab7324a6e>:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  subset.sort_values(by='Adjusted_P-value', ascending=True, inplace=True)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.50</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>3.156383e-12</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.50</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>4.370632e-04</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.50</td>\n",
       "      <td>Short stature</td>\n",
       "      <td>1.808020e-03</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.50</td>\n",
       "      <td>Neurodegenerative disease</td>\n",
       "      <td>5.007898e-03</td>\n",
       "      <td>Jensen_DISEASES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.50</td>\n",
       "      <td>neuropathy</td>\n",
       "      <td>5.197129e-03</td>\n",
       "      <td>OMIM_Disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.55</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>8.394258e-11</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.55</td>\n",
       "      <td>Autosomal recessive inheritance (HP:0000007)</td>\n",
       "      <td>3.454317e-04</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.55</td>\n",
       "      <td>Neurodegenerative disease</td>\n",
       "      <td>1.440183e-03</td>\n",
       "      <td>Jensen_DISEASES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0.55</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>2.308717e-03</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>0.55</td>\n",
       "      <td>Abnormal EKG (HP:0003115)</td>\n",
       "      <td>6.518371e-03</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>0.60</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>5.189207e-13</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>0.60</td>\n",
       "      <td>Neurodegenerative disease</td>\n",
       "      <td>4.711474e-05</td>\n",
       "      <td>Jensen_DISEASES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>0.60</td>\n",
       "      <td>Autosomal recessive inheritance (HP:0000007)</td>\n",
       "      <td>4.981381e-05</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>0.60</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>7.705421e-04</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>0.60</td>\n",
       "      <td>neuropathy</td>\n",
       "      <td>2.461203e-03</td>\n",
       "      <td>OMIM_Disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>0.65</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>4.683801e-16</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>0.65</td>\n",
       "      <td>Neurodegenerative disease</td>\n",
       "      <td>3.318617e-06</td>\n",
       "      <td>Jensen_DISEASES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>0.65</td>\n",
       "      <td>Autosomal recessive inheritance (HP:0000007)</td>\n",
       "      <td>3.358224e-05</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>0.65</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>4.583578e-04</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>0.65</td>\n",
       "      <td>Abnormal EKG (HP:0003115)</td>\n",
       "      <td>5.608645e-04</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>0.70</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>3.976306e-20</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>0.70</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>2.161136e-08</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>0.70</td>\n",
       "      <td>Neurodegenerative disease</td>\n",
       "      <td>9.139885e-08</td>\n",
       "      <td>Jensen_DISEASES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>0.70</td>\n",
       "      <td>Autosomal recessive inheritance (HP:0000007)</td>\n",
       "      <td>3.364939e-07</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>0.70</td>\n",
       "      <td>Pediatric failure to thrive</td>\n",
       "      <td>6.994659e-06</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>0.75</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>6.024297e-32</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>0.75</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>1.737115e-20</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>0.75</td>\n",
       "      <td>Autosomal recessive inheritance (HP:0000007)</td>\n",
       "      <td>3.887066e-11</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>0.75</td>\n",
       "      <td>Neurodegenerative disease</td>\n",
       "      <td>9.988713e-09</td>\n",
       "      <td>Jensen_DISEASES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>0.75</td>\n",
       "      <td>Mental deficiency</td>\n",
       "      <td>1.087223e-06</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>0.80</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>2.550165e-45</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>0.80</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>1.369238e-35</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>0.80</td>\n",
       "      <td>Mental deficiency</td>\n",
       "      <td>2.232324e-15</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>0.80</td>\n",
       "      <td>Mental Retardation</td>\n",
       "      <td>3.463721e-15</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>0.80</td>\n",
       "      <td>Poor school performance</td>\n",
       "      <td>3.540353e-15</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>0.85</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>2.047007e-77</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>0.85</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>1.957076e-74</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>0.85</td>\n",
       "      <td>Mental Retardation</td>\n",
       "      <td>2.964523e-29</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>0.85</td>\n",
       "      <td>Autosomal recessive inheritance (HP:0000007)</td>\n",
       "      <td>1.237603e-28</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>0.85</td>\n",
       "      <td>Mental deficiency</td>\n",
       "      <td>1.539864e-26</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>0.90</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>4.581076e-177</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>0.90</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>7.426349e-150</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>0.90</td>\n",
       "      <td>Mental Retardation</td>\n",
       "      <td>1.311514e-85</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>0.90</td>\n",
       "      <td>Poor school performance</td>\n",
       "      <td>2.282654e-84</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>0.90</td>\n",
       "      <td>Mental deficiency</td>\n",
       "      <td>1.817055e-82</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>0.95</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>0.95</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>4.037269e-297</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>0.95</td>\n",
       "      <td>Autosomal recessive inheritance (HP:0000007)</td>\n",
       "      <td>5.942928e-252</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>0.95</td>\n",
       "      <td>Mental Retardation</td>\n",
       "      <td>1.236648e-216</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>0.95</td>\n",
       "      <td>Mental deficiency</td>\n",
       "      <td>1.301659e-214</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>1.00</td>\n",
       "      <td>Autosomal recessive inheritance (HP:0000007)</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>1.00</td>\n",
       "      <td>Cognitive delay</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>1.00</td>\n",
       "      <td>Dull intelligence</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>1.00</td>\n",
       "      <td>Global developmental delay</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>1.00</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     AUC                                     Term_Name  Adjusted_P-value  \\\n",
       "0   0.50   Autosomal dominant inheritance (HP:0000006)      3.156383e-12   \n",
       "1   0.50                       Intellectual Disability      4.370632e-04   \n",
       "2   0.50                                 Short stature      1.808020e-03   \n",
       "3   0.50                     Neurodegenerative disease      5.007898e-03   \n",
       "4   0.50                                    neuropathy      5.197129e-03   \n",
       "5   0.55   Autosomal dominant inheritance (HP:0000006)      8.394258e-11   \n",
       "6   0.55  Autosomal recessive inheritance (HP:0000007)      3.454317e-04   \n",
       "7   0.55                     Neurodegenerative disease      1.440183e-03   \n",
       "8   0.55                       Intellectual Disability      2.308717e-03   \n",
       "9   0.55                     Abnormal EKG (HP:0003115)      6.518371e-03   \n",
       "10  0.60   Autosomal dominant inheritance (HP:0000006)      5.189207e-13   \n",
       "11  0.60                     Neurodegenerative disease      4.711474e-05   \n",
       "12  0.60  Autosomal recessive inheritance (HP:0000007)      4.981381e-05   \n",
       "13  0.60                       Intellectual Disability      7.705421e-04   \n",
       "14  0.60                                    neuropathy      2.461203e-03   \n",
       "15  0.65   Autosomal dominant inheritance (HP:0000006)      4.683801e-16   \n",
       "16  0.65                     Neurodegenerative disease      3.318617e-06   \n",
       "17  0.65  Autosomal recessive inheritance (HP:0000007)      3.358224e-05   \n",
       "18  0.65                       Intellectual Disability      4.583578e-04   \n",
       "19  0.65                     Abnormal EKG (HP:0003115)      5.608645e-04   \n",
       "20  0.70   Autosomal dominant inheritance (HP:0000006)      3.976306e-20   \n",
       "21  0.70                       Intellectual Disability      2.161136e-08   \n",
       "22  0.70                     Neurodegenerative disease      9.139885e-08   \n",
       "23  0.70  Autosomal recessive inheritance (HP:0000007)      3.364939e-07   \n",
       "24  0.70                   Pediatric failure to thrive      6.994659e-06   \n",
       "25  0.75   Autosomal dominant inheritance (HP:0000006)      6.024297e-32   \n",
       "26  0.75                       Intellectual Disability      1.737115e-20   \n",
       "27  0.75  Autosomal recessive inheritance (HP:0000007)      3.887066e-11   \n",
       "28  0.75                     Neurodegenerative disease      9.988713e-09   \n",
       "29  0.75                             Mental deficiency      1.087223e-06   \n",
       "30  0.80   Autosomal dominant inheritance (HP:0000006)      2.550165e-45   \n",
       "31  0.80                       Intellectual Disability      1.369238e-35   \n",
       "32  0.80                             Mental deficiency      2.232324e-15   \n",
       "33  0.80                            Mental Retardation      3.463721e-15   \n",
       "34  0.80                       Poor school performance      3.540353e-15   \n",
       "35  0.85                       Intellectual Disability      2.047007e-77   \n",
       "36  0.85   Autosomal dominant inheritance (HP:0000006)      1.957076e-74   \n",
       "37  0.85                            Mental Retardation      2.964523e-29   \n",
       "38  0.85  Autosomal recessive inheritance (HP:0000007)      1.237603e-28   \n",
       "39  0.85                             Mental deficiency      1.539864e-26   \n",
       "40  0.90                       Intellectual Disability     4.581076e-177   \n",
       "41  0.90   Autosomal dominant inheritance (HP:0000006)     7.426349e-150   \n",
       "42  0.90                            Mental Retardation      1.311514e-85   \n",
       "43  0.90                       Poor school performance      2.282654e-84   \n",
       "44  0.90                             Mental deficiency      1.817055e-82   \n",
       "45  0.95                       Intellectual Disability      0.000000e+00   \n",
       "46  0.95   Autosomal dominant inheritance (HP:0000006)     4.037269e-297   \n",
       "47  0.95  Autosomal recessive inheritance (HP:0000007)     5.942928e-252   \n",
       "48  0.95                            Mental Retardation     1.236648e-216   \n",
       "49  0.95                             Mental deficiency     1.301659e-214   \n",
       "50  1.00  Autosomal recessive inheritance (HP:0000007)      0.000000e+00   \n",
       "51  1.00                               Cognitive delay      0.000000e+00   \n",
       "52  1.00                             Dull intelligence      0.000000e+00   \n",
       "53  1.00                    Global developmental delay      0.000000e+00   \n",
       "54  1.00                       Intellectual Disability      0.000000e+00   \n",
       "\n",
       "                      Source  \n",
       "0   Human_Phenotype_Ontology  \n",
       "1                   DisGeNET  \n",
       "2                   DisGeNET  \n",
       "3            Jensen_DISEASES  \n",
       "4               OMIM_Disease  \n",
       "5   Human_Phenotype_Ontology  \n",
       "6   Human_Phenotype_Ontology  \n",
       "7            Jensen_DISEASES  \n",
       "8                   DisGeNET  \n",
       "9   Human_Phenotype_Ontology  \n",
       "10  Human_Phenotype_Ontology  \n",
       "11           Jensen_DISEASES  \n",
       "12  Human_Phenotype_Ontology  \n",
       "13                  DisGeNET  \n",
       "14              OMIM_Disease  \n",
       "15  Human_Phenotype_Ontology  \n",
       "16           Jensen_DISEASES  \n",
       "17  Human_Phenotype_Ontology  \n",
       "18                  DisGeNET  \n",
       "19  Human_Phenotype_Ontology  \n",
       "20  Human_Phenotype_Ontology  \n",
       "21                  DisGeNET  \n",
       "22           Jensen_DISEASES  \n",
       "23  Human_Phenotype_Ontology  \n",
       "24                  DisGeNET  \n",
       "25  Human_Phenotype_Ontology  \n",
       "26                  DisGeNET  \n",
       "27  Human_Phenotype_Ontology  \n",
       "28           Jensen_DISEASES  \n",
       "29                  DisGeNET  \n",
       "30  Human_Phenotype_Ontology  \n",
       "31                  DisGeNET  \n",
       "32                  DisGeNET  \n",
       "33                  DisGeNET  \n",
       "34                  DisGeNET  \n",
       "35                  DisGeNET  \n",
       "36  Human_Phenotype_Ontology  \n",
       "37                  DisGeNET  \n",
       "38  Human_Phenotype_Ontology  \n",
       "39                  DisGeNET  \n",
       "40                  DisGeNET  \n",
       "41  Human_Phenotype_Ontology  \n",
       "42                  DisGeNET  \n",
       "43                  DisGeNET  \n",
       "44                  DisGeNET  \n",
       "45                  DisGeNET  \n",
       "46  Human_Phenotype_Ontology  \n",
       "47  Human_Phenotype_Ontology  \n",
       "48                  DisGeNET  \n",
       "49                  DisGeNET  \n",
       "50  Human_Phenotype_Ontology  \n",
       "51                  DisGeNET  \n",
       "52                  DisGeNET  \n",
       "53                  DisGeNET  \n",
       "54                  DisGeNET  "
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fcdfe = pd.DataFrame(columns=cdfe.columns)\n",
    "for auc in cdfe['AUC'].unique():\n",
    "    subset = cdfe[cdfe['AUC'] == auc]\n",
    "    subset.sort_values(by='Adjusted_P-value', ascending=True, inplace=True)\n",
    "    subset = subset[:5]\n",
    "    fcdfe = fcdfe.append(subset, ignore_index=True)\n",
    "fcdfe.to_csv('./EnrichrAPIResults/combined_df_nondetailed_paper.csv', index=False)\n",
    "fcdfe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-55-81717ed15518>:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  subset.sort_values(by='Adjusted_P-value', ascending=True, inplace=True)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.50</td>\n",
       "      <td>1</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>YARS, RTN2, KCNJ5, NTRK2, CEACAM16, KCNK9, MAP...</td>\n",
       "      <td>3.156383e-12</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.50</td>\n",
       "      <td>1</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>RTN2, CD96, SLC46A1, UNC13A, CTDP1, KCNK9, TGD...</td>\n",
       "      <td>4.370632e-04</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.50</td>\n",
       "      <td>2</td>\n",
       "      <td>Short stature</td>\n",
       "      <td>CD96, DPP6, CTDP1, TGDS, GJB3, GPD1, MYH8, NRX...</td>\n",
       "      <td>1.808020e-03</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.50</td>\n",
       "      <td>1</td>\n",
       "      <td>Neurodegenerative disease</td>\n",
       "      <td>YARS, RTN2, DPP6, C10ORF2, ANG, TUBA4A, GIGYF2</td>\n",
       "      <td>5.007898e-03</td>\n",
       "      <td>Jensen_DISEASES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.50</td>\n",
       "      <td>1</td>\n",
       "      <td>neuropathy</td>\n",
       "      <td>CTDP1, GJB3, C10ORF2</td>\n",
       "      <td>5.197129e-03</td>\n",
       "      <td>OMIM_Disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.55</td>\n",
       "      <td>1</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>RTN2, KCNK9, KCNE3, KRT86, GIGYF2, KBTBD13, LM...</td>\n",
       "      <td>8.394258e-11</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.55</td>\n",
       "      <td>2</td>\n",
       "      <td>Autosomal recessive inheritance (HP:0000007)</td>\n",
       "      <td>CD96, SLC25A3, SLC46A1, CTDP1, TPMT, PRCD, NDU...</td>\n",
       "      <td>3.454317e-04</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.55</td>\n",
       "      <td>1</td>\n",
       "      <td>Neurodegenerative disease</td>\n",
       "      <td>RTN2, YARS, DPP6, GRID2, C10ORF2, ANG, SMN1, P...</td>\n",
       "      <td>1.440183e-03</td>\n",
       "      <td>Jensen_DISEASES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0.55</td>\n",
       "      <td>1</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>RTN2, SLC46A1, CTDP1, KCNK9, PRCD, NRXN3, AFF4...</td>\n",
       "      <td>2.308717e-03</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>0.55</td>\n",
       "      <td>3</td>\n",
       "      <td>Abnormal EKG (HP:0003115)</td>\n",
       "      <td>KCNJ5, CACNB2, KCNE3, TRPM4</td>\n",
       "      <td>6.518371e-03</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>0.60</td>\n",
       "      <td>1</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>RTN2, KCNE3, KCNK9, KRT86, GIGYF2, KBTBD13, LM...</td>\n",
       "      <td>5.189207e-13</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>0.60</td>\n",
       "      <td>1</td>\n",
       "      <td>Neurodegenerative disease</td>\n",
       "      <td>YARS, RTN2, DPP6, GRID2, C10ORF2, GARS, ANG, I...</td>\n",
       "      <td>4.711474e-05</td>\n",
       "      <td>Jensen_DISEASES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>0.60</td>\n",
       "      <td>2</td>\n",
       "      <td>Autosomal recessive inheritance (HP:0000007)</td>\n",
       "      <td>SLC46A1, SLC25A3, CTDP1, TPMT, PRCD, WISP3, SN...</td>\n",
       "      <td>4.981381e-05</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>0.60</td>\n",
       "      <td>1</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>RTN2, CEP104, SLC46A1, CTDP1, KCNK9, PRCD, NRX...</td>\n",
       "      <td>7.705421e-04</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>0.60</td>\n",
       "      <td>1</td>\n",
       "      <td>neuropathy</td>\n",
       "      <td>CTDP1, GJB3, C10ORF2, GARS</td>\n",
       "      <td>2.461203e-03</td>\n",
       "      <td>OMIM_Disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>0.65</td>\n",
       "      <td>1</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>RTN2, KCNE2, KCNE3, KCNK9, KRT86, GIGYF2, KBTB...</td>\n",
       "      <td>4.683801e-16</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>0.65</td>\n",
       "      <td>1</td>\n",
       "      <td>Neurodegenerative disease</td>\n",
       "      <td>YARS, RTN2, GRID2, IKBKAP, PDYN, TUBA4A, GIGYF...</td>\n",
       "      <td>3.318617e-06</td>\n",
       "      <td>Jensen_DISEASES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>0.65</td>\n",
       "      <td>2</td>\n",
       "      <td>Autosomal recessive inheritance (HP:0000007)</td>\n",
       "      <td>SLC46A1, SLC25A3, CTDP1, TPMT, PRCD, WISP3, CA...</td>\n",
       "      <td>3.358224e-05</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>0.65</td>\n",
       "      <td>1</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>RTN2, CEP104, SLC46A1, CTDP1, KCNK9, PRCD, NRX...</td>\n",
       "      <td>4.583578e-04</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>0.65</td>\n",
       "      <td>3</td>\n",
       "      <td>Abnormal EKG (HP:0003115)</td>\n",
       "      <td>KCNJ5, CACNB2, KCNE2, KCNE3, TRPM4</td>\n",
       "      <td>5.608645e-04</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>0.70</td>\n",
       "      <td>1</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>RTN2, RPL5, KCNE2, KCNE3, KCNK9, PCSK9, KRT86,...</td>\n",
       "      <td>3.976306e-20</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>0.70</td>\n",
       "      <td>1</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>SLC46A1, KCNK9, CHRDL1, AFF4, KBTBD13, HINT1, ...</td>\n",
       "      <td>2.161136e-08</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>0.70</td>\n",
       "      <td>1</td>\n",
       "      <td>Neurodegenerative disease</td>\n",
       "      <td>YARS, RTN2, GRID2, KIAA0196, TRIP4, IKBKAP, PD...</td>\n",
       "      <td>9.139885e-08</td>\n",
       "      <td>Jensen_DISEASES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>0.70</td>\n",
       "      <td>2</td>\n",
       "      <td>Autosomal recessive inheritance (HP:0000007)</td>\n",
       "      <td>SLC46A1, SLC25A3, CTDP1, TPMT, RBM8A, PRCD, WI...</td>\n",
       "      <td>3.364939e-07</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>0.70</td>\n",
       "      <td>4</td>\n",
       "      <td>Pediatric failure to thrive</td>\n",
       "      <td>CD96, RPL5, SLC46A1, SLC25A3, TGDS, SLC10A2, V...</td>\n",
       "      <td>6.994659e-06</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>0.75</td>\n",
       "      <td>1</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>RPL5, KCNK9, PCSK9, KBTBD13, LMBR1, JPH2, SPTL...</td>\n",
       "      <td>6.024297e-32</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>0.75</td>\n",
       "      <td>1</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>COL18A1, SLC46A1, KCNK9, CHRDL1, AFF4, KBTBD13...</td>\n",
       "      <td>1.737115e-20</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>0.75</td>\n",
       "      <td>2</td>\n",
       "      <td>Autosomal recessive inheritance (HP:0000007)</td>\n",
       "      <td>COL18A1, SLC46A1, SLC25A3, TPMT, WISP3, CASC5,...</td>\n",
       "      <td>3.887066e-11</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>0.75</td>\n",
       "      <td>1</td>\n",
       "      <td>Neurodegenerative disease</td>\n",
       "      <td>YARS, RTN2, GRID2, KIAA0196, TRIP4, IKBKAP, PD...</td>\n",
       "      <td>9.988713e-09</td>\n",
       "      <td>Jensen_DISEASES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>0.75</td>\n",
       "      <td>2</td>\n",
       "      <td>Mental deficiency</td>\n",
       "      <td>COL18A1, CEP104, SLC46A1, CTDP1, NFIX, RBM8A, ...</td>\n",
       "      <td>1.087223e-06</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>0.80</td>\n",
       "      <td>1</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>RPL5, GUCA1A, ACTG2, FGF8, FGF9, SCN11A, SCN9A...</td>\n",
       "      <td>2.550165e-45</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>0.80</td>\n",
       "      <td>1</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>SLC46A1, TUSC3, KIAA1109, COL4A3BP, STS, SCN11...</td>\n",
       "      <td>1.369238e-35</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>0.80</td>\n",
       "      <td>2</td>\n",
       "      <td>Mental deficiency</td>\n",
       "      <td>COL18A1, SLC46A1, DDX3X, KCNK9, TUSC3, PHF6, A...</td>\n",
       "      <td>2.232324e-15</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>0.80</td>\n",
       "      <td>4</td>\n",
       "      <td>Mental Retardation</td>\n",
       "      <td>SLC46A1, DDX3X, KCNK9, THRA, TUSC3, PHF6, AFF4...</td>\n",
       "      <td>3.463721e-15</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>0.80</td>\n",
       "      <td>3</td>\n",
       "      <td>Poor school performance</td>\n",
       "      <td>SLC46A1, DDX3X, KCNK9, TUSC3, PHF6, AFF4, IFIH...</td>\n",
       "      <td>3.540353e-15</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>0.85</td>\n",
       "      <td>1</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>SLC46A1, HNRNPU, RBPJ, GLE1, SOX2, KIAA1109, S...</td>\n",
       "      <td>2.047007e-77</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>0.85</td>\n",
       "      <td>1</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>RPL5, PRSS1, GUCA1A, RBPJ, ACTG2, ACTG1, SOX2,...</td>\n",
       "      <td>1.957076e-74</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>0.85</td>\n",
       "      <td>2</td>\n",
       "      <td>Mental Retardation</td>\n",
       "      <td>SLC46A1, AMHR2, TUSC3, HNRNPU, ACTG1, COL4A3BP...</td>\n",
       "      <td>2.964523e-29</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>0.85</td>\n",
       "      <td>2</td>\n",
       "      <td>Autosomal recessive inheritance (HP:0000007)</td>\n",
       "      <td>PRSS1, SLC46A1, IL1RN, TPMT, AMHR2, TUSC3, CAS...</td>\n",
       "      <td>1.237603e-28</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>0.85</td>\n",
       "      <td>3</td>\n",
       "      <td>Mental deficiency</td>\n",
       "      <td>SETD5, COL18A1, SLC46A1, DDX3X, KCNK9, TUSC3, ...</td>\n",
       "      <td>1.539864e-26</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>0.90</td>\n",
       "      <td>1</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>TRIO, PGAP3, MYT1L, HNRNPU, RBPJ, SMC3, GLE1, ...</td>\n",
       "      <td>4.581076e-177</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>0.90</td>\n",
       "      <td>1</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>RPL5, PRSS1, RBPJ, ACTB, SOX2, GJA1, GJA3, RPS...</td>\n",
       "      <td>7.426349e-150</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>0.90</td>\n",
       "      <td>2</td>\n",
       "      <td>Mental Retardation</td>\n",
       "      <td>SLC46A1, PGAP3, MYT1L, HNRNPU, SMC3, ACTB, GJA...</td>\n",
       "      <td>1.311514e-85</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>0.90</td>\n",
       "      <td>3</td>\n",
       "      <td>Poor school performance</td>\n",
       "      <td>SLC46A1, PGAP3, MYT1L, ZNF292, HNRNPU, SMC3, A...</td>\n",
       "      <td>2.282654e-84</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>0.90</td>\n",
       "      <td>4</td>\n",
       "      <td>Mental deficiency</td>\n",
       "      <td>SLC46A1, PGAP3, MYT1L, HNRNPU, SMC3, ACTB, GJA...</td>\n",
       "      <td>1.817055e-82</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>0.95</td>\n",
       "      <td>1</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>NUP107, PGAP3, NEXMIF, RBPJ, TSEN54, KIAA1109,...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>0.95</td>\n",
       "      <td>1</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>RB1, RPL5, PRSS1, RBPJ, MYLK, GJA1, SYNGAP1, G...</td>\n",
       "      <td>4.037269e-297</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>0.95</td>\n",
       "      <td>2</td>\n",
       "      <td>Autosomal recessive inheritance (HP:0000007)</td>\n",
       "      <td>PRSS1, SPINT2, ABCA12, ENO3, TSEN54, GLE1, GJA...</td>\n",
       "      <td>5.942928e-252</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>0.95</td>\n",
       "      <td>2</td>\n",
       "      <td>Mental Retardation</td>\n",
       "      <td>RB1, PGAP3, MYT1L, NEXMIF, HNRNPU, SMC3, GJA1,...</td>\n",
       "      <td>1.236648e-216</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>0.95</td>\n",
       "      <td>3</td>\n",
       "      <td>Mental deficiency</td>\n",
       "      <td>RB1, PGAP3, MYT1L, NEXMIF, HNRNPU, SMC3, TSEN5...</td>\n",
       "      <td>1.301659e-214</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>1.00</td>\n",
       "      <td>2</td>\n",
       "      <td>Autosomal recessive inheritance (HP:0000007)</td>\n",
       "      <td>IFT172, SPINT2, LIPA, TSEN54, ZFYVE26, LIPC, C...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>1.00</td>\n",
       "      <td>1</td>\n",
       "      <td>Cognitive delay</td>\n",
       "      <td>NUP107, CLPB, PGAP3, FRMPD4, MYT1L, NEXMIF, EH...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>1.00</td>\n",
       "      <td>2</td>\n",
       "      <td>Dull intelligence</td>\n",
       "      <td>RB1, CLPB, PGAP3, GLDC, FRMPD4, MYT1L, IFT172,...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>1.00</td>\n",
       "      <td>3</td>\n",
       "      <td>Global developmental delay</td>\n",
       "      <td>RB1, NUP107, CLPB, PGAP3, FRMPD4, MYT1L, NEXMI...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>1.00</td>\n",
       "      <td>4</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>NUP107, PGAP3, IFT172, NEXMIF, EHMT1, RBPJ, TS...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     AUC Rank                                     Term_Name  \\\n",
       "0   0.50    1   Autosomal dominant inheritance (HP:0000006)   \n",
       "1   0.50    1                       Intellectual Disability   \n",
       "2   0.50    2                                 Short stature   \n",
       "3   0.50    1                     Neurodegenerative disease   \n",
       "4   0.50    1                                    neuropathy   \n",
       "5   0.55    1   Autosomal dominant inheritance (HP:0000006)   \n",
       "6   0.55    2  Autosomal recessive inheritance (HP:0000007)   \n",
       "7   0.55    1                     Neurodegenerative disease   \n",
       "8   0.55    1                       Intellectual Disability   \n",
       "9   0.55    3                     Abnormal EKG (HP:0003115)   \n",
       "10  0.60    1   Autosomal dominant inheritance (HP:0000006)   \n",
       "11  0.60    1                     Neurodegenerative disease   \n",
       "12  0.60    2  Autosomal recessive inheritance (HP:0000007)   \n",
       "13  0.60    1                       Intellectual Disability   \n",
       "14  0.60    1                                    neuropathy   \n",
       "15  0.65    1   Autosomal dominant inheritance (HP:0000006)   \n",
       "16  0.65    1                     Neurodegenerative disease   \n",
       "17  0.65    2  Autosomal recessive inheritance (HP:0000007)   \n",
       "18  0.65    1                       Intellectual Disability   \n",
       "19  0.65    3                     Abnormal EKG (HP:0003115)   \n",
       "20  0.70    1   Autosomal dominant inheritance (HP:0000006)   \n",
       "21  0.70    1                       Intellectual Disability   \n",
       "22  0.70    1                     Neurodegenerative disease   \n",
       "23  0.70    2  Autosomal recessive inheritance (HP:0000007)   \n",
       "24  0.70    4                   Pediatric failure to thrive   \n",
       "25  0.75    1   Autosomal dominant inheritance (HP:0000006)   \n",
       "26  0.75    1                       Intellectual Disability   \n",
       "27  0.75    2  Autosomal recessive inheritance (HP:0000007)   \n",
       "28  0.75    1                     Neurodegenerative disease   \n",
       "29  0.75    2                             Mental deficiency   \n",
       "30  0.80    1   Autosomal dominant inheritance (HP:0000006)   \n",
       "31  0.80    1                       Intellectual Disability   \n",
       "32  0.80    2                             Mental deficiency   \n",
       "33  0.80    4                            Mental Retardation   \n",
       "34  0.80    3                       Poor school performance   \n",
       "35  0.85    1                       Intellectual Disability   \n",
       "36  0.85    1   Autosomal dominant inheritance (HP:0000006)   \n",
       "37  0.85    2                            Mental Retardation   \n",
       "38  0.85    2  Autosomal recessive inheritance (HP:0000007)   \n",
       "39  0.85    3                             Mental deficiency   \n",
       "40  0.90    1                       Intellectual Disability   \n",
       "41  0.90    1   Autosomal dominant inheritance (HP:0000006)   \n",
       "42  0.90    2                            Mental Retardation   \n",
       "43  0.90    3                       Poor school performance   \n",
       "44  0.90    4                             Mental deficiency   \n",
       "45  0.95    1                       Intellectual Disability   \n",
       "46  0.95    1   Autosomal dominant inheritance (HP:0000006)   \n",
       "47  0.95    2  Autosomal recessive inheritance (HP:0000007)   \n",
       "48  0.95    2                            Mental Retardation   \n",
       "49  0.95    3                             Mental deficiency   \n",
       "50  1.00    2  Autosomal recessive inheritance (HP:0000007)   \n",
       "51  1.00    1                               Cognitive delay   \n",
       "52  1.00    2                             Dull intelligence   \n",
       "53  1.00    3                    Global developmental delay   \n",
       "54  1.00    4                       Intellectual Disability   \n",
       "\n",
       "                                    Overlapping_Genes  Adjusted_P-value  \\\n",
       "0   YARS, RTN2, KCNJ5, NTRK2, CEACAM16, KCNK9, MAP...      3.156383e-12   \n",
       "1   RTN2, CD96, SLC46A1, UNC13A, CTDP1, KCNK9, TGD...      4.370632e-04   \n",
       "2   CD96, DPP6, CTDP1, TGDS, GJB3, GPD1, MYH8, NRX...      1.808020e-03   \n",
       "3      YARS, RTN2, DPP6, C10ORF2, ANG, TUBA4A, GIGYF2      5.007898e-03   \n",
       "4                                CTDP1, GJB3, C10ORF2      5.197129e-03   \n",
       "5   RTN2, KCNK9, KCNE3, KRT86, GIGYF2, KBTBD13, LM...      8.394258e-11   \n",
       "6   CD96, SLC25A3, SLC46A1, CTDP1, TPMT, PRCD, NDU...      3.454317e-04   \n",
       "7   RTN2, YARS, DPP6, GRID2, C10ORF2, ANG, SMN1, P...      1.440183e-03   \n",
       "8   RTN2, SLC46A1, CTDP1, KCNK9, PRCD, NRXN3, AFF4...      2.308717e-03   \n",
       "9                         KCNJ5, CACNB2, KCNE3, TRPM4      6.518371e-03   \n",
       "10  RTN2, KCNE3, KCNK9, KRT86, GIGYF2, KBTBD13, LM...      5.189207e-13   \n",
       "11  YARS, RTN2, DPP6, GRID2, C10ORF2, GARS, ANG, I...      4.711474e-05   \n",
       "12  SLC46A1, SLC25A3, CTDP1, TPMT, PRCD, WISP3, SN...      4.981381e-05   \n",
       "13  RTN2, CEP104, SLC46A1, CTDP1, KCNK9, PRCD, NRX...      7.705421e-04   \n",
       "14                         CTDP1, GJB3, C10ORF2, GARS      2.461203e-03   \n",
       "15  RTN2, KCNE2, KCNE3, KCNK9, KRT86, GIGYF2, KBTB...      4.683801e-16   \n",
       "16  YARS, RTN2, GRID2, IKBKAP, PDYN, TUBA4A, GIGYF...      3.318617e-06   \n",
       "17  SLC46A1, SLC25A3, CTDP1, TPMT, PRCD, WISP3, CA...      3.358224e-05   \n",
       "18  RTN2, CEP104, SLC46A1, CTDP1, KCNK9, PRCD, NRX...      4.583578e-04   \n",
       "19                 KCNJ5, CACNB2, KCNE2, KCNE3, TRPM4      5.608645e-04   \n",
       "20  RTN2, RPL5, KCNE2, KCNE3, KCNK9, PCSK9, KRT86,...      3.976306e-20   \n",
       "21  SLC46A1, KCNK9, CHRDL1, AFF4, KBTBD13, HINT1, ...      2.161136e-08   \n",
       "22  YARS, RTN2, GRID2, KIAA0196, TRIP4, IKBKAP, PD...      9.139885e-08   \n",
       "23  SLC46A1, SLC25A3, CTDP1, TPMT, RBM8A, PRCD, WI...      3.364939e-07   \n",
       "24  CD96, RPL5, SLC46A1, SLC25A3, TGDS, SLC10A2, V...      6.994659e-06   \n",
       "25  RPL5, KCNK9, PCSK9, KBTBD13, LMBR1, JPH2, SPTL...      6.024297e-32   \n",
       "26  COL18A1, SLC46A1, KCNK9, CHRDL1, AFF4, KBTBD13...      1.737115e-20   \n",
       "27  COL18A1, SLC46A1, SLC25A3, TPMT, WISP3, CASC5,...      3.887066e-11   \n",
       "28  YARS, RTN2, GRID2, KIAA0196, TRIP4, IKBKAP, PD...      9.988713e-09   \n",
       "29  COL18A1, CEP104, SLC46A1, CTDP1, NFIX, RBM8A, ...      1.087223e-06   \n",
       "30  RPL5, GUCA1A, ACTG2, FGF8, FGF9, SCN11A, SCN9A...      2.550165e-45   \n",
       "31  SLC46A1, TUSC3, KIAA1109, COL4A3BP, STS, SCN11...      1.369238e-35   \n",
       "32  COL18A1, SLC46A1, DDX3X, KCNK9, TUSC3, PHF6, A...      2.232324e-15   \n",
       "33  SLC46A1, DDX3X, KCNK9, THRA, TUSC3, PHF6, AFF4...      3.463721e-15   \n",
       "34  SLC46A1, DDX3X, KCNK9, TUSC3, PHF6, AFF4, IFIH...      3.540353e-15   \n",
       "35  SLC46A1, HNRNPU, RBPJ, GLE1, SOX2, KIAA1109, S...      2.047007e-77   \n",
       "36  RPL5, PRSS1, GUCA1A, RBPJ, ACTG2, ACTG1, SOX2,...      1.957076e-74   \n",
       "37  SLC46A1, AMHR2, TUSC3, HNRNPU, ACTG1, COL4A3BP...      2.964523e-29   \n",
       "38  PRSS1, SLC46A1, IL1RN, TPMT, AMHR2, TUSC3, CAS...      1.237603e-28   \n",
       "39  SETD5, COL18A1, SLC46A1, DDX3X, KCNK9, TUSC3, ...      1.539864e-26   \n",
       "40  TRIO, PGAP3, MYT1L, HNRNPU, RBPJ, SMC3, GLE1, ...     4.581076e-177   \n",
       "41  RPL5, PRSS1, RBPJ, ACTB, SOX2, GJA1, GJA3, RPS...     7.426349e-150   \n",
       "42  SLC46A1, PGAP3, MYT1L, HNRNPU, SMC3, ACTB, GJA...      1.311514e-85   \n",
       "43  SLC46A1, PGAP3, MYT1L, ZNF292, HNRNPU, SMC3, A...      2.282654e-84   \n",
       "44  SLC46A1, PGAP3, MYT1L, HNRNPU, SMC3, ACTB, GJA...      1.817055e-82   \n",
       "45  NUP107, PGAP3, NEXMIF, RBPJ, TSEN54, KIAA1109,...      0.000000e+00   \n",
       "46  RB1, RPL5, PRSS1, RBPJ, MYLK, GJA1, SYNGAP1, G...     4.037269e-297   \n",
       "47  PRSS1, SPINT2, ABCA12, ENO3, TSEN54, GLE1, GJA...     5.942928e-252   \n",
       "48  RB1, PGAP3, MYT1L, NEXMIF, HNRNPU, SMC3, GJA1,...     1.236648e-216   \n",
       "49  RB1, PGAP3, MYT1L, NEXMIF, HNRNPU, SMC3, TSEN5...     1.301659e-214   \n",
       "50  IFT172, SPINT2, LIPA, TSEN54, ZFYVE26, LIPC, C...      0.000000e+00   \n",
       "51  NUP107, CLPB, PGAP3, FRMPD4, MYT1L, NEXMIF, EH...      0.000000e+00   \n",
       "52  RB1, CLPB, PGAP3, GLDC, FRMPD4, MYT1L, IFT172,...      0.000000e+00   \n",
       "53  RB1, NUP107, CLPB, PGAP3, FRMPD4, MYT1L, NEXMI...      0.000000e+00   \n",
       "54  NUP107, PGAP3, IFT172, NEXMIF, EHMT1, RBPJ, TS...      0.000000e+00   \n",
       "\n",
       "                      Source  \n",
       "0   Human_Phenotype_Ontology  \n",
       "1                   DisGeNET  \n",
       "2                   DisGeNET  \n",
       "3            Jensen_DISEASES  \n",
       "4               OMIM_Disease  \n",
       "5   Human_Phenotype_Ontology  \n",
       "6   Human_Phenotype_Ontology  \n",
       "7            Jensen_DISEASES  \n",
       "8                   DisGeNET  \n",
       "9   Human_Phenotype_Ontology  \n",
       "10  Human_Phenotype_Ontology  \n",
       "11           Jensen_DISEASES  \n",
       "12  Human_Phenotype_Ontology  \n",
       "13                  DisGeNET  \n",
       "14              OMIM_Disease  \n",
       "15  Human_Phenotype_Ontology  \n",
       "16           Jensen_DISEASES  \n",
       "17  Human_Phenotype_Ontology  \n",
       "18                  DisGeNET  \n",
       "19  Human_Phenotype_Ontology  \n",
       "20  Human_Phenotype_Ontology  \n",
       "21                  DisGeNET  \n",
       "22           Jensen_DISEASES  \n",
       "23  Human_Phenotype_Ontology  \n",
       "24                  DisGeNET  \n",
       "25  Human_Phenotype_Ontology  \n",
       "26                  DisGeNET  \n",
       "27  Human_Phenotype_Ontology  \n",
       "28           Jensen_DISEASES  \n",
       "29                  DisGeNET  \n",
       "30  Human_Phenotype_Ontology  \n",
       "31                  DisGeNET  \n",
       "32                  DisGeNET  \n",
       "33                  DisGeNET  \n",
       "34                  DisGeNET  \n",
       "35                  DisGeNET  \n",
       "36  Human_Phenotype_Ontology  \n",
       "37                  DisGeNET  \n",
       "38  Human_Phenotype_Ontology  \n",
       "39                  DisGeNET  \n",
       "40                  DisGeNET  \n",
       "41  Human_Phenotype_Ontology  \n",
       "42                  DisGeNET  \n",
       "43                  DisGeNET  \n",
       "44                  DisGeNET  \n",
       "45                  DisGeNET  \n",
       "46  Human_Phenotype_Ontology  \n",
       "47  Human_Phenotype_Ontology  \n",
       "48                  DisGeNET  \n",
       "49                  DisGeNET  \n",
       "50  Human_Phenotype_Ontology  \n",
       "51                  DisGeNET  \n",
       "52                  DisGeNET  \n",
       "53                  DisGeNET  \n",
       "54                  DisGeNET  "
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fcdfed = pd.DataFrame(columns=cdfed.columns)\n",
    "for auc in cdfed['AUC'].unique():\n",
    "    subset = cdfed[cdfed['AUC'] == auc]\n",
    "    subset.sort_values(by='Adjusted_P-value', ascending=True, inplace=True)\n",
    "    subset = subset[:5]\n",
    "    fcdfed = fcdfed.append(subset, ignore_index=True)\n",
    "fcdfed.to_csv('./EnrichrAPIResults/combined_df_detailed_paper.csv', index=False)\n",
    "fcdfed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-56-6fd3d5e1d73c>:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  subset.sort_values(by='Adjusted_P-value', ascending=True, inplace=True)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.5</td>\n",
       "      <td>1</td>\n",
       "      <td>Autosomal dominant inheritance (HP:0000006)</td>\n",
       "      <td>YARS, RTN2, KCNJ5, NTRK2, CEACAM16, KCNK9, MAP...</td>\n",
       "      <td>3.156383e-12</td>\n",
       "      <td>Human_Phenotype_Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.5</td>\n",
       "      <td>1</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>RTN2, CD96, SLC46A1, UNC13A, CTDP1, KCNK9, TGD...</td>\n",
       "      <td>4.370632e-04</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.5</td>\n",
       "      <td>2</td>\n",
       "      <td>Short stature</td>\n",
       "      <td>CD96, DPP6, CTDP1, TGDS, GJB3, GPD1, MYH8, NRX...</td>\n",
       "      <td>1.808020e-03</td>\n",
       "      <td>DisGeNET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.5</td>\n",
       "      <td>1</td>\n",
       "      <td>Neurodegenerative disease</td>\n",
       "      <td>YARS, RTN2, DPP6, C10ORF2, ANG, TUBA4A, GIGYF2</td>\n",
       "      <td>5.007898e-03</td>\n",
       "      <td>Jensen_DISEASES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.5</td>\n",
       "      <td>1</td>\n",
       "      <td>neuropathy</td>\n",
       "      <td>CTDP1, GJB3, C10ORF2</td>\n",
       "      <td>5.197129e-03</td>\n",
       "      <td>OMIM_Disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3698</th>\n",
       "      <td>1.0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr16</td>\n",
       "      <td>ROGDI, XYLT1, GCSH, CDH3, SALL1, CDH1, TUBB3, ...</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3699</th>\n",
       "      <td>1.0</td>\n",
       "      <td>3</td>\n",
       "      <td>chr18</td>\n",
       "      <td>CTDP1, FECH, PIGN, LAMA1, TSHZ1, LAMA3, RAX, G...</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3700</th>\n",
       "      <td>1.0</td>\n",
       "      <td>2</td>\n",
       "      <td>chr17 ctg5 hap1</td>\n",
       "      <td>KANSL1, MAPT</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3701</th>\n",
       "      <td>1.0</td>\n",
       "      <td>10</td>\n",
       "      <td>chr9</td>\n",
       "      <td>INVS, LRSAM1, PIGO, LAMC3, GLDC, DOCK8, POMT1,...</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3702</th>\n",
       "      <td>1.0</td>\n",
       "      <td>18</td>\n",
       "      <td>chr6 apd hap1</td>\n",
       "      <td>TUBB, NEU1, PSMB8</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3703 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      AUC Rank                                    Term_Name  \\\n",
       "0     0.5    1  Autosomal dominant inheritance (HP:0000006)   \n",
       "1     0.5    1                      Intellectual Disability   \n",
       "2     0.5    2                                Short stature   \n",
       "3     0.5    1                    Neurodegenerative disease   \n",
       "4     0.5    1                                   neuropathy   \n",
       "...   ...  ...                                          ...   \n",
       "3698  1.0    4                                        chr16   \n",
       "3699  1.0    3                                        chr18   \n",
       "3700  1.0    2                              chr17 ctg5 hap1   \n",
       "3701  1.0   10                                         chr9   \n",
       "3702  1.0   18                                chr6 apd hap1   \n",
       "\n",
       "                                      Overlapping_Genes  Adjusted_P-value  \\\n",
       "0     YARS, RTN2, KCNJ5, NTRK2, CEACAM16, KCNK9, MAP...      3.156383e-12   \n",
       "1     RTN2, CD96, SLC46A1, UNC13A, CTDP1, KCNK9, TGD...      4.370632e-04   \n",
       "2     CD96, DPP6, CTDP1, TGDS, GJB3, GPD1, MYH8, NRX...      1.808020e-03   \n",
       "3        YARS, RTN2, DPP6, C10ORF2, ANG, TUBA4A, GIGYF2      5.007898e-03   \n",
       "4                                  CTDP1, GJB3, C10ORF2      5.197129e-03   \n",
       "...                                                 ...               ...   \n",
       "3698  ROGDI, XYLT1, GCSH, CDH3, SALL1, CDH1, TUBB3, ...      1.000000e+00   \n",
       "3699  CTDP1, FECH, PIGN, LAMA1, TSHZ1, LAMA3, RAX, G...      1.000000e+00   \n",
       "3700                                       KANSL1, MAPT      1.000000e+00   \n",
       "3701  INVS, LRSAM1, PIGO, LAMC3, GLDC, DOCK8, POMT1,...      1.000000e+00   \n",
       "3702                                  TUBB, NEU1, PSMB8      1.000000e+00   \n",
       "\n",
       "                        Source  \n",
       "0     Human_Phenotype_Ontology  \n",
       "1                     DisGeNET  \n",
       "2                     DisGeNET  \n",
       "3              Jensen_DISEASES  \n",
       "4                 OMIM_Disease  \n",
       "...                        ...  \n",
       "3698  Chromosome_Location_hg19  \n",
       "3699  Chromosome_Location_hg19  \n",
       "3700  Chromosome_Location_hg19  \n",
       "3701  Chromosome_Location_hg19  \n",
       "3702  Chromosome_Location_hg19  \n",
       "\n",
       "[3703 rows x 6 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fcdfede = pd.DataFrame(columns=cdfed.columns)\n",
    "for auc in cdfed['AUC'].unique():\n",
    "    subset = cdfed[cdfed['AUC'] == auc]\n",
    "    subset.sort_values(by='Adjusted_P-value', ascending=True, inplace=True)\n",
    "    fcdfede = fcdfede.append(subset, ignore_index=True)\n",
    "fcdfede.to_csv('./EnrichrAPIResults/combined_df_detailed_paper_extended.csv', index=False)\n",
    "fcdfede"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Interpretation\n",
    "It seems that a normal distribution like pattern is formed with a steeper slope after the maximum, starting at rougly AUC of 0.3 and ending at roughly 0.8.\n",
    "<br/><br/>\n",
    "- DisGeNET seems to outline an interesting picture with the top 10 term names:\n",
    "    - For it seems that in the really bad performing genes the main cause could be infant development related. Later on, more neurological causes are identified.\n",
    "        - Neurological seems clear to me... There is still so much unknown about the human mind.\n",
    "        - Infant development related is a bit less clear, since there is a growing amount of research towards this infant development.\n",
    "- GO_biological_process seems to have primarily the same genes for AUC 0.55-0.7 that are most enriched\n",
    "    - It seems linked to ion channeling\n",
    "- Feet and toe abnormalities seem to occur once or twice too\n",
    "- Could this be an indication to genes that are related to very small molecules? (like steroids, ions, neuro-molecules?)\n",
    "- Most significant: autosomal dominant inheritence\n",
    "    - Very interesting plot, the HPO bokeh plot of n=10\n",
    "- \n",
    "\n",
    "<br/><br/>\n",
    "TODO: make word counts of genes and term names?\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "position": {
    "height": "451.85px",
    "left": "1759px",
    "right": "20px",
    "top": "91px",
    "width": "780px"
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}