{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Importing some shizzle.\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from matplotlib import pyplot as plt\n",
    "from scipy import stats\n",
    "import time\n",
    "import requests\n",
    "import datetime\n",
    "import sys\n",
    "import os\n",
    "import pathlib\n",
    "import seaborn as sns\n",
    "from sklearn.metrics import roc_auc_score\n",
    "import math\n",
    "import json\n",
    "import glob\n",
    "import xlrd3 as xlrd\n",
    "import gzip\n",
    "from utilities import perform_stats, calc_z_scores,\\\n",
    "    get_enrichr_results, plot_results,\\\n",
    "    plot_results_in_singleplot, plot_results_in_singleplot_iteractive_legend,\\\n",
    "    count_words_in_df, plot_count_results, split_and_count_words\n",
    "\n",
    "# Defining some import and export locations\n",
    "location = 'rjsietsma'\n",
    "read_loc = '/home/'+location+'/shared/'\n",
    "data_expor_loc = '/home/'+location+'/Documents/School/Master_DSLS/Final_Thesis/Past_initial_data/'\n",
    "img_output_dir = '/home/'+location+'/PycharmProjects/dsls_master_thesis/side_scripts/output_img/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Some bokeh stuff\n",
    "\n",
    "from bokeh.plotting import figure, ColumnDataSource, output_file, show, save\n",
    "from bokeh.models import HoverTool, WheelZoomTool, PanTool,\\\n",
    "    BoxZoomTool, ResetTool, TapTool, SaveTool, FactorRange, Legend\n",
    "from bokeh.palettes import inferno, Spectral5, viridis, Colorblind\n",
    "from bokeh.transform import factor_cmap"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Sprint 28-04-2020 - 19-05-2020 (?)\n",
    "=====================\n",
    "\n",
    "- What I want to do:\n",
    "  1. [Focus on fixing missing CAPICE precomputed scores.](#Fixing-CAPICE-precomputed-scores) __Done__\n",
    "      - _Note: Find out how the script of Shuang works on a smaller scale, get an example where a CAPICE score is missing._\n",
    "  2. [Normalize BM_ratio](#Normalize-BM_ratio) for gene length, so $\\frac{n_{malignant}}{n_{benign}}$ per 1000 bases in a gene, instead of [$\\frac{n_{malign}}{n_{benign}} \\times n_{total}$](output_img/bm_ratio.png \"BM_ratio\").\n",
    "  3. _(questionable)_ Fix PCA by identifying categorical features _(Questionable because should I still put time and focus in on the input data?)_\n",
    "      - _Ask Shuang, Joeri and Krista in the PRU 28-04-2020 @ 13:00-14:00_\n",
    "  4. [See what is required](#Required-for-data-sources) for the [data sources](https://docs.google.com/document/d/1D5SiNbeDEfY2hTWquS88MGUrLv6IgZzfCzflppRGb5k/edit) to be implemented into the input data.\n",
    "      - _Note: as this is research, I can use sources that only have precomputed scores available, but just to keep in mind that I should mention that in the discussion._\n",
    "\n",
    "- Notes:\n",
    "  - If I find myself unfocussed:\n",
    "      - Win a game of [solitaire](https://play.google.com/store/apps/details?id=com.lemongame.klondike.solitaire&hl=nl) (game is rigged, not every match is win-able).\n",
    "      - Clean out my closet (it's messy, maybe to clean up the mess inside my head too).\n",
    "      - Go for a walk around the neighbourhood.\n",
    "\n",
    "<br/><br/>\n",
    "<br/><br/>\n",
    "### Fixing CAPICE precomputed scores\n",
    "\n",
    "Problem: CAPICE precomputed file of all single nucleotide level variants contains caps.\n",
    "\n",
    "Solution: Fix the underlying problem that causes these gaps.\n",
    "\n",
    "Required: \n",
    "- Script to make and batch the CAPICE computing __(positive check)__\n",
    "- Access to GCC cluster __(positive check)__\n",
    "- Knowledge of job scheduler on GCC cluster __(positive check)__\n",
    "- Smaller batches of SNV files to run locally  __(positive check)__\n",
    "\n",
    "Possible cause:\n",
    "- Incorrect batching on tabix.\n",
    "\n",
    "31-04-2020 Problem:\n",
    "- Got it working on Gearshift, but with a problem:\n",
    "    - AttributeError: Can't get attribute 'DeprecationDict' on <module 'sklearn.utils.deprecation' from '/apps/software/PythonPlus/3.7.4-foss-2018b-v20.02.1/lib/python3.7/site-packages/sklearn/utils/deprecation.py'>\n",
    "- Which let's me to beleive that the model was trained in a previous version of sklearn.\n",
    "\n",
    "<br/><br/>\n",
    "\n",
    "- Update 10:51 : Model compatibility issue. Might have to retrain CAPICE on Gearshift using the latest xgboost, sklearn, scikit-learn, numpy and pandas just to be safe.\n",
    "\n",
    "- Update 12:01 : Managed to get it working using python3.6 on calculon, essentially downgrading my own script (I only used python3.7 for F-strings)\n",
    "\n",
    "- Update 15:55 : Found a very weird bug in the chunking. Seems like the script doesn't like to start at 0 with skiprows and 1000 nrows, solution: nrows - 2 when first iteration, then + 1, then start += nrows + 1...\n",
    "\n",
    "- Update 04-05-2020 : Found out that the first iteration, an overhead of 2 variants was present. Adjusted the start with 2 to compensate. Seemed fine in the first 400000 CADD entries. (no duplicates nor missed entries). __Started full processing__\n",
    "\n",
    "- Update 04-05-2020 : \n",
    "    - Add functionality to continue where left off if it crashes.\n",
    "    - Make dummy script to check for duplicates or if the files are fully complete.\n",
    "    \n",
    "- Update 06-05-2020 : \n",
    "    - Functionaility to continue where left off is required. After 40 hours, it's still processing chromosome 1 after 7 million processed variants. \n",
    "        - Ideas: write json file with start and stop each iteration once that part is processed and added to the larger files. Should overwrite.\n",
    "        - As for now, since this idea is not implemented yet, count the number of rows in the already processed files and set that as start.\n",
    "\n",
    "- Update 11-05-2020:\n",
    "    - The 4 days have ended, chromosome 1 has not been finished yet.\n",
    "    - Output file of chromosome 1 has been renamed to a .tsv file and gzipped (saving about 5 gb of storage).\n",
    "    - Adjustments to the original script have to be made.\n",
    "   \n",
    "\n",
    "#### Version 2.0\n",
    "\n",
    "- Added gzip output by default\n",
    "- Currently no function yet to continue where left off.\n",
    "    - Ideas:\n",
    "        - for line in gzip.open(file.gz): nlines+=1 (to trace back where was left off)\n",
    "            - Do not know if this lead to any complications of duplicated entries. Saving the previous results in the RAM memory might solve this to compare to.\n",
    "    - Then in the future:\n",
    "        - Save a json of start and batch size each iteration to mark where the file was left of.\n",
    "\n",
    "- Update 15-05-2020:\n",
    "    - Program did not load in the previous results correctly, resulting in 12 duplicates (111400001-111400012) (including start and stop). Discuss with Joeri and Shuang what to do. It takes roughly 200 MB of ram to load in 2.1 million processed variants, calculating this for 111.4 million variants (that's how much there is in the file now), would come down to ~11GB of ram usage.\n",
    "\n",
    "- Update 18-05-2020:\n",
    "    - Script to remove duplicates from the CAPICE processed file has been run and duplicates have been removed.\n",
    "    - First test runs have been ran, only for me to realised I messed up. I kept copying the main class, instead of the calculate_capice class. Final release should be on this date.\n",
    "    \n",
    "\n",
    "<br/><br/>\n",
    "<br/><br/>\n",
    "### Normalize BM_ratio\n",
    "\n",
    "Problem: Larger genes have more benign and malignant samples, because they are larger and thus the chanses of variation on the reference read is higher.\n",
    "\n",
    "Solution: Make a $n_{malignant}$ and $n_{benign}$ per $x$ bases in a gene, where {$x \\in \\mathbb{N}\\ | x \\geq 1 | x \\leq 10000$}\n",
    "\n",
    "<br/><br/>\n",
    "<br/><br/>\n",
    "### Required for data sources\n",
    "\n",
    "Problem: Now that a lot of data sources have been found, annotation and merging has to be performed in order to use it as training data.\n",
    "- _Additional problem: once a source has been proven beneficial, it has te be implemented to be annotated before capice fits and predicts a new sample._\n",
    "\n",
    "Solution: Find out, feature by feature if genes that have a low AUC improve *__and that genes that have a high AUC do not perform worse__*\n",
    "\n",
    "Required:\n",
    "- Storage \n",
    "    - On GCC cluster for precomputed score files.\n",
    "    - Locally IF file isn't that big or an API can be used.\n",
    "- Data sources __([positive check](https://docs.google.com/document/d/1D5SiNbeDEfY2hTWquS88MGUrLv6IgZzfCzflppRGb5k/edit))__\n",
    "- Script to annotate / merge\n",
    "\n",
    "\n",
    "      "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CTEPG (Catching The Elusive Predictable Genes) 28-04-2020 - 04-05-2020\n",
    "<br/><br/>\n",
    "\n",
    "What have I done:\n",
    "\n",
    "- [Enrichr Genes < 0.6](https://amp.pharm.mssm.edu/Enrichr/enrich?dataset=b7a959357214069d1905719220ed3e3b)\n",
    "- [Enrichr Genes < 0.7](https://amp.pharm.mssm.edu/Enrichr/enrich?dataset=7d252a3e1df275859228fd562e964344)\n",
    "    - For reference: [Enrichr Genes < 0.85](https://amp.pharm.mssm.edu/Enrichr/enrich?dataset=60c74741c967c25331419bd2b5e7183e)\n",
    "- [Finished investigating initial data sources](https://docs.google.com/document/d/1D5SiNbeDEfY2hTWquS88MGUrLv6IgZzfCzflppRGb5k/edit)\n",
    "\n",
    "<br/><br/>\n",
    "<br/><br/>\n",
    "\n",
    "What I'm planning to do this week:\n",
    "\n",
    "- [Fixing missing values in precomputed CAPICE scoring file](#Fixing-CAPICE-precomputed-scores). __Done and not done, script has to have some additonal features__\n",
    "    - If stuck, do not keep walking into the wall, continue below and contact Shuang and Joeri\n",
    "- [Run enrichr on increasing cutoffs in AUC (.5, .6, .7 etc.)](#Enrichr-automated-API). __Done__\n",
    "    - Read into the [Enrichr API documentation](https://amp.pharm.mssm.edu/Enrichr/help#api)\n",
    "- Mail Tsjerk that a meeting is apreciated. __Done, no response yet__\n",
    "- Make [suggested plot](https://docs.google.com/presentation/d/1_woKoD9nqWnxueyFYfwyX6B4owg8vYF7rw_nKpMPux4/edit#slide=id.p). __Done__\n",
    "    - Select from enrichr all the genes involved in (lipid) metabolism and see if this explains all / most of the bad performing genes.\n",
    "- Make model on bad performing genes and investigate further.\n",
    "- Start investigating what is required for the data sources to be used in the training phase.\n",
    "- Look into the genes that do perform well without a CGD category, also look into the 1/2 genes that do not perform well __with__ a CGD category\n",
    "- _Optionally_ :\n",
    "    - Create outline for article.\n",
    "    - Adjust BM_ratio plot to normalize for gene length.\n",
    "    - Get the UMCG gene panel lists into python\n",
    "        - Then: investigate which gene panel is performing the worst.\n",
    "        - It would be nice to have this on the side for diagnostics, do __not__ focus on this\n",
    "    - Fix PCA.\n",
    "    - Find possible way to annotate genes for co-factors, sub-family etc.\n",
    "<br/><br/>\n",
    "\n",
    "# CTEPG (Catching The Elusive Predictable Genes) 04-05-2020 - 11-05-2020\n",
    "<br/><br/>\n",
    "\n",
    "What have I done:\n",
    "- [Fixing missing values in precomputed CAPICE scoring file](#Fixing-CAPICE-precomputed-scores).\n",
    "- [Run enrichr on increasing cutoffs in AUC (.5, .6, .7 etc.)](#Enrichr-automated-API).\n",
    "- Make [suggested plot](https://docs.google.com/presentation/d/1_woKoD9nqWnxueyFYfwyX6B4owg8vYF7rw_nKpMPux4/edit#slide=id.p).\n",
    "<br/><br/>\n",
    "\n",
    "Planned for this week:\n",
    "- Release precomputed CAPICE script v2.0\n",
    "- Investigate Bokeh plots on commonalities between Adjusted P-value of 0-0.2 and AUC of 0.5-~0.75 and try to find features / links that would be beneficial to train a separate model on. (or prehaps just train a new model without features and see if performance improves)\n",
    "    - Ion transport / transmembrane related genes seems to perform bad. \n",
    "    - Find commonalities between y-axis 0-0.2 and x-axis 0-0.7\n",
    "    - Make model on bad performing genes and investigate further.\n",
    "        - Make plots combined of all sources, but color them by source.\n",
    "- Start investigating what is required for the data sources to be used in the training phase.\n",
    "- Look into the genes that do perform well without a CGD category, also look into the 1/2 genes that do not perform well __with__ a CGD category\n",
    "- _Optionally_ :\n",
    "    - Create outline for article.\n",
    "        - For static plots: make labels on each point with the term name and the number of genes in (n=x) behind that.\n",
    "    - Adjust BM_ratio plot to normalize for gene length.\n",
    "    - Get the UMCG gene panel lists into python (calculate average AUC for each panel, take the worst performing onces and retain).\n",
    "        - Then: investigate which gene panel is performing the worst.\n",
    "        - It would be nice to have this on the side for diagnostics, do __not__ focus on this\n",
    "    - Find possible way to annotate genes for co-factors, sub-family etc.\n",
    "    - Interactive plot of all variants labeled as benign or pathogenic with their capice score (y-axis) and location (x-axis), to show the variant itself.\n",
    "<br/><br/>\n",
    "\n",
    "# CTEPG (Catching The Elusive Predictable Genes) 11-05-2020 - 18-05-2020\n",
    "<br/><br/>\n",
    "\n",
    "What have I done:\n",
    "- [Fixing missing values in precomputed CAPICE scoring file](#Fixing-CAPICE-precomputed-scores).\n",
    "- Make more bokeh plots on enrichr results\n",
    "    - ~[Investigate correlations between AUC's and Adjusted P-value](#Correlation-analysis-Enrichr-results).  (spearman / pearson, preverably spearman). Keep in mind that the distribution is not linear, but more like a normal distribution like.~\n",
    "        - ~Make legend interactive to confirm normal distribution like.~\n",
    "    - Chromosome X seems enriched for higher AUC thresholds. Bias in the source data? (Put in discussion). Lot's of developmental delay reseach is linked / correlated to chromosome X.\n",
    "\n",
    "Planned for this week:\n",
    "- ~Release precomputed CAPICE script v2.0.~\n",
    "- ~Run correlation test for all possible sources on Enrichr to maximize chanses of finding stuff.~\n",
    "    - ~Look into a good way to get all possible results from the enrichr API.~ __Not possible.__\n",
    "- ~Escalate mail contact with Tsjerk to Martijn and Fenna.~\n",
    "    - __Unneeded, Tsjerk replied.__\n",
    "- ~Investigate Bokeh plots on commonalities between Adjusted P-value of 0-0.2 and AUC of 0.5-0.75 and try to find features / links that would be beneficial to train a separate model on.~ (or prehaps just train a new model without features and see if performance improves)\n",
    "    - Ion transport / transmembrane related genes seems to perform bad. \n",
    "    - ~Find commonalities between y-axis 0-0.2 and x-axis 0-0.7~\n",
    "    - __Make model on bad performing genes and investigate further.__\n",
    "        - ~Make plots combined of all sources, but color them by source.~\n",
    "- Use regex `x[~x['INFO'].str.contains('CLNSIG=Uncertain_significance')]` to find all variants within the clinvar database that are not VUS'es\n",
    "- Start investigating what is required for the data sources to be used in the training phase.\n",
    "- Look into the genes that do perform well without a CGD category, also look into the 1/2 genes that do not perform well __with__ a CGD category\n",
    "- _Optionally_ :\n",
    "    - __Create outline for article.__\n",
    "        - For static plots: make labels on each point with the term name and the number of genes in (n=x) behind that.\n",
    "        - For first concept, it's due date is the last week of the resit exams on the 6th of July.\n",
    "    - Adjust BM_ratio plot to normalize for gene length.\n",
    "    - Get the UMCG gene panel lists into python (calculate average AUC for each panel, take the worst performing onces and retain).\n",
    "        - Then: investigate which gene panel is performing the worst.\n",
    "        - It would be nice to have this on the side for diagnostics, do __not__ focus on this\n",
    "    - Find possible way to annotate genes for co-factors, sub-family etc.\n",
    "    - Interactive plot of all variants labeled as benign or pathogenic with their capice score (y-axis) and location (x-axis), to show the variant itself.\n",
    "<br/><br/>\n",
    "    \n",
    "## For after I graduate:\n",
    "<br/><br/>\n",
    "Make a script that for every gene, for every gene makes a plot like [these](https://molgenis26.gcc.rug.nl/downloads/capice/plots/), but then interactive that if you hover over a point you see it's chromosomal location, ref and alt and maybe an RS number?\n",
    "\n",
    "- If input of CAPICE is not CADD annotated, annotate in capice locally and only return the capice scores.\n",
    " <br/><br/>\n",
    " \n",
    " # CTEPG (Catching The Elusive Predictable Genes) 02-06-2020 - 15-06-2020\n",
    "<br/><br/>\n",
    "\n",
    "What have I done:\n",
    "- Get the UMCG gene panel lists into python (calculate average AUC for each panel, take the worst performing onces and retain).\n",
    "     - Then: investigate which gene panel is performing the worst.\n",
    "        - It would be nice to have this on the side for diagnostics, do __not__ focus on this\n",
    "- Finished investigating Enrichr pathways. Only a couple are upregulated significantly when the AUC gets higher.\n",
    "\n",
    "Planned for this week:\n",
    "- ~Finish investigating Enrichr results.~\n",
    "    - ~Make [suggested plots](https://docs.google.com/presentation/d/1e4zBxhPAGORXYJG0bulR1ADb68haa58v-agkFcv_cxQ/edit#slide=id.p).~\n",
    "        - ~Plot 1: Color by main category and plot each panel average AUC in 1 bar plot to see if main categories cluster low / high.~\n",
    "        - ~Plot 2: Compare all genes in 1 main category to all other categories independently P-values using the wilcoxon test. Play around with the test to see if higher / lower / other is applicable.~\n",
    "- Make model on bad performing genes and investigate further.\n",
    "- ~__Think about the following:__~\n",
    "    - ~Work through the summer holiday as extra time to do research, potentially ending up with something amazing.~\n",
    "    - ~Rush to get article / report finished, have summer holidays 'off' to do other school related stuff and relax a bit, potentially missing something amazing.~\n",
    "- ~Make plot of Hart- en vaatziekten vs all other panels combined (minus 5GPM)~\n",
    "- ~Make plot of Hart- en vaatziekten vs 5GPM~\n",
    "- ~How many variants do I have in Hart- en vaatziekten, dyslipidimie (total)?~\n",
    "    - If not enough variants are found (benign, malignant or total), look into [this](https://databases.lovd.nl/shared/variants/).\n",
    "        - Usefull, [convert to vcf](http://grch37.ensembl.org/Homo_sapiens/Tools/VEP).\n",
    "- Usefull, [convert 37 to 38](https://genome.ucsc.edu/cgi-bin/hgLiftOver).\n",
    "- Start investigating what is required for the data sources to be used in the training phase.\n",
    "- Look into the genes that do perform well without a CGD category, also look into the 1/2 genes that do not perform well __with__ a CGD category.\n",
    "- Create outline for article.\n",
    "     - For static plots: make labels on each point with the term name and the number of genes in (n=x) behind that.\n",
    "     - For first concept, it's due date is the last week of the resit exams on the 6th of July.\n",
    "- _Optionally_ :\n",
    "    - Adjust BM_ratio plot to normalize for gene length.\n",
    "    - Find possible way to annotate genes for co-factors, sub-family etc.\n",
    "    - Interactive plot of all variants labeled as benign or pathogenic with their capice score (y-axis) and location (x-axis), to show the variant itself.\n",
    "<br/><br/>\n",
    " \n",
    " # CTEPG (Catching The Elusive Predictable Genes) 15-06-2020 - 22-06-2020\n",
    "<br/><br/>\n",
    "\n",
    "What have I done:\n",
    "\n",
    "\n",
    "Planned for this week:\n",
    "- Make model on bad performing genes and investigate further.\n",
    "    - At first, make models of [the original training data](https://zenodo.org/record/3564757#.XudiZ2ixUUE), models of various gene panels (but keep the variants of those panels within the bigger model). See if they improve and how much.\n",
    "    - Then make a single model of the original training data for CADD 1.6 annotations.\n",
    "    - Then make multiple models of the original training data for CADD 1.6 annotations.\n",
    "    - Investigate improvements.\n",
    "    - Then choose between:\n",
    "        - newer / larger dataset or \n",
    "        - Custom features that might be relevant to the cause (prefered)\n",
    "- Make single plot of panel vs all others combined (except for 5GPM).\n",
    "- Usefull, [convert 37 to 38](https://genome.ucsc.edu/cgi-bin/hgLiftOver).\n",
    "- Start investigating what is required for the data sources to be used in the training phase.\n",
    "- Create outline for article.\n",
    "     - For static plots: make labels on each point with the term name and the number of genes in (n=x) behind that.\n",
    "     - For first concept, it's due date is the last week of the resit exams on the 6th of July.\n",
    "- _Optionally_ :\n",
    "    - Find possible way to annotate genes for co-factors, sub-family etc.\n",
    "    - Interactive plot of all variants labeled as benign or pathogenic with their capice score (y-axis) and location (x-axis), to show the variant itself.\n",
    "<br/><br/>\n",
    "\n",
    "\n",
    "\n",
    "  \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Current Tickets:\n",
    "<br/><br/>\n",
    "\n",
    "## As student / Robert I want to map requirements for SNV and gene features.\n",
    "\n",
    "- Content:\n",
    "\n",
    "Overview of features has now been collected and processed (https://docs.google.com/document/d/1D5SiNbeDEfY2hTWquS88MGUrLv6IgZzfCzflppRGb5k/edit). In order to use these features, knowledge has to be obtained for the requirements on said features including, but not limited to merging to the trianing dataset and storage.\n",
    "\n",
    "- Acceptance criterea:\n",
    "    - Does not take up terrabytes of storage.\n",
    "        - _If it does, data sources might have to be merged 1 by 1, removing the source after the merge._\n",
    "    - Can be merged without 3rd party software (tabix, python etc. allowed).\n",
    "    - Data can easily be downloaded.\n",
    "\n",
    "<br/><br/>\n",
    "\n",
    "## ~As student / Robert I want to find commonalities between bad performing genes in CAPICE~\n",
    "\n",
    "- Content:\n",
    "\n",
    "The enrichr tool has proven to be quite beneficial to understanding a set of bad performing genes. Further processing of these enrichr results is required in order to fully understand the underlying mechanics and commonalities that these bad performing genes share.\n",
    "\n",
    "- Acceptance criterea:\n",
    "    - Enrichr is performed on multiple gene sets of different AUC thresholds (0.5, 0.6, 0.7 etc.).\n",
    "    - Results of various data sources (kegg, bioplanet, wikipathways etc.) have been obtained in an automated way using the Enrichr API.\n",
    "    - Results from the Enrichr API have been processed in an understandable plot, an mockup can be found [here](https://docs.google.com/presentation/d/1_woKoD9nqWnxueyFYfwyX6B4owg8vYF7rw_nKpMPux4/edit#slide=id.p).\n",
    "\n",
    "<br/><br/>\n",
    "\n",
    "## As CAPICE user I want to use the best possible performing model\n",
    "\n",
    "- Prerequisites:\n",
    "\n",
    "    - As Bioinformatician I want to update CAPICE by preprocessing train test datasets to include all features that CAPICE learns on.\n",
    "    - As researcher I want to know in detail how sample weights are applied.\n",
    "    \n",
    "\n",
    "- Content:\n",
    "\n",
    "Even though CAPICE performs really well, there is room for improvement. Subsetting genepanels in the model training and predicting phase might prove beneficial to the performance of CAPICE. Furthermore, the model must be updated a bit to fit certain quality of life improvements (see acceptance criterea).\n",
    "\n",
    "- Acceptance criterea:\n",
    "    - Performance of all gene panels must not worsen.\n",
    "    - Subsetting the variants of the different panels must happen within the model.\n",
    "    - The model must be able to run on python 3.6 and python 3.8 for future proofing.\n",
    "    - Performance must increase with as least additional features as possible.\n",
    "        - For this, sample size must remain within acceptable boundaries. (acceptable is discussable)\n",
    "        - Additional features shall focus primarily on gene level to limit storage and processing stress.\n",
    "        - When variant level features shall be added, this shall be performed one datasource at a time, again to limit stress.\n",
    "        - Additional features must be beneficial for all gene panels.\n",
    "            - Certain features might prove beneficial to a certain gene panel, but not to others, could indicate that the cause of worse performance for that certain panel lies within that feature.\n",
    "    - The model will be created for genome build 37 in the first place, but in such a way that build 38 can easily be added as supported.\n",
    "    - The model must support CADD 1.4 and CADD 1.6\n",
    "        - CADD 1.5 is basically CADD 1.4 but with fixed GERP annotations for build 38.\n",
    "    - The model must use precomputed scores first. When no precomputed score is found for the variant, it shall be computed.\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Enrichr automated API"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "enrichr_url = 'http://amp.pharm.mssm.edu/Enrichr/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>auc</th>\n",
       "      <th>default_f1</th>\n",
       "      <th>default_recall</th>\n",
       "      <th>default_fpr</th>\n",
       "      <th>default_spec</th>\n",
       "      <th>optimal_c</th>\n",
       "      <th>optimal_auc</th>\n",
       "      <th>optimal_f1</th>\n",
       "      <th>optimal_recall</th>\n",
       "      <th>optimal_fpr</th>\n",
       "      <th>optimal_spec</th>\n",
       "      <th>n_tot</th>\n",
       "      <th>n_benign</th>\n",
       "      <th>n_malign</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2775</th>\n",
       "      <td>PRKN</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1659</th>\n",
       "      <td>BUB1B</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>73</td>\n",
       "      <td>66</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2102</th>\n",
       "      <td>GALNT3</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1655</th>\n",
       "      <td>PNPLA1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.8</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.8</td>\n",
       "      <td>31</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2782</th>\n",
       "      <td>TMEM80</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2625</th>\n",
       "      <td>C10orf2</td>\n",
       "      <td>0.469697</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.469697</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>133</td>\n",
       "      <td>132</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2330</th>\n",
       "      <td>SPTLC2</td>\n",
       "      <td>0.455157</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.011</td>\n",
       "      <td>0.750374</td>\n",
       "      <td>0.170213</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>229</td>\n",
       "      <td>223</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2615</th>\n",
       "      <td>PLCD1</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2611</th>\n",
       "      <td>UBE2A</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2806</th>\n",
       "      <td>SOX9-AS1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2807 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene       auc  default_f1  default_recall  default_fpr  \\\n",
       "2775      PRKN  1.000000         1.0             1.0          0.0   \n",
       "1659     BUB1B  1.000000         1.0             1.0          0.0   \n",
       "2102    GALNT3  1.000000         1.0             1.0          0.0   \n",
       "1655    PNPLA1  1.000000         1.0             1.0          0.0   \n",
       "2782    TMEM80  1.000000         1.0             1.0          0.0   \n",
       "...        ...       ...         ...             ...          ...   \n",
       "2625   C10orf2  0.469697         0.0             0.0          1.0   \n",
       "2330    SPTLC2  0.455157         0.0             0.0          1.0   \n",
       "2615     PLCD1  0.500000         0.0             0.0          1.0   \n",
       "2611     UBE2A  0.500000         0.0             0.0          1.0   \n",
       "2806  SOX9-AS1  0.000000         0.0             0.0          1.0   \n",
       "\n",
       "      default_spec  optimal_c  optimal_auc  optimal_f1  optimal_recall  \\\n",
       "2775           1.0      0.020     1.000000    1.000000             1.0   \n",
       "1659           1.0      0.020     1.000000    1.000000             1.0   \n",
       "2102           1.0      0.020     1.000000    1.000000             1.0   \n",
       "1655           1.0      0.020     1.000000    1.000000             0.8   \n",
       "2782           1.0      0.020     1.000000    1.000000             1.0   \n",
       "...            ...        ...          ...         ...             ...   \n",
       "2625           0.0      0.020     0.469697    0.000000             0.0   \n",
       "2330           0.0      0.011     0.750374    0.170213             0.0   \n",
       "2615           0.0      0.020     0.500000    0.000000             0.0   \n",
       "2611           0.0      0.020     0.500000    0.000000             0.0   \n",
       "2806           0.0      0.020     0.000000    0.000000             0.0   \n",
       "\n",
       "      optimal_fpr  optimal_spec  n_tot  n_benign  n_malign  \n",
       "2775          0.0           1.0     11         7         4  \n",
       "1659          0.0           1.0     73        66         7  \n",
       "2102          0.0           1.0     12        10         2  \n",
       "1655          0.2           0.8     31        26         5  \n",
       "2782          0.0           1.0      5         4         1  \n",
       "...           ...           ...    ...       ...       ...  \n",
       "2625          1.0           0.0    133       132         1  \n",
       "2330          1.0           0.0    229       223         6  \n",
       "2615          1.0           0.0      8         7         1  \n",
       "2611          1.0           0.0      3         2         1  \n",
       "2806          1.0           0.0      3         2         1  \n",
       "\n",
       "[2807 rows x 15 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load in initial data\n",
    "data = pd.read_csv(read_loc+'optimal_f1_full_ds_v2.csv', header=0)\n",
    "data.rename(columns={\n",
    "    'default_auc': 'auc'\n",
    "}, inplace=True)\n",
    "data.sort_values(by='default_f1', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2807"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# data[['n_tot', 'n_benign', 'n_malign']].sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Make a dictionary of lists of genes of interest with as key the 0.x threshold\n",
    "\n",
    "auc_dict = {}\n",
    "thresholds = np.arange(0.5, 1.01, 0.05).round(decimals=2)\n",
    "for t in thresholds:\n",
    "    list_of_genes = data[data['auc'] < t]['gene'].tolist()\n",
    "    auc_dict[t.astype('str')] = '\\n'.join(list_of_genes)\n",
    "    del list_of_genes\n",
    "del thresholds\n",
    "\n",
    "# Save it to a file so I don't bombard the Enrichr API with requests\n",
    "if not os.path.isfile('./user_id_list.json'):\n",
    "    file = open('./user_id_list.json', 'w')\n",
    "    file.close()\n",
    "\n",
    "with open('./user_id_list.json') as json_file:\n",
    "    try:\n",
    "        user_list_ids = json.load(json_file)\n",
    "        json_was_empty = False\n",
    "    except json.JSONDecodeError:\n",
    "        user_list_ids = {}\n",
    "        json_was_empty = True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get list ID's for the different AUC's\n",
    "\n",
    "for threshold in auc_dict.keys():\n",
    "    if threshold not in user_list_ids.keys():\n",
    "        description = f'CAPICE performance of genes with an AUC lower than {threshold}'\n",
    "        payload = {\n",
    "            'list': (None, auc_dict[threshold]),\n",
    "            'description': (None, description)\n",
    "        }\n",
    "        response = requests.post(enrichr_url + '/addList', files=payload)\n",
    "        if not response.ok:\n",
    "            print(f'Something went wrong with {threshold}!')\n",
    "        else:\n",
    "            enrichr_addlist_response = json.loads(response.text)\n",
    "            user_list_ids[threshold] = enrichr_addlist_response['userListId']\n",
    "\n",
    "# Write it to a file in case I didn't have any.\n",
    "if json_was_empty:\n",
    "    with open('./user_id_list.json', 'w') as json_file:\n",
    "        json.dump(user_list_ids, json_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Now to get the enrichment results.\n",
    "\n",
    "enrichr_sources_oi = ['BioPlanet_2019', 'WikiPathways_2019_Human', 'KEGG_2019_Human', 'Elsevier_Pathway_Collection',\n",
    "                     'GO_Biological_Process_2018','Human_Phenotype_Ontology','Jensen_DISEASES',\n",
    "                     'PheWeb_2019', 'ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET','OMIM_Disease','OMIM_Expanded',\n",
    "                     'Human_Gene_Atlas', 'ARCHS4_Tissues',\n",
    "                     'HMDB_Metabolites','Chromosome_Location_hg19']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "output_source_link = {}\n",
    "for source in enrichr_sources_oi:\n",
    "    output_filename = source + '.csv'\n",
    "    output_source_link[source] = output_filename"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "glob_files_present = glob.glob('./EnrichrAPIResults/*.csv')\n",
    "list_of_files_present = []\n",
    "if len(glob_files_present) > 0:\n",
    "    for file in glob_files_present:\n",
    "        list_of_files_present.append(file.split('/')[-1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "sources_required = []\n",
    "for source, output_filename in output_source_link.items():\n",
    "    if output_filename not in list_of_files_present:\n",
    "        sources_required.append(source)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['BioPlanet_2019',\n",
       " 'WikiPathways_2019_Human',\n",
       " 'KEGG_2019_Human',\n",
       " 'Elsevier_Pathway_Collection',\n",
       " 'GO_Biological_Process_2018',\n",
       " 'Human_Phenotype_Ontology',\n",
       " 'Jensen_DISEASES',\n",
       " 'PheWeb_2019',\n",
       " 'ClinVar_2019',\n",
       " 'GWAS_Catalog_2019',\n",
       " 'DisGeNET',\n",
       " 'OMIM_Disease',\n",
       " 'OMIM_Expanded',\n",
       " 'Human_Gene_Atlas',\n",
       " 'ARCHS4_Tissues',\n",
       " 'HMDB_Metabolites',\n",
       " 'Chromosome_Location_hg19']"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sources_required"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Working on source: BioPlanet_2019\n",
      "Sources left: ['WikiPathways_2019_Human', 'KEGG_2019_Human', 'Elsevier_Pathway_Collection', 'GO_Biological_Process_2018', 'Human_Phenotype_Ontology', 'Jensen_DISEASES', 'PheWeb_2019', 'ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/rjsietsma/PycharmProjects/dsls_master_thesis/venv/lib/python3.8/site-packages/pandas/core/frame.py:4160: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  return super().drop(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource BioPlanet_2019 has been exported to csv (\n",
      "\t\t./EnrichrAPIResults/BioPlanet_2019.csv\n",
      "\t)\n",
      "Working on source: WikiPathways_2019_Human\n",
      "Sources left: ['KEGG_2019_Human', 'Elsevier_Pathway_Collection', 'GO_Biological_Process_2018', 'Human_Phenotype_Ontology', 'Jensen_DISEASES', 'PheWeb_2019', 'ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource WikiPathways_2019_Human has been exported to csv (\n",
      "\t\t./EnrichrAPIResults/WikiPathways_2019_Human.csv\n",
      "\t)\n",
      "Working on source: KEGG_2019_Human\n",
      "Sources left: ['Elsevier_Pathway_Collection', 'GO_Biological_Process_2018', 'Human_Phenotype_Ontology', 'Jensen_DISEASES', 'PheWeb_2019', 'ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource KEGG_2019_Human has been exported to csv (\n",
      "\t\t./EnrichrAPIResults/KEGG_2019_Human.csv\n",
      "\t)\n",
      "Working on source: Elsevier_Pathway_Collection\n",
      "Sources left: ['GO_Biological_Process_2018', 'Human_Phenotype_Ontology', 'Jensen_DISEASES', 'PheWeb_2019', 'ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource Elsevier_Pathway_Collection has been exported to csv (\n",
      "\t\t./EnrichrAPIResults/Elsevier_Pathway_Collection.csv\n",
      "\t)\n",
      "Working on source: GO_Biological_Process_2018\n",
      "Sources left: ['Human_Phenotype_Ontology', 'Jensen_DISEASES', 'PheWeb_2019', 'ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource GO_Biological_Process_2018 has been exported to csv (\n",
      "\t\t./EnrichrAPIResults/GO_Biological_Process_2018.csv\n",
      "\t)\n",
      "Working on source: Human_Phenotype_Ontology\n",
      "Sources left: ['Jensen_DISEASES', 'PheWeb_2019', 'ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource Human_Phenotype_Ontology has been exported to csv (\n",
      "\t\t./EnrichrAPIResults/Human_Phenotype_Ontology.csv\n",
      "\t)\n",
      "Working on source: Jensen_DISEASES\n",
      "Sources left: ['PheWeb_2019', 'ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource Jensen_DISEASES has been exported to csv (\n",
      "\t\t./EnrichrAPIResults/Jensen_DISEASES.csv\n",
      "\t)\n",
      "Working on source: PheWeb_2019\n",
      "Sources left: ['ClinVar_2019', 'GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource PheWeb_2019 has been exported to csv (\n",
      "\t\t./EnrichrAPIResults/PheWeb_2019.csv\n",
      "\t)\n",
      "Working on source: ClinVar_2019\n",
      "Sources left: ['GWAS_Catalog_2019', 'DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource ClinVar_2019 has been exported to csv (\n",
      "\t\t./EnrichrAPIResults/ClinVar_2019.csv\n",
      "\t)\n",
      "Working on source: GWAS_Catalog_2019\n",
      "Sources left: ['DisGeNET', 'OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource GWAS_Catalog_2019 has been exported to csv (\n",
      "\t\t./EnrichrAPIResults/GWAS_Catalog_2019.csv\n",
      "\t)\n",
      "Working on source: DisGeNET\n",
      "Sources left: ['OMIM_Disease', 'OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource DisGeNET has been exported to csv (\n",
      "\t\t./EnrichrAPIResults/DisGeNET.csv\n",
      "\t)\n",
      "Working on source: OMIM_Disease\n",
      "Sources left: ['OMIM_Expanded', 'Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource OMIM_Disease has been exported to csv (\n",
      "\t\t./EnrichrAPIResults/OMIM_Disease.csv\n",
      "\t)\n",
      "Working on source: OMIM_Expanded\n",
      "Sources left: ['Human_Gene_Atlas', 'ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource OMIM_Expanded has been exported to csv (\n",
      "\t\t./EnrichrAPIResults/OMIM_Expanded.csv\n",
      "\t)\n",
      "Working on source: Human_Gene_Atlas\n",
      "Sources left: ['ARCHS4_Tissues', 'HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource Human_Gene_Atlas has been exported to csv (\n",
      "\t\t./EnrichrAPIResults/Human_Gene_Atlas.csv\n",
      "\t)\n",
      "Working on source: ARCHS4_Tissues\n",
      "Sources left: ['HMDB_Metabolites', 'Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource ARCHS4_Tissues has been exported to csv (\n",
      "\t\t./EnrichrAPIResults/ARCHS4_Tissues.csv\n",
      "\t)\n",
      "Working on source: HMDB_Metabolites\n",
      "Sources left: ['Chromosome_Location_hg19']\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource HMDB_Metabolites has been exported to csv (\n",
      "\t\t./EnrichrAPIResults/HMDB_Metabolites.csv\n",
      "\t)\n",
      "Working on source: Chromosome_Location_hg19\n",
      "Sources left: []\n",
      "\tCurrently working on threshold: 0.5\n",
      "\tCurrently working on threshold: 0.55\n",
      "\tCurrently working on threshold: 0.6\n",
      "\tCurrently working on threshold: 0.65\n",
      "\tCurrently working on threshold: 0.7\n",
      "\tCurrently working on threshold: 0.75\n",
      "\tCurrently working on threshold: 0.8\n",
      "\tCurrently working on threshold: 0.85\n",
      "\tCurrently working on threshold: 0.9\n",
      "\tCurrently working on threshold: 0.95\n",
      "\tCurrently working on threshold: 1.0\n",
      "\tSource Chromosome_Location_hg19 has been exported to csv (\n",
      "\t\t./EnrichrAPIResults/Chromosome_Location_hg19.csv\n",
      "\t)\n"
     ]
    }
   ],
   "source": [
    "get_enrichr_results(sources_required, user_list_ids, 20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "enrichr_data_loc = './EnrichrAPIResults/'\n",
    "\n",
    "enrichr_dataset = {}\n",
    "\n",
    "for source in enrichr_sources_oi:\n",
    "    dataset = enrichr_data_loc + source + '.csv'\n",
    "    enrichr_dataset[source] = pd.read_csv(dataset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.5</td>\n",
       "      <td>3</td>\n",
       "      <td>Activation of TRKA receptors</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.5</td>\n",
       "      <td>4</td>\n",
       "      <td>MAP kinase pathway regulation through dual spe...</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.5</td>\n",
       "      <td>5</td>\n",
       "      <td>ERK inactivation</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.5</td>\n",
       "      <td>7</td>\n",
       "      <td>Nucleotide-like (purinergic) G-protein coupled...</td>\n",
       "      <td>P2RY12</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.5</td>\n",
       "      <td>8</td>\n",
       "      <td>Gap junction assembly</td>\n",
       "      <td>GJB3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   AUC  Rank                                          Term_Name  \\\n",
       "0  0.5     3                       Activation of TRKA receptors   \n",
       "1  0.5     4  MAP kinase pathway regulation through dual spe...   \n",
       "2  0.5     5                                   ERK inactivation   \n",
       "3  0.5     7  Nucleotide-like (purinergic) G-protein coupled...   \n",
       "4  0.5     8                              Gap junction assembly   \n",
       "\n",
       "  Overlapping_Genes  Adjusted_P-value          Source  \n",
       "0             NTRK2               1.0  BioPlanet_2019  \n",
       "1             DUSP6               1.0  BioPlanet_2019  \n",
       "2             DUSP6               1.0  BioPlanet_2019  \n",
       "3            P2RY12               1.0  BioPlanet_2019  \n",
       "4              GJB3               1.0  BioPlanet_2019  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enrichr_dataset['BioPlanet_2019'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "for key in enrichr_dataset.keys():\n",
    "    plot = plot_results(enrichr_dataset[key], 10)\n",
    "    save(plot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.5</td>\n",
       "      <td>3</td>\n",
       "      <td>Activation of TRKA receptors</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.5</td>\n",
       "      <td>4</td>\n",
       "      <td>MAP kinase pathway regulation through dual spe...</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.5</td>\n",
       "      <td>5</td>\n",
       "      <td>ERK inactivation</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.5</td>\n",
       "      <td>7</td>\n",
       "      <td>Nucleotide-like (purinergic) G-protein coupled...</td>\n",
       "      <td>P2RY12</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.5</td>\n",
       "      <td>8</td>\n",
       "      <td>Gap junction assembly</td>\n",
       "      <td>GJB3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1804</th>\n",
       "      <td>1.0</td>\n",
       "      <td>8</td>\n",
       "      <td>chr16</td>\n",
       "      <td>CTCF, LITAF, PKD1, SALL1, CDH1, TUBB3, CHMP1A,...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1805</th>\n",
       "      <td>1.0</td>\n",
       "      <td>6</td>\n",
       "      <td>chr6 mcf hap5</td>\n",
       "      <td>TNXB, CYP21A2, TUBB, COL11A2, SKIV2L, VARS2, C...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1806</th>\n",
       "      <td>1.0</td>\n",
       "      <td>9</td>\n",
       "      <td>chr15</td>\n",
       "      <td>GABRB3, MTFMT, HEXA, ETFA, UBE3A, CHD2, NR2E3,...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1807</th>\n",
       "      <td>1.0</td>\n",
       "      <td>7</td>\n",
       "      <td>chr6 dbb hap3</td>\n",
       "      <td>TNXB, CYP21A2, TUBB, COL11A2, SKIV2L, VARS2, C...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1808</th>\n",
       "      <td>1.0</td>\n",
       "      <td>11</td>\n",
       "      <td>chr10</td>\n",
       "      <td>RET, MRPS16, PTEN, PRF1, LDB3, KIF11, SMC3, LI...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1809 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      AUC Rank                                          Term_Name  \\\n",
       "0     0.5    3                       Activation of TRKA receptors   \n",
       "1     0.5    4  MAP kinase pathway regulation through dual spe...   \n",
       "2     0.5    5                                   ERK inactivation   \n",
       "3     0.5    7  Nucleotide-like (purinergic) G-protein coupled...   \n",
       "4     0.5    8                              Gap junction assembly   \n",
       "...   ...  ...                                                ...   \n",
       "1804  1.0    8                                              chr16   \n",
       "1805  1.0    6                                      chr6 mcf hap5   \n",
       "1806  1.0    9                                              chr15   \n",
       "1807  1.0    7                                      chr6 dbb hap3   \n",
       "1808  1.0   11                                              chr10   \n",
       "\n",
       "                                      Overlapping_Genes  Adjusted_P-value  \\\n",
       "0                                                 NTRK2               1.0   \n",
       "1                                                 DUSP6               1.0   \n",
       "2                                                 DUSP6               1.0   \n",
       "3                                                P2RY12               1.0   \n",
       "4                                                  GJB3               1.0   \n",
       "...                                                 ...               ...   \n",
       "1804  CTCF, LITAF, PKD1, SALL1, CDH1, TUBB3, CHMP1A,...               1.0   \n",
       "1805  TNXB, CYP21A2, TUBB, COL11A2, SKIV2L, VARS2, C...               1.0   \n",
       "1806  GABRB3, MTFMT, HEXA, ETFA, UBE3A, CHD2, NR2E3,...               1.0   \n",
       "1807  TNXB, CYP21A2, TUBB, COL11A2, SKIV2L, VARS2, C...               1.0   \n",
       "1808  RET, MRPS16, PTEN, PRF1, LDB3, KIF11, SMC3, LI...               1.0   \n",
       "\n",
       "                        Source  \n",
       "0               BioPlanet_2019  \n",
       "1               BioPlanet_2019  \n",
       "2               BioPlanet_2019  \n",
       "3               BioPlanet_2019  \n",
       "4               BioPlanet_2019  \n",
       "...                        ...  \n",
       "1804  Chromosome_Location_hg19  \n",
       "1805  Chromosome_Location_hg19  \n",
       "1806  Chromosome_Location_hg19  \n",
       "1807  Chromosome_Location_hg19  \n",
       "1808  Chromosome_Location_hg19  \n",
       "\n",
       "[1809 rows x 6 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df = pd.DataFrame(columns=enrichr_dataset['BioPlanet_2019'].columns)\n",
    "for key, value in enrichr_dataset.items():\n",
    "    subdf = pd.DataFrame(columns=combined_df.columns)\n",
    "    for auc in value['AUC'].unique():\n",
    "        subdf = subdf.append(value[value['AUC'] == auc][:10], ignore_index=True)\n",
    "    combined_df = combined_df.append(subdf, ignore_index=True)\n",
    "combined_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.50</td>\n",
       "      <td>3</td>\n",
       "      <td>Activation of TRKA receptors</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.50</td>\n",
       "      <td>4</td>\n",
       "      <td>MAP kinase pathway regulation through dual spe...</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.50</td>\n",
       "      <td>5</td>\n",
       "      <td>ERK inactivation</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.50</td>\n",
       "      <td>7</td>\n",
       "      <td>Nucleotide-like (purinergic) G-protein coupled...</td>\n",
       "      <td>P2RY12</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.50</td>\n",
       "      <td>8</td>\n",
       "      <td>Gap junction assembly</td>\n",
       "      <td>GJB3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.50</td>\n",
       "      <td>9</td>\n",
       "      <td>Interleukin-11 pathway</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.50</td>\n",
       "      <td>10</td>\n",
       "      <td>ADP signalling through P2Y purinoceptor 12</td>\n",
       "      <td>P2RY12</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.50</td>\n",
       "      <td>11</td>\n",
       "      <td>Gap junction trafficking and regulation</td>\n",
       "      <td>GJB3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0.50</td>\n",
       "      <td>12</td>\n",
       "      <td>Nuclear events mediated by MAP kinases</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>0.50</td>\n",
       "      <td>13</td>\n",
       "      <td>Signal amplification</td>\n",
       "      <td>P2RY12</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>0.55</td>\n",
       "      <td>2</td>\n",
       "      <td>Activation of TRKA receptors</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>0.55</td>\n",
       "      <td>4</td>\n",
       "      <td>MAP kinase pathway regulation through dual spe...</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>0.55</td>\n",
       "      <td>5</td>\n",
       "      <td>Opsins</td>\n",
       "      <td>OPN1MW</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>0.55</td>\n",
       "      <td>8</td>\n",
       "      <td>SUMOylation as a mechanism to modulate CtBP-de...</td>\n",
       "      <td>UBE2A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>0.55</td>\n",
       "      <td>9</td>\n",
       "      <td>ERK inactivation</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>0.55</td>\n",
       "      <td>10</td>\n",
       "      <td>Fibrinolysis pathway</td>\n",
       "      <td>UBE2A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>0.55</td>\n",
       "      <td>11</td>\n",
       "      <td>Hypoxia-inducible factor in the cardivascular ...</td>\n",
       "      <td>UBE2A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>0.55</td>\n",
       "      <td>12</td>\n",
       "      <td>Nucleotide-like (purinergic) G-protein coupled...</td>\n",
       "      <td>P2RY12</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>0.55</td>\n",
       "      <td>14</td>\n",
       "      <td>Gap junction assembly</td>\n",
       "      <td>GJB3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>0.55</td>\n",
       "      <td>16</td>\n",
       "      <td>Hypoxia and p53 in the cardiovascular system</td>\n",
       "      <td>UBE2A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     AUC Rank                                          Term_Name  \\\n",
       "0   0.50    3                       Activation of TRKA receptors   \n",
       "1   0.50    4  MAP kinase pathway regulation through dual spe...   \n",
       "2   0.50    5                                   ERK inactivation   \n",
       "3   0.50    7  Nucleotide-like (purinergic) G-protein coupled...   \n",
       "4   0.50    8                              Gap junction assembly   \n",
       "5   0.50    9                             Interleukin-11 pathway   \n",
       "6   0.50   10         ADP signalling through P2Y purinoceptor 12   \n",
       "7   0.50   11            Gap junction trafficking and regulation   \n",
       "8   0.50   12             Nuclear events mediated by MAP kinases   \n",
       "9   0.50   13                               Signal amplification   \n",
       "10  0.55    2                       Activation of TRKA receptors   \n",
       "11  0.55    4  MAP kinase pathway regulation through dual spe...   \n",
       "12  0.55    5                                             Opsins   \n",
       "13  0.55    8  SUMOylation as a mechanism to modulate CtBP-de...   \n",
       "14  0.55    9                                   ERK inactivation   \n",
       "15  0.55   10                               Fibrinolysis pathway   \n",
       "16  0.55   11  Hypoxia-inducible factor in the cardivascular ...   \n",
       "17  0.55   12  Nucleotide-like (purinergic) G-protein coupled...   \n",
       "18  0.55   14                              Gap junction assembly   \n",
       "19  0.55   16       Hypoxia and p53 in the cardiovascular system   \n",
       "\n",
       "   Overlapping_Genes  Adjusted_P-value          Source  \n",
       "0              NTRK2               1.0  BioPlanet_2019  \n",
       "1              DUSP6               1.0  BioPlanet_2019  \n",
       "2              DUSP6               1.0  BioPlanet_2019  \n",
       "3             P2RY12               1.0  BioPlanet_2019  \n",
       "4               GJB3               1.0  BioPlanet_2019  \n",
       "5              DUSP6               1.0  BioPlanet_2019  \n",
       "6             P2RY12               1.0  BioPlanet_2019  \n",
       "7               GJB3               1.0  BioPlanet_2019  \n",
       "8              DUSP6               1.0  BioPlanet_2019  \n",
       "9             P2RY12               1.0  BioPlanet_2019  \n",
       "10             NTRK2               1.0  BioPlanet_2019  \n",
       "11             DUSP6               1.0  BioPlanet_2019  \n",
       "12            OPN1MW               1.0  BioPlanet_2019  \n",
       "13             UBE2A               1.0  BioPlanet_2019  \n",
       "14             DUSP6               1.0  BioPlanet_2019  \n",
       "15             UBE2A               1.0  BioPlanet_2019  \n",
       "16             UBE2A               1.0  BioPlanet_2019  \n",
       "17            P2RY12               1.0  BioPlanet_2019  \n",
       "18              GJB3               1.0  BioPlanet_2019  \n",
       "19             UBE2A               1.0  BioPlanet_2019  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/combined_enrichr_plot_results_interactive_legend.html'"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "plot = plot_results_in_singleplot_iteractive_legend(combined_df)\n",
    "save(plot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "BokehDeprecationWarning: 'legend' keyword is deprecated, use explicit 'legend_label', 'legend_field', or 'legend_group' keywords instead\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/combined_enrichr_plot_results.html'"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "plot = plot_results_in_singleplot(combined_df)\n",
    "save(plot)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "## Interpretation\n",
    "It seems that a normal distribution like pattern is formed with a steeper slope after the maximum, starting at rougly AUC of 0.3 and ending at roughly 0.8.\n",
    "<br/><br/>\n",
    "- DisGeNET seems to outline an interesting picture with the top 10 term names:\n",
    "    - For it seems that in the really bad performing genes the main cause could be infant development related. Later on, more neurological causes are identified.\n",
    "        - Neurological seems clear to me... There is still so much unknown about the human mind.\n",
    "        - Infant development related is a bit less clear, since there is a growing amount of research towards this infant development.\n",
    "- GO_biological_process seems to have primarily the same genes for AUC 0.55-0.7 that are most enriched\n",
    "    - It seems linked to ion channeling\n",
    "- Feet and toe abnormalities seem to occur once or twice too\n",
    "- Could this be an indication to genes that are related to very small molecules? (like steroids, ions, neuro-molecules?)\n",
    "- Most significant: autosomal dominant inheritence\n",
    "    - Very interesting plot, the HPO bokeh plot of n=10\n",
    "- \n",
    "\n",
    "<br/><br/>\n",
    "TODO: make word counts of genes and term names?\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.5</td>\n",
       "      <td>3</td>\n",
       "      <td>Activation of TRKA receptors</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.5</td>\n",
       "      <td>4</td>\n",
       "      <td>MAP kinase pathway regulation through dual spe...</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.5</td>\n",
       "      <td>5</td>\n",
       "      <td>ERK inactivation</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.5</td>\n",
       "      <td>7</td>\n",
       "      <td>Nucleotide-like (purinergic) G-protein coupled...</td>\n",
       "      <td>P2RY12</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.5</td>\n",
       "      <td>8</td>\n",
       "      <td>Gap junction assembly</td>\n",
       "      <td>GJB3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3359</th>\n",
       "      <td>1.0</td>\n",
       "      <td>17</td>\n",
       "      <td>chr22</td>\n",
       "      <td>SMARCB1, TMPRSS6, PDGFB, TCF20, PLA2G6, TYMP, ...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3360</th>\n",
       "      <td>1.0</td>\n",
       "      <td>16</td>\n",
       "      <td>chr6 cox hap2</td>\n",
       "      <td>TNXB, CYP21A2, TUBB, COL11A2, SKIV2L, VARS2, C...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3361</th>\n",
       "      <td>1.0</td>\n",
       "      <td>18</td>\n",
       "      <td>chr21</td>\n",
       "      <td>COL18A1, KCNE2, TMPRSS3, ITGB2, DYRK1A, TSPEAR...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3362</th>\n",
       "      <td>1.0</td>\n",
       "      <td>20</td>\n",
       "      <td>chr3</td>\n",
       "      <td>SETD5, SETD2, GFM1, ZBTB20, PLOD2, SLC6A1, MYL...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3363</th>\n",
       "      <td>1.0</td>\n",
       "      <td>19</td>\n",
       "      <td>chr6 apd hap1</td>\n",
       "      <td>TUBB, PSMB8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3364 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      AUC Rank                                          Term_Name  \\\n",
       "0     0.5    3                       Activation of TRKA receptors   \n",
       "1     0.5    4  MAP kinase pathway regulation through dual spe...   \n",
       "2     0.5    5                                   ERK inactivation   \n",
       "3     0.5    7  Nucleotide-like (purinergic) G-protein coupled...   \n",
       "4     0.5    8                              Gap junction assembly   \n",
       "...   ...  ...                                                ...   \n",
       "3359  1.0   17                                              chr22   \n",
       "3360  1.0   16                                      chr6 cox hap2   \n",
       "3361  1.0   18                                              chr21   \n",
       "3362  1.0   20                                               chr3   \n",
       "3363  1.0   19                                      chr6 apd hap1   \n",
       "\n",
       "                                      Overlapping_Genes  Adjusted_P-value  \\\n",
       "0                                                 NTRK2               1.0   \n",
       "1                                                 DUSP6               1.0   \n",
       "2                                                 DUSP6               1.0   \n",
       "3                                                P2RY12               1.0   \n",
       "4                                                  GJB3               1.0   \n",
       "...                                                 ...               ...   \n",
       "3359  SMARCB1, TMPRSS6, PDGFB, TCF20, PLA2G6, TYMP, ...               1.0   \n",
       "3360  TNXB, CYP21A2, TUBB, COL11A2, SKIV2L, VARS2, C...               1.0   \n",
       "3361  COL18A1, KCNE2, TMPRSS3, ITGB2, DYRK1A, TSPEAR...               1.0   \n",
       "3362  SETD5, SETD2, GFM1, ZBTB20, PLOD2, SLC6A1, MYL...               1.0   \n",
       "3363                                        TUBB, PSMB8               1.0   \n",
       "\n",
       "                        Source  \n",
       "0               BioPlanet_2019  \n",
       "1               BioPlanet_2019  \n",
       "2               BioPlanet_2019  \n",
       "3               BioPlanet_2019  \n",
       "4               BioPlanet_2019  \n",
       "...                        ...  \n",
       "3359  Chromosome_Location_hg19  \n",
       "3360  Chromosome_Location_hg19  \n",
       "3361  Chromosome_Location_hg19  \n",
       "3362  Chromosome_Location_hg19  \n",
       "3363  Chromosome_Location_hg19  \n",
       "\n",
       "[3364 rows x 6 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df_ext = pd.DataFrame(columns=enrichr_dataset['BioPlanet_2019'].columns)\n",
    "for key, value in enrichr_dataset.items():\n",
    "    subdf = pd.DataFrame(columns=combined_df_ext.columns)\n",
    "    for auc in value['AUC'].unique():\n",
    "        subdf = subdf.append(value[value['AUC'] == auc], ignore_index=True)\n",
    "    combined_df_ext = combined_df_ext.append(subdf, ignore_index=True)\n",
    "combined_df_ext"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.5</td>\n",
       "      <td>3</td>\n",
       "      <td>Activation of TRKA receptors</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.5</td>\n",
       "      <td>4</td>\n",
       "      <td>MAP kinase pathway regulation through dual spe...</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.5</td>\n",
       "      <td>5</td>\n",
       "      <td>ERK inactivation</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.5</td>\n",
       "      <td>7</td>\n",
       "      <td>Nucleotide-like (purinergic) G-protein coupled...</td>\n",
       "      <td>P2RY12</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.5</td>\n",
       "      <td>8</td>\n",
       "      <td>Gap junction assembly</td>\n",
       "      <td>GJB3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.5</td>\n",
       "      <td>9</td>\n",
       "      <td>Interleukin-11 pathway</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.5</td>\n",
       "      <td>10</td>\n",
       "      <td>ADP signalling through P2Y purinoceptor 12</td>\n",
       "      <td>P2RY12</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.5</td>\n",
       "      <td>11</td>\n",
       "      <td>Gap junction trafficking and regulation</td>\n",
       "      <td>GJB3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0.5</td>\n",
       "      <td>12</td>\n",
       "      <td>Nuclear events mediated by MAP kinases</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>0.5</td>\n",
       "      <td>13</td>\n",
       "      <td>Signal amplification</td>\n",
       "      <td>P2RY12</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>0.5</td>\n",
       "      <td>2</td>\n",
       "      <td>NGF signaling via TRKA from the plasma membrane</td>\n",
       "      <td>NTRK2, DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>0.5</td>\n",
       "      <td>14</td>\n",
       "      <td>Wnt interactions in lipid metabolism and immun...</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>0.5</td>\n",
       "      <td>15</td>\n",
       "      <td>Post-translational regulation of adherens junc...</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>0.5</td>\n",
       "      <td>16</td>\n",
       "      <td>Arf6 integrin-mediated signaling pathway</td>\n",
       "      <td>MAPK8IP3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>0.5</td>\n",
       "      <td>17</td>\n",
       "      <td>Non-coding RNA metabolism</td>\n",
       "      <td>SMN1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>0.5</td>\n",
       "      <td>1</td>\n",
       "      <td>MAPK signaling pathway</td>\n",
       "      <td>NTRK2, MAPK8IP3, DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>0.5</td>\n",
       "      <td>18</td>\n",
       "      <td>JNK/MAPK pathway</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>0.5</td>\n",
       "      <td>19</td>\n",
       "      <td>SHP2 signaling</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>0.5</td>\n",
       "      <td>20</td>\n",
       "      <td>Neurotrophic factor-mediated Trk receptor sign...</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>0.5</td>\n",
       "      <td>21</td>\n",
       "      <td>FGF signaling pathway</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    AUC Rank                                          Term_Name  \\\n",
       "0   0.5    3                       Activation of TRKA receptors   \n",
       "1   0.5    4  MAP kinase pathway regulation through dual spe...   \n",
       "2   0.5    5                                   ERK inactivation   \n",
       "3   0.5    7  Nucleotide-like (purinergic) G-protein coupled...   \n",
       "4   0.5    8                              Gap junction assembly   \n",
       "5   0.5    9                             Interleukin-11 pathway   \n",
       "6   0.5   10         ADP signalling through P2Y purinoceptor 12   \n",
       "7   0.5   11            Gap junction trafficking and regulation   \n",
       "8   0.5   12             Nuclear events mediated by MAP kinases   \n",
       "9   0.5   13                               Signal amplification   \n",
       "10  0.5    2    NGF signaling via TRKA from the plasma membrane   \n",
       "11  0.5   14  Wnt interactions in lipid metabolism and immun...   \n",
       "12  0.5   15  Post-translational regulation of adherens junc...   \n",
       "13  0.5   16           Arf6 integrin-mediated signaling pathway   \n",
       "14  0.5   17                          Non-coding RNA metabolism   \n",
       "15  0.5    1                             MAPK signaling pathway   \n",
       "16  0.5   18                                   JNK/MAPK pathway   \n",
       "17  0.5   19                                     SHP2 signaling   \n",
       "18  0.5   20  Neurotrophic factor-mediated Trk receptor sign...   \n",
       "19  0.5   21                              FGF signaling pathway   \n",
       "\n",
       "         Overlapping_Genes  Adjusted_P-value          Source  \n",
       "0                    NTRK2               1.0  BioPlanet_2019  \n",
       "1                    DUSP6               1.0  BioPlanet_2019  \n",
       "2                    DUSP6               1.0  BioPlanet_2019  \n",
       "3                   P2RY12               1.0  BioPlanet_2019  \n",
       "4                     GJB3               1.0  BioPlanet_2019  \n",
       "5                    DUSP6               1.0  BioPlanet_2019  \n",
       "6                   P2RY12               1.0  BioPlanet_2019  \n",
       "7                     GJB3               1.0  BioPlanet_2019  \n",
       "8                    DUSP6               1.0  BioPlanet_2019  \n",
       "9                   P2RY12               1.0  BioPlanet_2019  \n",
       "10            NTRK2, DUSP6               1.0  BioPlanet_2019  \n",
       "11                   DUSP6               1.0  BioPlanet_2019  \n",
       "12                   NTRK2               1.0  BioPlanet_2019  \n",
       "13                MAPK8IP3               1.0  BioPlanet_2019  \n",
       "14                    SMN1               1.0  BioPlanet_2019  \n",
       "15  NTRK2, MAPK8IP3, DUSP6               1.0  BioPlanet_2019  \n",
       "16                   DUSP6               1.0  BioPlanet_2019  \n",
       "17                   NTRK2               1.0  BioPlanet_2019  \n",
       "18                   NTRK2               1.0  BioPlanet_2019  \n",
       "19                   DUSP6               1.0  BioPlanet_2019  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df_ext.where((combined_df_ext['AUC'] == 0.5) & (combined_df_ext['Source'] == 'BioPlanet_2019')).dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "word_count_df = count_words_in_df(combined_df_ext)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>count</th>\n",
       "      <th>auc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>signaling</td>\n",
       "      <td>19</td>\n",
       "      <td>0.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>pathway</td>\n",
       "      <td>14</td>\n",
       "      <td>0.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>disease</td>\n",
       "      <td>9</td>\n",
       "      <td>0.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>9</td>\n",
       "      <td>0.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>(bulk</td>\n",
       "      <td>9</td>\n",
       "      <td>0.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>498</th>\n",
       "      <td>signaling</td>\n",
       "      <td>13</td>\n",
       "      <td>0.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499</th>\n",
       "      <td>pathway</td>\n",
       "      <td>10</td>\n",
       "      <td>0.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500</th>\n",
       "      <td>disease</td>\n",
       "      <td>10</td>\n",
       "      <td>0.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501</th>\n",
       "      <td>regulation</td>\n",
       "      <td>9</td>\n",
       "      <td>0.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>9</td>\n",
       "      <td>0.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1055</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>14</td>\n",
       "      <td>0.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1056</th>\n",
       "      <td>signaling</td>\n",
       "      <td>13</td>\n",
       "      <td>0.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1057</th>\n",
       "      <td>pathway</td>\n",
       "      <td>10</td>\n",
       "      <td>0.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1058</th>\n",
       "      <td>disease</td>\n",
       "      <td>10</td>\n",
       "      <td>0.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059</th>\n",
       "      <td>regulation</td>\n",
       "      <td>8</td>\n",
       "      <td>0.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1634</th>\n",
       "      <td>pathway</td>\n",
       "      <td>13</td>\n",
       "      <td>0.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1635</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>13</td>\n",
       "      <td>0.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1636</th>\n",
       "      <td>signaling</td>\n",
       "      <td>11</td>\n",
       "      <td>0.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1637</th>\n",
       "      <td>disease</td>\n",
       "      <td>10</td>\n",
       "      <td>0.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1638</th>\n",
       "      <td>regulation</td>\n",
       "      <td>8</td>\n",
       "      <td>0.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2246</th>\n",
       "      <td>disease</td>\n",
       "      <td>14</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2247</th>\n",
       "      <td>pathway</td>\n",
       "      <td>13</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2248</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>11</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2249</th>\n",
       "      <td>signaling</td>\n",
       "      <td>9</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2250</th>\n",
       "      <td>metabolism</td>\n",
       "      <td>7</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2858</th>\n",
       "      <td>disease</td>\n",
       "      <td>15</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2859</th>\n",
       "      <td>pathway</td>\n",
       "      <td>14</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2860</th>\n",
       "      <td>metabolism</td>\n",
       "      <td>10</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2861</th>\n",
       "      <td>regulation</td>\n",
       "      <td>9</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2862</th>\n",
       "      <td>acid</td>\n",
       "      <td>8</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3490</th>\n",
       "      <td>disease</td>\n",
       "      <td>16</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3491</th>\n",
       "      <td>pathway</td>\n",
       "      <td>11</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3492</th>\n",
       "      <td>regulation</td>\n",
       "      <td>10</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3493</th>\n",
       "      <td>metabolism</td>\n",
       "      <td>9</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3494</th>\n",
       "      <td>signaling</td>\n",
       "      <td>9</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4124</th>\n",
       "      <td>pathway</td>\n",
       "      <td>20</td>\n",
       "      <td>0.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4125</th>\n",
       "      <td>signaling</td>\n",
       "      <td>18</td>\n",
       "      <td>0.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4126</th>\n",
       "      <td>disease</td>\n",
       "      <td>17</td>\n",
       "      <td>0.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4127</th>\n",
       "      <td>metabolism</td>\n",
       "      <td>13</td>\n",
       "      <td>0.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4128</th>\n",
       "      <td>cholesterol</td>\n",
       "      <td>12</td>\n",
       "      <td>0.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4741</th>\n",
       "      <td>signaling</td>\n",
       "      <td>18</td>\n",
       "      <td>0.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4742</th>\n",
       "      <td>pathway</td>\n",
       "      <td>16</td>\n",
       "      <td>0.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4743</th>\n",
       "      <td>disease</td>\n",
       "      <td>12</td>\n",
       "      <td>0.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4744</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>11</td>\n",
       "      <td>0.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4745</th>\n",
       "      <td>cholesterol</td>\n",
       "      <td>9</td>\n",
       "      <td>0.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5335</th>\n",
       "      <td>pathway</td>\n",
       "      <td>13</td>\n",
       "      <td>0.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5336</th>\n",
       "      <td>disease</td>\n",
       "      <td>13</td>\n",
       "      <td>0.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5337</th>\n",
       "      <td>proteins</td>\n",
       "      <td>11</td>\n",
       "      <td>0.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5338</th>\n",
       "      <td>metabolism</td>\n",
       "      <td>10</td>\n",
       "      <td>0.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5339</th>\n",
       "      <td>cardiomyopathy</td>\n",
       "      <td>10</td>\n",
       "      <td>0.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5890</th>\n",
       "      <td>cancer</td>\n",
       "      <td>29</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5891</th>\n",
       "      <td>cardiomyopathy</td>\n",
       "      <td>18</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5892</th>\n",
       "      <td>pathway</td>\n",
       "      <td>16</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5893</th>\n",
       "      <td>signaling</td>\n",
       "      <td>13</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5894</th>\n",
       "      <td>breast</td>\n",
       "      <td>12</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                word count   auc\n",
       "0          signaling    19  0.50\n",
       "1            pathway    14  0.50\n",
       "2            disease     9  0.50\n",
       "3           syndrome     9  0.50\n",
       "4              (bulk     9  0.50\n",
       "498        signaling    13  0.55\n",
       "499          pathway    10  0.55\n",
       "500          disease    10  0.55\n",
       "501       regulation     9  0.55\n",
       "502         syndrome     9  0.55\n",
       "1055        syndrome    14  0.60\n",
       "1056       signaling    13  0.60\n",
       "1057         pathway    10  0.60\n",
       "1058         disease    10  0.60\n",
       "1059      regulation     8  0.60\n",
       "1634         pathway    13  0.65\n",
       "1635        syndrome    13  0.65\n",
       "1636       signaling    11  0.65\n",
       "1637         disease    10  0.65\n",
       "1638      regulation     8  0.65\n",
       "2246         disease    14  0.70\n",
       "2247         pathway    13  0.70\n",
       "2248        syndrome    11  0.70\n",
       "2249       signaling     9  0.70\n",
       "2250      metabolism     7  0.70\n",
       "2858         disease    15  0.75\n",
       "2859         pathway    14  0.75\n",
       "2860      metabolism    10  0.75\n",
       "2861      regulation     9  0.75\n",
       "2862            acid     8  0.75\n",
       "3490         disease    16  0.80\n",
       "3491         pathway    11  0.80\n",
       "3492      regulation    10  0.80\n",
       "3493      metabolism     9  0.80\n",
       "3494       signaling     9  0.80\n",
       "4124         pathway    20  0.85\n",
       "4125       signaling    18  0.85\n",
       "4126         disease    17  0.85\n",
       "4127      metabolism    13  0.85\n",
       "4128     cholesterol    12  0.85\n",
       "4741       signaling    18  0.90\n",
       "4742         pathway    16  0.90\n",
       "4743         disease    12  0.90\n",
       "4744        syndrome    11  0.90\n",
       "4745     cholesterol     9  0.90\n",
       "5335         pathway    13  0.95\n",
       "5336         disease    13  0.95\n",
       "5337        proteins    11  0.95\n",
       "5338      metabolism    10  0.95\n",
       "5339  cardiomyopathy    10  0.95\n",
       "5890          cancer    29  1.00\n",
       "5891  cardiomyopathy    18  1.00\n",
       "5892         pathway    16  1.00\n",
       "5893       signaling    13  1.00\n",
       "5894          breast    12  1.00"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nwkdf = pd.DataFrame(columns=word_count_df.columns)\n",
    "for auc in word_count_df['auc'].unique():\n",
    "    nwkdf = nwkdf.append(word_count_df[word_count_df['auc'] == auc][:5])\n",
    "nwkdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/enrichr_word_count_bokeh.html'"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "plot = plot_count_results(nwkdf, item='word')\n",
    "save(plot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "gene_count_df = pd.DataFrame(columns=['gene', 'count', 'auc'])\n",
    "for auc in combined_df_ext['AUC'].unique():\n",
    "    subset_df = combined_df_ext[combined_df_ext['AUC'] == auc]\n",
    "    gene_list = []\n",
    "    unsplitted_gene_list = subset_df['Overlapping_Genes'].tolist()\n",
    "    for unsplitted_gene in unsplitted_gene_list:\n",
    "        gene_list += unsplitted_gene.split(', ')\n",
    "    gene_count_dict = {}\n",
    "    for gene in gene_list:\n",
    "        if gene not in gene_count_dict.keys():\n",
    "            gene_count_dict[gene] = 1\n",
    "        else:\n",
    "            gene_count_dict[gene] += 1\n",
    "    temp_df = pd.DataFrame(columns=gene_count_df.columns)\n",
    "    for key in gene_count_dict.keys():\n",
    "        temp_temp_df = pd.DataFrame({'gene':key, 'count': gene_count_dict[key],\n",
    "                                    'auc': auc}, index=[0])\n",
    "        temp_df = temp_df.append(temp_temp_df, ignore_index=True)\n",
    "    gene_count_df = gene_count_df.append(temp_df, ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "gene_count_df.sort_values(by=['auc','count'], ascending=[True, False], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "ngkdf = pd.DataFrame(columns=gene_count_df.columns)\n",
    "for auc in gene_count_df['auc'].unique():\n",
    "    ngkdf = ngkdf.append(gene_count_df[gene_count_df['auc'] == auc][:5])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/enrichr_gene_count_bokeh.html'"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "plot = plot_count_results(ngkdf, item='gene')\n",
    "save(plot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.5</td>\n",
       "      <td>3</td>\n",
       "      <td>Activation of TRKA receptors</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.5</td>\n",
       "      <td>4</td>\n",
       "      <td>MAP kinase pathway regulation through dual spe...</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.5</td>\n",
       "      <td>5</td>\n",
       "      <td>ERK inactivation</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.5</td>\n",
       "      <td>7</td>\n",
       "      <td>Nucleotide-like (purinergic) G-protein coupled...</td>\n",
       "      <td>P2RY12</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.5</td>\n",
       "      <td>8</td>\n",
       "      <td>Gap junction assembly</td>\n",
       "      <td>GJB3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3359</th>\n",
       "      <td>1.0</td>\n",
       "      <td>17</td>\n",
       "      <td>chr22</td>\n",
       "      <td>SMARCB1, TMPRSS6, PDGFB, TCF20, PLA2G6, TYMP, ...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3360</th>\n",
       "      <td>1.0</td>\n",
       "      <td>16</td>\n",
       "      <td>chr6 cox hap2</td>\n",
       "      <td>TNXB, CYP21A2, TUBB, COL11A2, SKIV2L, VARS2, C...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3361</th>\n",
       "      <td>1.0</td>\n",
       "      <td>18</td>\n",
       "      <td>chr21</td>\n",
       "      <td>COL18A1, KCNE2, TMPRSS3, ITGB2, DYRK1A, TSPEAR...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3362</th>\n",
       "      <td>1.0</td>\n",
       "      <td>20</td>\n",
       "      <td>chr3</td>\n",
       "      <td>SETD5, SETD2, GFM1, ZBTB20, PLOD2, SLC6A1, MYL...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3363</th>\n",
       "      <td>1.0</td>\n",
       "      <td>19</td>\n",
       "      <td>chr6 apd hap1</td>\n",
       "      <td>TUBB, PSMB8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3364 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      AUC Rank                                          Term_Name  \\\n",
       "0     0.5    3                       Activation of TRKA receptors   \n",
       "1     0.5    4  MAP kinase pathway regulation through dual spe...   \n",
       "2     0.5    5                                   ERK inactivation   \n",
       "3     0.5    7  Nucleotide-like (purinergic) G-protein coupled...   \n",
       "4     0.5    8                              Gap junction assembly   \n",
       "...   ...  ...                                                ...   \n",
       "3359  1.0   17                                              chr22   \n",
       "3360  1.0   16                                      chr6 cox hap2   \n",
       "3361  1.0   18                                              chr21   \n",
       "3362  1.0   20                                               chr3   \n",
       "3363  1.0   19                                      chr6 apd hap1   \n",
       "\n",
       "                                      Overlapping_Genes  Adjusted_P-value  \\\n",
       "0                                                 NTRK2               1.0   \n",
       "1                                                 DUSP6               1.0   \n",
       "2                                                 DUSP6               1.0   \n",
       "3                                                P2RY12               1.0   \n",
       "4                                                  GJB3               1.0   \n",
       "...                                                 ...               ...   \n",
       "3359  SMARCB1, TMPRSS6, PDGFB, TCF20, PLA2G6, TYMP, ...               1.0   \n",
       "3360  TNXB, CYP21A2, TUBB, COL11A2, SKIV2L, VARS2, C...               1.0   \n",
       "3361  COL18A1, KCNE2, TMPRSS3, ITGB2, DYRK1A, TSPEAR...               1.0   \n",
       "3362  SETD5, SETD2, GFM1, ZBTB20, PLOD2, SLC6A1, MYL...               1.0   \n",
       "3363                                        TUBB, PSMB8               1.0   \n",
       "\n",
       "                        Source  \n",
       "0               BioPlanet_2019  \n",
       "1               BioPlanet_2019  \n",
       "2               BioPlanet_2019  \n",
       "3               BioPlanet_2019  \n",
       "4               BioPlanet_2019  \n",
       "...                        ...  \n",
       "3359  Chromosome_Location_hg19  \n",
       "3360  Chromosome_Location_hg19  \n",
       "3361  Chromosome_Location_hg19  \n",
       "3362  Chromosome_Location_hg19  \n",
       "3363  Chromosome_Location_hg19  \n",
       "\n",
       "[3364 rows x 6 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df_ext = pd.DataFrame(columns=enrichr_dataset['BioPlanet_2019'].columns)\n",
    "for key, value in enrichr_dataset.items():\n",
    "    subdf = pd.DataFrame(columns=combined_df_ext.columns)\n",
    "    for auc in value['AUC'].unique():\n",
    "        subdf = subdf.append(value[value['AUC'] == auc], ignore_index=True)\n",
    "    combined_df_ext = combined_df_ext.append(subdf, ignore_index=True)\n",
    "combined_df_ext"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.5</td>\n",
       "      <td>3</td>\n",
       "      <td>Activation of TRKA receptors</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.5</td>\n",
       "      <td>4</td>\n",
       "      <td>MAP kinase pathway regulation through dual spe...</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.5</td>\n",
       "      <td>5</td>\n",
       "      <td>ERK inactivation</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.5</td>\n",
       "      <td>7</td>\n",
       "      <td>Nucleotide-like (purinergic) G-protein coupled...</td>\n",
       "      <td>P2RY12</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.5</td>\n",
       "      <td>8</td>\n",
       "      <td>Gap junction assembly</td>\n",
       "      <td>GJB3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.5</td>\n",
       "      <td>9</td>\n",
       "      <td>Interleukin-11 pathway</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.5</td>\n",
       "      <td>10</td>\n",
       "      <td>ADP signalling through P2Y purinoceptor 12</td>\n",
       "      <td>P2RY12</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.5</td>\n",
       "      <td>11</td>\n",
       "      <td>Gap junction trafficking and regulation</td>\n",
       "      <td>GJB3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0.5</td>\n",
       "      <td>12</td>\n",
       "      <td>Nuclear events mediated by MAP kinases</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>0.5</td>\n",
       "      <td>13</td>\n",
       "      <td>Signal amplification</td>\n",
       "      <td>P2RY12</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>0.5</td>\n",
       "      <td>2</td>\n",
       "      <td>NGF signaling via TRKA from the plasma membrane</td>\n",
       "      <td>NTRK2, DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>0.5</td>\n",
       "      <td>14</td>\n",
       "      <td>Wnt interactions in lipid metabolism and immun...</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>0.5</td>\n",
       "      <td>15</td>\n",
       "      <td>Post-translational regulation of adherens junc...</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>0.5</td>\n",
       "      <td>16</td>\n",
       "      <td>Arf6 integrin-mediated signaling pathway</td>\n",
       "      <td>MAPK8IP3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>0.5</td>\n",
       "      <td>17</td>\n",
       "      <td>Non-coding RNA metabolism</td>\n",
       "      <td>SMN1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>0.5</td>\n",
       "      <td>1</td>\n",
       "      <td>MAPK signaling pathway</td>\n",
       "      <td>NTRK2, MAPK8IP3, DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>0.5</td>\n",
       "      <td>18</td>\n",
       "      <td>JNK/MAPK pathway</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>0.5</td>\n",
       "      <td>19</td>\n",
       "      <td>SHP2 signaling</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>0.5</td>\n",
       "      <td>20</td>\n",
       "      <td>Neurotrophic factor-mediated Trk receptor sign...</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>0.5</td>\n",
       "      <td>21</td>\n",
       "      <td>FGF signaling pathway</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    AUC Rank                                          Term_Name  \\\n",
       "0   0.5    3                       Activation of TRKA receptors   \n",
       "1   0.5    4  MAP kinase pathway regulation through dual spe...   \n",
       "2   0.5    5                                   ERK inactivation   \n",
       "3   0.5    7  Nucleotide-like (purinergic) G-protein coupled...   \n",
       "4   0.5    8                              Gap junction assembly   \n",
       "5   0.5    9                             Interleukin-11 pathway   \n",
       "6   0.5   10         ADP signalling through P2Y purinoceptor 12   \n",
       "7   0.5   11            Gap junction trafficking and regulation   \n",
       "8   0.5   12             Nuclear events mediated by MAP kinases   \n",
       "9   0.5   13                               Signal amplification   \n",
       "10  0.5    2    NGF signaling via TRKA from the plasma membrane   \n",
       "11  0.5   14  Wnt interactions in lipid metabolism and immun...   \n",
       "12  0.5   15  Post-translational regulation of adherens junc...   \n",
       "13  0.5   16           Arf6 integrin-mediated signaling pathway   \n",
       "14  0.5   17                          Non-coding RNA metabolism   \n",
       "15  0.5    1                             MAPK signaling pathway   \n",
       "16  0.5   18                                   JNK/MAPK pathway   \n",
       "17  0.5   19                                     SHP2 signaling   \n",
       "18  0.5   20  Neurotrophic factor-mediated Trk receptor sign...   \n",
       "19  0.5   21                              FGF signaling pathway   \n",
       "\n",
       "         Overlapping_Genes  Adjusted_P-value          Source  \n",
       "0                    NTRK2               1.0  BioPlanet_2019  \n",
       "1                    DUSP6               1.0  BioPlanet_2019  \n",
       "2                    DUSP6               1.0  BioPlanet_2019  \n",
       "3                   P2RY12               1.0  BioPlanet_2019  \n",
       "4                     GJB3               1.0  BioPlanet_2019  \n",
       "5                    DUSP6               1.0  BioPlanet_2019  \n",
       "6                   P2RY12               1.0  BioPlanet_2019  \n",
       "7                     GJB3               1.0  BioPlanet_2019  \n",
       "8                    DUSP6               1.0  BioPlanet_2019  \n",
       "9                   P2RY12               1.0  BioPlanet_2019  \n",
       "10            NTRK2, DUSP6               1.0  BioPlanet_2019  \n",
       "11                   DUSP6               1.0  BioPlanet_2019  \n",
       "12                   NTRK2               1.0  BioPlanet_2019  \n",
       "13                MAPK8IP3               1.0  BioPlanet_2019  \n",
       "14                    SMN1               1.0  BioPlanet_2019  \n",
       "15  NTRK2, MAPK8IP3, DUSP6               1.0  BioPlanet_2019  \n",
       "16                   DUSP6               1.0  BioPlanet_2019  \n",
       "17                   NTRK2               1.0  BioPlanet_2019  \n",
       "18                   NTRK2               1.0  BioPlanet_2019  \n",
       "19                   DUSP6               1.0  BioPlanet_2019  "
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df_ext.where((combined_df_ext['AUC'] == 0.5) & (combined_df_ext['Source'] == 'BioPlanet_2019')).dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "word_count_df = count_words_in_df(combined_df_ext)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>count</th>\n",
       "      <th>auc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>signaling</td>\n",
       "      <td>19</td>\n",
       "      <td>0.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>pathway</td>\n",
       "      <td>14</td>\n",
       "      <td>0.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>disease</td>\n",
       "      <td>9</td>\n",
       "      <td>0.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>9</td>\n",
       "      <td>0.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>(bulk</td>\n",
       "      <td>9</td>\n",
       "      <td>0.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>498</th>\n",
       "      <td>signaling</td>\n",
       "      <td>13</td>\n",
       "      <td>0.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499</th>\n",
       "      <td>pathway</td>\n",
       "      <td>10</td>\n",
       "      <td>0.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500</th>\n",
       "      <td>disease</td>\n",
       "      <td>10</td>\n",
       "      <td>0.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501</th>\n",
       "      <td>regulation</td>\n",
       "      <td>9</td>\n",
       "      <td>0.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>9</td>\n",
       "      <td>0.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1055</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>14</td>\n",
       "      <td>0.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1056</th>\n",
       "      <td>signaling</td>\n",
       "      <td>13</td>\n",
       "      <td>0.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1057</th>\n",
       "      <td>pathway</td>\n",
       "      <td>10</td>\n",
       "      <td>0.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1058</th>\n",
       "      <td>disease</td>\n",
       "      <td>10</td>\n",
       "      <td>0.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059</th>\n",
       "      <td>regulation</td>\n",
       "      <td>8</td>\n",
       "      <td>0.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1634</th>\n",
       "      <td>pathway</td>\n",
       "      <td>13</td>\n",
       "      <td>0.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1635</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>13</td>\n",
       "      <td>0.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1636</th>\n",
       "      <td>signaling</td>\n",
       "      <td>11</td>\n",
       "      <td>0.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1637</th>\n",
       "      <td>disease</td>\n",
       "      <td>10</td>\n",
       "      <td>0.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1638</th>\n",
       "      <td>regulation</td>\n",
       "      <td>8</td>\n",
       "      <td>0.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2246</th>\n",
       "      <td>disease</td>\n",
       "      <td>14</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2247</th>\n",
       "      <td>pathway</td>\n",
       "      <td>13</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2248</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>11</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2249</th>\n",
       "      <td>signaling</td>\n",
       "      <td>9</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2250</th>\n",
       "      <td>metabolism</td>\n",
       "      <td>7</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2858</th>\n",
       "      <td>disease</td>\n",
       "      <td>15</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2859</th>\n",
       "      <td>pathway</td>\n",
       "      <td>14</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2860</th>\n",
       "      <td>metabolism</td>\n",
       "      <td>10</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2861</th>\n",
       "      <td>regulation</td>\n",
       "      <td>9</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2862</th>\n",
       "      <td>acid</td>\n",
       "      <td>8</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3490</th>\n",
       "      <td>disease</td>\n",
       "      <td>16</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3491</th>\n",
       "      <td>pathway</td>\n",
       "      <td>11</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3492</th>\n",
       "      <td>regulation</td>\n",
       "      <td>10</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3493</th>\n",
       "      <td>metabolism</td>\n",
       "      <td>9</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3494</th>\n",
       "      <td>signaling</td>\n",
       "      <td>9</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4124</th>\n",
       "      <td>pathway</td>\n",
       "      <td>20</td>\n",
       "      <td>0.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4125</th>\n",
       "      <td>signaling</td>\n",
       "      <td>18</td>\n",
       "      <td>0.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4126</th>\n",
       "      <td>disease</td>\n",
       "      <td>17</td>\n",
       "      <td>0.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4127</th>\n",
       "      <td>metabolism</td>\n",
       "      <td>13</td>\n",
       "      <td>0.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4128</th>\n",
       "      <td>cholesterol</td>\n",
       "      <td>12</td>\n",
       "      <td>0.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4741</th>\n",
       "      <td>signaling</td>\n",
       "      <td>18</td>\n",
       "      <td>0.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4742</th>\n",
       "      <td>pathway</td>\n",
       "      <td>16</td>\n",
       "      <td>0.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4743</th>\n",
       "      <td>disease</td>\n",
       "      <td>12</td>\n",
       "      <td>0.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4744</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>11</td>\n",
       "      <td>0.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4745</th>\n",
       "      <td>cholesterol</td>\n",
       "      <td>9</td>\n",
       "      <td>0.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5335</th>\n",
       "      <td>pathway</td>\n",
       "      <td>13</td>\n",
       "      <td>0.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5336</th>\n",
       "      <td>disease</td>\n",
       "      <td>13</td>\n",
       "      <td>0.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5337</th>\n",
       "      <td>proteins</td>\n",
       "      <td>11</td>\n",
       "      <td>0.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5338</th>\n",
       "      <td>metabolism</td>\n",
       "      <td>10</td>\n",
       "      <td>0.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5339</th>\n",
       "      <td>cardiomyopathy</td>\n",
       "      <td>10</td>\n",
       "      <td>0.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5890</th>\n",
       "      <td>cancer</td>\n",
       "      <td>29</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5891</th>\n",
       "      <td>cardiomyopathy</td>\n",
       "      <td>18</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5892</th>\n",
       "      <td>pathway</td>\n",
       "      <td>16</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5893</th>\n",
       "      <td>signaling</td>\n",
       "      <td>13</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5894</th>\n",
       "      <td>breast</td>\n",
       "      <td>12</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                word count   auc\n",
       "0          signaling    19  0.50\n",
       "1            pathway    14  0.50\n",
       "2            disease     9  0.50\n",
       "3           syndrome     9  0.50\n",
       "4              (bulk     9  0.50\n",
       "498        signaling    13  0.55\n",
       "499          pathway    10  0.55\n",
       "500          disease    10  0.55\n",
       "501       regulation     9  0.55\n",
       "502         syndrome     9  0.55\n",
       "1055        syndrome    14  0.60\n",
       "1056       signaling    13  0.60\n",
       "1057         pathway    10  0.60\n",
       "1058         disease    10  0.60\n",
       "1059      regulation     8  0.60\n",
       "1634         pathway    13  0.65\n",
       "1635        syndrome    13  0.65\n",
       "1636       signaling    11  0.65\n",
       "1637         disease    10  0.65\n",
       "1638      regulation     8  0.65\n",
       "2246         disease    14  0.70\n",
       "2247         pathway    13  0.70\n",
       "2248        syndrome    11  0.70\n",
       "2249       signaling     9  0.70\n",
       "2250      metabolism     7  0.70\n",
       "2858         disease    15  0.75\n",
       "2859         pathway    14  0.75\n",
       "2860      metabolism    10  0.75\n",
       "2861      regulation     9  0.75\n",
       "2862            acid     8  0.75\n",
       "3490         disease    16  0.80\n",
       "3491         pathway    11  0.80\n",
       "3492      regulation    10  0.80\n",
       "3493      metabolism     9  0.80\n",
       "3494       signaling     9  0.80\n",
       "4124         pathway    20  0.85\n",
       "4125       signaling    18  0.85\n",
       "4126         disease    17  0.85\n",
       "4127      metabolism    13  0.85\n",
       "4128     cholesterol    12  0.85\n",
       "4741       signaling    18  0.90\n",
       "4742         pathway    16  0.90\n",
       "4743         disease    12  0.90\n",
       "4744        syndrome    11  0.90\n",
       "4745     cholesterol     9  0.90\n",
       "5335         pathway    13  0.95\n",
       "5336         disease    13  0.95\n",
       "5337        proteins    11  0.95\n",
       "5338      metabolism    10  0.95\n",
       "5339  cardiomyopathy    10  0.95\n",
       "5890          cancer    29  1.00\n",
       "5891  cardiomyopathy    18  1.00\n",
       "5892         pathway    16  1.00\n",
       "5893       signaling    13  1.00\n",
       "5894          breast    12  1.00"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nwkdf = pd.DataFrame(columns=word_count_df.columns)\n",
    "for auc in word_count_df['auc'].unique():\n",
    "    nwkdf = nwkdf.append(word_count_df[word_count_df['auc'] == auc][:5])\n",
    "nwkdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/enrichr_word_count_bokeh.html'"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "plot = plot_count_results(nwkdf, item='word')\n",
    "save(plot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_count_df = pd.DataFrame(columns=['gene', 'count', 'auc'])\n",
    "for auc in combined_df_ext['AUC'].unique():\n",
    "    subset_df = combined_df_ext[combined_df_ext['AUC'] == auc]\n",
    "    gene_list = []\n",
    "    unsplitted_gene_list = subset_df['Overlapping_Genes'].tolist()\n",
    "    for unsplitted_gene in unsplitted_gene_list:\n",
    "        gene_list += unsplitted_gene.split(', ')\n",
    "    gene_count_dict = {}\n",
    "    for gene in gene_list:\n",
    "        if gene not in gene_count_dict.keys():\n",
    "            gene_count_dict[gene] = 1\n",
    "        else:\n",
    "            gene_count_dict[gene] += 1\n",
    "    temp_df = pd.DataFrame(columns=gene_count_df.columns)\n",
    "    for key in gene_count_dict.keys():\n",
    "        temp_temp_df = pd.DataFrame({'gene':key, 'count': gene_count_dict[key],\n",
    "                                    'auc': auc}, index=[0])\n",
    "        temp_df = temp_df.append(temp_temp_df, ignore_index=True)\n",
    "    gene_count_df = gene_count_df.append(temp_df, ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_count_df.sort_values(by=['auc','count'], ascending=[True, False], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "ngkdf = pd.DataFrame(columns=gene_count_df.columns)\n",
    "for auc in gene_count_df['auc'].unique():\n",
    "    ngkdf = ngkdf.append(gene_count_df[gene_count_df['auc'] == auc][:5])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/rjsietsma/PycharmProjects/dsls_master_thesis/side_scripts/EnrichrAPIResults/enrichr_gene_count_bokeh.html'"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "plot = plot_count_results(ngkdf, item='gene')\n",
    "save(plot)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "## Interpretation\n",
    "It seems that a normal distribution like pattern is formed with a steeper slope after the maximum, starting at rougly AUC of 0.3 and ending at roughly 0.8.\n",
    "<br/><br/>\n",
    "- DisGeNET seems to outline an interesting picture with the top 10 term names:\n",
    "    - For it seems that in the really bad performing genes the main cause could be infant development related. Later on, more neurological causes are identified.\n",
    "        - Neurological seems clear to me... There is still so much unknown about the human mind.\n",
    "        - Infant development related is a bit less clear, since there is a growing amount of research towards this infant development.\n",
    "- GO_biological_process seems to have primarily the same genes for AUC 0.55-0.7 that are most enriched\n",
    "    - It seems linked to ion channeling\n",
    "- Feet and toe abnormalities seem to occur once or twice too\n",
    "- Could this be an indication to genes that are related to very small molecules? (like steroids, ions, neuro-molecules?)\n",
    "- Most significant: autosomal dominant inheritence\n",
    "    - Very interesting plot, the HPO bokeh plot of n=10\n",
    "- \n",
    "\n",
    "<br/><br/>\n",
    "TODO: make word counts of genes and term names?\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "# Correlation analysis Enrichr results\n",
    "Investigate the interactive legend.\n",
    "\n",
    "- Bioplanet 2019:\n",
    "    - Anticorrelation possible, high adjusted-pvalue (APV) when auc is low, but low APV when AUC is high.\n",
    "    \n",
    "- Wikipathways:\n",
    "    - Shows simular anticorrelation to bioplanet, but peaks at 0.55 auc. Could be a normal distribution like.\n",
    "    \n",
    "- Kegg 2019:\n",
    "    - Pretty much all over the place. Some significant when low AUC, but also some not significant. Seems like 2 different distributions are present. One distribution is a linear like from auc 0.5 apv 0.2 to auc 0.7 apv 0.6, but above that a normally distributed curve. Interesting, investigate futher.\n",
    "    \n",
    "- Elsevier:\n",
    "    - Clear normal distribution like, with a peak at either 0.55 (everything is not significant) or 0.65 (largely not significant).\n",
    "\n",
    "- GO Biological:\n",
    "    - Similar to kegg, 2 distributions might be shown. A normal distribution like and an inverse exponent. Both seem to be crossing each other at auc 0.5 APV 0.4.\n",
    "\n",
    "- HPO:\n",
    "    - Is low and stays low. Tops at auc 0.55 and APV 0.15. Always significant.\n",
    "\n",
    "- Jensen Diseases:\n",
    "    - Not really following any distribution. Got some significant, some not and some in between. Tops at auc 0.55 and APV 1.\n",
    "    \n",
    "- PheWeb:\n",
    "    - Exponentially correlated. \"Low\" when auc is low, get higher when AUC get higher up to auc 0.95 where it becomes more significant.\n",
    "    \n",
    "- Clinvar:\n",
    "    - Multiple distributions present. One 'high' and one 'low'. Investigate futher.\n",
    "    \n",
    "- GWAS catalogue:\n",
    "    - Maybe anti-correlated with some outliers?\n",
    "    \n",
    "- DisGeNet:\n",
    "    - Always significant, simular to HPO.\n",
    "    \n",
    "- Omim Disease:\n",
    "    - Anti-correlated with some points that are always significant.\n",
    "    \n",
    "- Omim expanded:\n",
    "    - Not following omim disease pattern, more like a normal distribution that peaks at auc 0.7 APV 0.6.\n",
    "    \n",
    "- Human gene atlas:\n",
    "    - Like a half cosinus wave with a very sudden drop at auc 0.8.\n",
    "    \n",
    "- ARCHS4 tissues:\n",
    "    - Normal distribution like that tops over APV of 1 at AUC 0.65.\n",
    "    \n",
    "- HMDB Metabolites:\n",
    "    - Says insigificant up untill auc 0.75, where it drops steeply.\n",
    "    \n",
    "- Chromosome location:\n",
    "    - Always insigificant untill AUC 0.85, where chromosome X is always significant.\n",
    "    \n",
    "    \n",
    "# AUC cutoffs where there is still some variation\n",
    "\n",
    "- Bioplanet: 0.85\n",
    "\n",
    "- Wikipathways: 0.85\n",
    "\n",
    "- Kegg: 0.85\n",
    "\n",
    "- Elsevier pathway collection: 0.80\n",
    "\n",
    "- GO Biological: 0.80\n",
    "\n",
    "- HPO: 0.75\n",
    "\n",
    "- Jensen DISEASES: 0.80\n",
    "\n",
    "- PheWEB: 0.7 (after that some points become more significant again, but that is commonly seen throughout every database)\n",
    "\n",
    "- ClinVar: 0.85\n",
    "\n",
    "- GWAS Catalogue: 0.9\n",
    "\n",
    "- DisGeNET: 0.7\n",
    "\n",
    "- OMIM_Disease: 0.85\n",
    "\n",
    "- OMIM_Expanded: 0.80\n",
    "\n",
    "- Human Gene Atlas: 0.95\n",
    "\n",
    "- ARCHS4 tissues: 0.9\n",
    "\n",
    "- HMDB Metabolites: 0.9\n",
    "\n",
    "- Chromosome location: Odd one out (always insignificant except for auc 0.9-1.0 where chromosome X becomes significant)\n",
    "\n",
    "\n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.5</td>\n",
       "      <td>3</td>\n",
       "      <td>Activation of TRKA receptors</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.5</td>\n",
       "      <td>4</td>\n",
       "      <td>MAP kinase pathway regulation through dual spe...</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.5</td>\n",
       "      <td>5</td>\n",
       "      <td>ERK inactivation</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.5</td>\n",
       "      <td>7</td>\n",
       "      <td>Nucleotide-like (purinergic) G-protein coupled...</td>\n",
       "      <td>P2RY12</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.5</td>\n",
       "      <td>8</td>\n",
       "      <td>Gap junction assembly</td>\n",
       "      <td>GJB3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   AUC Rank                                          Term_Name  \\\n",
       "0  0.5    3                       Activation of TRKA receptors   \n",
       "1  0.5    4  MAP kinase pathway regulation through dual spe...   \n",
       "2  0.5    5                                   ERK inactivation   \n",
       "3  0.5    7  Nucleotide-like (purinergic) G-protein coupled...   \n",
       "4  0.5    8                              Gap junction assembly   \n",
       "\n",
       "  Overlapping_Genes  Adjusted_P-value          Source  \n",
       "0             NTRK2               1.0  BioPlanet_2019  \n",
       "1             DUSP6               1.0  BioPlanet_2019  \n",
       "2             DUSP6               1.0  BioPlanet_2019  \n",
       "3            P2RY12               1.0  BioPlanet_2019  \n",
       "4              GJB3               1.0  BioPlanet_2019  "
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df_ext.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Correlation coefficient for BioPlanet_2019 is:-0.8991204081865278\n",
      "P-value of BioPlanet_2019 is: 0.0001661839752544972\n",
      "Able to reject null hypothesis of BioPlanet_2019 being uncorrelated, there is a:\n",
      "negative correlation. \n",
      "\n",
      "Correlation coefficient for WikiPathways_2019_Human is:-0.6885583156948685\n",
      "P-value of WikiPathways_2019_Human is: 0.019136081775021636\n",
      "Able to reject null hypothesis of WikiPathways_2019_Human being uncorrelated, there is a:\n",
      "negative correlation. \n",
      "\n",
      "Correlation coefficient for KEGG_2019_Human is:-0.6102160571171791\n",
      "P-value of KEGG_2019_Human is: 0.046180632934565924\n",
      "Able to reject null hypothesis of KEGG_2019_Human being uncorrelated, there is a:\n",
      "negative correlation. \n",
      "\n",
      "Correlation coefficient for Elsevier_Pathway_Collection is:-0.9339431975353206\n",
      "P-value of Elsevier_Pathway_Collection is: 2.6028693621417825e-05\n",
      "Able to reject null hypothesis of Elsevier_Pathway_Collection being uncorrelated, there is a:\n",
      "negative correlation. \n",
      "\n",
      "Correlation coefficient for GO_Biological_Process_2018 is:-0.9211793682944862\n",
      "P-value of GO_Biological_Process_2018 is: 5.65647366848333e-05\n",
      "Able to reject null hypothesis of GO_Biological_Process_2018 being uncorrelated, there is a:\n",
      "negative correlation. \n",
      "\n",
      "Correlation coefficient for Human_Phenotype_Ontology is:-1.0\n",
      "P-value of Human_Phenotype_Ontology is: 0.0\n",
      "Able to reject null hypothesis of Human_Phenotype_Ontology being uncorrelated, there is a:\n",
      "negative correlation. \n",
      "\n",
      "Correlation coefficient for Jensen_DISEASES is:-0.9090909090909091\n",
      "P-value of Jensen_DISEASES is: 0.00010559337630360869\n",
      "Able to reject null hypothesis of Jensen_DISEASES being uncorrelated, there is a:\n",
      "negative correlation. \n",
      "\n",
      "Correlation coefficient for PheWeb_2019 is:0.10409313539028804\n",
      "P-value of PheWeb_2019 is: 0.7606881992787188\n",
      "Fail to reject null hypothesis of PheWeb_2019 being uncorrelated, meaning they are uncorrelated.\n",
      "\n",
      "Correlation coefficient for ClinVar_2019 is:-0.690909090909091\n",
      "P-value of ClinVar_2019 is: 0.018565033381595004\n",
      "Able to reject null hypothesis of ClinVar_2019 being uncorrelated, there is a:\n",
      "negative correlation. \n",
      "\n",
      "Correlation coefficient for GWAS_Catalog_2019 is:-0.9174698042719671\n",
      "P-value of GWAS_Catalog_2019 is: 6.919040788741452e-05\n",
      "Able to reject null hypothesis of GWAS_Catalog_2019 being uncorrelated, there is a:\n",
      "negative correlation. \n",
      "\n",
      "Correlation coefficient for DisGeNET is:-0.9703897613415771\n",
      "P-value of DisGeNET is: 7.419726550680055e-07\n",
      "Able to reject null hypothesis of DisGeNET being uncorrelated, there is a:\n",
      "negative correlation. \n",
      "\n",
      "Correlation coefficient for OMIM_Disease is:-0.6727272727272727\n",
      "P-value of OMIM_Disease is: 0.023313039833185972\n",
      "Able to reject null hypothesis of OMIM_Disease being uncorrelated, there is a:\n",
      "negative correlation. \n",
      "\n",
      "Correlation coefficient for OMIM_Expanded is:-0.6605782590758164\n",
      "P-value of OMIM_Expanded is: 0.026929310586705137\n",
      "Able to reject null hypothesis of OMIM_Expanded being uncorrelated, there is a:\n",
      "negative correlation. \n",
      "\n",
      "Correlation coefficient for Human_Gene_Atlas is:-0.8633970960424556\n",
      "P-value of Human_Gene_Atlas is: 0.0006163160723566311\n",
      "Able to reject null hypothesis of Human_Gene_Atlas being uncorrelated, there is a:\n",
      "negative correlation. \n",
      "\n",
      "Correlation coefficient for ARCHS4_Tissues is:-0.5545454545454546\n",
      "P-value of ARCHS4_Tissues is: 0.07665233333978251\n",
      "Fail to reject null hypothesis of ARCHS4_Tissues being uncorrelated, meaning they are uncorrelated.\n",
      "\n",
      "Correlation coefficient for HMDB_Metabolites is:-0.7006490497453708\n",
      "P-value of HMDB_Metabolites is: 0.02401850924875498\n",
      "Able to reject null hypothesis of HMDB_Metabolites being uncorrelated, there is a:\n",
      "negative correlation. \n",
      "\n",
      "Correlation coefficient for Chromosome_Location_hg19 is:-0.5\n",
      "P-value of Chromosome_Location_hg19 is: 0.11730680301423815\n",
      "Fail to reject null hypothesis of Chromosome_Location_hg19 being uncorrelated, meaning they are uncorrelated.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for source in combined_df_ext['Source'].unique():\n",
    "    subset = combined_df_ext[combined_df_ext['Source'] == source]\n",
    "    subset = subset.groupby(by='AUC').mean()\n",
    "    rho, pval = stats.spearmanr(subset.index, subset['Adjusted_P-value'])\n",
    "    print(f'Correlation coefficient for {source} is:{rho}')\n",
    "    print(f'P-value of {source} is: {pval}')\n",
    "    if pval >= 0.05:\n",
    "        print(f'Fail to reject null hypothesis of {source} being uncorrelated, meaning they are uncorrelated.\\n')\n",
    "    else:\n",
    "        print(f'Able to reject null hypothesis of {source} being uncorrelated, there is a:')\n",
    "        if rho > 0:\n",
    "            print('positive correlation. \\n')\n",
    "        else:\n",
    "            print('negative correlation. \\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      Adjusted_P-value\n",
      "AUC                   \n",
      "0.50      1.000000e+00\n",
      "0.55      1.000000e+00\n",
      "0.60      1.000000e+00\n",
      "0.65      9.580241e-01\n",
      "0.70      9.615924e-01\n",
      "0.75      9.699334e-01\n",
      "0.80      9.860955e-01\n",
      "0.85      5.848485e-01\n",
      "0.90      1.880091e-01\n",
      "0.95      2.542807e-02\n",
      "1.00      2.557615e-12\n"
     ]
    }
   ],
   "source": [
    "for source in combined_df_ext['Source'].unique():\n",
    "    subset = combined_df_ext[combined_df_ext['Source'] == source]\n",
    "    subset = subset.groupby(by='AUC').mean()\n",
    "    rho, pval = stats.spearmanr(subset.index, subset['Adjusted_P-value'])\n",
    "    print(subset)\n",
    "    break\n",
    "    print(f'Correlation coefficient for {source} is:{rho}')\n",
    "    print(f'P-value of {source} is: {pval}')\n",
    "    if pval >= 0.05:\n",
    "        print(f'Fail to reject null hypothesis of {source} being uncorrelated, meaning they are uncorrelated.\\n')\n",
    "    else:\n",
    "        print(f'Able to reject null hypothesis of {source} being uncorrelated, there is a:')\n",
    "        if rho > 0:\n",
    "            print('positive correlation. \\n')\n",
    "        else:\n",
    "            print('negative correlation. \\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.5</td>\n",
       "      <td>3</td>\n",
       "      <td>Activation of TRKA receptors</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.5</td>\n",
       "      <td>4</td>\n",
       "      <td>MAP kinase pathway regulation through dual spe...</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.5</td>\n",
       "      <td>5</td>\n",
       "      <td>ERK inactivation</td>\n",
       "      <td>DUSP6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.5</td>\n",
       "      <td>7</td>\n",
       "      <td>Nucleotide-like (purinergic) G-protein coupled...</td>\n",
       "      <td>P2RY12</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.5</td>\n",
       "      <td>8</td>\n",
       "      <td>Gap junction assembly</td>\n",
       "      <td>GJB3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BioPlanet_2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3359</th>\n",
       "      <td>1.0</td>\n",
       "      <td>17</td>\n",
       "      <td>chr22</td>\n",
       "      <td>SMARCB1, TMPRSS6, PDGFB, TCF20, PLA2G6, TYMP, ...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3360</th>\n",
       "      <td>1.0</td>\n",
       "      <td>16</td>\n",
       "      <td>chr6 cox hap2</td>\n",
       "      <td>TNXB, CYP21A2, TUBB, COL11A2, SKIV2L, VARS2, C...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3361</th>\n",
       "      <td>1.0</td>\n",
       "      <td>18</td>\n",
       "      <td>chr21</td>\n",
       "      <td>COL18A1, KCNE2, TMPRSS3, ITGB2, DYRK1A, TSPEAR...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3362</th>\n",
       "      <td>1.0</td>\n",
       "      <td>20</td>\n",
       "      <td>chr3</td>\n",
       "      <td>SETD5, SETD2, GFM1, ZBTB20, PLOD2, SLC6A1, MYL...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3363</th>\n",
       "      <td>1.0</td>\n",
       "      <td>19</td>\n",
       "      <td>chr6 apd hap1</td>\n",
       "      <td>TUBB, PSMB8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Chromosome_Location_hg19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3364 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      AUC Rank                                          Term_Name  \\\n",
       "0     0.5    3                       Activation of TRKA receptors   \n",
       "1     0.5    4  MAP kinase pathway regulation through dual spe...   \n",
       "2     0.5    5                                   ERK inactivation   \n",
       "3     0.5    7  Nucleotide-like (purinergic) G-protein coupled...   \n",
       "4     0.5    8                              Gap junction assembly   \n",
       "...   ...  ...                                                ...   \n",
       "3359  1.0   17                                              chr22   \n",
       "3360  1.0   16                                      chr6 cox hap2   \n",
       "3361  1.0   18                                              chr21   \n",
       "3362  1.0   20                                               chr3   \n",
       "3363  1.0   19                                      chr6 apd hap1   \n",
       "\n",
       "                                      Overlapping_Genes  Adjusted_P-value  \\\n",
       "0                                                 NTRK2               1.0   \n",
       "1                                                 DUSP6               1.0   \n",
       "2                                                 DUSP6               1.0   \n",
       "3                                                P2RY12               1.0   \n",
       "4                                                  GJB3               1.0   \n",
       "...                                                 ...               ...   \n",
       "3359  SMARCB1, TMPRSS6, PDGFB, TCF20, PLA2G6, TYMP, ...               1.0   \n",
       "3360  TNXB, CYP21A2, TUBB, COL11A2, SKIV2L, VARS2, C...               1.0   \n",
       "3361  COL18A1, KCNE2, TMPRSS3, ITGB2, DYRK1A, TSPEAR...               1.0   \n",
       "3362  SETD5, SETD2, GFM1, ZBTB20, PLOD2, SLC6A1, MYL...               1.0   \n",
       "3363                                        TUBB, PSMB8               1.0   \n",
       "\n",
       "                        Source  \n",
       "0               BioPlanet_2019  \n",
       "1               BioPlanet_2019  \n",
       "2               BioPlanet_2019  \n",
       "3               BioPlanet_2019  \n",
       "4               BioPlanet_2019  \n",
       "...                        ...  \n",
       "3359  Chromosome_Location_hg19  \n",
       "3360  Chromosome_Location_hg19  \n",
       "3361  Chromosome_Location_hg19  \n",
       "3362  Chromosome_Location_hg19  \n",
       "3363  Chromosome_Location_hg19  \n",
       "\n",
       "[3364 rows x 6 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df_ext"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Source Jensen_DISEASES has a significant positive correlation:\n",
      "Term: Ollier disease\n",
      "p-value: 0.0\n",
      "correlation coefficient: 1.0\n",
      "Running from 0.75 to 0.85\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for term in combined_df_ext['Term_Name'].unique():\n",
    "    subset = combined_df_ext[combined_df_ext['Term_Name'] == term]\n",
    "    min_thres = subset['AUC'].min()\n",
    "    max_thres = subset['AUC'].max()\n",
    "    if min_thres == max_thres:\n",
    "        continue\n",
    "    data_source = subset['Source'].unique()[0]\n",
    "    rho, pval = stats.spearmanr(subset['AUC'], subset['Adjusted_P-value'])\n",
    "    if pval <= 0.05:\n",
    "        if rho > 0:\n",
    "            print(f\"Source {data_source} has a significant positive correlation:\")\n",
    "            print(f\"Term: {term}\")\n",
    "            print(f\"p-value: {pval}\")\n",
    "            print(f\"correlation coefficient: {rho}\")\n",
    "            print(f\"Running from {min_thres} to {max_thres}\")\n",
    "            print(\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "pheweb = combined_df_ext.where((combined_df_ext['AUC'] <= 0.9) & (combined_df_ext['Source'] == 'PheWeb_2019')).dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Correlation coefficient for pheweb is:0.45643546458763845\n",
      "P-value of pheweb is: 0.2168375456189006\n",
      "Fail to reject null hypothesis of pheweb being uncorrelated, meaning they are uncorrelated.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "pheweb.index = pheweb.reindex(labels=np.arange(pheweb.shape[0]), axis=0).index\n",
    "grouped_pheweb = pheweb.groupby(by='AUC').mean()\n",
    "grouped_pheweb\n",
    "rho, pval = stats.spearmanr(grouped_pheweb.index, grouped_pheweb['Adjusted_P-value'])\n",
    "print(f'Correlation coefficient for pheweb is:{rho}')\n",
    "print(f'P-value of pheweb is: {pval}')\n",
    "if pval >= 0.05:\n",
    "    print(f'Fail to reject null hypothesis of pheweb being uncorrelated, meaning they are uncorrelated.\\n')\n",
    "else:\n",
    "    print(f'Able to reject null hypothesis of pheweb being uncorrelated, there is a:')\n",
    "    if rho > 0:\n",
    "        print('positive correlation. \\n')\n",
    "    else:\n",
    "        print('negative correlation. \\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Both 0.7 and 0.9 state that PheWeb is positively correlated when AUC cutoff is left at 0.9."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "kegg = combined_df_ext.where((combined_df_ext['AUC'] <= 0.85) & (combined_df_ext['Source'] == 'KEGG_2019_Human')).dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Correlation coefficient for kegg is:0.10910894511799618\n",
      "P-value of kegg is: 0.7970385741881917\n",
      "Fail to reject null hypothesis of kegg being uncorrelated, meaning they are uncorrelated.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "kegg.index = kegg.reindex(labels=np.arange(kegg.shape[0]), axis=0).index\n",
    "grouped_kegg = kegg.groupby(by='AUC').mean()\n",
    "grouped_kegg\n",
    "rho, pval = stats.spearmanr(grouped_kegg.index, grouped_kegg['Adjusted_P-value'])\n",
    "print(f'Correlation coefficient for kegg is:{rho}')\n",
    "print(f'P-value of kegg is: {pval}')\n",
    "if pval >= 0.05:\n",
    "    print(f'Fail to reject null hypothesis of kegg being uncorrelated, meaning they are uncorrelated.\\n')\n",
    "else:\n",
    "    print(f'Able to reject null hypothesis of kegg being uncorrelated, there is a:')\n",
    "    if rho > 0:\n",
    "        print('positive correlation. \\n')\n",
    "    else:\n",
    "        print('negative correlation. \\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Further confirms that KEGG might have multiple distributions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "jensen_d = combined_df_ext.where((combined_df_ext['AUC'] <= 0.8) & (combined_df_ext['Source'] == 'Jensen_DISEASES')).dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Correlation coefficient for jensen_d is:-0.642857142857143\n",
      "P-value of jensen_d is: 0.11939237342741094\n",
      "Fail to reject null hypothesis of jensen_d being uncorrelated, meaning they are uncorrelated.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "jensen_d.index = jensen_d.reindex(labels=np.arange(jensen_d.shape[0]), axis=0).index\n",
    "grouped_jensen_d = jensen_d.groupby(by='AUC').mean()\n",
    "rho, pval = stats.spearmanr(grouped_jensen_d.index, grouped_jensen_d['Adjusted_P-value'])\n",
    "print(f'Correlation coefficient for jensen_d is:{rho}')\n",
    "print(f'P-value of jensen_d is: {pval}')\n",
    "if pval >= 0.05:\n",
    "    print(f'Fail to reject null hypothesis of jensen_d being uncorrelated, meaning they are uncorrelated.\\n')\n",
    "else:\n",
    "    print(f'Able to reject null hypothesis of jensen_d being uncorrelated, there is a:')\n",
    "    if rho > 0:\n",
    "        print('positive correlation. \\n')\n",
    "    else:\n",
    "        print('negative correlation. \\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "- Look into pathways in the databases themselfs, not the complete database.\n",
    "    - Final enrichr analysis, investigate best hits, are these significant?\n",
    "    \n",
    "        - Is there something here?\n",
    "        \n",
    "            -No: No specific enrichment for any pathways.\n",
    "            \n",
    "            -Yes: Very good! Investigate further.\n",
    "            \n",
    "- Investigate CAPICE performance on UMCG gene panels.\n",
    "\n",
    "- Train new model on umcg gene panels.\n",
    "    - 1st option: Top 3 worst and top 3 best by Fingerspitsengefuhl (then put into discussion that more might have been better)\n",
    "    - 2nd option: Use bad performing genes as panel to train on.\n",
    "        - Slice out problem genes or genes that share some commonalities?\n",
    "    - First disease / gene panel specific model training without adding features.\n",
    "    - Then add new features.\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/rjsietsma/PycharmProjects/dsls_master_thesis/venv/lib/python3.8/site-packages/scipy/stats/stats.py:4196: SpearmanRConstantInputWarning: An input array is constant; the correlation coefficent is not defined.\n",
      "  warnings.warn(SpearmanRConstantInputWarning())\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>count</th>\n",
       "      <th>auc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ollier</td>\n",
       "      <td>1</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>disease</td>\n",
       "      <td>1</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ollier</td>\n",
       "      <td>1</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>disease</td>\n",
       "      <td>1</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ollier</td>\n",
       "      <td>1</td>\n",
       "      <td>0.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>disease</td>\n",
       "      <td>1</td>\n",
       "      <td>0.85</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      word count   auc\n",
       "0   ollier     1  0.75\n",
       "1  disease     1  0.75\n",
       "2   ollier     1  0.80\n",
       "3  disease     1  0.80\n",
       "4   ollier     1  0.85\n",
       "5  disease     1  0.85"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nlines = 0\n",
    "temp_df = pd.DataFrame(columns=combined_df_ext.columns)\n",
    "for pathway in combined_df_ext['Term_Name'].unique():\n",
    "    nlines += 1\n",
    "    subset = combined_df_ext[combined_df_ext['Term_Name'] == pathway]\n",
    "    x = subset['AUC']\n",
    "    y = subset['Adjusted_P-value']\n",
    "    rho, pval = stats.spearmanr(x, y)\n",
    "    if pval < 0.05 and rho > 0:\n",
    "        temp_df = temp_df.append(subset, ignore_index=True)\n",
    "second_word_count = count_words_in_df(temp_df)\n",
    "second_word_count.sort_values(by='count', ascending=False).head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "combined_df_ext['n_genes'] = combined_df_ext['Overlapping_Genes'].apply(lambda x: split_and_count_words(x))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/rjsietsma/PycharmProjects/dsls_master_thesis/venv/lib/python3.8/site-packages/scipy/stats/stats.py:4196: SpearmanRConstantInputWarning: An input array is constant; the correlation coefficent is not defined.\n",
      "  warnings.warn(SpearmanRConstantInputWarning())\n"
     ]
    }
   ],
   "source": [
    "nlines = 0\n",
    "for pathway in combined_df_ext['Term_Name'].unique():\n",
    "    nlines += 1\n",
    "    subset = combined_df_ext[combined_df_ext['Term_Name'] == pathway]\n",
    "    x = subset['AUC']\n",
    "    y = subset['Adjusted_P-value']\n",
    "    min_genes = subset['n_genes'].min()\n",
    "    max_genes = subset['n_genes'].max()\n",
    "    rho, pval = stats.spearmanr(x, y)\n",
    "    if pval < 0.1 and rho > 0:\n",
    "        genes = subset['Overlapping_Genes'].values.tolist()\n",
    "        output_genes = []\n",
    "        for gene in genes:\n",
    "            intermediate_genes = split_and_count_words(gene, return_value = None)\n",
    "            for g in intermediate_genes:\n",
    "                if g not in output_genes:\n",
    "                    output_genes.append(g)\n",
    "        if len(output_genes) >= 3:\n",
    "            output_genes = ', '.join(output_genes)\n",
    "            print(f'Able to reject null hypothesis of {pathway} for AUC: '\n",
    "                  f'{x.values[0]} - {x.values[-1]} and source: '\n",
    "                  f'{subset[\"Source\"].unique()[0]} for genes: {output_genes}')\n",
    "            print('\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transfer RNA aminoacylation\n",
      "      AUC Rank                    Term_Name Overlapping_Genes  \\\n",
      "108  0.75    3  Transfer RNA aminoacylation        YARS, GARS   \n",
      "\n",
      "     Adjusted_P-value          Source  n_genes  \n",
      "108               1.0  BioPlanet_2019        2  \n",
      "\n",
      "\n",
      "Oxytocin signaling pathway\n",
      "Empty DataFrame\n",
      "Columns: [AUC, Rank, Term_Name, Overlapping_Genes, Adjusted_P-value, Source, n_genes]\n",
      "Index: []\n",
      "\n",
      "\n",
      "Tinnitus\n",
      "       AUC Rank Term_Name     Overlapping_Genes  Adjusted_P-value  \\\n",
      "1658  0.90   11  Tinnitus          TRPC6, AGBL1               1.0   \n",
      "1668  0.95    2  Tinnitus  CACNB2, TRPC6, AGBL1               1.0   \n",
      "\n",
      "           Source  n_genes  \n",
      "1658  PheWeb_2019        2  \n",
      "1668  PheWeb_2019        3  \n",
      "\n",
      "\n",
      "Acne\n",
      "       AUC Rank Term_Name Overlapping_Genes  Adjusted_P-value       Source  \\\n",
      "1583  0.70   18      Acne             NRXN3               1.0  PheWeb_2019   \n",
      "1605  0.75   20      Acne             NRXN3               1.0  PheWeb_2019   \n",
      "\n",
      "      n_genes  \n",
      "1583        1  \n",
      "1605        1  \n",
      "\n",
      "\n",
      "COLONIC MUCOSA\n",
      "      AUC Rank       Term_Name Overlapping_Genes  Adjusted_P-value  \\\n",
      "2925  0.8  108  COLONIC MUCOSA            SCN10A          0.999888   \n",
      "\n",
      "              Source  n_genes  \n",
      "2925  ARCHS4_Tissues        1  \n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "interesting_pathway = ['Transfer RNA aminoacylation',\n",
    "                        'Oxytocin signaling pathway',\n",
    "                        'Tinnitus',\n",
    "                        'Acne',\n",
    "                        'COLONIC MUCOSA'\n",
    "                      ]\n",
    "for pathway in interesting_pathway:\n",
    "    print(pathway)\n",
    "    print(combined_df_ext[combined_df_ext['Term_Name'] == pathway])\n",
    "    print('\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/rjsietsma/PycharmProjects/dsls_master_thesis/venv/lib/python3.8/site-packages/scipy/stats/stats.py:4196: SpearmanRConstantInputWarning: An input array is constant; the correlation coefficent is not defined.\n",
      "  warnings.warn(SpearmanRConstantInputWarning())\n"
     ]
    }
   ],
   "source": [
    "nlines = 0\n",
    "for pathway in combined_df_ext['Term_Name'].unique():\n",
    "    nlines += 1\n",
    "    subset = combined_df_ext[combined_df_ext['Term_Name'] == pathway]\n",
    "    x = subset['AUC']\n",
    "    y = subset['Adjusted_P-value']\n",
    "    min_genes = subset['n_genes'].min()\n",
    "    max_genes = subset['n_genes'].max()\n",
    "    rho, pval = stats.spearmanr(x, y)\n",
    "    if pval < 0.05 and rho > 0:\n",
    "        genes = subset['Overlapping_Genes'].values.tolist()\n",
    "        output_genes = []\n",
    "        for gene in genes:\n",
    "            intermediate_genes = split_and_count_words(gene, return_value = None)\n",
    "            for g in intermediate_genes:\n",
    "                if g not in output_genes:\n",
    "                    output_genes.append(g)\n",
    "        if len(output_genes) >= 3:\n",
    "            output_genes = ', '.join(output_genes)\n",
    "            print(f'Able to reject null hypothesis of {pathway} for AUC: '\n",
    "                  f'{x.values[0]} - {x.values[-1]} and source: '\n",
    "                  f'{subset[\"Source\"].unique()[0]} for genes: {output_genes}')\n",
    "            print('\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "      <th>n_genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1658</th>\n",
       "      <td>0.90</td>\n",
       "      <td>11</td>\n",
       "      <td>Tinnitus</td>\n",
       "      <td>TRPC6, AGBL1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>PheWeb_2019</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1668</th>\n",
       "      <td>0.95</td>\n",
       "      <td>2</td>\n",
       "      <td>Tinnitus</td>\n",
       "      <td>CACNB2, TRPC6, AGBL1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>PheWeb_2019</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       AUC Rank Term_Name     Overlapping_Genes  Adjusted_P-value  \\\n",
       "1658  0.90   11  Tinnitus          TRPC6, AGBL1               1.0   \n",
       "1668  0.95    2  Tinnitus  CACNB2, TRPC6, AGBL1               1.0   \n",
       "\n",
       "           Source  n_genes  \n",
       "1658  PheWeb_2019        2  \n",
       "1668  PheWeb_2019        3  "
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df_ext[combined_df_ext['Term_Name'] == 'Tinnitus']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Term_Name</th>\n",
       "      <th>Overlapping_Genes</th>\n",
       "      <th>Adjusted_P-value</th>\n",
       "      <th>Source</th>\n",
       "      <th>n_genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1846</th>\n",
       "      <td>0.95</td>\n",
       "      <td>19</td>\n",
       "      <td>brugada syndrome</td>\n",
       "      <td>CACNB2, CACNA1C</td>\n",
       "      <td>0.078511</td>\n",
       "      <td>ClinVar_2019</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       AUC Rank         Term_Name Overlapping_Genes  Adjusted_P-value  \\\n",
       "1846  0.95   19  brugada syndrome   CACNB2, CACNA1C          0.078511   \n",
       "\n",
       "            Source  n_genes  \n",
       "1846  ClinVar_2019        2  "
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df_ext[combined_df_ext['Term_Name'].str.lower().str.startswith('brugada')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Genes involved in identifying brugada syndrome might be troublesome for CAPICE, but then again, if we look at the command 'combined_df_ext[combined_df_ext['Term_Name'].str.lower().str.startswith('brugada')]' there's not a lot of genes in that.\n",
    "- Colonic Mucosa is not really that interesting either. Starts at AUC 0.65 and ends at 0.75, only increasing slightly with an adjusted p-value of at least 0.999925 (very insignificant).\n",
    "    - Transfer RNA aminoacylation is a bit different, still insignificant but starts at 0.594633.\n",
    "    - Same goes for Oxytocin signaling pathway, starts at 0.941531\n",
    "    - Tinnitus is a bit different, only has 2 genes in the lower AUC values but starts out at 0.218952 and goes up to 1 from auc 0.5-0.9.\n",
    "    \n",
    "    \n",
    "- When a pad performing gene panels is found, make a decision tree on what to do next.\n",
    "- Keep in mind, I have 4 times the amount of benign samples than malignant samples.\n",
    "- Chi squared / regression correlation analysis between false positive / negative samples on a shared feature.\n",
    "- Turning in concept verion paper on 6th of July is possible with the 2 weeks deadline for feedback. \n",
    "- All methods to seperate python files and then import them to clean up notebook.\n",
    "- Phase 2 will produce graphs, order them to the story I want to tell.\n",
    "    - Thesis is a bit more than a paper, but a bit less than a bachelor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/home/rjsietsma/Documents/LAB-F0701 Excel van Genenlijst panels Genoomdiagnostiek UMCG_versie 4 mei.xlsx'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-61-745cd7f7f818>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mumcg_genepanels\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_excel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/home/rjsietsma/Documents/LAB-F0701 Excel van Genenlijst panels Genoomdiagnostiek UMCG_versie 4 mei.xlsx'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msheet_name\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/PycharmProjects/dsls_master_thesis/venv/lib/python3.8/site-packages/pandas/util/_decorators.py\u001b[0m in \u001b[0;36mwrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    294\u001b[0m                 )\n\u001b[1;32m    295\u001b[0m                 \u001b[0mwarnings\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwarn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmsg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mFutureWarning\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstacklevel\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mstacklevel\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 296\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    297\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    298\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mwrapper\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/PycharmProjects/dsls_master_thesis/venv/lib/python3.8/site-packages/pandas/io/excel/_base.py\u001b[0m in \u001b[0;36mread_excel\u001b[0;34m(io, sheet_name, header, names, index_col, usecols, squeeze, dtype, engine, converters, true_values, false_values, skiprows, nrows, na_values, keep_default_na, na_filter, verbose, parse_dates, date_parser, thousands, comment, skipfooter, convert_float, mangle_dupe_cols)\u001b[0m\n\u001b[1;32m    302\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    303\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mio\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mExcelFile\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 304\u001b[0;31m         \u001b[0mio\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mExcelFile\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mio\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mengine\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    305\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0mio\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    306\u001b[0m         raise ValueError(\n",
      "\u001b[0;32m~/PycharmProjects/dsls_master_thesis/venv/lib/python3.8/site-packages/pandas/io/excel/_base.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, path_or_buffer, engine)\u001b[0m\n\u001b[1;32m    865\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_io\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mstringify_path\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpath_or_buffer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    866\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 867\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engines\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_io\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    868\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    869\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__fspath__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/PycharmProjects/dsls_master_thesis/venv/lib/python3.8/site-packages/pandas/io/excel/_xlrd.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, filepath_or_buffer)\u001b[0m\n\u001b[1;32m     20\u001b[0m         \u001b[0merr_msg\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"Install xlrd >= 1.0.0 for Excel support\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     21\u001b[0m         \u001b[0mimport_optional_dependency\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"xlrd\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mextra\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merr_msg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 22\u001b[0;31m         \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     23\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     24\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0mproperty\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/PycharmProjects/dsls_master_thesis/venv/lib/python3.8/site-packages/pandas/io/excel/_base.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, filepath_or_buffer)\u001b[0m\n\u001b[1;32m    351\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbook\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mload_workbook\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    352\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 353\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbook\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mload_workbook\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    354\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbytes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    355\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbook\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mload_workbook\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mBytesIO\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/PycharmProjects/dsls_master_thesis/venv/lib/python3.8/site-packages/pandas/io/excel/_xlrd.py\u001b[0m in \u001b[0;36mload_workbook\u001b[0;34m(self, filepath_or_buffer)\u001b[0m\n\u001b[1;32m     35\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mopen_workbook\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfile_contents\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     36\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 37\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mopen_workbook\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     38\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     39\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0mproperty\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/PycharmProjects/dsls_master_thesis/venv/lib/python3.8/site-packages/xlrd/__init__.py\u001b[0m in \u001b[0;36mopen_workbook\u001b[0;34m(filename, logfile, verbosity, use_mmap, file_contents, encoding_override, formatting_info, on_demand, ragged_rows)\u001b[0m\n\u001b[1;32m    109\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    110\u001b[0m         \u001b[0mfilename\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexpanduser\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilename\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 111\u001b[0;31m         \u001b[0;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilename\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"rb\"\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    112\u001b[0m             \u001b[0mpeek\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpeeksz\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    113\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mpeek\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34mb\"PK\\x03\\x04\"\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;31m# a ZIP file\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/home/rjsietsma/Documents/LAB-F0701 Excel van Genenlijst panels Genoomdiagnostiek UMCG_versie 4 mei.xlsx'"
     ]
    }
   ],
   "source": [
    "umcg_genepanels = pd.read_excel('/home/rjsietsma/Documents/LAB-F0701 Excel van Genenlijst panels Genoomdiagnostiek UMCG_versie 4 mei.xlsx', sheet_name=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "start = '5GPM'\n",
    "output_genepanels = {}\n",
    "in_genepanels = False\n",
    "for value in umcg_genepanels.values():\n",
    "    subject = value.columns[0]\n",
    "    if subject == start:\n",
    "        in_genepanels = True\n",
    "    if in_genepanels:\n",
    "        specific_disease = value.iloc[0,0]\n",
    "        if isinstance(specific_disease, float):\n",
    "            specific_disease = subject\n",
    "        first_col = subject\n",
    "        real_col = 'Gen'\n",
    "        gen_index = value[value[first_col] == real_col].index\n",
    "        if gen_index.shape[0] < 1:\n",
    "            continue\n",
    "        if value.shape[1] > 1:\n",
    "            value = value.iloc[:,0]\n",
    "        value = value.shift(-(gen_index.values[0] + 1))\n",
    "        if not isinstance(value, pd.DataFrame):\n",
    "            value = pd.DataFrame(value)\n",
    "        value.columns = [real_col]\n",
    "        value.replace('Terug naar inhoudsopgave', np.NaN, inplace=True)     \n",
    "        value.dropna(inplace=True)\n",
    "        output = value.values.tolist()\n",
    "        real_output = []\n",
    "        for i in output:\n",
    "            real_output = real_output + i\n",
    "        if subject not in output_genepanels.keys():\n",
    "            temp = {specific_disease: real_output}\n",
    "            output_genepanels[subject] = temp\n",
    "        else:\n",
    "            temp = {specific_disease: real_output}\n",
    "            subject_dict = output_genepanels[subject]\n",
    "            subject_dict.update(temp)\n",
    "\n",
    "output_genepanels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "auc_data = data[['gene','default_auc']].copy()\n",
    "auc_data.columns = ['gene','auc']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "auc_data.set_index('gene', inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "auc_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "x = auc_data.index.values\n",
    "x2 = auc_data['auc'].values\n",
    "auc_data_series = pd.Series(x2, index=x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "auc_data_dict = auc_data_series.to_dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "umcg_genepanel_analysis = pd.DataFrame(columns=['category','panel','min_auc',\n",
    "                                                'max_auc','avg_auc',\n",
    "                                                'std','med_auc','used%','used_n',\n",
    "                                                'genes_no_data'])\n",
    "for key in output_genepanels.keys():\n",
    "    for key2, value in output_genepanels[key].items():\n",
    "        disease = \", \".join([key, key2])\n",
    "        n_genes_tot = len(value)\n",
    "        genes_no_auc_data = []\n",
    "        keep_track = np.full(shape=len(value), fill_value=np.NaN)\n",
    "        for iteration, gene in enumerate(value):\n",
    "            if gene in auc_data_dict.keys():\n",
    "                keep_track[iteration] = auc_data_dict[gene]\n",
    "            else:\n",
    "                genes_no_auc_data.append(gene)\n",
    "        keep_track = keep_track[~np.isnan(keep_track)]\n",
    "        min_auc = keep_track.min()\n",
    "        max_auc = keep_track.max()\n",
    "        average_auc = keep_track.mean()\n",
    "        std_auc = keep_track.std()\n",
    "        median_auc = np.median(keep_track)\n",
    "        n_used = keep_track.size\n",
    "        used_n = f'{n_used}/{n_genes_tot}'\n",
    "        p_used = round(n_used / n_genes_tot * 100)\n",
    "        x = pd.DataFrame({'category':[key],\n",
    "                          'panel':[key2],\n",
    "                          'min_auc':[min_auc],\n",
    "                         'max_auc':[max_auc],\n",
    "                         'avg_auc':[average_auc],\n",
    "                         'std':[std_auc],\n",
    "                         'med_auc':[median_auc],\n",
    "                         'used%':[p_used],\n",
    "                         'used_n':[used_n],\n",
    "                         'genes_no_data':[genes_no_auc_data]}\n",
    "                         , index=[1])\n",
    "        umcg_genepanel_analysis = umcg_genepanel_analysis.append(x, ignore_index=True)\n",
    "umcg_genepanel_analysis           "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "umcg_genepanel_analysis.sort_values(by='avg_auc', ascending=True).tail(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "categories = umcg_genepanel_analysis['category'].unique()\n",
    "pallette = inferno(len(categories) + 1)\n",
    "colormap = dict(zip(categories, pallette))\n",
    "umcg_genepanel_analysis['color'] = umcg_genepanel_analysis['category'].map(colormap)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "umcg_genepanel_analysis.sort_values(by='avg_auc', inplace=True)\n",
    "umcg_genepanel_analysis['placeholder_x'] = np.arange(umcg_genepanel_analysis.shape[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "placeholder_actual_dict = dict(zip(umcg_genepanel_analysis['placeholder_x'].tolist(),\n",
    "                                  umcg_genepanel_analysis['panel'].tolist()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "output_file(filename='./umcg_genepanel_barplot.html',\n",
    "            title='Boxplot UMCG Genepanels'\n",
    "           )\n",
    "title = 'Boxplot of UMCG genepanels'\n",
    "source = ColumnDataSource(umcg_genepanel_analysis)\n",
    "p = figure(title=title,\n",
    "           plot_width=1500,\n",
    "               plot_height=600,\n",
    "               toolbar_location='right',\n",
    "               toolbar_sticky=False)\n",
    "p.add_layout(Legend(), 'right')\n",
    "p.vbar(x='placeholder_x',\n",
    "      top='avg_auc',\n",
    "      color='color',\n",
    "      legend_field='category',\n",
    "      width=0.9,\n",
    "      source=source)\n",
    "p.add_tools(HoverTool(tooltips=[\n",
    "    ('Category','@category'),\n",
    "    ('Panel','@panel'),\n",
    "    ('AUC','@avg_auc')\n",
    "]))\n",
    "save(p)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "umcg_genepanel_analysis[umcg_genepanel_analysis['category'] == 'Primaire ImmuundeficiÃ«ntie']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "umcg_genepanel_analysis.sort_values(by='avg_auc', ascending=True, inplace=True)\n",
    "umcg_genepanel_analysis[umcg_genepanel_analysis['panel'].str.lower().str.startswith('dyslipid')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "wilcoxon_analysis = pd.DataFrame(columns=['category', 'compared_to',\n",
    "                                          'p_two_sided', 'p_lower',\n",
    "                                          'p_greater'])\n",
    "wilcoxon_alternatives = ['two-sided', 'greater', 'less']\n",
    "for category in umcg_genepanel_analysis['category'].unique():\n",
    "    subset = umcg_genepanel_analysis[umcg_genepanel_analysis['category'] == category]\n",
    "    subset_auc = np.array(subset['avg_auc'])\n",
    "    not_subset = umcg_genepanel_analysis[umcg_genepanel_analysis['category'] != category]\n",
    "    for n_sub in not_subset['category'].unique():\n",
    "        two = None\n",
    "        greater = None\n",
    "        less = None\n",
    "        sub_subset = not_subset[not_subset['category'] == n_sub]\n",
    "        sub_subauc = np.array(sub_subset['avg_auc'])\n",
    "        for alt in wilcoxon_alternatives:\n",
    "            _, p = stats.mannwhitneyu(subset_auc, sub_subauc, alternative=alt)\n",
    "            if alt == 'two-sided':\n",
    "                two = p\n",
    "            elif alt == 'greater':\n",
    "                greater = p\n",
    "            elif alt == 'less':\n",
    "                less = p\n",
    "        sub_cat_df = pd.DataFrame(\n",
    "            {\n",
    "                'category':category,\n",
    "                'compared_to':n_sub,\n",
    "                'p_two_sided':two,\n",
    "                'p_lower':less,\n",
    "                'p_greater':greater\n",
    "            }, index=[0]\n",
    "        )\n",
    "        wilcoxon_analysis = wilcoxon_analysis.append(sub_cat_df, ignore_index=True)\n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "categories = wilcoxon_analysis['category'].unique()\n",
    "pallette = inferno(len(categories) + 1)\n",
    "colormap = dict(zip(categories, pallette))\n",
    "wilcoxon_analysis['color'] = wilcoxon_analysis['compared_to'].map(colormap)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "to_be_placeholded = wilcoxon_analysis['compared_to'].unique()\n",
    "placehold_array = np.arange(len(to_be_placeholded))\n",
    "placeholder_dict = dict(zip(to_be_placeholded, placehold_array))\n",
    "wilcoxon_analysis['x_placeholder'] = wilcoxon_analysis['compared_to'].map(placeholder_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "wilcoxon_analysis[wilcoxon_analysis['category'] == 'Hart- en vaatziekten']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "x = np.random.randint(low=20,high=40,size=(20))\n",
    "y = np.random.randint(20,size=(20))\n",
    "stats.mannwhitneyu(x,y, alternative='greater')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for category in wilcoxon_analysis['category'].unique():\n",
    "    subset = wilcoxon_analysis[wilcoxon_analysis['category'] == category]\n",
    "    ax = subset.plot.bar(x='compared_to',y=['p_two_sided', 'p_lower', 'p_greater'],\n",
    "                         colormap='viridis',\n",
    "                         figsize=(10,5),\n",
    "                         title=f'Mann-Whitney analysis of {category}'\n",
    "                        )\n",
    "    ax.hlines(y=0.05,xmin=-10,xmax=100)\n",
    "    ax.set_ylabel('P-value')\n",
    "    ax.set_xlabel('Panel')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now what do the plots above mean?\n",
    "\n",
    "    - P_two_sided tests the null hypothesis that the distribution of title is different from the alternative. So when P is low, reject H0 (meaning there is a statisticial difference).\n",
    "    - P_lower tests the alternative hypothesis that the median difference is negative. When P is low, reject H0, meaning that the median difference is negative. (If this value is low, it is statistically proven that x is lower than y)\n",
    "    - P_higher tests the alternative hypothesis that the median difference is positive. When P is low, reject h0, meaning that the median difference is positive. (If this value is low, it is statistically proven that x is higher than y)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# From the Wilcoxon / Mannwhitney analysis\n",
    "##### Hart- en vaatziekten\n",
    "Every p_greater has a high P-value, indicating the category is not performing better than any other panel. Interestingly, Amyloidose, Leukemie-Lymfoom, 5GPM, OWA, Noonan syndroom, Cardiomyopathie bij kinderen, mitochondriele aandoeningen, hyper/hypofosfatemie, primaire cilliare dyskinesie en angio-oedeem also accepts H0 for two-sided and less, so it performs simular to them.\n",
    "##### Erfelijke Kanker\n",
    "Now quite the opposite of Hart- en vaatziekten, where p_lower accepts H0 except for maybe Angio-oedeem, meaning that everything is performing either worse or on par. Only Hart- en vaatziekten, epilepsie, metabole & leverziekten, aangeboren hartafwijkingen have some statistical proof of being better / worse than erfelijke kanker.\n",
    "##### Primaire Immuundeficientie\n",
    "Primaire immuundefientie is pretty much the average panel, only Hart- en vaatziekten performs statistically performs worse. Everything else is null hypothesis.\n",
    "###### Angio-oedeem\n",
    "Should be performing the best, a lot of high less p values is seen. But the null hypothesis for two-sided and greater can't be rejected in a single instance, so it might be performing better at sight, but it performs just average (which is good, this means CAPICE has a high average in the UMCG genepanels).\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "allvs1 = pd.DataFrame(columns=['two-sided', 'less', 'greater'])\n",
    "for category in umcg_genepanel_analysis['category'].unique():\n",
    "    if category == '5GPM':\n",
    "        continue\n",
    "    x = np.array(umcg_genepanel_analysis[umcg_genepanel_analysis['category'] == category]['avg_auc'])\n",
    "\n",
    "    y = np.array(umcg_genepanel_analysis.where(\n",
    "        ~(umcg_genepanel_analysis['category'] == category) & \n",
    "        ~(umcg_genepanel_analysis['category'] == '5GPM')\n",
    "    ).dropna()['avg_auc'])\n",
    "    output = {'category':'all',\n",
    "              'compared_to':category,\n",
    "              'two-sided':None,\n",
    "              'less':None,\n",
    "              'greater':None}\n",
    "    for alternative in wilcoxon_alternatives:\n",
    "        _, p = stats.mannwhitneyu(x, y, alternative=alternative)\n",
    "        output[alternative] = p\n",
    "    allvs1 = allvs1.append(\n",
    "        pd.DataFrame(output, index=[0]), ignore_index=True\n",
    "    )\n",
    "allvs1        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "allvs1.to_csv('./not_saving_directory/previousmodelauc.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ax = allvs1.plot(y=['two-sided','less','greater'], x='compared_to', kind='bar', colormap='viridis', figsize=(10,5),\n",
    "                title='Mann-Whitney analysis of panel vs all.')\n",
    "ax.hlines(y=0.05,xmin=-10,xmax=100)\n",
    "ax.set_ylabel('P-value')\n",
    "ax.set_xlabel('Panel')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "umcg_genepanel_analysis.where(~(umcg_genepanel_analysis['category'] == 'Hart- en vaatziekten') & \n",
    "                             ~(umcg_genepanel_analysis['category'] == '5GPM')).dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "havd = output_genepanels['Hart- en vaatziekten']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "lohavdg = [] # List Of Heart And Vascular Disease Genes\n",
    "for key in havd.keys():\n",
    "    if key.lower().startswith('dyslipid'):\n",
    "        lohavdg = lohavdg + havd[key]\n",
    "lohavdg = list(dict.fromkeys(lohavdg))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "n_dislipid = pd.DataFrame(columns=['total', 'benign', 'malign'])\n",
    "counting_dict = {\n",
    "    'total':0,\n",
    "    'benign':0,\n",
    "    'malignant':0\n",
    "}\n",
    "for gene in lohavdg:\n",
    "    subset = data[data['gene'] == gene]\n",
    "    if subset.shape[0] > 0:\n",
    "        counting_dict['total'] += subset['n_tot'].values[0]\n",
    "        counting_dict['benign'] += subset['n_benign'].values[0]\n",
    "        counting_dict['malignant'] += subset['n_malign'].values[0]\n",
    "print(f\"In dyslipidimie related genes, a total of {counting_dict['total']} variants are present, \"\n",
    "      f\"of which {counting_dict['benign']} are benign and {counting_dict['malignant']} are malignant.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Preparing ClinVar and VKGL databases\n",
    "What genome build does VKGL have?\n",
    "    - 37\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_clinvar_header(clinvar_loc):\n",
    "    header = None\n",
    "    with gzip.open(clinvar_loc) as clinvar_file:\n",
    "        while True:\n",
    "            line = clinvar_file.readline().decode('utf-8')\n",
    "            if line.startswith('#CHROM'):\n",
    "                header = line.strip().split('\\t')\n",
    "                break\n",
    "    return header\n",
    "clinvar_loc = '/run/media/rjsietsma/evo2tb/linux/Datafiles/clinvar.vcf.gz'\n",
    "header = get_clinvar_header(clinvar_loc)\n",
    "clinvar = pd.read_csv(clinvar_loc, comment='#', names=header, sep='\\t', compression='gzip')\n",
    "clinvar = clinvar[~clinvar['INFO'].str.lower().str.contains('clnsig=uncertain')]\n",
    "clinvar = clinvar.reset_index().drop('index', axis=1)\n",
    "clinvar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar.loc[0,'INFO'].split(';')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar.loc[0,:].str.contains('benign')['INFO']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def clinvar_extract_clnsig(x):\n",
    "    return_value = np.NaN\n",
    "    info_list = x.split(';')\n",
    "    for y in info_list:\n",
    "        if y.startswith('CLNSIG'):\n",
    "            label = y.split('=')[1].lower()\n",
    "            if (label.find('benign') != -1):\n",
    "                return_value = 'LB'\n",
    "            elif (label.find('pathogenic') != -1):\n",
    "                return_value = 'LP'\n",
    "    return return_value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar['label'] = clinvar['INFO'].apply(lambda x: clinvar_extract_clnsig(x))\n",
    "clinvar.dropna(axis=0, subset=['label'], inplace=True)\n",
    "clinvar = clinvar.reset_index().drop('index', axis=1)\n",
    "clinvar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "vkgl = pd.read_csv('/run/media/rjsietsma/evo2tb/linux/Datafiles/VKGL_public_consensus_jun2020.csv.gz', compression='gzip')\n",
    "vkgl = vkgl[vkgl['classification'] != 'VUS']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "vkgl.drop_duplicates(subset=['chromosome','start','ref','alt'], inplace=True)\n",
    "vkgl = vkgl.reset_index().drop('index',axis=1)[['chromosome','start','ref','alt','classification']]\n",
    "vkgl_cols = ['#CHROM','POS','REF','ALT', 'label']\n",
    "vkgl.columns = vkgl_cols\n",
    "vkgl['INFO'] = '.'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_vkgl = clinvar.append(vkgl, ignore_index=True).drop(['ID','QUAL','FILTER'], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def build_nondupe_clinvarvkgl(x):\n",
    "    dupe_check = vkgl_cols[:-1]\n",
    "    label_check = vkgl_cols\n",
    "    if (vkgl[vkgl_cols] == x[vkgl_cols]).all(axis=1).sum() > 0:\n",
    "        print(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "position": {
    "height": "451.85px",
    "left": "1759px",
    "right": "20px",
    "top": "91px",
    "width": "780px"
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
