{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Importing some shizzle.\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from matplotlib import pyplot as plt\n",
    "from scipy import stats\n",
    "import time\n",
    "import requests\n",
    "import datetime\n",
    "import sys\n",
    "import os\n",
    "import pathlib\n",
    "import seaborn as sns\n",
    "from sklearn.metrics import roc_auc_score\n",
    "\n",
    "location = 'rjsietsma'\n",
    "\n",
    "read_loc = '/home/'+location+'/Documents/School/Master_DSLS/Final_Thesis/Initial_Data_exploration/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define function to perform Shapiro-Wilk, Kolmogorov-Smirnov, Wilcoxon-/Mann-Whitney U -test, and a pearson correlation test.\n",
    "\n",
    "def perform_stats(x, y):\n",
    "    shapiro_x = stats.shapiro(x)[1]\n",
    "    shapiro_y = stats.shapiro(y)[1]\n",
    "    if shapiro_x >= 0.05:\n",
    "        print(\"X is likely normally distributed.\")\n",
    "    else:\n",
    "        print(\"X is likely not normally distributed.\")\n",
    "    if shapiro_y >= 0.05:\n",
    "        print(\"Y is likely normally distributed.\")\n",
    "    else:\n",
    "        print(\"Y is likely not normally distributed.\")\n",
    "    try:\n",
    "        p_ks_xy = stats.ks_2samp(x, y)[1]\n",
    "        print(f\"The Kolmogorov-Smirnov test p-value: {p_ks_xy}\")\n",
    "    except Exception:\n",
    "        print(\"Kolmogorov-Smirnov could not be performed!\")\n",
    "    try:\n",
    "        p_wc_xy = stats.wilcoxon(x, y)[1]\n",
    "        print(f\"The Wilcoxon test p-value: {p_wc_xy}\")\n",
    "    except ValueError:\n",
    "        p_mw_xy = stats.mannwhitneyu(x, y)[1]\n",
    "        print(f\"Wilcoxon could not be performed, \\n\"\n",
    "             f\"Using Mann-Whitney rank test p-value: {p_mw_xy}\")\n",
    "    except Exception:\n",
    "        print(\"Neither Wilcoxon nor Mann-Whitney tests could be performed!\")\n",
    "    try:\n",
    "        p_pears_xy = stats.pearsonr(x, y)\n",
    "        print(f\"The Pearson correlation: {p_pears_xy[0]},\\n\"\n",
    "             f\"p-value: {p_pears_xy[1]}\")\n",
    "    except Exception:\n",
    "        print(\"Pearson correlation could not be performed!\")\n",
    "\n",
    "# Define function to calculate the Z-scores of given data.\n",
    "\n",
    "def calc_z_scores(data):\n",
    "    centered = data - data.mean(axis=0)\n",
    "    return centered / centered.std(axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>label</th>\n",
       "      <th>gene</th>\n",
       "      <th>consequence</th>\n",
       "      <th>capice</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14</td>\n",
       "      <td>68196054</td>\n",
       "      <td>GCCCTG</td>\n",
       "      <td>G</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>RDH12</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>0.988552</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20</td>\n",
       "      <td>10626717</td>\n",
       "      <td>TCA</td>\n",
       "      <td>T</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>JAG1</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>0.990329</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20</td>\n",
       "      <td>10625898</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>JAG1</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>0.991696</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20</td>\n",
       "      <td>10628741</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>JAG1</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>0.983607</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20</td>\n",
       "      <td>10625509</td>\n",
       "      <td>ACT</td>\n",
       "      <td>A</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>JAG1</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>0.987999</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>456004</th>\n",
       "      <td>17</td>\n",
       "      <td>29556342</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>NF1</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>0.000139</td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>456005</th>\n",
       "      <td>11</td>\n",
       "      <td>5248177</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>HBB</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>0.069934</td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>456006</th>\n",
       "      <td>15</td>\n",
       "      <td>48787324</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>FBN1</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>0.909190</td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>456007</th>\n",
       "      <td>19</td>\n",
       "      <td>17947957</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>JAK3</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>0.001238</td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>456008</th>\n",
       "      <td>16</td>\n",
       "      <td>223597</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>HBA2</td>\n",
       "      <td>STOP_LOST</td>\n",
       "      <td>0.971616</td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>456009 rows Ã— 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr       pos     ref alt label   gene  consequence    capice source\n",
       "0       14  68196054  GCCCTG   G  LP/P  RDH12  FRAME_SHIFT  0.988552  train\n",
       "1       20  10626717     TCA   T  LP/P   JAG1  STOP_GAINED  0.990329  train\n",
       "2       20  10625898     CTG   C  LP/P   JAG1  FRAME_SHIFT  0.991696  train\n",
       "3       20  10628741      AC   A  LP/P   JAG1  FRAME_SHIFT  0.983607  train\n",
       "4       20  10625509     ACT   A  LP/P   JAG1   DOWNSTREAM  0.987999  train\n",
       "...     ..       ...     ...  ..   ...    ...          ...       ...    ...\n",
       "456004  17  29556342       G   A  LP/P    NF1   SYNONYMOUS  0.000139   test\n",
       "456005  11   5248177       A   T  LP/P    HBB   SYNONYMOUS  0.069934   test\n",
       "456006  15  48787324       T   C  LP/P   FBN1   SYNONYMOUS  0.909190   test\n",
       "456007  19  17947957       G   A  LP/P   JAK3   SYNONYMOUS  0.001238   test\n",
       "456008  16    223597       T   C  LP/P   HBA2    STOP_LOST  0.971616   test\n",
       "\n",
       "[456009 rows x 9 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read in the data\n",
    "full_snv_dataset = pd.read_csv(read_loc+'full_dataset.csv')\n",
    "full_snv_dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def identify_incorrect_genes():\n",
    "    possibly_incorrect_genes = pd.DataFrame(columns=['chr', 'pos', 'ref', 'alt', 'gene', 'correct_chr'])\n",
    "\n",
    "    done = 0\n",
    "    total = full_snv_dataset['gene'].unique().shape[0]\n",
    "    reset_timer = time.time()\n",
    "    for gene in full_snv_dataset['gene'].unique():\n",
    "        time_ifl = time.time()\n",
    "        if time_ifl - reset_timer > 10:\n",
    "            print(f'Still busy, done: {round(done/total*100, ndigits=2)}%')\n",
    "            reset_timer = time.time()\n",
    "        subset_ds = full_snv_dataset[full_snv_dataset['gene'] == gene]\n",
    "        if subset_ds['chr'].unique().shape[0] > 1:\n",
    "            print(f'Gene {gene} lies on multiple chromosomes!: {subset_ds[\"chr\"].unique()}')\n",
    "            value_counts = subset_ds['chr'].value_counts()\n",
    "            print(value_counts)\n",
    "            possibly_incorrect_index = value_counts.sort_values().index[0]\n",
    "            correct_index = value_counts.sort_values().index[1]\n",
    "            print(subset_ds[subset_ds['chr'] == possibly_incorrect_index])\n",
    "            exclude_genes = ['Y-RNA', 'snoU13', 'Y_RNA',\n",
    "                             'SNORA43', 'SNORA73', 'U3', 'U6']\n",
    "            if gene not in exclude_genes:\n",
    "                incorrect_chr = subset_ds[subset_ds['chr'] == str(possibly_incorrect_index)]['chr'].values\n",
    "                pos = subset_ds[subset_ds['chr'] == str(possibly_incorrect_index)]['pos'].values\n",
    "                ref = subset_ds[subset_ds['chr'] == str(possibly_incorrect_index)]['ref'].values\n",
    "                alt = subset_ds[subset_ds['chr'] == str(possibly_incorrect_index)]['alt'].values\n",
    "                output = pd.DataFrame({'chr':incorrect_chr,\n",
    "                                       'pos':pos,\n",
    "                                       'ref':ref,\n",
    "                                       'alt':alt,\n",
    "                                       'gene': gene, \n",
    "                                       'correct_chr': str(correct_index)},\n",
    "                                      index=[0])\n",
    "                possibly_incorrect_genes = possibly_incorrect_genes.append(output, ignore_index=True)\n",
    "        done += 1\n",
    "    return possibly_incorrect_genes\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# possibly_incorrect_genes = identify_incorrect_genes()\n",
    "# possibly_incorrect_genes.to_csv(read_loc+'VKGL_incorrect_annotated_snvs_14nov2019.txt', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'19'"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_ds = full_snv_dataset[full_snv_dataset['gene'] == 'RYR1']\n",
    "subset_ds['chr'].unique()[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Conclusion: dataset is not clean!\n",
    "now it is"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>default_auc</th>\n",
       "      <th>default_f1</th>\n",
       "      <th>default_recall</th>\n",
       "      <th>default_fpr</th>\n",
       "      <th>default_spec</th>\n",
       "      <th>optimal_c</th>\n",
       "      <th>optimal_auc</th>\n",
       "      <th>optimal_f1</th>\n",
       "      <th>optimal_recall</th>\n",
       "      <th>optimal_fpr</th>\n",
       "      <th>optimal_spec</th>\n",
       "      <th>n_tot</th>\n",
       "      <th>n_benign</th>\n",
       "      <th>n_malign</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2775</th>\n",
       "      <td>PRKN</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1659</th>\n",
       "      <td>BUB1B</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>73</td>\n",
       "      <td>66</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2102</th>\n",
       "      <td>GALNT3</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1655</th>\n",
       "      <td>PNPLA1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.8</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.8</td>\n",
       "      <td>31</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2782</th>\n",
       "      <td>TMEM80</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2625</th>\n",
       "      <td>C10orf2</td>\n",
       "      <td>0.469697</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.469697</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>133</td>\n",
       "      <td>132</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2330</th>\n",
       "      <td>SPTLC2</td>\n",
       "      <td>0.455157</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.011</td>\n",
       "      <td>0.750374</td>\n",
       "      <td>0.170213</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>229</td>\n",
       "      <td>223</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2615</th>\n",
       "      <td>PLCD1</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2611</th>\n",
       "      <td>UBE2A</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2806</th>\n",
       "      <td>SOX9-AS1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2807 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene  default_auc  default_f1  default_recall  default_fpr  \\\n",
       "2775      PRKN     1.000000         1.0             1.0          0.0   \n",
       "1659     BUB1B     1.000000         1.0             1.0          0.0   \n",
       "2102    GALNT3     1.000000         1.0             1.0          0.0   \n",
       "1655    PNPLA1     1.000000         1.0             1.0          0.0   \n",
       "2782    TMEM80     1.000000         1.0             1.0          0.0   \n",
       "...        ...          ...         ...             ...          ...   \n",
       "2625   C10orf2     0.469697         0.0             0.0          1.0   \n",
       "2330    SPTLC2     0.455157         0.0             0.0          1.0   \n",
       "2615     PLCD1     0.500000         0.0             0.0          1.0   \n",
       "2611     UBE2A     0.500000         0.0             0.0          1.0   \n",
       "2806  SOX9-AS1     0.000000         0.0             0.0          1.0   \n",
       "\n",
       "      default_spec  optimal_c  optimal_auc  optimal_f1  optimal_recall  \\\n",
       "2775           1.0      0.020     1.000000    1.000000             1.0   \n",
       "1659           1.0      0.020     1.000000    1.000000             1.0   \n",
       "2102           1.0      0.020     1.000000    1.000000             1.0   \n",
       "1655           1.0      0.020     1.000000    1.000000             0.8   \n",
       "2782           1.0      0.020     1.000000    1.000000             1.0   \n",
       "...            ...        ...          ...         ...             ...   \n",
       "2625           0.0      0.020     0.469697    0.000000             0.0   \n",
       "2330           0.0      0.011     0.750374    0.170213             0.0   \n",
       "2615           0.0      0.020     0.500000    0.000000             0.0   \n",
       "2611           0.0      0.020     0.500000    0.000000             0.0   \n",
       "2806           0.0      0.020     0.000000    0.000000             0.0   \n",
       "\n",
       "      optimal_fpr  optimal_spec  n_tot  n_benign  n_malign  \n",
       "2775          0.0           1.0     11         7         4  \n",
       "1659          0.0           1.0     73        66         7  \n",
       "2102          0.0           1.0     12        10         2  \n",
       "1655          0.2           0.8     31        26         5  \n",
       "2782          0.0           1.0      5         4         1  \n",
       "...           ...           ...    ...       ...       ...  \n",
       "2625          1.0           0.0    133       132         1  \n",
       "2330          1.0           0.0    229       223         6  \n",
       "2615          1.0           0.0      8         7         1  \n",
       "2611          1.0           0.0      3         2         1  \n",
       "2806          1.0           0.0      3         2         1  \n",
       "\n",
       "[2807 rows x 15 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_mapped_ds = pd.read_csv(read_loc+'optimal_f1_full_ds_v2.csv', header=0)\n",
    "full_mapped_ds.sort_values(by='default_f1', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>default_auc</th>\n",
       "      <th>default_f1</th>\n",
       "      <th>default_recall</th>\n",
       "      <th>default_fpr</th>\n",
       "      <th>default_spec</th>\n",
       "      <th>optimal_c</th>\n",
       "      <th>optimal_auc</th>\n",
       "      <th>optimal_f1</th>\n",
       "      <th>optimal_recall</th>\n",
       "      <th>optimal_fpr</th>\n",
       "      <th>optimal_spec</th>\n",
       "      <th>n_tot</th>\n",
       "      <th>n_benign</th>\n",
       "      <th>n_malign</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2330</th>\n",
       "      <td>SPTLC2</td>\n",
       "      <td>0.455157</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.011</td>\n",
       "      <td>0.750374</td>\n",
       "      <td>0.170213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>229.0</td>\n",
       "      <td>223.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2376</th>\n",
       "      <td>SCARB1</td>\n",
       "      <td>0.487032</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.487032</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>355.0</td>\n",
       "      <td>347.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2455</th>\n",
       "      <td>GARS</td>\n",
       "      <td>0.561587</td>\n",
       "      <td>0.120000</td>\n",
       "      <td>0.230769</td>\n",
       "      <td>0.769231</td>\n",
       "      <td>0.230769</td>\n",
       "      <td>0.016</td>\n",
       "      <td>0.630599</td>\n",
       "      <td>0.175439</td>\n",
       "      <td>0.230769</td>\n",
       "      <td>0.769231</td>\n",
       "      <td>0.230769</td>\n",
       "      <td>329.0</td>\n",
       "      <td>316.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2399</th>\n",
       "      <td>MUT</td>\n",
       "      <td>0.578992</td>\n",
       "      <td>0.075949</td>\n",
       "      <td>0.428571</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>0.428571</td>\n",
       "      <td>0.010</td>\n",
       "      <td>0.724650</td>\n",
       "      <td>0.102564</td>\n",
       "      <td>0.428571</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>0.428571</td>\n",
       "      <td>262.0</td>\n",
       "      <td>255.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2442</th>\n",
       "      <td>SCN10A</td>\n",
       "      <td>0.598381</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>0.800000</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>0.023</td>\n",
       "      <td>0.598785</td>\n",
       "      <td>0.266667</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>0.800000</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>1245.0</td>\n",
       "      <td>1235.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>CYP7A1</td>\n",
       "      <td>0.646646</td>\n",
       "      <td>0.300000</td>\n",
       "      <td>0.352941</td>\n",
       "      <td>0.647059</td>\n",
       "      <td>0.352941</td>\n",
       "      <td>0.018</td>\n",
       "      <td>0.703715</td>\n",
       "      <td>0.372093</td>\n",
       "      <td>0.352941</td>\n",
       "      <td>0.647059</td>\n",
       "      <td>0.352941</td>\n",
       "      <td>302.0</td>\n",
       "      <td>285.0</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269</th>\n",
       "      <td>PCSK9</td>\n",
       "      <td>0.657343</td>\n",
       "      <td>0.313725</td>\n",
       "      <td>0.363636</td>\n",
       "      <td>0.636364</td>\n",
       "      <td>0.363636</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.657343</td>\n",
       "      <td>0.313725</td>\n",
       "      <td>0.340909</td>\n",
       "      <td>0.659091</td>\n",
       "      <td>0.340909</td>\n",
       "      <td>902.0</td>\n",
       "      <td>858.0</td>\n",
       "      <td>44.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271</th>\n",
       "      <td>STAP1</td>\n",
       "      <td>0.680432</td>\n",
       "      <td>0.260870</td>\n",
       "      <td>0.428571</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>0.428571</td>\n",
       "      <td>0.024</td>\n",
       "      <td>0.688244</td>\n",
       "      <td>0.300000</td>\n",
       "      <td>0.428571</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>0.428571</td>\n",
       "      <td>199.0</td>\n",
       "      <td>192.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>COL18A1</td>\n",
       "      <td>0.702941</td>\n",
       "      <td>0.538462</td>\n",
       "      <td>0.411765</td>\n",
       "      <td>0.588235</td>\n",
       "      <td>0.411765</td>\n",
       "      <td>0.019</td>\n",
       "      <td>0.761765</td>\n",
       "      <td>0.642857</td>\n",
       "      <td>0.411765</td>\n",
       "      <td>0.588235</td>\n",
       "      <td>0.411765</td>\n",
       "      <td>357.0</td>\n",
       "      <td>340.0</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1135</th>\n",
       "      <td>PDE11A</td>\n",
       "      <td>0.704272</td>\n",
       "      <td>0.196721</td>\n",
       "      <td>0.857143</td>\n",
       "      <td>0.142857</td>\n",
       "      <td>0.857143</td>\n",
       "      <td>0.035</td>\n",
       "      <td>0.807076</td>\n",
       "      <td>0.307692</td>\n",
       "      <td>0.857143</td>\n",
       "      <td>0.142857</td>\n",
       "      <td>0.857143</td>\n",
       "      <td>114.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene  default_auc  default_f1  default_recall  default_fpr  \\\n",
       "2330   SPTLC2     0.455157    0.000000        0.000000     1.000000   \n",
       "2376   SCARB1     0.487032    0.000000        0.000000     1.000000   \n",
       "2455     GARS     0.561587    0.120000        0.230769     0.769231   \n",
       "2399      MUT     0.578992    0.075949        0.428571     0.571429   \n",
       "2442   SCN10A     0.598381    0.250000        0.200000     0.800000   \n",
       "110    CYP7A1     0.646646    0.300000        0.352941     0.647059   \n",
       "1269    PCSK9     0.657343    0.313725        0.363636     0.636364   \n",
       "271     STAP1     0.680432    0.260870        0.428571     0.571429   \n",
       "429   COL18A1     0.702941    0.538462        0.411765     0.588235   \n",
       "1135   PDE11A     0.704272    0.196721        0.857143     0.142857   \n",
       "\n",
       "      default_spec  optimal_c  optimal_auc  optimal_f1  optimal_recall  \\\n",
       "2330      0.000000      0.011     0.750374    0.170213        0.000000   \n",
       "2376      0.000000      0.020     0.487032    0.000000        0.000000   \n",
       "2455      0.230769      0.016     0.630599    0.175439        0.230769   \n",
       "2399      0.428571      0.010     0.724650    0.102564        0.428571   \n",
       "2442      0.200000      0.023     0.598785    0.266667        0.200000   \n",
       "110       0.352941      0.018     0.703715    0.372093        0.352941   \n",
       "1269      0.363636      0.020     0.657343    0.313725        0.340909   \n",
       "271       0.428571      0.024     0.688244    0.300000        0.428571   \n",
       "429       0.411765      0.019     0.761765    0.642857        0.411765   \n",
       "1135      0.857143      0.035     0.807076    0.307692        0.857143   \n",
       "\n",
       "      optimal_fpr  optimal_spec   n_tot  n_benign  n_malign  \n",
       "2330     1.000000      0.000000   229.0     223.0       6.0  \n",
       "2376     1.000000      0.000000   355.0     347.0       8.0  \n",
       "2455     0.769231      0.230769   329.0     316.0      13.0  \n",
       "2399     0.571429      0.428571   262.0     255.0       7.0  \n",
       "2442     0.800000      0.200000  1245.0    1235.0      10.0  \n",
       "110      0.647059      0.352941   302.0     285.0      17.0  \n",
       "1269     0.659091      0.340909   902.0     858.0      44.0  \n",
       "271      0.571429      0.428571   199.0     192.0       7.0  \n",
       "429      0.588235      0.411765   357.0     340.0      17.0  \n",
       "1135     0.142857      0.857143   114.0     107.0       7.0  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filter1 = full_mapped_ds['n_benign'] > 5\n",
    "filter2 = full_mapped_ds['n_malign'] > 5\n",
    "filter3 = full_mapped_ds['n_tot'] > 100\n",
    "\n",
    "full_mapped_ds.where(filter1 & filter2 & filter3).dropna().sort_values(by='default_auc', ascending=True).head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Consequence: FRAME_SHIFT\n",
      "LP/P    17804\n",
      "LB/B     8069\n",
      "Name: label, dtype: int64\n",
      "Consequence: STOP_GAINED\n",
      "LP/P    14842\n",
      "LB/B     4458\n",
      "Name: label, dtype: int64\n",
      "Consequence: DOWNSTREAM\n",
      "LB/B    22319\n",
      "LP/P     1224\n",
      "Name: label, dtype: int64\n",
      "Consequence: CANONICAL_SPLICE\n",
      "LP/P    5788\n",
      "LB/B    2817\n",
      "Name: label, dtype: int64\n",
      "Consequence: INTRONIC\n",
      "LB/B    61132\n",
      "LP/P      620\n",
      "Name: label, dtype: int64\n",
      "Consequence: INFRAME\n",
      "LB/B    4053\n",
      "LP/P     824\n",
      "Name: label, dtype: int64\n",
      "Consequence: SPLICE_SITE\n",
      "LB/B    8092\n",
      "LP/P     839\n",
      "Name: label, dtype: int64\n",
      "Consequence: REGULATORY\n",
      "LB/B    13540\n",
      "LP/P      874\n",
      "Name: label, dtype: int64\n",
      "Consequence: 3PRIME_UTR\n",
      "LB/B    688\n",
      "LP/P     11\n",
      "Name: label, dtype: int64\n",
      "Consequence: UPSTREAM\n",
      "LB/B    14500\n",
      "LP/P      958\n",
      "Name: label, dtype: int64\n",
      "Consequence: NONCODING-CHANGE\n",
      "LB/B    1113\n",
      "LP/P       2\n",
      "Name: label, dtype: int64\n",
      "Consequence: NON_SYNONYMOUS\n",
      "LB/B    62811\n",
      "LP/P    19845\n",
      "Name: label, dtype: int64\n",
      "Consequence: NONCODING_CHANGE\n",
      "LB/B    592\n",
      "LP/P     82\n",
      "Name: label, dtype: int64\n",
      "Consequence: SPLICE-SITE\n",
      "LB/B    9682\n",
      "LP/P       8\n",
      "Name: label, dtype: int64\n",
      "Consequence: SYNONYMOUS\n",
      "LB/B    88398\n",
      "LP/P      111\n",
      "Name: label, dtype: int64\n",
      "Consequence: STOP_LOST\n",
      "LB/B    52\n",
      "LP/P    47\n",
      "Name: label, dtype: int64\n",
      "Consequence: NON-SYNONYMOUS\n",
      "LB/B    78999\n",
      "LP/P       70\n",
      "Name: label, dtype: int64\n",
      "Consequence: STOP-GAINED\n",
      "LB/B    199\n",
      "LP/P      8\n",
      "Name: label, dtype: int64\n",
      "Consequence: CANONICAL-SPLICE\n",
      "LB/B    81\n",
      "LP/P     1\n",
      "Name: label, dtype: int64\n",
      "Consequence: 3PRIME-UTR\n",
      "LB/B    8604\n",
      "Name: label, dtype: int64\n",
      "Consequence: FRAME-SHIFT\n",
      "LB/B    227\n",
      "Name: label, dtype: int64\n",
      "Consequence: 5PRIME-UTR\n",
      "LB/B    1244\n",
      "Name: label, dtype: int64\n",
      "Consequence: 5PRIME_UTR\n",
      "LB/B    329\n",
      "LP/P      4\n",
      "Name: label, dtype: int64\n",
      "Consequence: UNKNOWN\n",
      "LB/B    10\n",
      "Name: label, dtype: int64\n",
      "Consequence: INTERGENIC\n",
      "LB/B    30\n",
      "Name: label, dtype: int64\n",
      "Consequence: STOP-LOST\n",
      "LB/B    7\n",
      "Name: label, dtype: int64\n",
      "Consequence: start_retained_variant\n",
      "LB/B    1\n",
      "Name: label, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "for consequence in full_snv_dataset['consequence'].unique():\n",
    "    print(f'Consequence: {consequence}')\n",
    "    print(full_snv_dataset[full_snv_dataset['consequence'] == consequence]['label'].value_counts())\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_snv_dataset[full_snv_dataset['consequence'] == consequence]['label'].value_counts()['LB/B']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Notes meeting 24-03-2020\n",
    "- Get all genes that perform bad from the histograms earlier in this notebook\n",
    "- Add FPR, recall, sensitivity and specificity to mapper function\n",
    "- Add vkgl and test data to optimum AUC mapper\n",
    "- *see school bookmark for gene panels* calculate auc and compare auc's\n",
    "    - Immunodeficiency (big chunk)\n",
    "    - Developmental delay\n",
    "    - 5GPM everything except late onset, but is very large\n",
    "    - More popular panels will have more information, possibly easier to interpret.\n",
    " - Phase 1:\n",
    "     - Make a list of biases and other easy to test statistics per gene / chromosome\n",
    "     - Grab genes that are difficult to predict and label them.\n",
    "     - Apply (NaN)GMLVQ on (to predict) to show correlations / causations.\n",
    "\n",
    "\n",
    "[google docs](https://docs.google.com/document/d/1D5SiNbeDEfY2hTWquS88MGUrLv6IgZzfCzflppRGb5k/edit) of what is interesting\n",
    "\n",
    "To add onto the idea of novel annotations, below are some of the ideas I find interesting:\n",
    "- pLI (loss of intolerance)\n",
    "- Popmax filtering AF (we might be already using it in benchmarking but I guess there is no harm in explicitly testing it?)\n",
    "- [pext](https://www.biorxiv.org/content/10.1101/554444v1)\n",
    "- [Enhancer features](https://www.sciencedirect.com/science/article/abs/pii/S0002929720300124)\n",
    "- Features included in this [paper](https://www.nature.com/articles/s41467-019-13212-3)\n",
    "\n",
    "Interesting papers:\n",
    "\n",
    "https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6369448/\n",
    "\n",
    "https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5618255/ (supplement 5)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Notes on training data on GCC cluster\n",
    "- Some gene panels seem to be included\n",
    "    - Allergy/Immunology/Infectious\n",
    "    - Audiologic/Otolaryngologic\n",
    "    - Biochemical\n",
    "    - Cardiovascular\n",
    "    - Craniofacial (facial bones)\n",
    "    - Dermatologic\n",
    "    - Endocrine\n",
    "    - Gastrointestinal\n",
    "    - Genitourinary\n",
    "    - Hematologic\n",
    "    - Musculoskeletal\n",
    "    - Neurologic\n",
    "    - Obstetric\n",
    "    - Oncologic\n",
    "    - Ophthalmologic\n",
    "    - Pulmonary\n",
    "    - Renal\n",
    "        - Documentation is key, that shows when I looked up the training script.\n",
    "        - CAPICE on github could use some per method documentation.\n",
    "        - __Origin of these panels is the Clinical Genomic Database (CGD)__\n",
    "- Some really unclear columns are in the real training data:\n",
    "    - General\n",
    "    - inTest (is just False)\n",
    "- Lots of commented out lines of code.\n",
    "\n",
    "From the umcg-sli/variant_prioritization/data:\n",
    "    - train_results.txt seems to be the same, somewhat.\n",
    "    - train.txt doesn't seem at all what I have.\n",
    "    - test_results seems to be the same somewhat.\n",
    "\n",
    "Looking in past-paper:\n",
    "    - test_results is a bit more detailed on cluster.\n",
    "    - Same for train_results.\n",
    "    - train.txt is the same as in /data.\n",
    "    \n",
    "Take home note: Ask Shuang what needs to happen if I want to retrain the model from scratch.\n",
    " - step4_xgboost_finetune_hyperparameters_sklearnWrapper.py\n",
    " - impute_preprocess.py\n",
    " \n",
    "I'm able to retrain the model from scratch I think.\n",
    " \n",
    "\n",
    "# Investigate:\n",
    "- What was used in CADD to make that top of the line at that time?\n",
    "    - Or just CADD just a search engine that searches the Alt through different pathogenicity predictors?\n",
    "\n",
    "- CADD already takes into account the phylogenetic tree.\n",
    "    - But not how new a gene is within the human race\n",
    "- CADD is a logistic regression model\n",
    "- Shuang mentioned that RawScore and PHRED are not used in CAPICE, so can I conclude that CAPICE is sort of a CADD 2.0?\n",
    "- What is known about the genes that do not perform well and what is known about the genes that do perform well?\n",
    "\n",
    "\n",
    "__do not forget: make presentation before thursday 12:00 about project__\n",
    "\n",
    "# Notes EOS 02-04-2020\n",
    "\n",
    "- Voor Sander: als je op een pubmed ID klikt, ga je naar molgenis.gcc.rug.nl/pubmedID waar een abstract te lezen is van die pubmed ID... (Is te schrijven in Python, heb ik ervaring mee)\n",
    "- Voor Sander, als het nog geen feature request, verander diseased C nummer met de titel van de hyperlink.\n",
    "\n",
    "Mine \n",
    "\n",
    "- Investigate whenever genes that do not perform well are in multiple gene panels\n",
    "- Possible reduced penetrance in genes that do not perform well\n",
    "- Pathways perform better or worse\n",
    "- Number of pathogenic variants per 1000 (or something) bases (normalizing the variants per gene)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "    \n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
